{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "toc_visible": true,
      "authorship_tag": "ABX9TyNW3sR/HvniEZD2gzN2i3AM",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/oghzzang/AMC_NGS1/blob/master/02_linux.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# 코드 실행시 참고할 점\n",
        "\n",
        "- code block의 맨 앞에 \"!\" 붙은 경우 terminal 에서 같이 실행할 예정임. \"!\" 만 제외하고 붙여넣기\n",
        "\n",
        "- 터미널에서만, 노트북에서만 등 표시가 있는 것은 실행에 유의하기 (순서대로 같이 진행)\n",
        "\n",
        "- 궁금한 점, 나중에 확인하고 싶은 것 등 필기를 하고싶으면 code block에 주석달기 혹은 댓글 추가 기능 활용\n",
        "\n",
        "- **경로확인 => 터미널의 파란색 글씨**"
      ],
      "metadata": {
        "id": "BCGRp-fEQPxj"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# 필기"
      ],
      "metadata": {
        "id": "AI21iSGXYI4M"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Setting"
      ],
      "metadata": {
        "id": "i8_50j6BQxZO"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "아래 ```drive.mount('/content/drive')``` 실행 시 팝업이 뜹니다. `구글 drive 연결` 누르시고 계정 선택 후 확인 누르시면됩니다."
      ],
      "metadata": {
        "id": "FKrDZD8-QzUD"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#runtime: 30s\n",
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "EvYhEWAZ50Qg",
        "outputId": "cb413047-01e8-4a24-bcce-4485c1af2103"
      },
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from contextlib import contextmanager\n",
        "import os\n",
        "\n",
        "def cd(newdir):\n",
        "  os.chdir(newdir)"
      ],
      "metadata": {
        "id": "b6JK468rI46o"
      },
      "execution_count": 2,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir -p /content/drive/MyDrive/AMC_NGS1/02_linux"
      ],
      "metadata": {
        "id": "9zvf5qSvpFeD"
      },
      "execution_count": 3,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "이번 실습의 모든 분석은 /content/drive/MyDrive/AMC_NGS1/02_linux 에서 수행합니다. 링크 클릭하면 왼쪽에 패널이 활성화 됩니다."
      ],
      "metadata": {
        "id": "8CUBVgskovGP"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# 폴더 이동시에만 주의\n",
        "!cd /content/drive/MyDrive/AMC_NGS1/02_linux\n",
        "cd(\"/content/drive/MyDrive/AMC_NGS1/02_linux\") # 노트북에서만 실행"
      ],
      "metadata": {
        "id": "6gq-ADoBReLc"
      },
      "execution_count": 4,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 리눅스 환경 확인"
      ],
      "metadata": {
        "id": "vPptVF137hnP"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!cat /etc/issue"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "I2lDw54S5q2o",
        "outputId": "8fb13465-3ccb-4130-9a2b-09fca038842c"
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Ubuntu 22.04.4 LTS \\n \\l\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 기본 리눅스 명령어"
      ],
      "metadata": {
        "id": "8KrZN7QNSrVU"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "list files: ```ls```"
      ],
      "metadata": {
        "id": "U9kpJc4YSuhu"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!ls -al"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "807lL4GW-Vsb",
        "outputId": "335fa1be-0e56-405f-9b12-e3af192a49a7"
      },
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "total 0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "show file structure: `tree`"
      ],
      "metadata": {
        "id": "Czjul3r2Um1A"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!tree --help"
      ],
      "metadata": {
        "id": "2sqCspVrWgcR"
      },
      "execution_count": 9,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#runtime: 8s\n",
        "!apt-get install tree"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Hx4xW7oNWfKx",
        "outputId": "5c20154b-5232-4c7b-94bc-58d1aeca2932"
      },
      "execution_count": 8,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "tree is already the newest version (2.0.2-1).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 35 not upgraded.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!tree --help"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "IkzwVXmv1mmh",
        "outputId": "389b1db8-b6d1-4876-aad6-74e0c68102de"
      },
      "execution_count": 10,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "usage: tree [-acdfghilnpqrstuvxACDFJQNSUX] [-L level [-R]] [-H  baseHREF]\n",
            "\t[-T title] [-o filename] [-P pattern] [-I pattern] [--gitignore]\n",
            "\t[--matchdirs] [--metafirst] [--ignore-case] [--nolinks] [--inodes]\n",
            "\t[--device] [--sort[=]<name>] [--dirsfirst] [--filesfirst]\n",
            "\t[--filelimit #] [--si] [--du] [--prune] [--charset X]\n",
            "\t[--timefmt[=]format] [--fromfile] [--noreport] [--version] [--help]\n",
            "\t[--] [directory ...]\n",
            "  ------- Listing options -------\n",
            "  -a            All files are listed.\n",
            "  -d            List directories only.\n",
            "  -l            Follow symbolic links like directories.\n",
            "  -f            Print the full path prefix for each file.\n",
            "  -x            Stay on current filesystem only.\n",
            "  -L level      Descend only level directories deep.\n",
            "  -R            Rerun tree when max dir level reached.\n",
            "  -P pattern    List only those files that match the pattern given.\n",
            "  -I pattern    Do not list files that match the given pattern.\n",
            "  --gitignore   Filter by using .gitignore files.\n",
            "  --ignore-case Ignore case when pattern matching.\n",
            "  --matchdirs   Include directory names in -P pattern matching.\n",
            "  --metafirst   Print meta-data at the beginning of each line.\n",
            "  --info        Print information about files found in .info files.\n",
            "  --noreport    Turn off file/directory count at end of tree listing.\n",
            "  --charset X   Use charset X for terminal/HTML and indentation line output.\n",
            "  --filelimit # Do not descend dirs with more than # files in them.\n",
            "  -o filename   Output to file instead of stdout.\n",
            "  ------- File options -------\n",
            "  -q            Print non-printable characters as '?'.\n",
            "  -N            Print non-printable characters as is.\n",
            "  -Q            Quote filenames with double quotes.\n",
            "  -p            Print the protections for each file.\n",
            "  -u            Displays file owner or UID number.\n",
            "  -g            Displays file group owner or GID number.\n",
            "  -s            Print the size in bytes of each file.\n",
            "  -h            Print the size in a more human readable way.\n",
            "  --si          Like -h, but use in SI units (powers of 1000).\n",
            "  -D            Print the date of last modification or (-c) status change.\n",
            "  --timefmt <f> Print and format time according to the format <f>.\n",
            "  -F            Appends '/', '=', '*', '@', '|' or '>' as per ls -F.\n",
            "  --inodes      Print inode number of each file.\n",
            "  --device      Print device ID number to which each file belongs.\n",
            "  ------- Sorting options -------\n",
            "  -v            Sort files alphanumerically by version.\n",
            "  -t            Sort files by last modification time.\n",
            "  -c            Sort files by last status change time.\n",
            "  -U            Leave files unsorted.\n",
            "  -r            Reverse the order of the sort.\n",
            "  --dirsfirst   List directories before files (-U disables).\n",
            "  --filesfirst  List files before directories (-U disables).\n",
            "  --sort X      Select sort: name,version,size,mtime,ctime.\n",
            "  ------- Graphics options -------\n",
            "  -i            Don't print indentation lines.\n",
            "  -A            Print ANSI lines graphic indentation lines.\n",
            "  -S            Print with CP437 (console) graphics indentation lines.\n",
            "  -n            Turn colorization off always (-C overrides).\n",
            "  -C            Turn colorization on always.\n",
            "  ------- XML/HTML/JSON options -------\n",
            "  -X            Prints out an XML representation of the tree.\n",
            "  -J            Prints out an JSON representation of the tree.\n",
            "  -H baseHREF   Prints out HTML format with baseHREF as top directory.\n",
            "  -T string     Replace the default HTML title and H1 header with string.\n",
            "  --nolinks     Turn off hyperlinks in HTML output.\n",
            "  ------- Input options -------\n",
            "  --fromfile    Reads paths from files (.=stdin)\n",
            "  ------- Miscellaneous options -------\n",
            "  --version     Print version and exit.\n",
            "  --help        Print usage and this help message and exit.\n",
            "  --            Options processing terminator.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!tree -L 1"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5YLGPI3LUjZc",
        "outputId": "2a50e058-edd2-4fcd-a48c-3064416bfe43"
      },
      "execution_count": 11,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[01;34m.\u001b[0m\n",
            "\n",
            "0 directories, 0 files\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "'p'athname of the current 'w'orking 'd'irectory: `pwd`"
      ],
      "metadata": {
        "id": "3xkBA8TqaFi7"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!pwd"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "LApIO8JJaE2-",
        "outputId": "24aa53fd-ee9e-4081-f5f2-e23bbad97a49"
      },
      "execution_count": 12,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/drive/MyDrive/AMC_NGS1/02_linux\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "**m**a**k**e **dir**ectory: mkdir"
      ],
      "metadata": {
        "id": "PNJpOrCWfGXG"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir test"
      ],
      "metadata": {
        "id": "3n2ykROffGBW"
      },
      "execution_count": 16,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!ls"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dCvTKEhYfdFC",
        "outputId": "5cd9df8d-43ca-4455-e223-70e3d6f9fe68"
      },
      "execution_count": 17,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "test\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "**r**e**m**ove **dir**ectory: rmdir"
      ],
      "metadata": {
        "id": "uW2OdQ1ofW6x"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!rmdir test"
      ],
      "metadata": {
        "id": "7bv6gfibfU1_"
      },
      "execution_count": 18,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!ls"
      ],
      "metadata": {
        "id": "5aqAODtPffB3"
      },
      "execution_count": 19,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "텍스트 파일 읽기: ```less/cat```"
      ],
      "metadata": {
        "id": "ku5S-pq7Syrg"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!echo \"Start linux tutorial at `date`\" > start.log"
      ],
      "metadata": {
        "id": "0u7rFq51SHkG"
      },
      "execution_count": 20,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "- `less` 사용법\n",
        "  - space bar 사용시 다음 장으로 넘어감\n",
        "  - 맨 끝 줄의 \":\" 뒤에 q 입력시 종료"
      ],
      "metadata": {
        "id": "g0_Goy10TSv-"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# terminal 에서만 실행\n",
        "# less start.log"
      ],
      "metadata": {
        "id": "ZKyMWOPNSXX9"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!cat start.log"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "CJiSmOmPSi9o",
        "outputId": "f0943968-8848-4f82-868c-2019bceec323"
      },
      "execution_count": 21,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Start linux tutorial at Wed Aug 20 11:36:21 PM UTC 2025\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "**r**e**m**ove: rm"
      ],
      "metadata": {
        "id": "Fo0loNcQqD2V"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!rm start.log"
      ],
      "metadata": {
        "id": "RwHdhZn_qD9z"
      },
      "execution_count": 24,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# 본 실습의 목적"
      ],
      "metadata": {
        "id": "ruLMO3aA6d1d"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "본 repository에는 강의에 필요한 파일들과 각종 커맨드 들을 소개하여 여러분들께서 직접 바닥부터 DNA Genome Analysis Variant Calling Pipeline을 만들어 볼 수 있는 tutorial을 제공합니다.\n",
        "특히 본 강의에서는 **BWA**, **GATK4**와 같이 유전체 분석에서 주로 사용되는 프로그램을 사용하여 pipeline을 만들어보는 강의를 진행합니다.\n",
        "\n",
        "1. 리눅스 환경에서 DNA mutation (SNV, Indel)을 검출하는 파이프라인을 제작\n",
        "2. 파이프라인을 구성하는 각 툴에서 들어가는 input과 output을 학습\n",
        "3. 파이프라인을 구성하는 각 툴을 설치하는 방법을 학습\n",
        "4. 파이프라인을 구성하는 각 툴들을 엮어서 파이프라인을 제작하는 방법을 학습"
      ],
      "metadata": {
        "id": "TWpskn356dyj"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "# 실습"
      ],
      "metadata": {
        "id": "71N5wSiWW3GM"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 1. 준비하기"
      ],
      "metadata": {
        "id": "NmdwIXdy9RmH"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 1.1 샘플 소개\n",
        "실습에 사용할 샘플은 public data인 NA12878 샘플입니다. 모든 염색체의 데이터를 사용하기에는 사용할 컴퓨터 자원이 많이 들고\n",
        ", 시간 또한 많이 들기에 금번 실습에서는 축소한 데이터를 사용하여 전체 파이프라인의 워크플로우를 익히는데 집중해보겠습니다\n",
        ". 데이터는 NA12878 chromosome 21번 데이터를 사용하겠습니다. 이에 맞춰 기준서열인 reference sequence 데이터도 chromosome 2\n",
        "1번만 가지고 만든 데이터를 사용하겠습니다. 모든 자료는 github에서 받으신 자료의 data와 resource 디렉터리에 준비하였습니다."
      ],
      "metadata": {
        "id": "AH4aBzL8b3Uq"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 10s\n",
        "!pip install gdown"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "baXUfziv9O6V",
        "outputId": "4b5648b2-b7b7-4879-ab7e-ccb6c83b7241"
      },
      "execution_count": 25,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: gdown in /usr/local/lib/python3.12/dist-packages (5.2.0)\n",
            "Requirement already satisfied: beautifulsoup4 in /usr/local/lib/python3.12/dist-packages (from gdown) (4.13.4)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.12/dist-packages (from gdown) (3.19.1)\n",
            "Requirement already satisfied: requests[socks] in /usr/local/lib/python3.12/dist-packages (from gdown) (2.32.4)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.12/dist-packages (from gdown) (4.67.1)\n",
            "Requirement already satisfied: soupsieve>1.2 in /usr/local/lib/python3.12/dist-packages (from beautifulsoup4->gdown) (2.7)\n",
            "Requirement already satisfied: typing-extensions>=4.0.0 in /usr/local/lib/python3.12/dist-packages (from beautifulsoup4->gdown) (4.14.1)\n",
            "Requirement already satisfied: charset_normalizer<4,>=2 in /usr/local/lib/python3.12/dist-packages (from requests[socks]->gdown) (3.4.3)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.12/dist-packages (from requests[socks]->gdown) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.12/dist-packages (from requests[socks]->gdown) (2.5.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.12/dist-packages (from requests[socks]->gdown) (2025.8.3)\n",
            "Requirement already satisfied: PySocks!=1.5.7,>=1.5.6 in /usr/local/lib/python3.12/dist-packages (from requests[socks]->gdown) (1.7.1)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 60s\n",
        "# download data\n",
        "!gdown --folder 1VK0BxAPCFx3CB3z8vDZL5CnSa1ecCigp \\\n",
        "  -O /content/drive/MyDrive/AMC_NGS1/02_linux/data\n",
        "\n",
        "# download resource file\n",
        "!gdown --folder 1TtZIZ2yUa-FEwDaTCHREdOHvALcsEH8U \\\n",
        "  -O /content/drive/MyDrive/AMC_NGS1/02_linux/resource"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "E9HFdP_A-EJp",
        "outputId": "5a95527e-9f68-4257-81f0-c78b4f6b7deb"
      },
      "execution_count": 26,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Retrieving folder contents\n",
            "Processing file 12sn8SPvZYftHAqqNM2DG-kvByuVOHXCk normal_R1.fastq.gz\n",
            "Processing file 1xGkB1HXnvTousgEJcPbyK5TTCA1c44OQ normal_R2.fastq.gz\n",
            "Processing file 1_1i4Io1Mufw4kyabiOXuJjLXq7WL-u2w sample_1.fastq.gz\n",
            "Processing file 1qzO7CLEPhoBSWSvmrV8KYZISKMsrQzyn sample_2.fastq.gz\n",
            "Processing file 1VbQ549LQQKjzKlrsLbqGzM88I02Fz-ta tumor_R1.fastq.gz\n",
            "Processing file 1xq9o9xRNmk6WmxQwq8FmwWXwUzCQh-Nv tumor_R2.fastq.gz\n",
            "Retrieving folder contents completed\n",
            "Building directory structure\n",
            "Building directory structure completed\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=12sn8SPvZYftHAqqNM2DG-kvByuVOHXCk\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/data/normal_R1.fastq.gz\n",
            "100% 16.0M/16.0M [00:00<00:00, 96.1MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1xGkB1HXnvTousgEJcPbyK5TTCA1c44OQ\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/data/normal_R2.fastq.gz\n",
            "100% 16.7M/16.7M [00:00<00:00, 163MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1_1i4Io1Mufw4kyabiOXuJjLXq7WL-u2w\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/data/sample_1.fastq.gz\n",
            "100% 4.37M/4.37M [00:00<00:00, 127MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1qzO7CLEPhoBSWSvmrV8KYZISKMsrQzyn\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/data/sample_2.fastq.gz\n",
            "100% 4.38M/4.38M [00:00<00:00, 61.1MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1VbQ549LQQKjzKlrsLbqGzM88I02Fz-ta\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/data/tumor_R1.fastq.gz\n",
            "100% 16.1M/16.1M [00:00<00:00, 98.6MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1xq9o9xRNmk6WmxQwq8FmwWXwUzCQh-Nv\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/data/tumor_R2.fastq.gz\n",
            "100% 16.9M/16.9M [00:00<00:00, 141MB/s]\n",
            "Download completed\n",
            "Retrieving folder contents\n",
            "Retrieving folder 1sT9eSMWh--w3MLwVqbcswvG1XbkziEQy knownsites\n",
            "Processing file 1aPdKUcjtWzvhnH47sUEudsP9_bBhPPvz af-only-gnomad.hg38.chr21.vcf.gz\n",
            "Processing file 1HPxvd8sl5PFtLAsgQXCgV9BciI-_Blno af-only-gnomad.hg38.chr21.vcf.gz.tbi\n",
            "Processing file 13eRtlo46_r_4WRoGy_YNKrrLcX3LgZk8 hg38_v0_Homo_sapiens_assembly38.known_indels.chr21.vcf.gz\n",
            "Processing file 1Q4k6dMKqSF6wXQ9HVrIvdvr0s9C1ymED hg38_v0_Homo_sapiens_assembly38.known_indels.chr21.vcf.gz.tbi\n",
            "Retrieving folder 1cL2oy_oZO4YE8Al1G12-YrOgTFEL3cna reference\n",
            "Processing file 12btfbvD7EEXVjzjJhQpZrPIn755QUiwY hg38.chr21.dict\n",
            "Processing file 1iBu_cSFibXr1jQZcV40aH5OC6d5z_Uyf hg38.chr21.fa\n",
            "Processing file 1ebomaSnoh7jjbQkw15oqQfcfeKaLaZOs hg38.chr21.fa.0123\n",
            "Processing file 1tGHn5OoVXBUMy77ymh7JQkQafFXI9Sd7 hg38.chr21.fa.amb\n",
            "Processing file 1JgejtLUn15F3A-77PQc0FbrVHRzDTZgK hg38.chr21.fa.ann\n",
            "Processing file 10VWLFPqbLdyJohdWe5HMAlLV35JxAfNq hg38.chr21.fa.bwt.2bit.64.gz\n",
            "Processing file 1XmZOxUV7p0BT0-aSnVD7EEQOFQeTgFPh hg38.chr21.fa.fai\n",
            "Processing file 1aSQhOTXmKrWi8Tzv3QOfTMvHC7wcAiZ6 hg38.chr21.fa.pac\n",
            "Retrieving folder contents completed\n",
            "Building directory structure\n",
            "Building directory structure completed\n",
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=1aPdKUcjtWzvhnH47sUEudsP9_bBhPPvz\n",
            "From (redirected): https://drive.google.com/uc?id=1aPdKUcjtWzvhnH47sUEudsP9_bBhPPvz&confirm=t&uuid=ad27a12c-ace8-4e76-a494-c5f3a292da19\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/resource/knownsites/af-only-gnomad.hg38.chr21.vcf.gz\n",
            "100% 39.3M/39.3M [00:00<00:00, 179MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1HPxvd8sl5PFtLAsgQXCgV9BciI-_Blno\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/resource/knownsites/af-only-gnomad.hg38.chr21.vcf.gz.tbi\n",
            "100% 24.7k/24.7k [00:00<00:00, 37.6MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=13eRtlo46_r_4WRoGy_YNKrrLcX3LgZk8\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/resource/knownsites/hg38_v0_Homo_sapiens_assembly38.known_indels.chr21.vcf.gz\n",
            "100% 939k/939k [00:00<00:00, 67.7MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1Q4k6dMKqSF6wXQ9HVrIvdvr0s9C1ymED\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/resource/knownsites/hg38_v0_Homo_sapiens_assembly38.known_indels.chr21.vcf.gz.tbi\n",
            "100% 19.7k/19.7k [00:00<00:00, 39.7MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=12btfbvD7EEXVjzjJhQpZrPIn755QUiwY\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.dict\n",
            "100% 140/140 [00:00<00:00, 567kB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1iBu_cSFibXr1jQZcV40aH5OC6d5z_Uyf\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.fa\n",
            "100% 47.5M/47.5M [00:00<00:00, 171MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1ebomaSnoh7jjbQkw15oqQfcfeKaLaZOs\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.fa.0123\n",
            "100% 93.4M/93.4M [00:00<00:00, 141MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1tGHn5OoVXBUMy77ymh7JQkQafFXI9Sd7\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.fa.amb\n",
            "100% 815/815 [00:00<00:00, 3.25MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1JgejtLUn15F3A-77PQc0FbrVHRzDTZgK\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.fa.ann\n",
            "100% 43.0/43.0 [00:00<00:00, 65.8kB/s]\n",
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=10VWLFPqbLdyJohdWe5HMAlLV35JxAfNq\n",
            "From (redirected): https://drive.google.com/uc?id=10VWLFPqbLdyJohdWe5HMAlLV35JxAfNq&confirm=t&uuid=8fa6cf04-ab00-4d13-91fb-d86a8da367f0\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.fa.bwt.2bit.64.gz\n",
            "100% 92.3M/92.3M [00:00<00:00, 107MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1XmZOxUV7p0BT0-aSnVD7EEQOFQeTgFPh\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.fa.fai\n",
            "100% 23.0/23.0 [00:00<00:00, 60.3kB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1aSQhOTXmKrWi8Tzv3QOfTMvHC7wcAiZ6\n",
            "To: /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.fa.pac\n",
            "100% 11.7M/11.7M [00:00<00:00, 171MB/s]\n",
            "Download completed\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!tree /content/drive/MyDrive/AMC_NGS1/02_linux/data"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "i_80K_BBXL3u",
        "outputId": "21186137-06c1-46df-9153-6618615a1b91"
      },
      "execution_count": 27,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[01;34m/content/drive/MyDrive/AMC_NGS1/02_linux/data\u001b[0m\n",
            "├── \u001b[01;31mnormal_R1.fastq.gz\u001b[0m\n",
            "├── \u001b[01;31mnormal_R2.fastq.gz\u001b[0m\n",
            "├── \u001b[01;31msample_1.fastq.gz\u001b[0m\n",
            "├── \u001b[01;31msample_2.fastq.gz\u001b[0m\n",
            "├── \u001b[01;31mtumor_R1.fastq.gz\u001b[0m\n",
            "└── \u001b[01;31mtumor_R2.fastq.gz\u001b[0m\n",
            "\n",
            "0 directories, 6 files\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!tree /content/drive/MyDrive/AMC_NGS1/02_linux/resource"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Lg__Do9lXCiC",
        "outputId": "afb8bbf4-2567-4d94-e3d3-2a71aa8ff8c2"
      },
      "execution_count": 28,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[01;34m/content/drive/MyDrive/AMC_NGS1/02_linux/resource\u001b[0m\n",
            "├── \u001b[01;34mknownsites\u001b[0m\n",
            "│   ├── \u001b[01;31maf-only-gnomad.hg38.chr21.vcf.gz\u001b[0m\n",
            "│   ├── \u001b[00maf-only-gnomad.hg38.chr21.vcf.gz.tbi\u001b[0m\n",
            "│   ├── \u001b[01;31mhg38_v0_Homo_sapiens_assembly38.known_indels.chr21.vcf.gz\u001b[0m\n",
            "│   └── \u001b[00mhg38_v0_Homo_sapiens_assembly38.known_indels.chr21.vcf.gz.tbi\u001b[0m\n",
            "└── \u001b[01;34mreference\u001b[0m\n",
            "    ├── \u001b[00mhg38.chr21.dict\u001b[0m\n",
            "    ├── \u001b[00mhg38.chr21.fa\u001b[0m\n",
            "    ├── \u001b[00mhg38.chr21.fa.0123\u001b[0m\n",
            "    ├── \u001b[00mhg38.chr21.fa.amb\u001b[0m\n",
            "    ├── \u001b[00mhg38.chr21.fa.ann\u001b[0m\n",
            "    ├── \u001b[01;31mhg38.chr21.fa.bwt.2bit.64.gz\u001b[0m\n",
            "    ├── \u001b[00mhg38.chr21.fa.fai\u001b[0m\n",
            "    └── \u001b[00mhg38.chr21.fa.pac\u001b[0m\n",
            "\n",
            "2 directories, 12 files\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "받은 파일 중 ```hg38.chr21.fa.bwt.2bit.64``` 파일의 크기가 커서 gzip으로 압축된 상태의 파일이 올라가 있습니다. 우리가 실습에 사용하기 위해 이 파일의 압축을 풀어야 합니다.\n",
        "\n",
        "파일은 ```/content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference``` 에 위치해있습니다. 다음 커맨드를 사용하여 압축을 풀어봅시다."
      ],
      "metadata": {
        "id": "m4Ip0Bsu_n-L"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!pwd"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "uOrC6X4YqU-O",
        "outputId": "2dc27673-3113-43da-a453-f1bd1034bd74"
      },
      "execution_count": 29,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/drive/MyDrive/AMC_NGS1/02_linux\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cd /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference\n",
        "cd(\"/content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference\") # 노트북에서만 실행"
      ],
      "metadata": {
        "id": "V3pOeo5MUSdE"
      },
      "execution_count": 30,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!ls"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "m0XIkC5gUbKZ",
        "outputId": "3d0d5a24-36a5-4abf-9207-a3526b5de9f7"
      },
      "execution_count": 31,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "hg38.chr21.dict     hg38.chr21.fa.amb\t\t  hg38.chr21.fa.fai\n",
            "hg38.chr21.fa\t    hg38.chr21.fa.ann\t\t  hg38.chr21.fa.pac\n",
            "hg38.chr21.fa.0123  hg38.chr21.fa.bwt.2bit.64.gz\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!gunzip hg38.chr21.fa.bwt.2bit.64.gz"
      ],
      "metadata": {
        "id": "kKKtUW9A_7iA"
      },
      "execution_count": 32,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!ls"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "I0YRW4UXUduO",
        "outputId": "29eb7cd0-6357-45f1-918e-c34ca2504054"
      },
      "execution_count": 33,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "hg38.chr21.dict     hg38.chr21.fa.amb\t       hg38.chr21.fa.fai\n",
            "hg38.chr21.fa\t    hg38.chr21.fa.ann\t       hg38.chr21.fa.pac\n",
            "hg38.chr21.fa.0123  hg38.chr21.fa.bwt.2bit.64\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 1.2 프로그램 설치\n",
        "\n",
        "- 방법1. apt-get install 로 설치하는 방법: ```tree```, ```bwa```, ```samtools```"
      ],
      "metadata": {
        "id": "hWdl4RdMXjO2"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 9s\n",
        "!apt-get install bwa"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "c5ILAm5qpznT",
        "outputId": "e692cde5-e6c1-4644-8430-0d8c33078ced"
      },
      "execution_count": 34,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "bwa is already the newest version (0.7.17-6).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 35 not upgraded.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "- 방법2. 프로그램 홈페이지에서 패키지를 다운받는 방법: ```gatk```, ```fastqc```\n",
        "  - 다운로드 받은 패키지는 /content/drive/MyDrive/AMC_NGS1/02_linux/tools 에 저장\n",
        "\n",
        "  - GATK4의 경우 다운 받아 압축 해제 후 사용가능함.\n",
        "  \n",
        "  - GATK4 다운로드 페이지: https://github.com/broadinstitute/gatk/releases/tag/4.3.0.0\n",
        "\n"
      ],
      "metadata": {
        "id": "biX8WoM5Y7HB"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir /content/drive/MyDrive/AMC_NGS1/02_linux/tools"
      ],
      "metadata": {
        "id": "ULjP9E9uqR4H"
      },
      "execution_count": 35,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!cd /content/drive/MyDrive/AMC_NGS1/02_linux/tools\n",
        "cd(\"/content/drive/MyDrive/AMC_NGS1/02_linux/tools\") #노트북에서만 실행"
      ],
      "metadata": {
        "id": "uwzew-8Zpvce"
      },
      "execution_count": 36,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 17s\n",
        "!wget https://github.com/broadinstitute/gatk/releases/download/4.3.0.0/gatk-4.3.0.0.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4s1TccVPrJ4t",
        "outputId": "620f2c0c-6605-49af-c2fe-ae81952d8227"
      },
      "execution_count": 37,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-08-20 23:38:40--  https://github.com/broadinstitute/gatk/releases/download/4.3.0.0/gatk-4.3.0.0.zip\n",
            "Resolving github.com (github.com)... 140.82.121.3\n",
            "Connecting to github.com (github.com)|140.82.121.3|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://release-assets.githubusercontent.com/github-production-release-asset/27452807/2f011c35-f7ee-4b53-8236-75558badeb82?sp=r&sv=2018-11-09&sr=b&spr=https&se=2025-08-21T00%3A19%3A16Z&rscd=attachment%3B+filename%3Dgatk-4.3.0.0.zip&rsct=application%2Foctet-stream&skoid=96c2d410-5711-43a1-aedd-ab1947aa7ab0&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skt=2025-08-20T23%3A18%3A31Z&ske=2025-08-21T00%3A19%3A16Z&sks=b&skv=2018-11-09&sig=j5nBNNKkpjASl3ANS9uwSLik9xwQajecDNMhq%2F%2Bwg%2FE%3D&jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmVsZWFzZS1hc3NldHMuZ2l0aHVidXNlcmNvbnRlbnQuY29tIiwia2V5Ijoia2V5MSIsImV4cCI6MTc1NTczMzQyMCwibmJmIjoxNzU1NzMzMTIwLCJwYXRoIjoicmVsZWFzZWFzc2V0cHJvZHVjdGlvbi5ibG9iLmNvcmUud2luZG93cy5uZXQifQ.fvLKA--nP2T3zbyf9MaouJWvHlBfXkwb8O9YqCaJCY0&response-content-disposition=attachment%3B%20filename%3Dgatk-4.3.0.0.zip&response-content-type=application%2Foctet-stream [following]\n",
            "--2025-08-20 23:38:40--  https://release-assets.githubusercontent.com/github-production-release-asset/27452807/2f011c35-f7ee-4b53-8236-75558badeb82?sp=r&sv=2018-11-09&sr=b&spr=https&se=2025-08-21T00%3A19%3A16Z&rscd=attachment%3B+filename%3Dgatk-4.3.0.0.zip&rsct=application%2Foctet-stream&skoid=96c2d410-5711-43a1-aedd-ab1947aa7ab0&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skt=2025-08-20T23%3A18%3A31Z&ske=2025-08-21T00%3A19%3A16Z&sks=b&skv=2018-11-09&sig=j5nBNNKkpjASl3ANS9uwSLik9xwQajecDNMhq%2F%2Bwg%2FE%3D&jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmVsZWFzZS1hc3NldHMuZ2l0aHVidXNlcmNvbnRlbnQuY29tIiwia2V5Ijoia2V5MSIsImV4cCI6MTc1NTczMzQyMCwibmJmIjoxNzU1NzMzMTIwLCJwYXRoIjoicmVsZWFzZWFzc2V0cHJvZHVjdGlvbi5ibG9iLmNvcmUud2luZG93cy5uZXQifQ.fvLKA--nP2T3zbyf9MaouJWvHlBfXkwb8O9YqCaJCY0&response-content-disposition=attachment%3B%20filename%3Dgatk-4.3.0.0.zip&response-content-type=application%2Foctet-stream\n",
            "Resolving release-assets.githubusercontent.com (release-assets.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.111.133, ...\n",
            "Connecting to release-assets.githubusercontent.com (release-assets.githubusercontent.com)|185.199.109.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 464953239 (443M) [application/octet-stream]\n",
            "Saving to: ‘gatk-4.3.0.0.zip’\n",
            "\n",
            "gatk-4.3.0.0.zip    100%[===================>] 443.41M  45.9MB/s    in 9.3s    \n",
            "\n",
            "2025-08-20 23:38:49 (47.7 MB/s) - ‘gatk-4.3.0.0.zip’ saved [464953239/464953239]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 20s\n",
        "!unzip gatk-4.3.0.0.zip\n",
        "!rm gatk-4.3.0.0.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "94-HEziMp_np",
        "outputId": "c8b12631-623a-4260-81ce-dc1dbcfdd9b9"
      },
      "execution_count": 38,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  gatk-4.3.0.0.zip\n",
            "   creating: gatk-4.3.0.0/\n",
            "   creating: gatk-4.3.0.0/gatkdoc/\n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_rnaseq_ASEReadCounter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_replicates_CollectIndependentReplicateMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_dragstr_CalibrateDragstrModel.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ChromosomeCounts.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_LibraryReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_DumpTabixIndex.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_FilterVariantTranches.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectWgsMetricsWithNonZeroCoverage.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonChimericOriginalAlignmentReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_MeanQualityByCycle.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_quality_CollectHiSeqXPfFailMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_PrintReadsSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_GenotypeConcordance.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_CheckTerminatorBlock.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadLengthReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_qc_Pileup.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_OverclippedReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_CleanSam.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_mutect_CreateSomaticPanelOfNormals.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_GtcToVcf.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_qc_CheckPileup.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_BwaMemIndexImageCreator.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_flow_HmerMotifs.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_OrientationBiasReadCounts.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_RenameSampleInVcf.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_ReadAnonymizer.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappingQualityAvailableReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_CountNs.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_AssemblyComplexity.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectOxoGMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_VerifyIDIntensityContaminationMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotProperlyPairedReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_FingerprintingDetailMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ExcessiveEndClippedReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_fasta_FastaReferenceMaker.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_BarcodeMetric.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_WellformedReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_AlleleFraction.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_QualityScoreDistributionSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_DepthPerSampleHC.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_SampleReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_bqsr_AnalyzeCovariates.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_RMSMappingQuality.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$FirstOfPairReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_artifacts_ConvertSequencingArtifactToOxoG.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_CountBasesSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_CountReadsSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_funcotator_Funcotator.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectQualityYieldMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_CreateReadCountPanelOfNormals.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqPipelineSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_evidence_CalcMetadataSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$SecondOfPairReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_AlignmentAgreesWithHeaderReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_markduplicates_CollectDuplicateMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_StrandOddsRatio.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_filter_FilterVcf.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectWgsMetricsWithNonZeroCoverage$WgsMetricsWithNonZeroCoverage.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_ParallelCopyGCSDirectoryIntoHDFSSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SamErrorMetric_IndelErrorMetric.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_fasta_ShiftFasta.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_varianteval_VariantEval.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_GcBiasSummaryMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectRrbsMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_RnaSeqMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_MendelianViolations_FindMendelianViolations.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotSecondaryAlignmentReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_CountReads.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_FixVcfHeader.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_directed_CollectHsMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_util_AccumulateQualityYieldMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_IlluminaBasecallingMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_gnarlyGenotyper_GnarlyGenotyper.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_JumpingLibraryMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ValidAlignmentEndReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_markduplicates_EstimateLibraryComplexity.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ReferenceBases.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_PrintReadsHeader.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_ViewSam.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_AnnotateVcfWithExpectedAlleleFraction.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SamErrorMetric_ErrorMetric.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_CountVariants.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_IndexFeatureFile.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$AllowAllReadsReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_util_LiftOverIntervalList.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_MetricsReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotDuplicateReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_evidence_FindBadGenomicKmersSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SamErrorMetric_OverlappingErrorMetric.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_fastq_BamToBfq.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_AddCommentsToBam.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_QualByDepth.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_AnalyzeSaturationMutagenesis.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_CollectVariantCallingMetrics$VariantCallingSummaryMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_GenotypeGVCFs.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterSummaryMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_TandemRepeat.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_CollectArraysVariantCallingMetrics$ArraysControlCodesSummaryMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_BuildBamIndex.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_VariantAnnotator.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_FilterSamReads.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_BaseQualityRankSumTest.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_MergePedIntoVcf.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_Concordance.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_FragmentLengthReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_rnaseq_SplitNCigarReads.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PrimaryLineReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonZeroFragmentLengthReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_BafRegressMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectMultipleMetricsSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_reference_CompareReferences.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_StructuralVariationDiscoveryPipelineSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_EvaluateInfoFieldConcordance.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_CountBases.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_StrandOddsRatio.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_SplitVcfs.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/index.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SplitSamByLibrary.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_funcotator_FilterFuncotations.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_MendelianViolations_MendelianViolationMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_fingerprint_CheckFingerprint.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_CNNVariantTrain.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$HasReadGroupReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_fingerprint_CrosscheckFingerprints.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_coverage_DepthOfCoverage.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_sv_JointGermlineCNVSegmentation.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SamErrorMetric_BaseErrorMetric.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_PileupSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_DetermineGermlineContigPloidy.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_evidence_FindBreakpointEvidenceSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_UniqueAltReadCount.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ProperlyPairedReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_MappingQuality.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$CigarContainsNoNOperator.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_plotting_PlotDenoisedCopyRatios.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_QualByDepth.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_featuremapping_FlowFeatureMapper.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_flow_FlowBasedTPAttributeSymetricReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_VariantsToTable.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqFilterSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_VariantRecalibrator.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectBaseDistributionByCycleSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_utils_ExtractOriginalAlignmentRecordsByNameSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantrecalling_HaplotypeBasedVariantRecaller.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_reference_NormalizeFasta.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_flow_CycleSkipStatus.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_UnmarkDuplicates.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_flow_HmerIndelNuc.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_RMSMappingQuality.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailSummaryMetric.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_CNNVariantWriteTensors.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_SplitIntervals.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_fasta_FastaAlternateReferenceMaker.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_CountVariantsSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_UpdateVCFSequenceDictionary.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_IlluminaBasecallsToFastq.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SamFormatConverter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_qc_PostProcessReadsForRSEM.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_FisherStrand.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_rnaseq_GeneExpressionEvaluation.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_fingerprint_ClusterCrosscheckMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_ClipReads.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ReadPosRankSumTest.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SamErrorMetric_CollectSamErrorMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_flow_ReadGroupHasFlowOrderReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_SoftClippedReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_GetSampleName.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_sv_CollectSVEvidence.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_MappingQualityRankSumTest.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_sv_SiteDepthtoBAF.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_IntervalOverlapReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_transforms_markduplicates_MarkDuplicatesSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_DepthPerAlleleBySample.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonZeroReferenceLengthAlignmentReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_CalculateReadGroupChecksum.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_groundtruth_GroundTruthReadsBuilder.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_haplotypecaller_RampedHaplotypeCaller.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_HaplotypeFilteringAnnotation.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_BaseQuality.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$SeqIsStoredReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SetNmMdAndUqTags.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_metrics_GcBiasMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CompareMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_funcotator_FuncotateSegments.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_GatherBamFiles.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasSummaryMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_fingerprint_CalculateFingerprintMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_funcotator_FuncotatorDataSourceDownloader.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_ConvertHeaderlessHadoopBamShardToBam.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBuildReferenceTaxonomy.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_FisherStrand.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_validation_CompareBaseQualities.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_replicates_IndependentReplicateMetric.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_bwa_BwaSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadGroupBlackListReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SamComparisonMetric.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_FastqToSam.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectAlignmentSummaryMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_WgsMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_CollectIlluminaBasecallingMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_GcBiasDetailMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SortSam.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_AnnotateVcfWithBamDepth.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SamToFastq.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_artifacts_CollectSequencingArtifactMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqScoreSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_RevertSam.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_GenotypeConcordanceContingencyMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_GatherVcfs.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappedReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_SplitReads.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PassesVendorQualityCheckReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_reference_ExtractSequences.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_markduplicates_MarkDuplicatesWithMateCigar.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectRnaSeqMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_bqsr_GatherBQSRReports.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_contamination_CalculateContamination.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_MergeSamFiles.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_bqsr_ApplyBQSR.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_GatherVcfsCloud.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_discovery_inference_CpxVariantReInterpreterSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_AnnotateIntervals.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBuildKmers.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_MakeSitesOnlyVcf.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_util_IntervalListTools.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_InbreedingCoeff.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_IlluminaPhasingMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectInsertSizeMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ReadLengthEqualsCigarLengthReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_CombineGenotypingArrayVcfs.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_StrandBiasBySample.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_util_BaitDesigner.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_FragmentDepthPerAlleleBySample.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_readorientation_LearnReadOrientationModel.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_ValidateSamFile.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_CollectReadCounts.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_illumina_BpmToNormalizationManifestCsv.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_flow_FlowBasedTPAttributeValidReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_InsertSizeMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_SortSamSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_LeftAlignIndels.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_FilterIntervals.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_MappingQualityRankSumTest.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappingQualityNotZeroReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_SortVcf.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_mutect_GetNormalArtifactData.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_FixMateInformation.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_PrintVariantsSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_HaplotypeCallerSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectJumpingLibraryMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SplitSamByNumberOfReads.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_directed_HsMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_dragstr_ComposeSTRTableFile.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_MethylationTypeCaller.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectOxoGMetrics$CpcgMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateDifferentStrandReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_reference_NonNFastaSize.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_mutect_filtering_FilterMutectCalls.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_fingerprint_FingerprintMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_BaseRecalibratorSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasDetailMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_CalculateMixingFractions.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_SelectVariants.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_AccumulateVariantCallingMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_IlluminaBasecallsToSam.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_sv_SVCluster.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_FlagStat.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_PrintDistantMates.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_FingerprintingSummaryMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_scalable_ScoreVariantAnnotations.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_PreprocessIntervals.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_QualityScoreDistribution.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadGroupReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_MappingQualityReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/gatkDoc.css  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_flow_WellformedFlowBasedReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectMultipleMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_CalculateGenotypePosteriors.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_genomicsdb_GenomicsDBImport.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_MergeVcfs.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectQualityYieldMetricsSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_PlatformUnitReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_CNNScoreVariants.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_RevertOriginalBaseQualitiesAndAddMateCigar.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_CreateBafRegressMetricsFile.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_flow_IndelLength.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_evidence_ExtractSVEvidenceSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_ValidateVariants.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_RawGtCount.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_CollectAllelicCounts.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_CreateHadoopBamSplittingIndex.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_PrintReads.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadTagValueFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingSummaryMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_GenotypeConcordanceSummaryMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_LiftoverVcf.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_flow_HmerQualitySymetricReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_BwaAndMarkDuplicatesPipelineSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_AddOATag.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_MateDistantReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_haplotypecaller_HaplotypeCaller.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_AlignmentSummaryMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_ApplyBQSRSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_CreateVerifyIDIntensityContaminationMetricsFile.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ReadPosition.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_Coverage.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ValidAlignmentStartReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateOnSameContigOrNoMappedMateReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_ReadsPipelineSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadStrandFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_directed_TargetedPcrMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_ReplaceSamHeader.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_CountFalsePositives.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateUnmappedAndUnmappedReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_sv_SVAnnotate.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_directed_TargetMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_AddOrReplaceReadGroups.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_VcfFormatConverter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_flow_VariantType.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_BamIndexStats.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_IlluminaLaneMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_mutect_Mutect2.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_CollectArraysVariantCallingMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_CollectIlluminaLaneMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_fasta_CountBasesInReference.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_fingerprint_ClusteredCrosscheckMetric.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_CombineGVCFs.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_JexlExpressionReadTagValueFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectGcBiasMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_LocalAssembler.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_sv_PrintReadCounts.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_VcfToIntervalList.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SetNmAndUqTags.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBwaSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_TheoreticalSensitivityMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_plotting_PlotModeledSegments.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_MarkIlluminaAdapters.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_CollectVariantCallingMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_SampleList.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_UpdateVcfSequenceDictionary.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MatchingBasesAndQualsReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_discovery_DiscoverVariantsFromContigAlignmentsSAMSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_GermlineCNVCaller.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_MeanQualityByCycleSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_FragmentLength.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_CheckIlluminaDirectory.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadNameReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_directed_CollectTargetedPcrMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectBaseDistributionByCycle.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_ApplyVQSR.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_RrbsSummaryMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_ReblockGVCF.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_RrbsCpgDetailMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_artifacts_ErrorSummaryMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_CallCopyRatioSegments.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_InbreedingCoeff.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_PlatformReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_util_ScatterIntervalsByNs.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectQualityYieldMetrics$QualityYieldMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_scalable_TrainVariantAnnotationsModel.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_MergeBamAlignment.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_scalable_ExtractVariantAnnotations.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_FixMisencodedBaseQualityReads.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_MappingQualityZero.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_LeftAlignAndTrimVariants.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_ModelSegments.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_FeaturizedReadSets.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_bqsr_BaseRecalibrator.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_DuplicationMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_GatherTranches.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_RevertSamSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_GenotypeConcordanceDetailMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_ExtractIlluminaBarcodes.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_RevertBaseQualityScores.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailDetailedMetric.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_DownsampleSam.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ClippingRankSumTest.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_AmbiguousBaseReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_util_BedToIntervalList.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ExcessHet.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_RemoveNearbyIndels.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_filters_VariantFiltration.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_LikelihoodRankSumTest.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_validation_CompareDuplicatesSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectRawWgsMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_PossibleDeNovo.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_flow_IndelClassify.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingDetailMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_qc_TransferReadTags.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectInsertSizeMetricsSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_BaseQualityRankSumTest.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_discovery_SvDiscoverFromLocalAssemblyContigAlignmentsSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_DenoiseReadCounts.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_util_IntervalListToBed.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_flow_HmerIndelLength.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_CompareSAMs.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_CollectVariantCallingMetrics$VariantCallingDetailMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_basicshortmutpileup_ValidateBasicSomaticShortMutations.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_OriginalAlignment.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_BQSRPipelineSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_markduplicates_MarkDuplicates.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_markduplicates_UmiAwareMarkDuplicatesWithMateCigar.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_flow_GcContent.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_CreateSequenceDictionary.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_VcfToAdpc.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$GoodCigarReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_BaseDistributionByCycleMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_realignmentfilter_FilterAlignmentArtifacts.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_markduplicates_SimpleMarkDuplicatesWithMateCigar.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_GenotypeSummaries.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_FlagStatSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectWgsMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_markduplicates_UmiMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_ReorderSam.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectRawWgsMetrics$RawWgsMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_AllelePseudoDepth.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_PostprocessGermlineCNVCalls.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotSupplementaryAlignmentReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_ReadPosRankSumTest.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_util_FifoBuffer.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_PositionBasedDownsampleSam.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_contamination_GetPileupSummaries.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_reference_CheckReferenceCompatibility.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PairedReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterDetailMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_ReorderSam.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectRawWgsMetrics$RawWgsMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_AllelePseudoDepth.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_PostprocessGermlineCNVCalls.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotSupplementaryAlignmentReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_BaseDistributionByCycleMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_GenotypeSummaries.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_FlagStatSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_markduplicates_SimpleMarkDuplicatesWithMateCigar.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_realignmentfilter_FilterAlignmentArtifacts.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectWgsMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_markduplicates_UmiMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_reference_CheckReferenceCompatibility.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PairedReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterDetailMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_ReadPosRankSumTest.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_contamination_GetPileupSummaries.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_util_FifoBuffer.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_PositionBasedDownsampleSam.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_AmbiguousBaseReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_util_BedToIntervalList.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ExcessHet.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_RemoveNearbyIndels.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_filters_VariantFiltration.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectRawWgsMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_validation_CompareDuplicatesSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_LikelihoodRankSumTest.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_bqsr_BaseRecalibrator.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_DuplicationMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_ExtractIlluminaBarcodes.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_RevertBaseQualityScores.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_RevertSamSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_GenotypeConcordanceDetailMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_GatherTranches.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailDetailedMetric.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_DownsampleSam.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ClippingRankSumTest.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_basicshortmutpileup_ValidateBasicSomaticShortMutations.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_CollectVariantCallingMetrics$VariantCallingDetailMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_CompareSAMs.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_BQSRPipelineSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_OriginalAlignment.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_markduplicates_UmiAwareMarkDuplicatesWithMateCigar.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_markduplicates_MarkDuplicates.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_VcfToAdpc.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$GoodCigarReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_flow_GcContent.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_CreateSequenceDictionary.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingDetailMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_qc_TransferReadTags.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_flow_IndelClassify.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_PossibleDeNovo.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectInsertSizeMetricsSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_DenoiseReadCounts.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_BaseQualityRankSumTest.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_discovery_SvDiscoverFromLocalAssemblyContigAlignmentsSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_flow_HmerIndelLength.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_util_IntervalListToBed.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_ApplyVQSR.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_RrbsSummaryMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_ReblockGVCF.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_MeanQualityByCycleSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_discovery_DiscoverVariantsFromContigAlignmentsSAMSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_GermlineCNVCaller.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_UpdateVcfSequenceDictionary.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MatchingBasesAndQualsReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_CheckIlluminaDirectory.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_FragmentLength.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_directed_CollectTargetedPcrMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadNameReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectBaseDistributionByCycle.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_scalable_TrainVariantAnnotationsModel.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_FixMisencodedBaseQualityReads.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_scalable_ExtractVariantAnnotations.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_MergeBamAlignment.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_MappingQualityZero.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_FeaturizedReadSets.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_ModelSegments.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_LeftAlignAndTrimVariants.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_RrbsCpgDetailMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_artifacts_ErrorSummaryMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_CallCopyRatioSegments.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_util_ScatterIntervalsByNs.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_InbreedingCoeff.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_PlatformReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectQualityYieldMetrics$QualityYieldMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_IlluminaLaneMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_BamIndexStats.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_mutect_Mutect2.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_CollectArraysVariantCallingMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_CountFalsePositives.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateUnmappedAndUnmappedReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_directed_TargetMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_sv_SVAnnotate.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_flow_VariantType.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_VcfFormatConverter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_AddOrReplaceReadGroups.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBwaSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SetNmAndUqTags.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_plotting_PlotModeledSegments.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_CollectVariantCallingMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_MarkIlluminaAdapters.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_TheoreticalSensitivityMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_SampleList.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_CombineGVCFs.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_fingerprint_ClusteredCrosscheckMetric.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_fasta_CountBasesInReference.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_CollectIlluminaLaneMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_LocalAssembler.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_sv_PrintReadCounts.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectGcBiasMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_JexlExpressionReadTagValueFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_VcfToIntervalList.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_LiftoverVcf.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_GenotypeConcordanceSummaryMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_CollectArraysVariantCallingMetrics$ArraysVariantCallingSummaryMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_BwaAndMarkDuplicatesPipelineSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_flow_HmerQualitySymetricReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_AddOATag.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_MateDistantReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_CreateHadoopBamSplittingIndex.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_PrintReads.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadTagValueFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ReadPosition.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_ReadsPipelineSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ValidAlignmentStartReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateOnSameContigOrNoMappedMateReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_Coverage.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_ReplaceSamHeader.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadStrandFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_directed_TargetedPcrMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_haplotypecaller_HaplotypeCaller.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_AlignmentSummaryMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_ApplyBQSRSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_CreateVerifyIDIntensityContaminationMetricsFile.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_CalculateGenotypePosteriors.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_genomicsdb_GenomicsDBImport.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_MergeVcfs.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectMultipleMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_flow_WellformedFlowBasedReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_CreateBafRegressMetricsFile.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_flow_IndelLength.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_ValidateVariants.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_RawGtCount.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_evidence_ExtractSVEvidenceSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_CollectAllelicCounts.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_PlatformUnitReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectQualityYieldMetricsSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_CNNScoreVariants.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_RevertOriginalBaseQualitiesAndAddMateCigar.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_SelectVariants.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_IlluminaBasecallsToSam.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_AccumulateVariantCallingMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_dragstr_ComposeSTRTableFile.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectJumpingLibraryMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SplitSamByNumberOfReads.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_directed_HsMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MateDifferentStrandReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_reference_NonNFastaSize.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectOxoGMetrics$CpcgMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_MethylationTypeCaller.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_BaseRecalibratorSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_mutect_filtering_FilterMutectCalls.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_fingerprint_FingerprintMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_CalculateMixingFractions.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasDetailMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_FingerprintingSummaryMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_PreprocessIntervals.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_scalable_ScoreVariantAnnotations.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_MappingQualityReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_QualityScoreDistribution.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadGroupReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_sv_SVCluster.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_PrintDistantMates.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_FlagStat.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_CombineGenotypingArrayVcfs.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_util_BaitDesigner.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_StrandBiasBySample.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_MakeSitesOnlyVcf.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBuildKmers.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_util_IntervalListTools.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_IlluminaPhasingMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_InbreedingCoeff.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ReadLengthEqualsCigarLengthReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectInsertSizeMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_MappingQualityRankSumTest.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_FilterIntervals.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_LeftAlignIndels.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappingQualityNotZeroReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_FixMateInformation.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_mutect_GetNormalArtifactData.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_SortVcf.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_HaplotypeCallerSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_PrintVariantsSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_CollectReadCounts.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_ValidateSamFile.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_readorientation_LearnReadOrientationModel.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_FragmentDepthPerAlleleBySample.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_flow_FlowBasedTPAttributeValidReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_illumina_BpmToNormalizationManifestCsv.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_SortSamSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_InsertSizeMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_replicates_IndependentReplicateMetric.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_FastqToSam.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SamComparisonMetric.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_bwa_BwaSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadGroupBlackListReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_WgsMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectAlignmentSummaryMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SamToFastq.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SortSam.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_AnnotateVcfWithBamDepth.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_GcBiasDetailMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_CollectIlluminaBasecallingMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_GatherBamFiles.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_artifacts_SequencingArtifactMetrics$BaitBiasSummaryMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_funcotator_FuncotatorDataSourceDownloader.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_fingerprint_CalculateFingerprintMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_FisherStrand.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqBuildReferenceTaxonomy.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_ConvertHeaderlessHadoopBamShardToBam.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_validation_CompareBaseQualities.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_markduplicates_MarkDuplicatesWithMateCigar.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_MergeSamFiles.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_bqsr_ApplyBQSR.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_contamination_CalculateContamination.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectRnaSeqMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_bqsr_GatherBQSRReports.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_GatherVcfsCloud.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_AnnotateIntervals.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_discovery_inference_CpxVariantReInterpreterSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_artifacts_CollectSequencingArtifactMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqScoreSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_GenotypeConcordanceContingencyMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_RevertSam.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_GatherVcfs.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_reference_ExtractSequences.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PassesVendorQualityCheckReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_SplitReads.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappedReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_flow_ReadGroupHasFlowOrderReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_SoftClippedReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SamErrorMetric_CollectSamErrorMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_MappingQualityRankSumTest.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_GetSampleName.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_sv_CollectSVEvidence.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_sv_SiteDepthtoBAF.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_qc_PostProcessReadsForRSEM.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SamFormatConverter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_FisherStrand.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_rnaseq_GeneExpressionEvaluation.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_fingerprint_ClusterCrosscheckMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_ClipReads.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ReadPosRankSumTest.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_BaseQuality.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$SeqIsStoredReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SetNmMdAndUqTags.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_metrics_GcBiasMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CompareMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_funcotator_FuncotateSegments.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_DepthPerAlleleBySample.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_transforms_markduplicates_MarkDuplicatesSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_IntervalOverlapReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_CalculateReadGroupChecksum.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonZeroReferenceLengthAlignmentReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_haplotypecaller_RampedHaplotypeCaller.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_groundtruth_GroundTruthReadsBuilder.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_HaplotypeFilteringAnnotation.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectBaseDistributionByCycleSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_VariantRecalibrator.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqFilterSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantrecalling_HaplotypeBasedVariantRecaller.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_utils_ExtractOriginalAlignmentRecordsByNameSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_reference_NormalizeFasta.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_plotting_PlotDenoisedCopyRatios.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_QualByDepth.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_featuremapping_FlowFeatureMapper.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_flow_FlowBasedTPAttributeSymetricReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_VariantsToTable.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_quality_CollectHiSeqXPfFailMetrics$PFFailSummaryMetric.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_SplitIntervals.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_CNNVariantWriteTensors.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_fasta_FastaAlternateReferenceMaker.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_variantutils_UpdateVCFSequenceDictionary.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_IlluminaBasecallsToFastq.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_CountVariantsSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_flow_CycleSkipStatus.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_UnmarkDuplicates.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_flow_HmerIndelNuc.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_RMSMappingQuality.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_CollectMultipleMetricsSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_reference_CompareReferences.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_StructuralVariationDiscoveryPipelineSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_StrandOddsRatio.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_CountBases.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_EvaluateInfoFieldConcordance.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_rnaseq_SplitNCigarReads.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$PrimaryLineReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonZeroFragmentLengthReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_BafRegressMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_sv_JointGermlineCNVSegmentation.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_coverage_DepthOfCoverage.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_DetermineGermlineContigPloidy.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_evidence_FindBreakpointEvidenceSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_UniqueAltReadCount.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_PileupSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SamErrorMetric_BaseErrorMetric.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ProperlyPairedReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$CigarContainsNoNOperator.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_MappingQuality.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_SplitVcfs.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_funcotator_FilterFuncotations.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_MendelianViolations_MendelianViolationMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SplitSamByLibrary.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_fingerprint_CrosscheckFingerprints.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$HasReadGroupReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_fingerprint_CheckFingerprint.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_CNNVariantTrain.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_CollectVariantCallingMetrics$VariantCallingSummaryMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_GenotypeGVCFs.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_CollectArraysVariantCallingMetrics$ArraysControlCodesSummaryMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_TandemRepeat.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_artifacts_SequencingArtifactMetrics$PreAdapterSummaryMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_evidence_FindBadGenomicKmersSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SamErrorMetric_OverlappingErrorMetric.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_fastq_BamToBfq.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_AddCommentsToBam.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_AnalyzeSaturationMutagenesis.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_QualByDepth.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_VariantAnnotator.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_FilterSamReads.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_BaseQualityRankSumTest.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_MergePedIntoVcf.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_Concordance.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_FragmentLengthReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_BuildBamIndex.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_RnaSeqMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectRrbsMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotSecondaryAlignmentReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_MendelianViolations_FindMendelianViolations.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_directed_CollectHsMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_FixVcfHeader.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_CountReads.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectWgsMetricsWithNonZeroCoverage$WgsMetricsWithNonZeroCoverage.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_filter_FilterVcf.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_ParallelCopyGCSDirectoryIntoHDFSSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SamErrorMetric_IndelErrorMetric.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_fasta_ShiftFasta.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_varianteval_VariantEval.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_GcBiasSummaryMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ReferenceBases.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_PrintReadsHeader.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_ViewSam.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_validation_AnnotateVcfWithExpectedAlleleFraction.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_IndexFeatureFile.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$AllowAllReadsReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_CountVariants.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_SamErrorMetric_ErrorMetric.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_util_LiftOverIntervalList.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_MetricsReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotDuplicateReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_util_AccumulateQualityYieldMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_IlluminaBasecallingMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_JumpingLibraryMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_gnarlyGenotyper_GnarlyGenotyper.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_markduplicates_EstimateLibraryComplexity.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$ValidAlignmentEndReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_FingerprintingDetailMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ExcessiveEndClippedReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_BarcodeMetric.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_fasta_FastaReferenceMaker.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_SampleReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_metrics_QualityScoreDistributionSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_DepthPerSampleHC.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_WellformedReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_AlleleFraction.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_artifacts_ConvertSequencingArtifactToOxoG.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$FirstOfPairReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_RMSMappingQuality.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_bqsr_AnalyzeCovariates.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_OrientationBiasReadCounts.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_AssemblyComplexity.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_CountNs.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_RenameSampleInVcf.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_ReadAnonymizer.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$MappingQualityAvailableReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectOxoGMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NotProperlyPairedReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_VerifyIDIntensityContaminationMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_copynumber_CreateReadCountPanelOfNormals.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_sv_evidence_CalcMetadataSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pathseq_PathSeqPipelineSpark.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_markduplicates_CollectDuplicateMetrics.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_AlignmentAgreesWithHeaderReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$SecondOfPairReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_allelespecific_AS_StrandOddsRatio.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_funcotator_Funcotator.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_CountReadsSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_CountBasesSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectQualityYieldMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_MeanQualityByCycle.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_illumina_quality_CollectHiSeqXPfFailMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_vcf_GenotypeConcordance.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_spark_pipelines_PrintReadsSpark.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_replicates_CollectIndependentReplicateMetrics.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_rnaseq_ASEReadCounter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_ChromosomeCounts.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_dragstr_CalibrateDragstrModel.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_DumpTabixIndex.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_vqsr_FilterVariantTranches.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_LibraryReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadFilterLibrary$NonChimericOriginalAlignmentReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_analysis_CollectWgsMetricsWithNonZeroCoverage.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_mutect_CreateSomaticPanelOfNormals.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_qc_CheckPileup.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_arrays_GtcToVcf.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_annotator_flow_HmerMotifs.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_BwaMemIndexImageCreator.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_CheckTerminatorBlock.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_ReadLengthReadFilter.html  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_engine_filters_OverclippedReadFilter.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/org_broadinstitute_hellbender_tools_walkers_qc_Pileup.json  \n",
            "  inflating: gatk-4.3.0.0/gatkdoc/picard_sam_CleanSam.html  \n",
            "  inflating: gatk-4.3.0.0/gatkPythonPackageArchive.zip  \n",
            "  inflating: gatk-4.3.0.0/gatk-completion.sh  \n",
            "  inflating: gatk-4.3.0.0/gatk-package-4.3.0.0-spark.jar  \n",
            "  inflating: gatk-4.3.0.0/gatkcondaenv.yml  \n",
            "  inflating: gatk-4.3.0.0/README.md  \n",
            "  inflating: gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar  \n",
            "   creating: gatk-4.3.0.0/scripts/\n",
            "   creating: gatk-4.3.0.0/scripts/sv/\n",
            "  inflating: gatk-4.3.0.0/scripts/sv/create_cluster.sh  \n",
            "  inflating: gatk-4.3.0.0/scripts/sv/delete_cluster.sh  \n",
            "  inflating: gatk-4.3.0.0/scripts/sv/sanity_checks.sh  \n",
            "  inflating: gatk-4.3.0.0/scripts/sv/manage_sv_pipeline.sh  \n",
            "  inflating: gatk-4.3.0.0/scripts/sv/run_whole_pipeline.sh  \n",
            "  inflating: gatk-4.3.0.0/scripts/sv/copy_sv_results.sh  \n",
            "  inflating: gatk-4.3.0.0/scripts/sv/default_init.sh  \n",
            "   creating: gatk-4.3.0.0/scripts/sv/stepByStep/\n",
            "  inflating: gatk-4.3.0.0/scripts/sv/stepByStep/scanBam.sh  \n",
            "  inflating: gatk-4.3.0.0/scripts/sv/stepByStep/discoverVariants.sh  \n",
            "  inflating: gatk-4.3.0.0/scripts/sv/stepByStep/svDiscover.sh  \n",
            "   creating: gatk-4.3.0.0/scripts/cnv_wdl/\n",
            "  inflating: gatk-4.3.0.0/scripts/cnv_wdl/cnv_common_tasks.wdl  \n",
            "   creating: gatk-4.3.0.0/scripts/cnv_wdl/germline/\n",
            "  inflating: gatk-4.3.0.0/scripts/cnv_wdl/germline/cnv_germline_cohort_workflow.wdl  \n",
            "  inflating: gatk-4.3.0.0/scripts/cnv_wdl/germline/README.md  \n",
            "  inflating: gatk-4.3.0.0/scripts/cnv_wdl/germline/cnv_germline_case_workflow.wdl  \n",
            "  inflating: gatk-4.3.0.0/scripts/cnv_wdl/germline/Structs.wdl  \n",
            "  inflating: gatk-4.3.0.0/scripts/cnv_wdl/germline/cnv_germline_case_scattered_workflow.wdl  \n",
            "  inflating: gatk-4.3.0.0/scripts/cnv_wdl/germline/joint_call_exome_cnvs.wdl  \n",
            "   creating: gatk-4.3.0.0/scripts/cnv_wdl/somatic/\n",
            "  inflating: gatk-4.3.0.0/scripts/cnv_wdl/somatic/cnv_somatic_pair_workflow.wdl  \n",
            "  inflating: gatk-4.3.0.0/scripts/cnv_wdl/somatic/cnv_somatic_oncotator_workflow.wdl  \n",
            "  inflating: gatk-4.3.0.0/scripts/cnv_wdl/somatic/cnv_somatic_panel_workflow.wdl  \n",
            "  inflating: gatk-4.3.0.0/scripts/cnv_wdl/somatic/README.md  \n",
            "  inflating: gatk-4.3.0.0/scripts/cnv_wdl/somatic/cnv_somatic_funcotate_seg_workflow.wdl  \n",
            "  inflating: gatk-4.3.0.0/scripts/dataproc-cluster-ui  \n",
            "   creating: gatk-4.3.0.0/scripts/mutect2_wdl/\n",
            "  inflating: gatk-4.3.0.0/scripts/mutect2_wdl/mutect2.wdl  \n",
            "   creating: gatk-4.3.0.0/scripts/mutect2_wdl/mutect_resources_json/\n",
            "  inflating: gatk-4.3.0.0/scripts/mutect2_wdl/mutect_resources_json/mutect_resources_process_gnomAD_2.1.json  \n",
            "  inflating: gatk-4.3.0.0/scripts/mutect2_wdl/README.md  \n",
            "  inflating: gatk-4.3.0.0/scripts/mutect2_wdl/mutect2_pon.wdl  \n",
            "  inflating: gatk-4.3.0.0/scripts/mutect2_wdl/mutect_resources.wdl  \n",
            "  inflating: gatk-4.3.0.0/gatk       \n",
            "  inflating: gatk-4.3.0.0/GATKConfig.EXAMPLE.properties  \n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "**압축을 푼** 디렉터리 내부에서 실행시켰을 때 다음과 같은 help message가 나오면 설치완료.\n",
        "\n",
        "```\n",
        "USAGE:  <program name> [-h]\n",
        "\n",
        "Available Programs:\n",
        "\n",
        "Base Calling:                                    Tools that process sequencing machine data, e.g. Illumina base calls, and detect sequencing level attributes, e.g. adapters\n",
        "    CheckIlluminaDirectory (Picard)              Asserts the validity for specified Illumina basecalling data.\n",
        "    ....\n",
        "```"
      ],
      "metadata": {
        "id": "dVvtHhZtsKxe"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!cd /content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0\n",
        "cd(\"/content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0\") # 노트북에서만 실행"
      ],
      "metadata": {
        "id": "vtHKUrnlr5tE"
      },
      "execution_count": 39,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!java -jar gatk-package-4.3.0.0-local.jar"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "htI-jSh2qEm0",
        "outputId": "dd8632ad-f8eb-4c98-ca58-939bbef48333"
      },
      "execution_count": 40,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[1m\u001b[31mUSAGE:  \u001b[32m<program name>\u001b[1m\u001b[31m [-h]\n",
            "\n",
            "\u001b[0m\u001b[1m\u001b[31mAvailable Programs:\n",
            "\u001b[0m\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mBase Calling:                                    Tools that process sequencing machine data, e.g. Illumina base calls, and detect sequencing level attributes, e.g. adapters\u001b[0m\n",
            "\u001b[32m    CheckIlluminaDirectory (Picard)              \u001b[36mAsserts the validity for specified Illumina basecalling data.  \u001b[0m\n",
            "\u001b[32m    CollectIlluminaBasecallingMetrics (Picard)   \u001b[36mCollects Illumina Basecalling metrics for a sequencing run.  \u001b[0m\n",
            "\u001b[32m    CollectIlluminaLaneMetrics (Picard)          \u001b[36mCollects Illumina lane metrics for the given BaseCalling analysis directory.\u001b[0m\n",
            "\u001b[32m    ExtractIlluminaBarcodes (Picard)             \u001b[36mTool determines the barcode for each read in an Illumina lane.  \u001b[0m\n",
            "\u001b[32m    IlluminaBasecallsToFastq (Picard)            \u001b[36mGenerate FASTQ file(s) from Illumina basecall read data.  \u001b[0m\n",
            "\u001b[32m    IlluminaBasecallsToSam (Picard)              \u001b[36mTransforms raw Illumina sequencing data into an unmapped SAM, BAM or CRAM file.\u001b[0m\n",
            "\u001b[32m    MarkIlluminaAdapters (Picard)                \u001b[36mReads a SAM/BAM/CRAM file and rewrites it with new adapter-trimming tags.  \u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mCopy Number Variant Discovery:                   Tools that analyze read coverage to detect copy number variants.\u001b[0m\n",
            "\u001b[32m    AnnotateIntervals                            \u001b[36mAnnotates intervals with GC content, mappability, and segmental-duplication content\u001b[0m\n",
            "\u001b[32m    CallCopyRatioSegments                        \u001b[36mCalls copy-ratio segments as amplified, deleted, or copy-number neutral\u001b[0m\n",
            "\u001b[32m    CombineSegmentBreakpoints                    \u001b[31m(EXPERIMENTAL Tool) \u001b[36mCombine the breakpoints of two segment files and annotate the resulting intervals with chosen columns from each file.\u001b[0m\n",
            "\u001b[32m    CreateReadCountPanelOfNormals                \u001b[36mCreates a panel of normals for read-count denoising\u001b[0m\n",
            "\u001b[32m    DenoiseReadCounts                            \u001b[36mDenoises read counts to produce denoised copy ratios\u001b[0m\n",
            "\u001b[32m    DetermineGermlineContigPloidy                \u001b[36mDetermines the baseline contig ploidy for germline samples given counts data\u001b[0m\n",
            "\u001b[32m    FilterIntervals                              \u001b[36mFilters intervals based on annotations and/or count statistics\u001b[0m\n",
            "\u001b[32m    GermlineCNVCaller                            \u001b[36mCalls copy-number variants in germline samples given their counts and the output of DetermineGermlineContigPloidy\u001b[0m\n",
            "\u001b[32m    MergeAnnotatedRegions                        \u001b[31m(EXPERIMENTAL Tool) \u001b[36mMerge annotated genomic regions based entirely on touching/overlapping intervals.\u001b[0m\n",
            "\u001b[32m    MergeAnnotatedRegionsByAnnotation            \u001b[31m(EXPERIMENTAL Tool) \u001b[36mMerge annotated genomic regions within specified distance if annotation value(s) are exactly the same.\u001b[0m\n",
            "\u001b[32m    ModelSegments                                \u001b[36mModels segmented copy ratios from denoised copy ratios and segmented minor-allele fractions from allelic counts\u001b[0m\n",
            "\u001b[32m    PlotDenoisedCopyRatios                       \u001b[36mCreates plots of denoised copy ratios\u001b[0m\n",
            "\u001b[32m    PlotModeledSegments                          \u001b[36mCreates plots of denoised and segmented copy-ratio and minor-allele-fraction estimates\u001b[0m\n",
            "\u001b[32m    PostprocessGermlineCNVCalls                  \u001b[36mPostprocesses the output of GermlineCNVCaller and generates VCFs and denoised copy ratios\u001b[0m\n",
            "\u001b[32m    TagGermlineEvents                            \u001b[31m(EXPERIMENTAL Tool) \u001b[36mDo a simplistic tagging of germline events in a tumor segment file.\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mCoverage Analysis:                               Tools that count coverage, e.g. depth per allele\u001b[0m\n",
            "\u001b[32m    ASEReadCounter                               \u001b[36mGenerates table of filtered base counts at het sites for allele specific expression\u001b[0m\n",
            "\u001b[32m    AnalyzeSaturationMutagenesis                 \u001b[31m(BETA Tool) \u001b[36m(EXPERIMENTAL) Processes reads from a MITESeq or other saturation mutagenesis experiment.\u001b[0m\n",
            "\u001b[32m    CollectAllelicCounts                         \u001b[36mCollects reference and alternate allele counts at specified sites\u001b[0m\n",
            "\u001b[32m    CollectAllelicCountsSpark                    \u001b[36mCollects reference and alternate allele counts at specified sites\u001b[0m\n",
            "\u001b[32m    CollectF1R2Counts                            \u001b[36mCollect F1R2 read counts for the Mutect2 orientation bias mixture model filter\u001b[0m\n",
            "\u001b[32m    CollectReadCounts                            \u001b[36mCollects read counts at specified intervals\u001b[0m\n",
            "\u001b[32m    CountBases                                   \u001b[36mCount bases in a SAM/BAM/CRAM file\u001b[0m\n",
            "\u001b[32m    CountBasesSpark                              \u001b[36mCounts bases in the input SAM/BAM\u001b[0m\n",
            "\u001b[32m    CountReads                                   \u001b[36mCount reads in a SAM/BAM/CRAM file\u001b[0m\n",
            "\u001b[32m    CountReadsSpark                              \u001b[36mCounts reads in the input SAM/BAM\u001b[0m\n",
            "\u001b[32m    DepthOfCoverage                              \u001b[31m(BETA Tool) \u001b[36mGenerate coverage summary information for reads data\u001b[0m\n",
            "\u001b[32m    GatherNormalArtifactData                     \u001b[36mCombine output files from GetNormalArtifactData in the order defined by a sequence dictionary\u001b[0m\n",
            "\u001b[32m    GeneExpressionEvaluation                     \u001b[31m(BETA Tool) \u001b[36mEvaluate gene expression from RNA-seq reads aligned to genome.\u001b[0m\n",
            "\u001b[32m    GetNormalArtifactData                        \u001b[36mCollects data for training normal artifact filter\u001b[0m\n",
            "\u001b[32m    GetPileupSummaries                           \u001b[36mTabulates pileup metrics for inferring contamination\u001b[0m\n",
            "\u001b[32m    LocalAssembler                               \u001b[31m(BETA Tool) \u001b[36mLocal assembler for SVs\u001b[0m\n",
            "\u001b[32m    Pileup                                       \u001b[36mPrints read alignments in samtools pileup format\u001b[0m\n",
            "\u001b[32m    PileupSpark                                  \u001b[31m(BETA Tool) \u001b[36mPrints read alignments in samtools pileup format\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mDiagnostics and Quality Control:                 Tools that collect sequencing quality related and comparative metrics\u001b[0m\n",
            "\u001b[32m    AccumulateQualityYieldMetrics (Picard)       \u001b[36mCombines multiple QualityYieldMetrics files into a single file.\u001b[0m\n",
            "\u001b[32m    AccumulateVariantCallingMetrics (Picard)     \u001b[36mCombines multiple Variant Calling Metrics files into a single file\u001b[0m\n",
            "\u001b[32m    AnalyzeCovariates                            \u001b[36mEvaluate and compare base quality score recalibration (BQSR) tables\u001b[0m\n",
            "\u001b[32m    BamIndexStats (Picard)                       \u001b[36mGenerate index statistics from a BAM file\u001b[0m\n",
            "\u001b[32m    CalcMetadataSpark                            \u001b[31m(BETA Tool) \u001b[36m(Internal) Collects read metrics relevant to structural variant discovery\u001b[0m\n",
            "\u001b[32m    CalculateContamination                       \u001b[36mCalculate the fraction of reads coming from cross-sample contamination\u001b[0m\n",
            "\u001b[32m    CalculateFingerprintMetrics (Picard)         \u001b[36mCalculate statistics on fingerprints, checking their viability\u001b[0m\n",
            "\u001b[32m    CalculateReadGroupChecksum (Picard)          \u001b[36mCreates a hash code based on the read groups (RG).  \u001b[0m\n",
            "\u001b[32m    CheckDuplicateMarking (Picard)               \u001b[36mChecks the consistency of duplicate markings.\u001b[0m\n",
            "\u001b[32m    CheckFingerprint (Picard)                    \u001b[36mComputes a fingerprint from the supplied input (SAM/BAM/CRAM or VCF) file and compares it to the provided genotypes\u001b[0m\n",
            "\u001b[32m    CheckPileup                                  \u001b[36mCompare GATK's internal pileup to a reference Samtools mpileup\u001b[0m\n",
            "\u001b[32m    CheckTerminatorBlock (Picard)                \u001b[36mAsserts the provided gzip file's (e.g., BAM) last block is well-formed; RC 100 otherwise\u001b[0m\n",
            "\u001b[32m    ClusterCrosscheckMetrics (Picard)            \u001b[36mClusters the results of a CrosscheckFingerprints run by LOD score\u001b[0m\n",
            "\u001b[32m    CollectAlignmentSummaryMetrics (Picard)      \u001b[36m<b>Produces a summary of alignment metrics from a SAM or BAM file.</b>  \u001b[0m\n",
            "\u001b[32m    CollectArraysVariantCallingMetrics (Picard)  \u001b[36mCollects summary and per-sample from the provided arrays VCF file\u001b[0m\n",
            "\u001b[32m    CollectBaseDistributionByCycle (Picard)      \u001b[36mChart the nucleotide distribution per cycle in a SAM or BAM file\u001b[0m\n",
            "\u001b[32m    CollectBaseDistributionByCycleSpark          \u001b[31m(BETA Tool) \u001b[36mCollects base distribution per cycle in SAM/BAM/CRAM file(s).\u001b[0m\n",
            "\u001b[32m    CollectGcBiasMetrics (Picard)                \u001b[36mCollect metrics regarding GC bias. \u001b[0m\n",
            "\u001b[32m    CollectHiSeqXPfFailMetrics (Picard)          \u001b[36mClassify PF-Failing reads in a HiSeqX Illumina Basecalling directory into various categories.\u001b[0m\n",
            "\u001b[32m    CollectHsMetrics (Picard)                    \u001b[36mCollects hybrid-selection (HS) metrics for a SAM or BAM file.  \u001b[0m\n",
            "\u001b[32m    CollectIndependentReplicateMetrics (Picard)  \u001b[31m(EXPERIMENTAL Tool) \u001b[36mEstimates the rate of independent replication rate of reads within a bam. \n",
            "\u001b[0m\n",
            "\u001b[32m    CollectInsertSizeMetrics (Picard)            \u001b[36mCollect metrics about the insert size distribution of a paired-end library. \u001b[0m\n",
            "\u001b[32m    CollectInsertSizeMetricsSpark                \u001b[31m(BETA Tool) \u001b[36mCollects insert size distribution information on alignment data\u001b[0m\n",
            "\u001b[32m    CollectJumpingLibraryMetrics (Picard)        \u001b[36mCollect jumping library metrics. \u001b[0m\n",
            "\u001b[32m    CollectMultipleMetrics (Picard)              \u001b[36mCollect multiple classes of metrics. \u001b[0m\n",
            "\u001b[32m    CollectMultipleMetricsSpark                  \u001b[31m(BETA Tool) \u001b[36mRuns multiple metrics collection modules for a given alignment file\u001b[0m\n",
            "\u001b[32m    CollectOxoGMetrics (Picard)                  \u001b[36mCollect metrics to assess oxidative artifacts.\u001b[0m\n",
            "\u001b[32m    CollectQualityYieldMetrics (Picard)          \u001b[36mCollect metrics about reads that pass quality thresholds and Illumina-specific filters.  \u001b[0m\n",
            "\u001b[32m    CollectQualityYieldMetricsSpark              \u001b[31m(BETA Tool) \u001b[36mCollects quality yield metrics from SAM/BAM/CRAM file(s).\u001b[0m\n",
            "\u001b[32m    CollectRawWgsMetrics (Picard)                \u001b[36mCollect whole genome sequencing-related metrics.  \u001b[0m\n",
            "\u001b[32m    CollectRnaSeqMetrics (Picard)                \u001b[36mProduces RNA alignment metrics for a SAM or BAM file.  \u001b[0m\n",
            "\u001b[32m    CollectRrbsMetrics (Picard)                  \u001b[36m<b>Collects metrics from reduced representation bisulfite sequencing (Rrbs) data.</b>  \u001b[0m\n",
            "\u001b[32m    CollectSamErrorMetrics (Picard)              \u001b[36mProgram to collect error metrics on bases stratified in various ways.\u001b[0m\n",
            "\u001b[32m    CollectSequencingArtifactMetrics (Picard)    \u001b[36mCollect metrics to quantify single-base sequencing artifacts.  \u001b[0m\n",
            "\u001b[32m    CollectTargetedPcrMetrics (Picard)           \u001b[36mCalculate PCR-related metrics from targeted sequencing data. \u001b[0m\n",
            "\u001b[32m    CollectVariantCallingMetrics (Picard)        \u001b[36mCollects per-sample and aggregate (spanning all samples) metrics from the provided VCF file\u001b[0m\n",
            "\u001b[32m    CollectWgsMetrics (Picard)                   \u001b[36mCollect metrics about coverage and performance of whole genome sequencing (WGS) experiments.\u001b[0m\n",
            "\u001b[32m    CollectWgsMetricsWithNonZeroCoverage (Picard)\u001b[31m(EXPERIMENTAL Tool) \u001b[36mCollect metrics about coverage and performance of whole genome sequencing (WGS) experiments.  \u001b[0m\n",
            "\u001b[32m    CompareBaseQualities                         \u001b[36mCompares the base qualities of two SAM/BAM/CRAM files\u001b[0m\n",
            "\u001b[32m    CompareDuplicatesSpark                       \u001b[31m(BETA Tool) \u001b[36mDetermine if two potentially identical BAMs have the same duplicate reads\u001b[0m\n",
            "\u001b[32m    CompareMetrics (Picard)                      \u001b[36mCompare two metrics files.\u001b[0m\n",
            "\u001b[32m    CompareSAMs (Picard)                         \u001b[36mCompare two input SAM/BAM/CRAM files.  \u001b[0m\n",
            "\u001b[32m    ConvertHaplotypeDatabaseToVcf (Picard)       \u001b[36mConvert Haplotype database file to vcf\u001b[0m\n",
            "\u001b[32m    ConvertSequencingArtifactToOxoG (Picard)     \u001b[36mExtract OxoG metrics from generalized artifacts metrics.  \u001b[0m\n",
            "\u001b[32m    CrosscheckFingerprints (Picard)              \u001b[36mChecks that all data in the input files appear to have come from the same individual\u001b[0m\n",
            "\u001b[32m    CrosscheckReadGroupFingerprints (Picard)     \u001b[36mDEPRECATED: USE CrosscheckFingerprints. \u001b[0m\n",
            "\u001b[32m    DumpTabixIndex                               \u001b[36mDumps a tabix index file.\u001b[0m\n",
            "\u001b[32m    EstimateLibraryComplexity (Picard)           \u001b[36mEstimates the numbers of unique molecules in a sequencing library.  \u001b[0m\n",
            "\u001b[32m    ExtractFingerprint (Picard)                  \u001b[36mComputes a fingerprint from the input file.\u001b[0m\n",
            "\u001b[32m    FlagStat                                     \u001b[36mAccumulate flag statistics given a BAM file\u001b[0m\n",
            "\u001b[32m    FlagStatSpark                                \u001b[36mSpark tool to accumulate flag statistics\u001b[0m\n",
            "\u001b[32m    GatherPileupSummaries                        \u001b[36mCombine output files from GetPileupSummary in the order defined by a sequence dictionary\u001b[0m\n",
            "\u001b[32m    GetSampleName                                \u001b[36mEmit a single sample name\u001b[0m\n",
            "\u001b[32m    IdentifyContaminant (Picard)                 \u001b[36mComputes a fingerprint from the supplied SAM/BAM file, given a contamination estimate.\u001b[0m\n",
            "\u001b[32m    LiftOverHaplotypeMap (Picard)                \u001b[36mLifts over a haplotype database from one reference to another\u001b[0m\n",
            "\u001b[32m    MeanQualityByCycle (Picard)                  \u001b[36mCollect mean quality by cycle.\u001b[0m\n",
            "\u001b[32m    MeanQualityByCycleSpark                      \u001b[31m(BETA Tool) \u001b[36mMeanQualityByCycle on Spark\u001b[0m\n",
            "\u001b[32m    QualityScoreDistribution (Picard)            \u001b[36mChart the distribution of quality scores.  \u001b[0m\n",
            "\u001b[32m    QualityScoreDistributionSpark                \u001b[31m(BETA Tool) \u001b[36mQualityScoreDistribution on Spark\u001b[0m\n",
            "\u001b[32m    ValidateSamFile (Picard)                     \u001b[36mValidates a SAM/BAM/CRAM file.\u001b[0m\n",
            "\u001b[32m    ViewSam (Picard)                             \u001b[36mPrints a SAM or BAM file to the screen\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mExample Tools:                                   Example tools that show developers how to implement new tools\u001b[0m\n",
            "\u001b[32m    ExampleMultiFeatureWalker                    \u001b[36mExample of a MultiFeatureWalker subclass.\u001b[0m\n",
            "\u001b[32m    HtsgetReader                                 \u001b[31m(EXPERIMENTAL Tool) \u001b[36mDownload a file using htsget\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mFlow Based Tools:                                Tools designed specifically to operate on flow based data\u001b[0m\n",
            "\u001b[32m    CalculateAverageCombinedAnnotations          \u001b[31m(EXPERIMENTAL Tool) \u001b[36mDivides annotations that were summed by genomicsDB by number of samples to calculate average.\u001b[0m\n",
            "\u001b[32m    FlowFeatureMapper                            \u001b[31m(EXPERIMENTAL Tool) \u001b[36mMap/find features in BAM file, output VCF. Initially mapping SNVs\u001b[0m\n",
            "\u001b[32m    GroundTruthReadsBuilder                      \u001b[31m(EXPERIMENTAL Tool) \u001b[36mProduces a flexible and robust ground truth set for base calling training\u001b[0m\n",
            "\u001b[32m    SplitCRAM                                    \u001b[31m(EXPERIMENTAL Tool) \u001b[36mSplit CRAM files to smaller files efficiently\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mGenotyping Arrays Manipulation:                  Tools that manipulate data generated by Genotyping arrays\u001b[0m\n",
            "\u001b[32m    BpmToNormalizationManifestCsv (Picard)       \u001b[36mProgram to convert an Illumina bpm file into a bpm.csv file.\u001b[0m\n",
            "\u001b[32m    CombineGenotypingArrayVcfs (Picard)          \u001b[36mProgram to combine multiple genotyping array VCF files into one VCF.\u001b[0m\n",
            "\u001b[32m    CompareGtcFiles (Picard)                     \u001b[36mCompares two GTC files.\u001b[0m\n",
            "\u001b[32m    CreateBafRegressMetricsFile (Picard)         \u001b[36mProgram to generate a picard metrics file from the output of the bafRegress tool.\u001b[0m\n",
            "\u001b[32m    CreateExtendedIlluminaManifest (Picard)      \u001b[36mCreate an Extended Illumina Manifest for usage by the Picard tool GtcToVcf\u001b[0m\n",
            "\u001b[32m    CreateVerifyIDIntensityContaminationMetricsFile (Picard)    \u001b[36mProgram to generate a picard metrics file from the output of the VerifyIDIntensity tool.\u001b[0m\n",
            "\u001b[32m    GtcToVcf (Picard)                            \u001b[36mProgram to convert an Illumina GTC file to a VCF\u001b[0m\n",
            "\u001b[32m    MergePedIntoVcf (Picard)                     \u001b[36mProgram to merge a single-sample ped file from zCall into a single-sample VCF.\u001b[0m\n",
            "\u001b[32m    VcfToAdpc (Picard)                           \u001b[36mProgram to convert an Arrays VCF to an ADPC file.\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mIntervals Manipulation:                          Tools that process genomic intervals in various formats\u001b[0m\n",
            "\u001b[32m    BedToIntervalList (Picard)                   \u001b[36mConverts a BED file to a Picard Interval List.  \u001b[0m\n",
            "\u001b[32m    CompareIntervalLists                         \u001b[36mCompare two interval lists for equality\u001b[0m\n",
            "\u001b[32m    IntervalListToBed (Picard)                   \u001b[36mConverts an Picard IntervalList file to a BED file.\u001b[0m\n",
            "\u001b[32m    IntervalListTools (Picard)                   \u001b[36mA tool for performing various IntervalList manipulations\u001b[0m\n",
            "\u001b[32m    LiftOverIntervalList (Picard)                \u001b[36mLifts over an interval list from one reference build to another. \u001b[0m\n",
            "\u001b[32m    PreprocessIntervals                          \u001b[36mPrepares bins for coverage collection\u001b[0m\n",
            "\u001b[32m    SplitIntervals                               \u001b[36mSplit intervals into sub-interval files.\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mMetagenomics:                                    Tools that perform metagenomic analysis, e.g. microbial community composition and pathogen detection\u001b[0m\n",
            "\u001b[32m    PathSeqBuildKmers                            \u001b[36mBuilds set of host reference k-mers\u001b[0m\n",
            "\u001b[32m    PathSeqBuildReferenceTaxonomy                \u001b[36mBuilds a taxonomy datafile of the microbe reference\u001b[0m\n",
            "\u001b[32m    PathSeqBwaSpark                              \u001b[36mStep 2: Aligns reads to the microbe reference\u001b[0m\n",
            "\u001b[32m    PathSeqFilterSpark                           \u001b[36mStep 1: Filters low quality, low complexity, duplicate, and host reads\u001b[0m\n",
            "\u001b[32m    PathSeqPipelineSpark                         \u001b[36mCombined tool that performs all steps: read filtering, microbe reference alignment, and abundance scoring\u001b[0m\n",
            "\u001b[32m    PathSeqScoreSpark                            \u001b[36mStep 3: Classifies pathogen-aligned reads and generates abundance scores\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mMethylation-Specific Tools:                      Tools that perform methylation calling, processing bisulfite sequenced, methylation-aware aligned BAM\u001b[0m\n",
            "\u001b[32m    MethylationTypeCaller                        \u001b[31m(EXPERIMENTAL Tool) \u001b[36mIdentify methylated bases from bisulfite sequenced, methylation-aware BAMs\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mOther:                                           Miscellaneous tools, e.g. those that aid in data streaming\u001b[0m\n",
            "\u001b[32m    CreateHadoopBamSplittingIndex                \u001b[31m(BETA Tool) \u001b[36mCreate a Hadoop BAM splitting index\u001b[0m\n",
            "\u001b[32m    FifoBuffer (Picard)                          \u001b[36mProvides a large, FIFO buffer that can be used to buffer input and output streams between programs.\u001b[0m\n",
            "\u001b[32m    GatherBQSRReports                            \u001b[36mGathers scattered BQSR recalibration reports into a single file\u001b[0m\n",
            "\u001b[32m    GatherTranches                               \u001b[31m(BETA Tool) \u001b[36mGathers scattered VQSLOD tranches into a single file\u001b[0m\n",
            "\u001b[32m    IndexFeatureFile                             \u001b[36mCreates an index for a feature file, e.g. VCF or BED file.\u001b[0m\n",
            "\u001b[32m    ParallelCopyGCSDirectoryIntoHDFSSpark        \u001b[31m(BETA Tool) \u001b[36mParallel copy a file or directory from Google Cloud Storage into the HDFS file system used by Spark\u001b[0m\n",
            "\u001b[32m    PrintBGZFBlockInformation                    \u001b[31m(EXPERIMENTAL Tool) \u001b[36mPrint information about the compressed blocks in a BGZF format file\u001b[0m\n",
            "\u001b[32m    ReadAnonymizer                               \u001b[31m(EXPERIMENTAL Tool) \u001b[36mReplace bases in reads with reference bases.\u001b[0m\n",
            "\u001b[32m    ReblockGVCF                                  \u001b[36mCondenses homRef blocks in a single-sample GVCF\u001b[0m\n",
            "\u001b[32m    SortGff (Picard)                             \u001b[36mSorts a gff3 file, and adds flush directives\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mRead Data Manipulation:                          Tools that manipulate read data in SAM, BAM or CRAM format\u001b[0m\n",
            "\u001b[32m    AddCommentsToBam (Picard)                    \u001b[36mAdds comments to the header of a BAM file.\u001b[0m\n",
            "\u001b[32m    AddOATag (Picard)                            \u001b[36mRecord current alignment information to OA tag.\u001b[0m\n",
            "\u001b[32m    AddOrReplaceReadGroups (Picard)              \u001b[36mAssigns all the reads in a file to a single new read-group.\u001b[0m\n",
            "\u001b[32m    AddOriginalAlignmentTags                     \u001b[31m(EXPERIMENTAL Tool) \u001b[36mAdds Original Alignment tag and original mate contig tag\u001b[0m\n",
            "\u001b[32m    ApplyBQSR                                    \u001b[36mApply base quality score recalibration\u001b[0m\n",
            "\u001b[32m    ApplyBQSRSpark                               \u001b[31m(BETA Tool) \u001b[36mApply base quality score recalibration on Spark\u001b[0m\n",
            "\u001b[32m    BQSRPipelineSpark                            \u001b[31m(BETA Tool) \u001b[36mBoth steps of BQSR (BaseRecalibrator and ApplyBQSR) on Spark\u001b[0m\n",
            "\u001b[32m    BamToBfq (Picard)                            \u001b[36mConverts a BAM file into a BFQ (binary fastq formatted) file\u001b[0m\n",
            "\u001b[32m    BaseRecalibrator                             \u001b[36mGenerates recalibration table for Base Quality Score Recalibration (BQSR)\u001b[0m\n",
            "\u001b[32m    BaseRecalibratorSpark                        \u001b[31m(BETA Tool) \u001b[36mGenerate recalibration table for Base Quality Score Recalibration (BQSR) on Spark\u001b[0m\n",
            "\u001b[32m    BuildBamIndex (Picard)                       \u001b[36mGenerates a BAM index \".bai\" file.  \u001b[0m\n",
            "\u001b[32m    BwaAndMarkDuplicatesPipelineSpark            \u001b[31m(BETA Tool) \u001b[36mTakes name-sorted file and runs BWA and MarkDuplicates.\u001b[0m\n",
            "\u001b[32m    BwaSpark                                     \u001b[31m(BETA Tool) \u001b[36mAlign reads to a given reference using BWA on Spark\u001b[0m\n",
            "\u001b[32m    CleanSam (Picard)                            \u001b[36mCleans a SAM/BAM/CRAM files, soft-clipping beyond-end-of-reference alignments and setting MAPQ to 0 for unmapped reads\u001b[0m\n",
            "\u001b[32m    ClipReads                                    \u001b[36mClip reads in a SAM/BAM/CRAM file\u001b[0m\n",
            "\u001b[32m    CollectDuplicateMetrics (Picard)             \u001b[36mCollect Duplicate metrics from marked file.\u001b[0m\n",
            "\u001b[32m    ConvertHeaderlessHadoopBamShardToBam         \u001b[31m(BETA Tool) \u001b[36mConvert a headerless BAM shard into a readable BAM\u001b[0m\n",
            "\u001b[32m    DownsampleByDuplicateSet                     \u001b[31m(BETA Tool) \u001b[36mDiscard a set fraction of duplicate sets from a UMI-grouped bam\u001b[0m\n",
            "\u001b[32m    DownsampleSam (Picard)                       \u001b[36mDownsample a SAM or BAM file.\u001b[0m\n",
            "\u001b[32m    ExtractOriginalAlignmentRecordsByNameSpark   \u001b[31m(BETA Tool) \u001b[36mSubsets reads by name\u001b[0m\n",
            "\u001b[32m    FastqToSam (Picard)                          \u001b[36mConverts a FASTQ file to an unaligned BAM or SAM file\u001b[0m\n",
            "\u001b[32m    FilterSamReads (Picard)                      \u001b[36mSubsets reads from a SAM/BAM/CRAM file by applying one of several filters.\u001b[0m\n",
            "\u001b[32m    FixMateInformation (Picard)                  \u001b[36mVerify mate-pair information between mates and fix if needed.\u001b[0m\n",
            "\u001b[32m    FixMisencodedBaseQualityReads                \u001b[36mFix Illumina base quality scores in a SAM/BAM/CRAM file\u001b[0m\n",
            "\u001b[32m    GatherBamFiles (Picard)                      \u001b[36mConcatenate efficiently BAM files that resulted from a scattered parallel analysis\u001b[0m\n",
            "\u001b[32m    LeftAlignIndels                              \u001b[36mLeft-aligns indels from reads in a SAM/BAM/CRAM file\u001b[0m\n",
            "\u001b[32m    MarkDuplicates (Picard)                      \u001b[36mIdentifies duplicate reads.  \u001b[0m\n",
            "\u001b[32m    MarkDuplicatesSpark                          \u001b[36mMarkDuplicates on Spark\u001b[0m\n",
            "\u001b[32m    MarkDuplicatesWithMateCigar (Picard)         \u001b[36mIdentifies duplicate reads, accounting for mate CIGAR.  \u001b[0m\n",
            "\u001b[32m    MergeBamAlignment (Picard)                   \u001b[36mMerge alignment data from a SAM or BAM with data in an unmapped BAM file.  \u001b[0m\n",
            "\u001b[32m    MergeSamFiles (Picard)                       \u001b[36mMerges multiple SAM/BAM/CRAM (and/or) files into a single file.  \u001b[0m\n",
            "\u001b[32m    PositionBasedDownsampleSam (Picard)          \u001b[36mDownsample a SAM or BAM file to retain a subset of the reads based on the reads location in each tile in the flowcell.\u001b[0m\n",
            "\u001b[32m    PostProcessReadsForRSEM                      \u001b[31m(BETA Tool) \u001b[36mReorder reads before running RSEM\u001b[0m\n",
            "\u001b[32m    PrintDistantMates                            \u001b[36mUnmaps reads with distant mates.\u001b[0m\n",
            "\u001b[32m    PrintReads                                   \u001b[36mPrint reads in the SAM/BAM/CRAM file\u001b[0m\n",
            "\u001b[32m    PrintReadsHeader                             \u001b[36mPrint the header from a SAM/BAM/CRAM file\u001b[0m\n",
            "\u001b[32m    PrintReadsSpark                              \u001b[36mPrintReads on Spark\u001b[0m\n",
            "\u001b[32m    ReorderSam (Picard)                          \u001b[36mReorders reads in a SAM or BAM file to match ordering in a second reference file.\u001b[0m\n",
            "\u001b[32m    ReplaceSamHeader (Picard)                    \u001b[36mReplaces the SAMFileHeader in a SAM/BAM/CRAM file.  \u001b[0m\n",
            "\u001b[32m    RevertBaseQualityScores                      \u001b[36mRevert Quality Scores in a SAM/BAM/CRAM file\u001b[0m\n",
            "\u001b[32m    RevertOriginalBaseQualitiesAndAddMateCigar (Picard)\u001b[36mReverts the original base qualities and adds the mate cigar tag to read-group files\u001b[0m\n",
            "\u001b[32m    RevertSam (Picard)                           \u001b[36mReverts SAM/BAM/CRAM files to a previous state.  \u001b[0m\n",
            "\u001b[32m    RevertSamSpark                               \u001b[31m(BETA Tool) \u001b[36mReverts SAM, BAM or CRAM files to a previous state.\u001b[0m\n",
            "\u001b[32m    SamFormatConverter (Picard)                  \u001b[36mConvert a BAM file to a SAM file, or a SAM to a BAM\u001b[0m\n",
            "\u001b[32m    SamToFastq (Picard)                          \u001b[36mConverts a SAM/BAM/CRAM file to FASTQ.\u001b[0m\n",
            "\u001b[32m    SamToFastqWithTags (Picard)                  \u001b[36mConverts a SAM or BAM file to FASTQ alongside FASTQs created from tags.\u001b[0m\n",
            "\u001b[32m    SetNmAndUqTags (Picard)                      \u001b[36mDEPRECATED: Use SetNmMdAndUqTags instead.\u001b[0m\n",
            "\u001b[32m    SetNmMdAndUqTags (Picard)                    \u001b[36mFixes the NM, MD, and UQ tags in a SAM/BAM/CRAM file \u001b[0m\n",
            "\u001b[32m    SimpleMarkDuplicatesWithMateCigar (Picard)   \u001b[31m(EXPERIMENTAL Tool) \u001b[36mExamines aligned records in the supplied SAM or BAM file to locate duplicate molecules.\u001b[0m\n",
            "\u001b[32m    SortSam (Picard)                             \u001b[36mSorts a SAM, BAM or CRAM file.  \u001b[0m\n",
            "\u001b[32m    SortSamSpark                                 \u001b[31m(BETA Tool) \u001b[36mSortSam on Spark (works on SAM/BAM/CRAM)\u001b[0m\n",
            "\u001b[32m    SplitNCigarReads                             \u001b[36mSplit Reads with N in Cigar\u001b[0m\n",
            "\u001b[32m    SplitReads                                   \u001b[36mOutputs reads from a SAM/BAM/CRAM by read group, sample and library name\u001b[0m\n",
            "\u001b[32m    SplitSamByLibrary (Picard)                   \u001b[36mSplits a SAM/BAM/CRAM file into individual files by library\u001b[0m\n",
            "\u001b[32m    SplitSamByNumberOfReads (Picard)             \u001b[36mSplits a SAM/BAM/CRAM file to multiple files.\u001b[0m\n",
            "\u001b[32m    TransferReadTags                             \u001b[31m(EXPERIMENTAL Tool) \u001b[36mIncorporate read tags in a SAM file to that of a matching SAM file\u001b[0m\n",
            "\u001b[32m    UmiAwareMarkDuplicatesWithMateCigar (Picard) \u001b[31m(EXPERIMENTAL Tool) \u001b[36mIdentifies duplicate reads using information from read positions and UMIs. \u001b[0m\n",
            "\u001b[32m    UnmarkDuplicates                             \u001b[36mClears the 0x400 duplicate SAM flag\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mReference:                                       Tools that analyze and manipulate FASTA format references\u001b[0m\n",
            "\u001b[32m    BaitDesigner (Picard)                        \u001b[36mDesigns oligonucleotide baits for hybrid selection reactions.\u001b[0m\n",
            "\u001b[32m    BwaMemIndexImageCreator                      \u001b[36mCreate a BWA-MEM index image file for use with GATK BWA tools\u001b[0m\n",
            "\u001b[32m    CheckReferenceCompatibility                  \u001b[31m(EXPERIMENTAL Tool) \u001b[36mCheck a BAM/VCF for compatibility against specified references.\u001b[0m\n",
            "\u001b[32m    CompareReferences                            \u001b[31m(EXPERIMENTAL Tool) \u001b[36mDisplay reference comparison as a tab-delimited table and summarize reference differences.\u001b[0m\n",
            "\u001b[32m    ComposeSTRTableFile                          \u001b[36mComposes a genome-wide STR location table used for DragSTR model auto-calibration\u001b[0m\n",
            "\u001b[32m    CountBasesInReference                        \u001b[36mCount the numbers of each base in a reference file\u001b[0m\n",
            "\u001b[32m    CreateSequenceDictionary (Picard)            \u001b[36mCreates a sequence dictionary for a reference sequence.  \u001b[0m\n",
            "\u001b[32m    ExtractSequences (Picard)                    \u001b[36mSubsets intervals from a reference sequence to a new FASTA file.\u001b[0m\n",
            "\u001b[32m    FastaAlternateReferenceMaker                 \u001b[36mCreate an alternative reference by combining a fasta with a vcf.\u001b[0m\n",
            "\u001b[32m    FastaReferenceMaker                          \u001b[36mCreate snippets of a fasta file\u001b[0m\n",
            "\u001b[32m    FindBadGenomicKmersSpark                     \u001b[31m(BETA Tool) \u001b[36mIdentifies sequences that occur at high frequency in a reference\u001b[0m\n",
            "\u001b[32m    NonNFastaSize (Picard)                       \u001b[36mCounts the number of non-N bases in a fasta file.\u001b[0m\n",
            "\u001b[32m    NormalizeFasta (Picard)                      \u001b[36mNormalizes lines of sequence in a FASTA file to be of the same length.\u001b[0m\n",
            "\u001b[32m    ScatterIntervalsByNs (Picard)                \u001b[36mWrites an interval list created by splitting a reference at Ns.\u001b[0m\n",
            "\u001b[32m    ShiftFasta                                   \u001b[31m(BETA Tool) \u001b[36mCreates a shifted fasta file and shift_back file\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mShort Variant Discovery:                         Tools that perform variant calling and genotyping for short variants (SNPs, SNVs and Indels)\u001b[0m\n",
            "\u001b[32m    CalibrateDragstrModel                        \u001b[36mestimates the parameters for the DRAGstr model\u001b[0m\n",
            "\u001b[32m    CombineGVCFs                                 \u001b[36mMerges one or more HaplotypeCaller GVCF files into a single GVCF with appropriate annotations\u001b[0m\n",
            "\u001b[32m    GenomicsDBImport                             \u001b[36mImport VCFs to GenomicsDB\u001b[0m\n",
            "\u001b[32m    GenotypeGVCFs                                \u001b[36mPerform joint genotyping on one or more samples pre-called with HaplotypeCaller\u001b[0m\n",
            "\u001b[32m    GnarlyGenotyper                              \u001b[31m(BETA Tool) \u001b[36mPerform \"quick and dirty\" joint genotyping on one or more samples pre-called with HaplotypeCaller\u001b[0m\n",
            "\u001b[32m    HaplotypeBasedVariantRecaller                \u001b[31m(EXPERIMENTAL Tool) \u001b[36mCalculate likelihood matrix for each Allele in VCF against a set of Reads limited by a set of Haplotypes\u001b[0m\n",
            "\u001b[32m    HaplotypeCaller                              \u001b[36mCall germline SNPs and indels via local re-assembly of haplotypes\u001b[0m\n",
            "\u001b[32m    HaplotypeCallerSpark                         \u001b[31m(BETA Tool) \u001b[36mHaplotypeCaller on Spark\u001b[0m\n",
            "\u001b[32m    LearnReadOrientationModel                    \u001b[36mGet the maximum likelihood estimates of artifact prior probabilities in the orientation bias mixture model filter\u001b[0m\n",
            "\u001b[32m    MergeMutectStats                             \u001b[36mMerge the stats output by scatters of a single Mutect2 job\u001b[0m\n",
            "\u001b[32m    Mutect2                                      \u001b[36mCall somatic SNVs and indels via local assembly of haplotypes\u001b[0m\n",
            "\u001b[32m    RampedHaplotypeCaller                        \u001b[31m(EXPERIMENTAL Tool) \u001b[36mCall germline SNPs and indels via local re-assembly of haplotypes (ramped version)\u001b[0m\n",
            "\u001b[32m    ReadsPipelineSpark                           \u001b[31m(BETA Tool) \u001b[36mRuns BWA (if specified), MarkDuplicates, BQSR, and HaplotypeCaller on unaligned or aligned reads to generate a VCF.\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mStructural Variant Discovery:                    Tools that detect structural variants        \u001b[0m\n",
            "\u001b[32m    CollectSVEvidence                            \u001b[31m(BETA Tool) \u001b[36mGathers paired-end and split read evidence files for use in the GATK-SV pipeline.\u001b[0m\n",
            "\u001b[32m    CondenseDepthEvidence                        \u001b[31m(EXPERIMENTAL Tool) \u001b[36mMerges adjacent DepthEvidence records.\u001b[0m\n",
            "\u001b[32m    CpxVariantReInterpreterSpark                 \u001b[31m(BETA Tool) \u001b[36m(Internal) Tries to extract simple variants from a provided GATK-SV CPX.vcf\u001b[0m\n",
            "\u001b[32m    DiscoverVariantsFromContigAlignmentsSAMSpark \u001b[31m(BETA Tool) \u001b[36m(Internal) Examines aligned contigs from local assemblies and calls structural variants\u001b[0m\n",
            "\u001b[32m    ExtractSVEvidenceSpark                       \u001b[31m(BETA Tool) \u001b[36m(Internal) Extracts evidence of structural variations from reads\u001b[0m\n",
            "\u001b[32m    FindBreakpointEvidenceSpark                  \u001b[31m(BETA Tool) \u001b[36m(Internal) Produces local assemblies of genomic regions that may harbor structural variants\u001b[0m\n",
            "\u001b[32m    JointGermlineCNVSegmentation                 \u001b[31m(BETA Tool) \u001b[36mCombine segmented gCNV VCFs.\u001b[0m\n",
            "\u001b[32m    PrintReadCounts                              \u001b[31m(EXPERIMENTAL Tool) \u001b[36mPrints count files for CNV determination.\u001b[0m\n",
            "\u001b[32m    PrintSVEvidence                              \u001b[31m(EXPERIMENTAL Tool) \u001b[36mMerges SV evidence records.\u001b[0m\n",
            "\u001b[32m    SVAnnotate                                   \u001b[36mAdds gene overlap and variant consequence annotations to SV VCF from GATK-SV pipeline\u001b[0m\n",
            "\u001b[32m    SVCluster                                    \u001b[31m(BETA Tool) \u001b[36mClusters structural variants\u001b[0m\n",
            "\u001b[32m    SiteDepthtoBAF                               \u001b[31m(EXPERIMENTAL Tool) \u001b[36mConvert SiteDepth to BafEvidence\u001b[0m\n",
            "\u001b[32m    StructuralVariantDiscoverer                  \u001b[31m(BETA Tool) \u001b[36m(Internal) Examines aligned contigs from local assemblies and calls structural variants or their breakpoints\u001b[0m\n",
            "\u001b[32m    StructuralVariationDiscoveryPipelineSpark    \u001b[31m(BETA Tool) \u001b[36mRuns the structural variation discovery workflow on a single sample\u001b[0m\n",
            "\u001b[32m    SvDiscoverFromLocalAssemblyContigAlignmentsSpark    \u001b[31m(BETA Tool) \u001b[36m(Internal) Examines aligned contigs from local assemblies and calls structural variants or their breakpoints\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mVariant Evaluation and Refinement:               Tools that evaluate and refine variant calls, e.g. with annotations not offered by the engine\u001b[0m\n",
            "\u001b[32m    AlleleFrequencyQC                            \u001b[31m(BETA Tool) \u001b[36mGeneral-purpose tool for variant evaluation (% in dbSNP, genotype concordance, Ti/Tv ratios, and a lot more)\u001b[0m\n",
            "\u001b[32m    AnnotateVcfWithBamDepth                      \u001b[36m(Internal) Annotate a vcf with a bam's read depth at each variant locus\u001b[0m\n",
            "\u001b[32m    AnnotateVcfWithExpectedAlleleFraction        \u001b[36m(Internal) Annotate a vcf with expected allele fractions in pooled sequencing\u001b[0m\n",
            "\u001b[32m    CalculateGenotypePosteriors                  \u001b[36mCalculate genotype posterior probabilities given family and/or known population genotypes\u001b[0m\n",
            "\u001b[32m    CalculateMixingFractions                     \u001b[36m(Internal) Calculate proportions of different samples in a pooled bam\u001b[0m\n",
            "\u001b[32m    Concordance                                  \u001b[36mEvaluate concordance of an input VCF against a validated truth VCF\u001b[0m\n",
            "\u001b[32m    CountFalsePositives                          \u001b[31m(BETA Tool) \u001b[36mCount PASS variants\u001b[0m\n",
            "\u001b[32m    CountVariants                                \u001b[36mCounts variant records in a VCF file, regardless of filter status.\u001b[0m\n",
            "\u001b[32m    CountVariantsSpark                           \u001b[36mCountVariants on Spark\u001b[0m\n",
            "\u001b[32m    EvaluateInfoFieldConcordance                 \u001b[31m(BETA Tool) \u001b[36mEvaluate concordance of info fields in an input VCF against a validated truth VCF\u001b[0m\n",
            "\u001b[32m    FilterFuncotations                           \u001b[31m(EXPERIMENTAL Tool) \u001b[36mFilter variants based on clinically-significant Funcotations.\u001b[0m\n",
            "\u001b[32m    FindMendelianViolations (Picard)             \u001b[36mFinds mendelian violations of all types within a VCF\u001b[0m\n",
            "\u001b[32m    FuncotateSegments                            \u001b[31m(BETA Tool) \u001b[36mFunctional annotation for segment files.  The output formats are not well-defined and subject to change.\u001b[0m\n",
            "\u001b[32m    Funcotator                                   \u001b[36mFunctional Annotator\u001b[0m\n",
            "\u001b[32m    FuncotatorDataSourceDownloader               \u001b[36mData source downloader for Funcotator.\u001b[0m\n",
            "\u001b[32m    GenotypeConcordance (Picard)                 \u001b[36mCalculates the concordance between genotype data of one sample in each of two VCFs - truth (or reference) vs. calls.\u001b[0m\n",
            "\u001b[32m    MergeMutect2CallsWithMC3                     \u001b[31m(EXPERIMENTAL Tool) \u001b[36mUNSUPPORTED.  FOR EVALUATION ONLY. Merge M2 calls with MC\u001b[0m\n",
            "\u001b[32m    ReferenceBlockConcordance                    \u001b[36mEvaluate GVCF reference block concordance of an input GVCF against a truth GVCF\u001b[0m\n",
            "\u001b[32m    ValidateBasicSomaticShortMutations           \u001b[31m(EXPERIMENTAL Tool) \u001b[36mCheck variants against tumor-normal bams representing the same samples, though not the ones from the actual calls.\u001b[0m\n",
            "\u001b[32m    ValidateVariants                             \u001b[36mValidate VCF\u001b[0m\n",
            "\u001b[32m    VariantEval                                  \u001b[31m(BETA Tool) \u001b[36mGeneral-purpose tool for variant evaluation (% in dbSNP, genotype concordance, Ti/Tv ratios, and a lot more)\u001b[0m\n",
            "\u001b[32m    VariantsToTable                              \u001b[36mExtract fields from a VCF file to a tab-delimited table\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mVariant Filtering:                               Tools that filter variants by annotating the FILTER column\u001b[0m\n",
            "\u001b[32m    ApplyVQSR                                    \u001b[36m Apply a score cutoff to filter variants based on a recalibration table\u001b[0m\n",
            "\u001b[32m    CNNScoreVariants                             \u001b[36mApply a Convolutional Neural Net to filter annotated variants\u001b[0m\n",
            "\u001b[32m    CNNVariantTrain                              \u001b[31m(EXPERIMENTAL Tool) \u001b[36mTrain a CNN model for filtering variants\u001b[0m\n",
            "\u001b[32m    CNNVariantWriteTensors                       \u001b[31m(EXPERIMENTAL Tool) \u001b[36mWrite variant tensors for training a CNN to filter variants\u001b[0m\n",
            "\u001b[32m    CreateSomaticPanelOfNormals                  \u001b[31m(BETA Tool) \u001b[36mMake a panel of normals for use with Mutect2\u001b[0m\n",
            "\u001b[32m    ExtractVariantAnnotations                    \u001b[31m(BETA Tool) \u001b[36mExtracts site-level variant annotations, labels, and other metadata from a VCF file to HDF5 files\u001b[0m\n",
            "\u001b[32m    FilterAlignmentArtifacts                     \u001b[31m(EXPERIMENTAL Tool) \u001b[36mFilter alignment artifacts from a vcf callset.\u001b[0m\n",
            "\u001b[32m    FilterMutectCalls                            \u001b[36mFilter somatic SNVs and indels called by Mutect2\u001b[0m\n",
            "\u001b[32m    FilterVariantTranches                        \u001b[36mApply tranche filtering\u001b[0m\n",
            "\u001b[32m    FilterVcf (Picard)                           \u001b[36mHard filters a VCF.\u001b[0m\n",
            "\u001b[32m    MTLowHeteroplasmyFilterTool                  \u001b[36mIf too many low het sites, filter all low het sites\u001b[0m\n",
            "\u001b[32m    NuMTFilterTool                               \u001b[36mUses the median autosomal coverage and the allele depth to determine whether the allele might be a NuMT\u001b[0m\n",
            "\u001b[32m    ScoreVariantAnnotations                      \u001b[31m(BETA Tool) \u001b[36mScores variant calls in a VCF file based on site-level annotations using a previously trained model\u001b[0m\n",
            "\u001b[32m    TrainVariantAnnotationsModel                 \u001b[31m(BETA Tool) \u001b[36mTrains a model for scoring variant calls based on site-level annotations\u001b[0m\n",
            "\u001b[32m    VariantFiltration                            \u001b[36mFilter variant calls based on INFO and/or FORMAT annotations\u001b[0m\n",
            "\u001b[32m    VariantRecalibrator                          \u001b[36mBuild a recalibration model to score variant quality for filtering purposes\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\u001b[31mVariant Manipulation:                            Tools that manipulate variant call format (VCF) data\u001b[0m\n",
            "\u001b[32m    FixVcfHeader (Picard)                        \u001b[36mReplaces or fixes a VCF header.\u001b[0m\n",
            "\u001b[32m    GatherVcfs (Picard)                          \u001b[36mGathers multiple VCF files from a scatter operation into a single VCF file\u001b[0m\n",
            "\u001b[32m    GatherVcfsCloud                              \u001b[31m(BETA Tool) \u001b[36mGathers multiple VCF files from a scatter operation into a single VCF file\u001b[0m\n",
            "\u001b[32m    LeftAlignAndTrimVariants                     \u001b[36mLeft align and trim vairants\u001b[0m\n",
            "\u001b[32m    LiftoverVcf (Picard)                         \u001b[36mLifts over a VCF file from one reference build to another.  \u001b[0m\n",
            "\u001b[32m    MakeSitesOnlyVcf (Picard)                    \u001b[36mCreates a VCF that contains all the site-level information for all records in the input VCF but no genotype information.\u001b[0m\n",
            "\u001b[32m    MakeVcfSampleNameMap (Picard)                \u001b[36mCreates a TSV from sample name to VCF/GVCF path, with one line per input.\u001b[0m\n",
            "\u001b[32m    MergeVcfs (Picard)                           \u001b[36mCombines multiple variant files into a single variant file\u001b[0m\n",
            "\u001b[32m    PrintVariantsSpark                           \u001b[36mPrints out variants from the input VCF.\u001b[0m\n",
            "\u001b[32m    RemoveNearbyIndels                           \u001b[36m(Internal) Remove indels from the VCF file that are close to each other.\u001b[0m\n",
            "\u001b[32m    RenameSampleInVcf (Picard)                   \u001b[36mRenames a sample within a VCF or BCF.\u001b[0m\n",
            "\u001b[32m    SelectVariants                               \u001b[36mSelect a subset of variants from a VCF file\u001b[0m\n",
            "\u001b[32m    SortVcf (Picard)                             \u001b[36mSorts one or more VCF files.  \u001b[0m\n",
            "\u001b[32m    SplitVcfs (Picard)                           \u001b[36mSplits SNPs and INDELs into separate files.  \u001b[0m\n",
            "\u001b[32m    UpdateVCFSequenceDictionary                  \u001b[36mUpdates the sequence dictionary in a variant file.\u001b[0m\n",
            "\u001b[32m    UpdateVcfSequenceDictionary (Picard)         \u001b[36mTakes a VCF and a second file that contains a sequence dictionary and updates the VCF with the new sequence dictionary.\u001b[0m\n",
            "\u001b[32m    VariantAnnotator                             \u001b[36mTool for adding annotations to VCF files\u001b[0m\n",
            "\u001b[32m    VcfFormatConverter (Picard)                  \u001b[36mConverts VCF to BCF or BCF to VCF.  \u001b[0m\n",
            "\u001b[32m    VcfToIntervalList (Picard)                   \u001b[36mConverts a VCF or BCF file to a Picard Interval List\u001b[0m\n",
            "\n",
            "\u001b[37m--------------------------------------------------------------------------------------\n",
            "\u001b[0m\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 2. 전체 워크플로우\n",
        "\n",
        "![GATK-process-banner.png](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAB5IAAAESCAYAAAA/h5lhAAEOzklEQVR4AezdA3Ck2deA8XRnbNuTDSbp7nvPfbvHXJu9/7Vt27Zt27Ztj9a2rfOdd9JTlS/rnQ428/yqnroxy6fOfUswbwAAAAAAAAAAAAAAAAAAAAAAAAAgYSWt0nq1+pNKCyULnw8A+A8CAAAAAAAAAAAMi0tDCK3LysraDhgwoEM6ne5YWVnZU0TKQggp59x47/3kXC43LYqiZa2VQgj5usVvsxZ3zk2JPzaTyXgzavTo0YNCCF3jrzts2LB21dXVbQoD56TVxAAAAAAAAAAAAAAAyblD43ioa4PjLjbwHRgPf6MoWt97f1AI4Vw7b7TX77WXnxORV+z8yPpcRL6y8xd7n/5eIYTvrS+sz603rFki8pj3/g47rwghHBdF0bbWivGAury8vFc8tI4H2I0zXAYAAAAAAAAAAAAAJPL5fGk8qE2lUt1taFthG8JTRWRj7/1JIYRHrXdEJB7+/mD9atUfDv+/RORPCyH8bnO/noj8ZH1jL39i5wzrWnt5HztXsJ9P7OfrG28vx5vLxRksAwAAAAAAAAAAAAASU6ZMaRUPj733vbPZ7KQQwsEicoOdM61v6w2Gm6wQQv3e994/YOdp1mY1NTVpO7sWhsql1j8BAAAAAAAAAAAAAGwex9dVp9PpQSKymHPucDsfsT6qex21iDTb6m4v2/mdiLxr5512bmO/1xgbiveLf8c/3lQGAAAAAAAAAAAAACTjzePy8vKBIrKotXf8bGPv/VdNv3FcvMFy4eXP7bzVzu2895ENlLvZ79/aSlgAAAAAAAAAAAAAYNg+7ua9j8zuIvJgPGi1mv3WcZGuwX5PRK7PZrMb2cuV6XS6I1vKAAAAAAAAAAAAAOZHiRBC66qqqv5RFC0hIqdYb1q/NP3mcZNtKv9ivWQd4L0f65yLt5RbWS0aAAAAAAAAAAAAACSqq6vbFK6vXse6JYTwaRGGxy1tU/lN66Ioipa1v1UvBsotFgAAAAAAAAAAAMAAOZVKjRCRza2HQgjfMUD+yy3lr+y83M4VysrKejNQbgkAAAAAAAAAAAAAJKZMmdLKOTdMRLaIoigeIH/DAPnvV3hW9Od2XpPJZJavrKzsyTOUAQAAAAAAAAAAAPwn5fP50pqamr7OudVE5PYQwtcMkOf5yutPRORs59yUESNGdI0H9RYAAAAAAAAAAAAANHvJYcOGtctkMlNF5GLv/SdNN4T1Kr5e4uq9zf1O/s9r4qGynW9774+ws8r+3m0tAAAAAAAAAAAAAGieCtdYLyAi+1gvW7822eBYLOc0zCmjUTo1J5Ggks2q5EarGzdRMxOm/G4yeqxKLle7CZyxz51TWiOXUYnz9b5X4w+Tf7HzWe/9+vHmN9ddAwAAAAAAAAAAAGhukmVlZV1EZCnrphDCtw0+UK07xLXTW0FEJcic0+XGaGrqQlq11Iq6QH4dHb72ljp0o521/zYHat9djtI+ux+nPQ4+R7sdfvFv6n7YBdp775O0z67HaP/tD9LBm+6mw9fZSstW3Ugrl11VaxZeUt3Y8RqioBInot67uj9PXGMNlD+3TrWXRw8YMKAD110DAAAAAAAAAAAAaBZbyN77UdbBIYQ3oihq+AGyWL5209hL0IxtFqemLqzlK62jgzffU3vse7p2Ov46bXf6Xdrm/Ie09LKnteTqGVpy/atacsMrdlrXvawl18bN/t3s/dYr8edYr2ni2pmavPI5bX3RY9r27Pu040k3ao9DztP+2x2iI9bYXKsXW0Yzk6aqZHPxzxU3d6jcKM9PFpHZURRtkc1m+zXVdjIAAAAAAAAAAAAAJCsqKjo755YTkTtCCD80/PaxaHBOfZTTmgUXtw3h9XXA1vtq98Mv0jbnPaTJK57XxNXTa4fG18y2Zlkz49frNH0emmHNtGbVDpvtayfsbYkrX9DSi5/Uzidcp313O1qHr7WljlpiBfVjJ9RuSYuv93s02HbyV3aem0qlgm2It7VKiIho/ok/AhERERERERERERE1afl8vjSdTg8PIewRRdErdjbYADk4pyJefTanqamL6JBNdtXuB5+n7U+/Q0uveC7eHC4MjYs1LJ6XIfOs2p/HXo83obscc4X23+FQrVh+dU1PmKQhG1S8s3xcg20ne++ftlYdMmRI90a56hoAAAAAAAAAAAAAV1lXV1dnReS8KIq+KP4Q2dcOkOMzBE1PXlBHrrah9tnrBG1/5t1actWL8aZxveHx9OZXYWs5cc10Lb30Ge144vW2Pb2fViy7irox4zTU3VIWX/SBchRF7zvn9q6pqRncUFddAwAAAAAAAAAAAEAihNAhvso6iqJ7ReQnS4uac3NOnxutVUutqP23O0g72nXRyUufqXdN9fT/YPG28mxtfcEj2vXwi3TohjtqzbRFNYRs7e/tiz9M9t5/H0XRJblcLmP/v9YWAAAAAAAAAAAAABRNorKysqeIbBRCeKHoW8jOa/Be3biJusDK62jvfU7RNufeX/u842tnN78BchEGyskrX7DruW/Tgdvsr1WLL6+Szc299rroV12LyB3Wor169epslRARUcuMPwIRERERERERERERNVomWVVVNdR7v38I4d2ibs06p8F7zYyfqCPX2ly7HXahtrrocRu2zqx/bXXLrPB7trWheZ89j9fKZVaOt7GLvqFc2E6emclk1h00aFAPq4SIiFpe/BGIiIiIiIiIiIiIqFHK5/OlIlIVQjjW+qhom8jeqzinbuw4LVt1I+1+6PlaeslT9beP558Kz3tuc+6D2nfXo7VqyRVVomxRN5QL/7v37NwqnU73ibfMLQAAAAAAAAAAAAD4++Ihsg0cxXt/YQjhm2INkIN3Ktmclq+whvbc70xtfeGjdTaQp8/nzbBmartz7tf+Oxym1QsvqT4EFe+LOVD+xHt/UDabHfy3h8kAAAAAAAAAAAAAUNhEHhNCuDyKom+L8xxkZ6fXmoWX0IHbHqhtz74n3j6OY4Bcr9rN7Ona8ZSbdfj626obP1mD95Yr1mbyt865A8rLywf+rWEyAAAAAAAAAAAAAIbI3vux1jUi8kNRtpCdUzdmvI5cbSPtcvQVmrjqxb9zhTVdM0tLr3jONrfP0MqlV1bJZlXExxVjmPyVtXdFRcWAPx0mAwAAAAAAAAAAAGCI7Jwb572/NoTwY3G2kEVHLbac9tnjeG190WMMkP/lM5TbnXOfDtp8T01PmqrBexVflGHyN977ff9omAwAAAAAAAAAAAAAyXQ6Pdo5d7WIzOsQuXYLeex4HbHW5trp+Gs0cfX0ebzGmu3k5JXPa48Dz9SqpfMqIVLxriibyVEU7cMwGQAAAAAAAAAAAEB9yUwmUy0i54UQ5vE6a6fBe01NW1T773Cwtjn/EbaQi7yd3OH0O/6PvbuAjura/zZ+EiLF3VNIQohNZuZsmSg0FIpDHSrUlRp1d3dvofTS4lZoCVTo/3pdqUtC7ZXr7i238nv3mdJ5U88MMpPyPGt91uTeKif179pnS/GhJ4mqqRPjR7fKmOw+L6qoqBjoeACA9omHsNUAAAAAAAAAAFxZkUikzFo71xjzzy0aJZXvPk38Pt9eV86X7FWvbaMRmTE5Z/mLbqi/VsKjxopRW3ZvsjFGrLV/VkqdVFpa2sfx2h8AAA8BAAAAAAAAALBVuLJqamoKjDHXW2v/vqX3IatYtQw7aKZ0mf1wMHhu0wEZzZK15i3pfdk9UjlpT1HGbPGY7PzOOSkUCvVyvPYFAMBDAAAAAAAAAABsMVeWUqqvtfZ8Y8xvtsZ9yENmniP58x/jFPJ20xzX5Y4mKd3nENGxatFqi8fk95VS+0Qikc6O134AAHgIAAAAAAAAAIAtVlZW1tUYc7zz3pa+zjrcOFoGnXqV5C55jhE5HZo2yk73/lKKZxwvKlazNcbk5yKRyOhQKJTneAAATiQDAAAAAAAAAHYA06ZN62Ct3d15wRizBSeRo1I1eoL0P/dmyVm+Qby1Gxl10zgmd7z3MRl61Oni1zaIVv4WjcmxWGxdNBoNBSfXHSIiIiIiIiIiIiIiIiL6gZdtrVVa6/udz7bkddahMROk74V3SIf7Xg1OImfguJqiNe2UG/LzljwrhUeeLqqmPuWTycaY4HOT+7za3aU92PEyHwCAhwAAAAAAAAAASJkxZpgzW2v9ry05iRzabZL0uXiOZK/KkBG56QvNmz9bPj8hve6d77Y2sFG8pkBz4Cu/nvZ3MjlvyTNSePSZ4temPiYHrLV/dp8n+L7fw/EyGwCAhwAAAAAAAAAASEkkEukci8VmGWN+5aQ8IofHTHQnke+U7HSdRG76Js2S5YbhrAffkewH3pbclRskf+FT0mnez6Xzneuly50Pf7M566XjPb+U/MXPxk9WZ61tcb+edzePy8Fdz057O7HctPlk8lGnbfHJZOe14L5k95nreJkLAMBDAAAAAAAAAAAkLbgX2Y3IU1O9FzlxJ/KYSW5Enp2Wk8iJk8JuMA7G0mz3dYfVb0jOsuel6y1rpN8Ft8mQEy6UYYecIMP3PkQqpu4robFTJNo4WqIjG79RxP200PgpUr77/lIy7QgpPOJUGXT61dLr6oWykxuYc9yPM9v9tr9+arl9nEwectSWnUw2xoi1dlU0Gi39Yd2XTERERERERERERERERERZxpiwUmqF1vrjlAZF5Utk5K4y8PRrJGf5S8Gwuh3H47cSJ46D4bjj/Mek+433yaAzrpXh+x0l4bGTJDqiUVR1jRilJOZHxEajYv2oGN8Xo76D/znrxH+54JexRvz6BonsOlZK9z1MBp16hfR0v72dFj7hRuW3gh97+zil7H4/O977mJTsf6QoWy1apzYmK6U+ci6IRCL9HC8zAQB4CAAAAAAAAACApFRXV/c2xlxorf1DqiOyXz9CBp90ieQufT447brdTh9nNbVI1poWyV/0tPS6aoEUHXGyhMZNFf35aJwYgt3XiaF0yynRiV+3+1R+/LdXOWlPGXrUWdLr2iWSu8KN6Ws2bj6VncH3KbsxudPcH0vpPgeJNjblU8nu81fW2n0LCgo6Oh4AIPPwENoMAAAAAAAAAODK9n1/D2PMi6mNyEr86ho3oJ4h+fOf3PYnkZuctW5AXrsxfmdxlzsfksGnXC6VU6eJqm0Q40fFOjqF4Th1qtXp5aioESOlfK8Z0t+9Snsndzo6Oz4kO5k6KLtn2e3WNVIxeW8xSm/JfclN4XC4vH2/4pqIiIiIiIiIiIiIiIiIsmKxWMgYsyylV1orJdoYGeZeH9159iPb/k5kN8BmNW2UnJWvSI/rV0rR4bMk3DhajDZilJ+45zfdjPLjtDFSOWGv4HXfblB+PPgxZPAdyi3S8+pFUjV2smjfT/XH/pEx5gKlVF/HAwBkFh5CmwAAAAAAAAAAIpFIZ631Kc6vUxpMfV/Kp+7rRt3lm8fI5m04IjdL9v1vxE/OFh9xikSDAVnp1gNy5lEqfkLZmJhUutO+fS+cI3lLntt8h3JLho3JzZK15i0ZeOqVEm0YmdKYbIwJtFhrx7rPXMcDAGQOHgIAAAAAAAAA4Hu5sqy1De7rH6d26jYq4V3HunH0Tsle9do2P43cwf02el/+I6mcsKcYpdI9IKcwKPvi19ZJyYHHSPeb7pMOq1/LwNPJLZK77HkpOnSWqOpaMVqlMiZ/qpRaFA6Hix0PAJA5eAgAAAAAAAAAgO9ljOljrb3Sff49pXuRa+ql4MSL3fC4YVvfi+w0S/Z9r0mfi+dI1W4TxfibR+SA1u3H5nuUo7vsKoNPvlzyFz4tXtPGzBqT170jXe76sZTtub8YbZJ+xsaYwN9isdhBwYl3xwMAZAYeAgAAAAAAAADgO7myjTF7OhtSHUVL9jtSOs15VLy1Ldvt1cvZ7hRvz2sXS+neB4myMdFKtctBOT6Ex2IyfP+jpOsdD0rWA28nxuSMGJTXviN9L50r4VFjRavUXnFtrV1bVVVV4XgAgMzAQwAAAAAAAAAAfKfq6uoirfVcZ1PyI2hUKibuJT2uW5aekdMNrvkLn5CBZ14nlRP2EG1N+xyTN7/uOjR2svS74A7JW/5i4lXXmXAyuYM7AT7YnTj360eI0cmPyc6/rLVnuj/WejseACD9eAjfCgAAAAAAAAAQCoXytNbTjTGvOEmPn9GGkTLo9KskZ8XLidPI211Ti2Tf/4Z0va1JCo84Nf6qaGN0exuTE8O8P7JRBp1ypeQvfjr48WXGoLxuo3S855cyfJ+DRRsrWqtUTiU/7ft+zPO8Dg4RERERERERERERERERZWrGmGFKqXmpnEYOlBx4jHSa+5PEvchp5YbsnOUbpNcV82X4tMNF1dWLVn47PJ3si19dLUOOPUd2mv945ozJ7vn2vnyehEdPEK1TesX1h9baU8PhcE/HAwCkFw/hGwEAAAAAAAAAjDG5WuuD3GdzKq9irhw72Q2L90jW/W+mfURufTo5K3jd9fzHZOBZ10uVGz2NNZvHZNXO7k2ulqLDT5GO834q3gPBM07/mNxh5csy9OizRNXUp3pX8mPRaFR5npflEBERERERUXuurq6uYywWG1BfXz80uDertra2MGCMGYLU1NTUFIwaNarHtGnTOjgeAAAAAKSDMaZca73I2ZTsiKxiNbLzcedJ7uJnE6eRM0uzZLuBu+stD7gxdpZER476/PdfqXZ1MtkYI8UzZrpT3z8OxuTNQ3Iax2T3ve5y54NSPnmflJ6lUmpTLBY7PRQK9XI8AED68BAAAEDS3MCZE4vF9jPGPOT80RFsU79yfm6tvd3Z3w33gx0PAAAAALal4N/9fN+fobVuTv5uZF/Kp+wj3W9YGT8B7AbGDNUs3rp3JHf5i9Lr8nuldN/DRNXWfX7aV7WTV137vihjZMjRZ0r+wicz4hXXWWvekkGnXCHRuhFiVPJ3JTtPONrzvGyHiIiIiIiI2kPBKVlr7S8Yd9PuTWvtle4EuHW8rQ0AAAAAlFKDrLU3GWP+ndSwqZT4bkAcPOsyyVn6QmJIzmzB+NoSH2IHnn6tVEzcS3Sspt3cn2x8X/yaOhlwzs3SYcVLGXBX8kbpfNePpXTvGaKMSWVI/pv7+uiysrKujgcASA8eAgAASIox5k5HAo2NjXLaaafJLbfcIvPnz5elS5fK8uXLZdmyZfGvlyxZghQtXrxY7rrrLrn++uvllFNOkb322ktisdi3jcqvuJ92zLhx4zo7HgAAAABsKVeWtXaSMeZZJ+lRMxgQu9y2Vry1iRG5fWhqcZql890/loITzpfw6PFijG03Y3Ll+KnS9dY14t0ff8V12p9lv/NulUjjaDE6pVPJq3zfH+54AID04CEAAICkWGv/YYyJD5u//vWvZftFf/3rX2X9+vVy8cUXi/sPO18blK21f3duc/dTVzgeAAAAAKTKGNPdudT5e7J3I0drG2TQqVdKh+UvJU4jtztNGyV79evS/cZVUnzw8RIZMUpMpt+frJTjy5BjzpbcJc8lRvF0nvLOX/iElEw/QpSxSQ/J7vOP1to9SkpK8h1v+wMA8BDaDAAA1NfX9zPGSCQclorhpVJVURk/Mfvhhx/K9o9aWlrk1ltvlTFjxnzTKeVHHe14AAAAAJAsa22D1vp/HEl2SC7bY3/pdssD8dcbu0GxfVv7juQte1F6X3a3lO25f/x11yaDX3dtohGp2m1S/FRy1gNvpf+uZKfPxXdJODiVnMIz833/ZqXUIMfb/gAAPIQ2AwAAsVhsZ2OMVJSVybChhXElhUVilI6/zvqTTz6R7R9t2rRJHnzwQTnkkEO+OiZ/Zq1dUVdXV+J4bQEAAAAAoVAoT2t9rDHmfzlJnYiN1jXIoFOukJwVG1IYMTP7ddcd7/2FFJx0qVSNnSw6FsvMMVkp0cbIgLNvlA73vZL+E+FrN0r+gidk+D6Hikq8IjypU8mvKKXqPM/LdoiIiIiIiChTSwzJpYkhOaGkqFhGNTbKz3/+c6H09eabb8r5558v1trWr7z+2H3Odt+/AY73XQAAAABAKTVUaz3X+a8jbWW0kvKp06XbTatSGTAznxuTs90p3663NUnxISdItH6kGKUyblA2kbAUHjZL8tyAmylj/sDTr5VoQ6NonfSz+pfv+8cXFxd3d7ztCwDAQ2grAADQekhOnEYuLR4WfJ1QOmyY7L3X3vLqq68Kpa+NGzfKrFmzvnpC+d/W2vPdZ67jfRMAAAAA0FqPdp+PJ3saWVXXSsEJF0nOsuczf0je4tddPyf9zrtVQpP3Fm0z63Sy8X0pn7yXdJyzPjGAp/tEd+e5P5byqdOS/7EYI9baheFwuNjxti8AAA+hzQAAwNeG5KJiCf53VSgUH5VbD8rlpaUyc+ZM+eCDD4TS14YNG+TQQw/90qBsrX29pqam1vFaAwAAAICSkpJ8Y8zJWuvfJzdeRqVywp7S85qlkrXmrWBA/IFrjg+0nWevl6JDTxJVP1K08jPm9dZ+3QjpduNK8R54MyNOJWevfk0KTrhQ/LqGpF9v7byvlBozatSoHMfbfgAAPIQ2AwAA3zYkB7TWUlle0XpQjn9dVRmSSy65RP74xz8Kpa/gleNTp05tPSh/aq29vaGhoavjAQAAAEDA9/1CpdQ8rfUnjiSj0A2q+fc+Jt66d9I3WqbhDuC8xc/I4FOulPDocWJU8CxU+ofk6mrpdcndkr3q1YwYkt0fE27Yvk9Cu01KZXD/xPf9k50ejrf9AAB4CG0FAAC+a0hOUL4vZcOHf+3+ZK2UzJ49W/79738LpacPP/xQbrrpJqmurm79Pfvf7vvayD1wAAAAAFxZ1tqxWuunHGkro5REGnaR/mfd4IbL19I/XKbh7uQO7sfd59IfSeWkvUUbk+ZTyUqUjcmAs66XnBUbMuP70dQieUuek+IZx4q21amcSl7lvi5xvO0HAMBDaDMAANCGITkhEg5/4/3JIxoaZOXKlfLxxx8Lpae3335bDjzwwC+dTnau3rFfEwYAAACgsLBwJ6XULGPMb5IdksunTpduN6+Kn9B1w+EOqFmy1rZI95tWScXu08UYm94x2RgZPOsyyVn2QuYM++s2yoAzr5fIiFGidXJDslLqHd/36z3Py3IyLSIiIiIiIkpmSHa+/f7k4aUyceJEefjhR+TRRx+NW7xokdw1Z06bzXGnm292p2uvvOIKue7aa2XevHkJr7/+Or7HK6+8Ile4Z/eV08mPOwMdDwAAAMCOx92RXKC1vt19/WFSo6U1MvSoMyRv0dPBydP0D5ZpPp3c6+rFUjl+97S+4loFQ/IJF0nu0uczakjucvs6KZ+4p2htkj2R/Hdr7RFlZWVdHW/7AADwENoKAACkNCS3uj/5a6+7PvbYmTJq1K5ilEn8/6nSSge/rjhjTOrwu9jneQAAAAB2LNXV1dYY85CT3Gutdxkt/c6/TbJXvx6czHXe3qF1cM9h8EmXSrSuQYxKz5hsfF+GHnZK/P5mb21L5jyblS9J8YyZomwsuR+PMZ+5z5uUUoMcb/sAAPAQ2goAAKQ6JCcoX0lZSeL+ZDnk4EPjw29VReUWD8lVFRVbb0jGx9XV1dMcLz0AAIAry+lQUlKSH7xqNtC3b98uoVBomwCA4KSnMeZA541khmTrR6VszxnS5ba1X5xGxoPvSY8bVkh49Li0DslFh5wYnBLPqCE5y+l/3q0S2aVRTJL3JFtr10cikSpeb92eIiIiIiIiXm2dhEg4ss2H5PXr17cBHnroIVm0aJFcfvnlMmHChK9+rz6z1p7oeNsWAABw5QRDsRuNu5WXl/d2nwVuTC6tqKiodePOPlVVVXu7z0PC4fCp7vMs97/P3AYA4FylVJPW+u+OtIlSoo2VoUeeJnkLnxJv3TvBWIh170rnuY9K2ZRporVJ45B8QsYNycEfI13ueEgqx+8hWvnJvt76/0aj0f3cmNyvoKCgo+d5eU6207aIiIiIiIiIIZlS67HHHpPp06e3fiX5Z+57Pt3xAADA1rX5pHEPN9zs7MbhEcaYadFo9LJIJLLcfT5njHnT/XzvO78PWGv/bIz50PloGwGATc4nSY2Vypdo/UgZcOZ1kr3q1S/u4sW6jbLTvY9J6V4HiTbpG5ILD5uVeUPy2o3x120XH3C0aBsTncSpZKXUp87Tvu/Pdv/72EgkMsH9/TRkjOnTv3//zsGbPBwiIiIiIiLaFjEk08cffxycUG79Pfv3iBEjejoeAADYMq6c4NWxwXjs7OUG42uUUo9orf+PE4zEHztirRWTpuEBAJIdkivHTpEeVy3+yljJkNzxRz+Vst0PEG3S9/0ZMvNcyV36XMa9cjzbGXzSJeLX1CVeb53kK64/c19vcv6qlHrO+VE4HD4xFApFS0tL+7hxOZ/XXxMREREREW2jGJLp2muvTXzPrLVzHC81AAAgeG21UqqvG45HRCKRS9znz621vzHGSHsfjgGgZPoR0nnO+tZDMta9J11uXycVE/cUrdL0vTFGBs+6THKWvZh5J8XdHdJ9L50r4cbdRGvl6BQkRuU49/U/nFej0ei8SCSyrxuVS1zdePU1ERERERHRVo4hmT799FM57LDDvvgX84/dZ7njtR0AAHDlhUKhAZWVldOMMQud/xv8fdX5YQxIAGCtFBx7juQsfaH1WImH3pee1y+TqtHjxCiV5iE5A783azdK59mPSNmUfbfuCXljAp9Za//mPn+hlDrb/T045OrCCWUiIiIiIqKtGEMyvfbaa61PJd/keAAA4Pu5OpSXl/eORqPjfd9fEIvFfmWM+UGdOgYAo5REG0ZK/7NvkqzVb3xprESz9DvnZonU7yJG+QzJ36DDypekZP8jRdnY1v9j05gv/Mv978ej0eiJkUikzPO8TgzKREREREREDMm0lZoxY8YX37sPHO/bAQAAV1afPn26+r4fU0rdoLVmQAbww74fedxU6XnVQvHWvMV4nNAsOctflMLDTxFVXStacyL52+x8wkXi1zWI3kantlv9Pfifznql1AENDQ2DPM/LcbYkIiIiIiIiYkimuXPnJr53wb9wOx4AAPg6V65SapAx5njnpR/6gAwAJhqV0n0Ols53PtT6fmQ0tUinOeulYtJeos13PUOG5L4XzpZI4+iUh+QU7lP+TSwWuzsSidQ6nbfgdDIRERERERExJNP69esT37vY53lfBgAAgv8Y7ft+vVJqkTHm7wzIAHYERispOvhEyZ//uHjrNjIgt7r/d8DZN0pk5K6itWJI/o7n1O3m1VI1ZuJ2u0d689+fP3Oe8X3/MKVUcDq5g5NsRERERERExJBMzz77bOJ7V11dPcXxPgcAAFxZgwcP7q21nmGtfZkBGcCOdD+yitVIwfEXSM7S58Vby5DsxAfb3CXPyTB396/+3rt/GZI7zvuZlE3Zd8tOJKc+KP/BWnuzUqoyeKuIk0xERERERETEkEwvvPBC4ntnrd3d8QAAsBaurLKyskGRSORcY8yvGJEB7Gj3I0dGNEr/s26U7FWvitcUf7U1HnxHel85X6pGj2vDOMqQnLfwKSmddphoY9M1Jn+klFodCoVGFBQUdHS8LwMA8BDaCgAAMCRvjiEZAABG5OCfi9zXtxlj/sGIDGBHY/yohBvHSp9L7mY8TmiWvOUbpOjQWaKqa8RkwpB8UgYPye73KWfFBik87GRRseq0De/W2k+MMU+6z0nBVRWOBwD4/3gIbQYAABiSgxiSAQDYkbmyjTFDYrHYbPf5X0ZkADvqkBwaN0V6XrtUvHXvMCI7wXPoed1SqdptYuJu5LQyWnY+7nzJXfJ8xp4Yz179Rvz16H51bVqH983/vvtsLBabwJgMAAzJAACAIZkYkgEASJora/OIPFdrvYkRGcCOfEdy6Z4HStfbmsRbx/3IgZwVLyVOI2fG2O9L4aEnSt6ip8Vbm5lDcvB66wGnXS1+bb0Y5af3eRkTeDH4995QKNTF8QAAIY+H0GYAAIAhOYghGQCAHXVErqmpKXBf326MYUQGsGMzRkr2P0o6zf2f+BjocRpZel29WKrGTMygU+O+FB9wrOQveCpzh+SH35MB594i0foRYnw//c/MmODzaff3+3HcmQwA3JEMAAC4I3krx5AMAKVDh00sGVq0xFmVqYYNLTrd8doKcGWVlpb28X3/Eq31R4zIABiSrRS607f59/wyMVLusJpaJG/J0zJsxnHxu36NVpkxJEfDUrbHfrLT5u9RRt6T/OB70uuyeRIeNTbx+50hY/Ivfd9v8Dwv12lDRERERERExJD8rTEkA8CoUaNyhg0pWun+mvyZI5mupKREO14bAd183z/aWvsnw4gsAKCslZ1nniO5S57dugNlU7Oz+dNpD0Ny1ppm6X/erRLZZVTint9MGZJD43eXTrMfTjzbzDvJ/a47yb1AQqMnJO6VzhCfOoudCs/zsh0iIiIiIiJiSE42hmQAqCgqGjqssPjaksLiTV/8NblsWInUV9dkHBP1JTEkFxWNczzge5WU5Gutp1hr32NE/hwAKBuTwcefLznLXth6A2VTsxtlncTpWacpIWNfad3ljgelfMq+orVxdEa92jrcOEZ6XLtEsla/kaFD8jvS44YVEho3NfPuATfmP87VVVVVOwdvJnGIqFXDCwtrSoYOfapkaOHfg3+/am9KCos2lRQWXuR43w08hDYDAAAMyUEMyQAwbdr/Y+8soJtIu/+flnV3XzIzmUysjU2aeou7W3EtDuuKywru8rLoClpoF1th8fX94/YWX8FZgRdtkft/7hySbfOjqSXtpL3fcz4nnJZMwoQ+zTyf3HubVxA5rqFRr/+arcE3DDrdDc96HB8dA7t27gQ1ZuP6DSSSiULBUiGShb0H+vFfiUwQBEE43NHw7OsfQtiyvYGr7F22B+6f/iU8/t5cJmdXwZ2Lt0H45yqWydjSeuFW4Dq/ys5HjKqqkRUcdrC73fDkgMlQIW0XPl8ViuQD+FqrUcR7ZPIZl8vVh70VeJShKe8QBHbpEbVCLx3H7woZaexfJp9haAj/BOGghCQIr+m0/Alf9LzuFN6GKnqef5OhCRWwtZ4sy08wNARBEERgIJGMIZFcfiEIwiIIFUVOGGoU9WckQXejWeMm13Ou4S2aNoM/z54FDIlkoizgcDietNvt43JLZIIgCMIRFQ3PBFIkL2dS9pPvQdvlNXDExEFktVogtOsNjw+fBfd8/B2Epe9TnVAOz8hkknYKWJOr+s5FVlV76xd6D8DK8VsiOVN9InkyVnQ3VatIRnaw9wPJGo3mDg2FUk4j8bxb5LjZ7DrqEgNyIul0MKj/ABg+dFjIUKNqVc/z/5uh8Q8RhIMSolY3M+cPksVk+ndQv8MBEWYzftIBvxdacNxchkaFYJXc9+w5XkXYub1qjYi8LjudN/GcO+z2m0x4ZOP3Qg2eLdAMDUEQhFogkYwhkUwQRDmsPub5+kwef4XVx6ziOGv61GkwdtRo0POCdx3u/24/uHbtGmBIJBNlAY7j7nG73a1xLjJJI4IIPZj4CRQ+xyZQmtpiE+Cpt8eBJj1QIvkA3DN3MwhtekBUZATIdhu47HZwRMeC1KQdPDF0Btz92U8Qhn9XBUIZBejD45eCsXZjkNX8WtmtoG/WHu6dudbTPpxEcuFl8g3GLKvVypenFtcUClYf6zmhB3Mv2/Pai6uSlAS/Hj0KoZY3Xn2NRDKJZPWI5CiX67ab6DHR0ZAUn6B6cI5YSIhkLXfI8zztNtttz7lRL4WevNfpKjM0aoEgCIJEcrkJiWSCIHDj4AWR4wabRP0pJoxvdO/a9drmTZvg8uXL0LtHT+/6ywQzLFywADwhkUyUBVjCnE6nyH4WNjFIHBFECItku90eCHIcm8AWzpGVqsITg6czkbwvYCL57k9+AG3n19hjOLAYxzvnN4rJUHt8Euja9oTHRn4Gdy7axoTy/lKTyTjX995534KuVVdwRLnx+ar39XLYwcbO3aOjFhR9TjKJZLw9y+hosVgeYGgIoixjEASXqOVnMudy0XfvLecHiXt07QYXL16EYoREMolkEsn4Q+XKQyQjf/zxB6g9+zMzQ0okW4ymPM93bGwstG/TVvVUq1yFRDJBEKUCiWQKiWSCIIYMGRKuF4S6Bp1+tcjxN+Lc7qtTJ0+Bs2fOAub333+HOjVreddel8MBW9g6gSGRTJQlcKNYluV3GFmhKXoIgiRyXFwcNGvWDDp37gydOnUqMh06dIA6deoAuxai6uRcIrlaQEUyCs47mCB+9rUPwBaXCE6U9z6P6WJYk6uANvUNeHBiBs79LfHqZBTedy7cAhW7vwuO2DiQQ+D/BJ7L5/sOxfbWJJKLhud6eAP7s6zRaMI1FEoZi8FgeFDH891Ejtvmu9/msjugfp26ub42Ydx4CEBIJJNIJpFsEPV+N9JnzpxJIjlQaPmD+Bwddrvfc759+3ZQez6eN69ERTJBEASJZAqJZIIgJEl6XuS4QSZROqlUH6emXtu0YSPcvHkTPPnu229Bttm9626dGjXh5MmTUNBkZ2cDHnPl8hWlxsgPRvwrkityH+g5oVVuuGiGpnxDsITHxMSYnE7nXpqNTBB5ouqW0ZGRkdCoUSPIyMiA3377DY4cOVIkjh49CpmZmTBmzBiIj48P4nMmkYyEZeyDx4fNhogqNbEl8/99XMSudCIEc60G8Mybo+GeeZsgTBHJ+xnBbd2Mj1FhyS549s1RYE2sHBIS2VPVbWjQHO6bscYj7VUnkg3qFskeLrM1oK/dbn+EoSGIsoAoik4dx89gFcgXfPfZWjZvAQvmz4dWLVp4vxZpMsOar9dAgEIimUQyiWRJJ/rdSGezxqBkQyJ51KhRJJIJgiBIJFNIJBMEcav6WNLqahtF/UqsPo6Ncl+dMmkSnD59Gnwze+YskASdd81lba7hypUrUNAcO3YMqiQl431VD6vIrsfQlF8InAfndDr7ybJ8nYQRQeRZnYcEqmVswImIiICUlBT4ln0IqrjB33dz5syBpKSkIItkEslY7fvA1FUgNW2Hra39zmj2zE/Wp3SBx9+fB3fP/wXClOMEp0IZjxfObp8c9B+IrFLT8/xCRiTbY+LgySH/gfClu1CIU0Vy0derbxhWmpVMCeVg5x0dz3dlTmWL77UQfnD4vWHD4NDBQ7B3z14cf+r9XmJcPBw4cAACHBLJJJJpRjK2U/YrNkeOhJkffaRaPnjvvdASyb7zkX2oXbs2iWSCIAgSyZTyLJIJgsC2Zc/peX6AUS+dQIHctXOXa9jyGauPfXP16lV49eWXvWstyuSRH3wAN27cgILkxPHj8Nbrb3hmaYUIfH+GpnxCsIRHR0dbZVnefXvBRRBUhZyYmAgDBw6EtLS0YrFkyRJ47bXXwO12B00kb968GYqby5cvw+zZs/HfTSI5l0iuzqRqYEUySmBsv/x874Fgd0ejrM33eWCFMs4A5jv0gYfGpcEdaTtRlN6SyYGUyPvgieFzIKJGXVAeN9ReM7sVhLa94O65mzyiXTUi+aHxy8BUu1GonMvLsixTVTIRkoha0cFcynSR4//new2U0qw5ZKSnQ9bVLMBgNyeLwfjv91lR5Plz5yEIIZFMIplEcqQlwu9musVkUv9GSgjNSDbpJUhMSPB7zvv06g0D+vVXLS2aNA24SCYIgiCRTCGRTBBUfazjdLUMorhc5PjrMa6orEkTJsCpU6f8SuCGdet511lHRCQsXrgIrl+/Dv6Cknnd2rWQ2qkziJzA4L3HeKl3H9ixfbvq2LRxE4lkBcJqtd6PbSudTucNBhAEkRu2EQ3Vq1eHlStXQiAyb948LMIgkRyCItmakAxP9psYUJGMhDEeGz4XLNXqeOYk54tLaXfthAh2n+dfGgb3T/8KKizbw2TpgeLPUEaJnL73lkSuj//2EH3N7GCPjYMnBk6D8MU71COTVxyCxz78BCxVa7Pn6QiVbgzr2BpDVcmUkAi+txV5vouO43/x3UdzWm0wbMhQOHjwYK5ruREffJjr773z5lvK14MUEskkkkkkG/T+5yTbbLYAyl4SyYjFbPZ7zvH7Kj/XARHJBEEQJJIpJJIJgrBwlmdELd/PJEnHsfq4S4dO19avW6dUH/vLzz/9BG6n7F1jq1WuDBvXr4dr167lLZ5PnICJ4ydAfHQMGHQiJCckeu+PMnn61Gmg1vz1119AIplgCZNluaLT6dxI1ch5Q5BIrlatmjJ7uLjBTWlsGU0iOTRxuKPhmdc/hLBlewPeQvremWtB16Y7isUCt5DGdtfYwll2ucBUpzE8+9ZouHfWBghXjrm/aOJ0+X6okH6rErl6PY9EDlmwetvYMAXun7YaPwCgEpF8GB57fx5EVK4RSiL5L/bntijoGBqCUCMSz9uYQJ4qcvx53/0zLGRLX7ZM6T6VM+fPn4eO7dp7/55ZL8HMGTOCIZFJJJNIJoxG4+OCILiEisI7uGGCb3j8bagvXboU1q1bpyqwajeHzKyC/x5RFJ9kaFQInvdMxk3EIPqfTY1v+lUukG96YP+2ZDWdZ4IgCBLJ6g+JZIIgACBM5PkaBlH8HKuPo2VXFgrekydPQkHy2SefKiLYu1a3aQNbt26F7Ozs20oAFNPduqRi22smjxNg1IhR0Lh+A+/97ZGRsGnDRlBxSCQTCM6Lu8vpdNZn14xXSRIRhH+RTCKZcERFw7MekRxgIVlh8XZ27BFgi0vAquRCV0tjhbKTiW6paRvWfnsq3DP3OwhHceoVyl7ylNmKRF6yHZ7qP5lJ5NrsuGXgdVNacstQscc7cOdnP6ujKnnVYXim/0Swxidj1XQozYifxmTdCyQsCTXB/l/ep+eETiLH/+S7Z+awWmHooMFwYP9+uF3w61WSkr1/PyYqClavWuXpRkUiOViQSCbEimILpb11hP/21vPnz4fi5oP334d+774bMOrVru1dNASlEpmbUdLoKwojGZrCIHLCq8rC6F/eq3IofJOGjW5Va3A/MjQEQRBqhESyekMimSCICJ5/WuSEd0x66ZjI8Tc7tm+fvfabbwr8CfKsrCx4+803vesqzjUeNGAA7Nq1S/lezmBL7MkTJ0JibJwinXt266a0h96+bRvEuaO9x6hRpSr8evQoqD0kkgkkMjLyUafT+R+qRiYIEsn5i2TC4XLDC32HQIVF2wIvI5crc3PBWL85yIUUyb7zkx3RsSA1aQtPvzueVSivx5bXHqHsufX+GUGBHMZu752zibXJHgy2pMoewVkmwPNpi0uEJ4Z+BOFpu0pfJq9kIvmNkWCLiQslkYzXxnvYehDLupmEa0o5FIpRECJFLT9Fx/HnfPfKmjVuDMvSlirVx3nlqy+/BKvZ4r1Pw3r14acffyxJiUwimUQyiWSjJPndUE9NTYXiBjd5PC3jDKIezAYj3uLXQheOyy5si2uPSI6wWPye8xkzZkBx8s8//8D4seNyMaD/AOjZvQe8+sorMOLDD/FrhSLOHePZuPqDMaw0kLS6xgxNXhAEQZBIVl9IJBMEVR9LglDNqNOn63nhmtspZ+F7S5xvXJicPn0amjZq7F1TcSNhzqxZsGvnTu+mA7bD3rh+A3RP7apUHyfFJ8DUyZPhzJkzgFm6JA3Mt34/IKmdOsHFixchFEIimWAJYxXJoizL29jPFkkigiCRTCI5X5Hsghe7vQV3fPYTCtiAykU83l3suC/27A92dwxW0hZLnEbZreCMjgFjgxbw/MvD4JEJ6XA3O36F9N0QrojjAxDG5HU4E6p3LdwCj4xeCPoWHZX23S6P3CxDoGA3126syPqw9L2lKpOxxfbzfQaBLTo2lIQ9rgVZLperC3vv8ABDU9IQBPvdea/EcR1EjvvBd3/MHmmFIQMHwf7MTMgn6CXQKXnvi51q8RoQRxqVYEgkk0gmkYzkt6mO88MqJyYVGZE9Bm7m2G028J3BrBeE8iaSUaL7Pd+tW7eG4gSrKnLOe/OtOscLC6M+5EQ+VkPPoYpLgiCoIplCIpkg1I9Vp3uKrctvm/XS71h93KFdu+w1X68p0vyqbVu3QWyU27ueVkpMUtqY7dyxA65cuaJIZhTGiXHxeM2BIhmFsnfOMn5SfdjgId77IyM/HAGFDLbOLjWwwtrbkakiNwhbHOdkyJAh4QwNUXYRRfFu9j6niSzL50kQFQCCRDKJZMIlA9fxZbhn7mas4g24YESx+cjI+Ux4NgrIbGI8hgurm3G/NKES6Ju2ZxXHQ+HJYR/Bw+PS4JFRC+Hp/pNA16Yn2BOSlb8vO8vu642C3cAqtR+Y/iWEZWSWikzGx8M25hU7v6pIe2cI/Xzd6l4yl62JHENTUhCESaeL0Gv5ScyF/OO7L4YdT9MWL1Gu4fLLhQsXoGvnLt77ossYN2YM7N61S7k+KoWQSCaRTCLZGhnpb1PdsyFfLHwlsodWrVrBnj17QoZWLVKKLZIRzy/1PPBUcRebCLP5dsfHNz/4GCSSCYIgkUwimUQyQRABqz426HRVjKJ+KVYfu+yOLLzYP37sGBQ1ixctApNe8q6lrVJS4Ltvv1VaVKOYxpbV2LoaJTK2skbh6lvJ2zql5b9rsdEEGenpUJgsz/gcXA6nyj/kyp8Teb4BQ1M2IQwGw4NMEA0sYDUyQZBIJpFMyE4QWneDe2esCY5IxqrkT3+Cij3eDahk9MxQxiplzyxla3JVsCVVAll2KV9z2Wxl/vXD6l+8xhM69Ib7Zq1jVdn/LXmZrFSe/whiy67gdEWF0vnzFBJtYWMxZOxqoqFQghisPtbxfHudlv/u/zihiEhlHNF/9+2DgubI4cNQvUpV7zHcThmWLFqElciekUYkkomSEsmEKIrJ7PYUwjbcb/jbVLdZrWAS9UVG0unA3/H34UISGsEKh4CI5PxmU2P770BsKsXHx+f5GNOnT8dP96gdbF1IIpkgCBLJ6gyJZIIg8LriST3Pv2nWG37D6uP2bdpmr/nq62LNrMJWZYMHDsy1jvZ7+23AmcoD3u0HCbFxSvVxty6psH7dOqX62Dd79+xVBLP3fXFMLGz5f1ugoDl+/DikNGseOh+61PJpDE3ZhGDS6Wn285ZWMJFMECSSSSQTKHalRq3gwUnLmUg+GBTRGJaxT6lKNtVqhO2Yg1aZ61Kw+a98LqNVyVhZznd6Ge5FmZxRsjIZZ2HfO2cjSA1SQHY4Q00k4+3x6OjohtjVhKEhiCBg1nHChFuyE3LSuEFD/FCw8juqMMFrO5TPnuPUrVUb1q9di92oPCONSCSXFiSSCT0vnPe3qV6lShUoTlCK+jv+6NGjy4lI5pIZ43Va7SSj3v9sahTA9evWLRI1q1f3tLX2+xhNmzaFUIjNEkEimSAIEskqC4lkgiB0Wl1lSadbzITuNdluzxozahQc++NY8ecB//kntGzeArwfwDSZ4V0mkVunpCjyOD46BiaOnwAnT56EvLLi8+VgyfHBzKaNG8PRw0cKJA2wLXaXjp1yzeLCYw0fOgxGfPCh2sj5XjmD3guUTVjCmRwyO53OA7c2hgmCIJFMIrkAc3YtNesrLaE1Kw8GTTTe/ckPoO3yOjhkV1BaH9PrqFQmA9+xL5PJ60GTXnIyGV/fByZmQETV2qEq8a8wXmHr4iOBaltMEJW4SveIPN+WVR9/67vvhdclA/r1Vz7QW5RMnTwl1zVYj65d4eeffoId27d72mGTSC5lSCQT57Dq2N/G+pEjR6CoMen9zwSuXr06bjqFBKzCosgi2QN+Eiw/yVujRo1AzEjGCwm/j3Po0CESyaqAIAgSyeoPiWSCICRJeoJtGrzO1tmjbG272bZV6+yvvvxSqT4ORHDmVUJMrHftdETaIFp2KRsKKHfXrV3rd84yfg/lquf+yOuvvKK0vPYX/L4yZxkrnXU6lNfe+ycnJMC+vftApcHnXMZFMiHL8p1MDlV1Op3XSCoUHIJEMolkEsmRlarB40NnMikYPPEYxnh0xCdMWjfAdsx07oNamfwSyuSSa3PNHuex4bPw/1EoimTPdfJYti48x9AUB4KQOMkoctw4Ucv95bvf1bBefVi0cCFcunQJihK8X68ePbzHw9FFo0aMgC1svwclMn5fHSGRTCKZOGcxmfxurG/cuBGKGonXgcPu8Ht8gyiGStu4gIhkPEZ+8h4XyeKK5CaNGvt9jIEDB8L3332nZrAChEQyQRAkkksxJJIJgsDOOkywLmIVwdlOmz1r9MiR8Pvvv0Mgg7OLzQZl/VauH/S8ALFRbhg/dhycOHEC8su5c+egQ9t23nXXLBlwbnKelcvYDnvTxk3KnGWsdE6KT4BX+vTNJZHbtmoFf//9N5RySCSTSL6PbeD1JJlAECSSSSQXbsauNSEZnn5zNISn7cR5t0Gdo6vt8Y7PrOSAQzJZziGT0/eViEx+/uXhYItLDMVz5rlGXmqz2SSGhiAKCzoMvVZoLXL8Jt89LhxF2f+dd2H37t1QnPz2229Qu0ZN73FdNjss+Gy+IpG3b9sGFy9eBPWERDKJZOIcbrr721j/+uuviyGSBbCYzX6PbzGZy51IjrT4n5OMm1bFFckG0X81uN1mC5lzTiKZIAgSyZSSE8kEQRiNxsdFTniVralHsPq4dUrL7C9Wr1bmFwcyuFE/sF8/Za00inql+rhDu3bwzZo1yvcKkgP790PlxCTvmosCOn3pUjhx/Dj45vTp09g2Decne+csb1i/HoV1rlZqQwcP8VRak0guVQg23/AhWZZHkEgoOASJZBLJBFaQOtwx8EKvAXDHZz+DZrkikoNWlfzImMVgrNfsVlUyvQ7Bq0x2Ad9Rkcns3AdXJldYthuEdr3AGRUVyiL5e/ZnGcdkaAoYCsXA8waR58ey6uM/ffe2GtStCwvnzw+I4P1282ZwWv91E7WqVVeuAVEib9u6FS5cuADqColkEsnEOaMkBU0kv/n669CuTTu/x3e73VC3Zm1VUru651MxCJ+p0wrpolaYw9AUCY9Ijgi6SEY8bx7yBDfQSCQjBEGQSCaRTCKZIAiJ5xMlQbeAke2w2rJGfjhC+aR4MLKGXWPEuKKU6mOX3QHjxoyF48eOQWGCrbWxijhnazUUw3/88Ueu6uPNmzZBz+7d8b2vIpEnTZiAUhk3KFAme+9v1OthGZPQagmJZIJdKz/OfjYXFmM+MkGgYAwoJJJJJIdKVTLfvg/cM28zaFYcCGIL5P1w58It8ELfIWCPiQOnnVpcB78y+WW4d/YGlPhBkcl4vHtmrgOpYQowDxuqIhlvD7HbypUqVbqDockLgrBYLHdJnK6lyHEbfPez8Fqr39vvwK6dOyFQmfnRR7mcRGrnzvDjDz94JfL58+ehnIVEMolkEsmexMTE+H2MzMxMUGPwTblXZvL8+wxNsQiySMbnu37dOgWTXoJIi8Xv43RNTYUJ48apipf79PWecz3P99cLQj2J520MTV4QBEGQSC5cSCQTBMHz/NOCVvupjuMzRa3wmlmSDmH1casWKdmrVq6E7OxsCHT++usvmDljBiQxmYsVwO1atYE1X31d6OpflMMonnNWEfft3Rt++flnr/g+c+YMTJsyVWlbjRsVXTt3UeYs430xR48cgRpVq3rvj/OYcQ6XmkIimbBarU/JsryOUW7lZ2FFqO99yxl5njOUrDab7baw/2c5yfl1vMX7evAcS32CmUQyiWQfXDYrGBq3hgcmr0DZG9Q2yHjsBydmgNS4DVYkB7kqmWSywxUFSmXyR2uD0+Z6xSF4bOSnYK5WB1/LUBbJ/zDqoSRkaAjCF5Mg6PVafrSo5c/67mPVr1MH5n/6WUArg69evQqvvPSS9zHwGu294cNRICtsZRIZRxaV45BIJpFMIrlWzZp+H2P69OkkknOAm2zFTYTRBEa9/9e2Y8eOoLKgBM/xC0WoxtDkA0EQBInkAoZEMkEQgiC0FjjhvyKvuymKRtCLxpuSTroxuP9ApbtNMPLjDz/ihwWVDzq6ZRnGjh4Nx/44BkXJhf/9D1I7dc5VRTx+7FjcfEA5rLRJ692jJxh0oiJaJ4wb7zsrWalatkdac2yU1IUzrEK5oLly5QoT2WPweZQqZsno+Tec1Gn5VTkRtXyanueTGJrQhZBl+Vmn07nj/7N3FtBRXWvYpna1V3t7pe1fJpmJE5uZeIgnSIpbIEChuBMcKrg7t7hDi5aEAAWKU9zdocilSq/Urq6/73/evTjTJD+EyBlJ+L613jVh5Ox9lHP2s9/vq+wgWYeTBVUQiOrrz9dHqSBELbK80vSl3HI1cC+6ntwW2v0xkpOTUb9+fbRq1Qrt2rVDr1690KdPH6X+/ftj5MiRGDt2LMaMGUOpv0eNGoWBAwciJydHfa+nNuj7muYWys7ORr169ZCamsp7r4LbvGgfKKMh8SMVHBys+pabm2sISCag1dLLG7D/Ky5ILnp8OUl6O4antw5Kq4nnRsxHldwLTq+n+9SaM/jjoMkIrZ4stZJd4Da3aa9er/bATxZ+iCpGO5M3XMMLOaMRGhsHW8Xel99r1+rGHIvWVIUSiTixwGwyNdOeH3Zq+l4TdJElDBowAGdOn4bRwcxTTI2ttxUeHIJlS5b8AJGPH1fGusc8BCQLSBaQ3KxJ02LbSElJwbgxYz1Oo0aMqNAg2WwyMXX4Q9vhw9+9e/cqMUgWiUQCkgUkC0gWiUQWi+V5s8m0xOLt8y9elwLtKd/HtJuAjAkHkTpsC0IikxFhjcDJEydhVHAggGnLUrVrn6p93LIVU1GXq/bwjes3kJ5S2EW8ZvVq7Ni2HWNGjUZSfHU1s71d27bYsX37A+ss06XMdNr6Mrp16apmx5c0OMgRabNXkPIwXqc1VamwEukg+az2WmkdxwWdscxkRvBZW3PBtNXOY8LMSZMmYf78+ViyZAny8/OxadOmQtq4cSPWrVunYNq0adMwZMgQ9dvMzEzCNd4j0FlbEljFzwgjyyu25xJ3qO4a5vGhpaZEnTp10KFDB4wbN45uWrVtCCtPnTqF69ev4+bNmxxrYMpIii4j1r7ntZLXZof4Husf6t/78ssvcefOHVy5ckUta9++fdiwYQOWLl2KiRMnomvXrmjUqBH3HcGrvj+N2gY6JH6kAgMDkZiYaBRI5nHH44fgtzT7n9/n+ld4kMzlUTzGuF5OlH6OGn+diYrBC33G4sk1Z5wOklmHmbV7WVc3nO2HS4prp6e5jozAi93ewI+XHzIMJvP3T606AdOrPREeGVnRQTInFDUVkEyJzGazxWzyHm8xeX1R9Jkhs2YtLF+2TE3YdUYcPnQYEeFWR3vpScn4YMuWQhD5r9q9RgUPAckCkgUkb926FeWNdm3a6jOJHyZ9QMfTVGFBcpCfciggMiKy2LYaN2zImnEeo8YNGghIFolEApINCAHJIpHI8rKlqQaQL9B97GPx+95ap+P3SYPfQ40pxwuJQNlaszX8zD5Y8c67KE8cOXxYpSyj+zjKHoGJ48ermsXlDaalDqsWXGjAY+aMGcjOaq7gcVxUtOZMnoJPPvnkoYPt3bt0dfyejuXJkyYRmJQIKLD9RvUd96lKEVYbUhISPU163WiC5DsVF6KKtHhCmxTsZbVaL1cmkFwQ5tIxSwdp8+bNMWDAAAXEdu3ahYsXL+Lu3buckMJzl+cg09I/MuU9J4XwNx9//DHOnz9P8EyoqjJxZWRkKCjIdh/mUCaMI4SmA7dHjx5lVuvWrTmh26nuUK5Lenq6Asfjtevs5s2bud1UBgZCYAJhZwe3+XfffYfPPvsMV69exe7duzF37lzlYm7QoAHTQhuS/prr2aVLF7qpi93u3bp1UxMJDh48aAhI3rZtG/r27Yvu3buXeN/z2OncuTPq1q3LfcR1rpAgWd9fcXFxyMrK4ro5S9yvPO9oMDH+mmOzw9S2N3685EOng2TqydyLeG7kQgSlZwpIdhFMDouJwYs9R+DH7xgDk1lP+6dzt8GvThPYKn6Kcv5/0V5zoD4raZwfT9lstmc0HtHEXNVrx4PcxwP69cOpk6fgzFi6ZAmfuxztcnLxwQMHHBCZuvfFF5AQkKwkIFlAMmuSBQcVD08D74NPYyUgmfu3eHAS7IHbW0CySCQSkFyOEJAsEon7eLHF2+efZuU+Ttbcx+ORPv4AoXGxim49HD5eZgzo3x//+c9/UNL4+9//joXzF9AtTHiJ1tktsUWDGoS0RsTMP/+5UD3kWhpUIDjme21ffRXbt21Tg/4PCzrpateo6fg9Bx7Xrl37yP4RyEyfOg3x0THcLoX60L9PX/zn3/+BJwYnZwpIrvjS4kkN5IRardbblQUk6y7NhIQEpkpW0HHPnj1qsgmvOQSTRgfrvX/++eeEdnQsE1rT+fzAlNeEe+fOnWNfCKXLJKa+P3LkCNeRyzc8/Tf73qRJE0yePBn79+/n2IEB0NhYuEynM0H+ihUrOEGAEF8HGqVeZ0426NevHycJlXi/GLQ9uJwyHQs83qZOnYrY2NgKCZJ1iEx3N1OdX7t2jeulbwvDxOXxfOH5P2/ePGYicEp6awLBX05bpwBhlTznwmQCzGdWHFNgMzQmXmCyS2ByKMIjovBij2EFncnl2NeX8duRCxGSkApbxd9/fFYeqAHF3wpUfbzk6+vrbTGZxppNXp8XHZ+qlVEDy5YuxddffQ1nBq/xBNV6uzQQDn3zTYLjQuL/mRICkgUkC0gudAO+ffv24trhDFPOZvUIzZg27QeA/IIlnPUDtIe1pzRVKY9cCZLTklNYF47OjUfWkPI1W+jkcKf0gTm+/lfTHUqDQ/ECyEQikYBkCQHJIlHJ5F3Va6a5qtf3ZpP392G1232fNGgtAXGplNBnKQKCwlAns44auC8ujmqwpE+v3gj09VPpnsePHYfbt2/DqOB9edfOnQu5iHnfGG2PUDWKP9Fch4+K/VoKVltomGMZNVLTsGf3bsKlh7rQWEO5Y7v2qq3EuHjUqVXb8Xu+t2jBQrgiBCQLSLZarSGablVkkKxDKa5DrVq1CAXpFCY8KmYyh/MGNS9duqTAW4sWLZiGuRBMbtq0KS5cuIDyxsmTJwnhDAPJXA4dznSGzpw5k31UANHTg9dTOsT37t2L119/vSBQLhEw5He43jxmmF67ggTHczhpgY7sCgeS9QkLaWlpXAeep04/H+moZ2YCtms8SA5DcEIafvfWbA0QXnS+K5lafwU/n70V5qwO7IPUS3YVTI6MVgD/R8sPEgaXeV8/mXsBL3cdjLDouMqwbficPEBA8uPjPvbx9m5kMXltK+o+ptmMk2BPnDgBVwQzlTSsV9/RfmhQNSxauLAoRFaTdiUEJAtIFpBcNDjTsLh6vUyZ4zEzaWfPnOW42Pm/5B+sqYoRciFI1oPpBAlHim2PD7rujjatWt8f8DJd0lSlNBKJRCIByQwByY+7RCKzybTacQ8bGoO0UTsJh0ut1BEfIDQmA9ZQq6ppVTD+8Y9/YPHCRaiRlkZQiZbNW+D9TZsMB0IE0hkpKWpdAiw+jjrL27Z+oOBESYIuaYJffZu81qYNa3A90G3NAY8/T5+uJmISWHfp2BF5uXmo90odx+8JpA+VOGWqgGSRgGQdDDFlLesds7Yu3aruDk4kuXz5MqZMmUK4qfeTIJlO2vI6cjlQawhI1iE8IRtTdBO68fpV0YLbhP937Ny5k6mfiwD8koHkL774oiKBZDqSKxxI5u8ppiVftmyZ07Y5jwf2mxMMWF+brvOiNaWNdCQTML7U9XU8veIoquRfdTpIpp7QU1ynZcIaJq5kVzqT/9RnvOYKP16mFNf8/k8W7YFv/SymRReQLKoQ8vPz87KYTGPMJtOn/5/7OD0DSxcvxldffQVXBe+BoiMiHH1Irp6gnhWLQmSWIZEQkCwguSLJZPrvo0BydEQkU8cZIusj6iTzYQ4Ckjkjhw+IhmjcmDEI8g8otj0+JAhIFolEFVkCkhkCkkUiqYnstbbg9cfHJwBJg9eWCSZnTDwEW2Y7+HibCY7VA3/f3jkI9PNDpNWGcWPG4tatW3BGzJszV82cZwo0e1g4Jk+ciI/v3kVJgykz6ZQu6CIeNXw4zpw+zc8KDWbv2bUbnTt0VN9JiItnGm2VYu3Y0aN0WRdKA3f3LyXtg4BkkYBkHQKyXnBeXh7u3bsHTwtC7Q0bNjDdNfvMlNEeA5J1F3fjxo2xbt06BWIrenASECcJzZgxg3WPdZjsZpAsIFl3Ievp3XlO8Nxw0jHA7cNrArMCsF3KuYAxNBTmZu3xs3nbCZJd5kpmiusXeo1EWGx1cSW7SLbQEITGJ+IPQ6bj6ZUnSg+TN17T3OtzUC0xTVteuIBkj5UoKSnpad+q5gYWk9fWou5jPqsRRPLZzdWxcsUKBPj4OvqS3SwLB/bvLwqRVbYLCQHJApIrml72+t+jQDI/N6rmbWDxQJMzlCs1SH7hhRd+pqBCUFCx24EAxNhaw+Zi25s0aZKAZJFIJCBZQLKAZFElk4BkJZM3YjtMKRNMjmk/AaGpzRBg8VXLatGsGTZt2Eg3n1MG3ufPm4eo+/C2ZVYLfLBla6kzFjHddb3MVxzrHx4cgqVLluL0qVOOVLAExQTG1WPj1H1yp/YdCJQd9VlXvPMu/C0+jmV06dhJZVcqWQhIFglI1l3II0aMYBpmBY88NXg9Y4aujh07EiSzRrJHgGTtmo6WLVsq1yavg5Uo6Ixi/WQdJrsRJAtI5m/07dypUyd1vHHClTOCmUsID+bMmcN6yHQhF9NnY9NbB6bXVnVvq6y74BqQTOVfwc8W7ILXqz0Rbo8QmOxCZ3JYXAJ+P3hGqWHyk++dQ9V2fZWL3RYuINnzJPI3+Zt8vLxGmauaPtGEgmK2KE7+dcfEM17f33z9dUdfmElqyKDBhMZFpSaUVdIQkCwgWUByWPVMhKZmGaJq1phi29qzZw+MDwHJPo8AyRMmTBCQLBKJBCQLSBaQLKokEpBMeGxrqBy5DtnqdS1ZWuvhW1AtrjYsFm/1O38fP4weORI3P/oIzgimzu7ds6cCt/bwcEyaMKHMs9SPHD5Mt7RjndOSk7F182ac0iDyt99+i7179jJlNVNXM4U1U1kzpXUhoDRk4KBCgyBTJk1GyUNAskhAMsFQnTp1sHz5clUftwIEz33CX6aPdjtI1p3IdEkTNFbS4EA3M6Pp28lNIFlAMrc9+9m/f3+cPn3aaZMWeI5xUgknlyQkJLgMIjtks+P/dBmEp9896jKQTD2htfWL6fnwa5hNkCww2UWiCz00IeW+M/m4I2118W7k6/j5rC3we6WRx0FkAcniPjabzPXNJtNmc1XT/y1k2PP1YxYmlUnJXfHlvXto1riJo0/BAYGYP3fuAyHyrZs3UYlDQLKAZAHJST1ncVDJENlrty4BSBaQPLxLIMb3CjBEjdItApI9WCJRfHz8bzQI2tJut0/VtErTDu283FlqifYLSBaQLBI5JCBZ3XvHdpwCi9nnh+tRbG1kTDioPiuq6FbD4BsQDLOXCT7+3rC2syHq9Rj4+/sgWwMaf//732FUEC6xfnFacoqCta2zW2KLBnzLM4C9XAMSBMSO63h2Ng4eOIBdO3di2pSpSIyPV+7jju3aq/cIewoGIUWThg0dvw8ODMSG/A0oXQhIFglIbtSoEfLz8ytaLV+CLl4HeK1zO0iuW7cuNm3apJzclTm4vSdOnMj7OG4rAckuBsnc5hqgwOjRo3HlyhUeu06rh7x//350796d+7r4eshOdCX7NcjGszPylVO4Sp6rYPJlPJl7Ab8btQhBGa/AFhYmMNlF4ramM5kw+Zk1px7tTF5/CX8cOAUhcQkCkj1DIm/vl81VvUZYqpo+LjrGlJ6Symcpdc/izjh75gzio2Mc/UqIjcXG/PwHQuSPbtyAhIBkAckGS0CygOS/7rIBx43RxJwAAckeKJEoKirKRzv/1tjt9v9prxAZIQHJApILSiQSkEwlDlipAeIwxzXJr5oVKcO3qs9Sh21FcFwmfO67j4NSAxAzLA6p+TWR9n4tpcRFKagWE4iYmGhcOH8B5YmjR44gp1cvNXs+yh6BCePG486dOyhPEFYNGjDAsX6sqzz0jTewaMECtMjKUvCYAxzTp07Dp59+igcF017HRkY5lpEYF69AUOlCQLJIQHL9+vVZX1VB2UofxoNkHeyxhrAz0wtz/1BM2V9UbJfitZXf5QQfrpPT6uWePXuWKbwJFwUkuwgk6zWRMzIy8Pbbb+OTTz5x2v7lxDFOLMnOztaPcb66XASDIfFJWrrjaXjivXMudSWzradXndQg5WQEJ6WJM9nFNZODk9Lx3MiFeHrN2WJh8o+XHYRv49Z0r7u1zwKSRT5VfQLMVU0XHuQ+ZhYnZmDyhMh97z3WY3b0j67kD/fufSBEvn79OjwrBCQLSBYJSBaQLCDZ8yUScbD/9aIAWXMlq4f5pk2bcmChjBJFWG0CkgUkF5BIJCCZSh2xDYERqT9cm7ws8PEL0r5n0l69YX3NSmBMcPxApayrgbD6ofD39UFebi5KE5wpv2jBQs6cJ9hEqxbZ2Pz++wpQlDdY67hR/QaFUqm1b9OWUFi11U77e+eOHcW6+tauXsOBGccymjdpUsoZ8wKSRQKSCaZY83TlypVlhMgCkrkN+TzQpUsX3Lx503B4zPT+vGYS3NIdum3bNuTl5SE3N7eQ1q1bh40bN6qxmUOHDuHixYsKNH799deElkanPuZyedzo0NXJIFlAMr9DNWjQgNvdaW42HnOsgzl79mzUqlWraCprN8kGr9Y98JPFH6LKhmsuhcl0QT+z8jj+lDMaofGJsIUJTHalMzk4pQaeGz4fT609/UCYzOPh+WFzEZKYCps1XECyyK2ymLy2FhxPYganBfPme0y5EN4HjBw+AgX7OKBvvwcCZOra1avOmZAmIFkkIFlAsutDQPL48eMFJItELhBrm9jt9pUFz7/evXtj9+7dygEgUf7o2ztHQLKA5IISiQQkU5OPIaL5YPj4BsJi9oJyHycH3ncf1yAsLpHs3SMUoB0+dOgjYQJrdbFmFyFtpM2O8WPHqUFlo+LkiZOIiYh0XGtDg6oxtbV6b+rkKY90WLH/w98ayt86NKh//1I6swQkiwQkE5ay5unkyZPxzTffoOwhIDk5ORlLly7lMgxzhLIW8fHjxzF37lzk5OQQYBL6K+czjyv2sai0zElIT09nrWvlFmZK4pEjR2LFihWsIU34aChQvnTpEmtCs20ngmQByfr2Zdvr168nxHdWmnhOQOAxU6QesruBYigCatTDr8e9gyq5F10Lkqn8q/jJkn2o2nEAwqLjYBNnsstkDwtBtdRaeG7UYjy5togzWTnGT8D71R4Ij4jyhP4KSBaQfKTg8wmfazwlCLNbNm/u6FuQnz9mvz3zYRC5jGUTJAQkC0gWkCwgWUCygGSROJFn6OddzZo11aCOhLFx9epVpvsRkCwguYBEIgHJ8T3mFrouxbwVRzBcJsVOSIB/oC8aN2qIL+/dQ8EgsFi8cBFqpKUp4NyyeQts2rhROZOMjNWrViHA4gN1r669sq02rVtj+7ZtBCclGuRv0Szrh+u0fwDmzJqFu3/5CxgCkkUCkksOP1nztH///oZOFCkOjHLy5VdffYUvv/yS6eoL6bPPPqPrluc4IRnTM3MAs0KAZLqRu3XrZth25LqfP3+ez/rKEcp9pQ1QIzg4GCEhIezbQ0Xwp4vf5+8CAwMVYGb95qFDhxKCGjZx4N69e5g0aRL79UCQzOOLIJnbtgQyen+WWjw2p02b5jEgme9T8fHxahLzgQMHeB455djnubd161Z06tSJx7SestwjRKdpeEQkqrbvhx8v2a8BxCsuBcm6M/knS/fjpR5DEZJA96vVDTCZAPsBslKV2ZkciuCUmnhuxHw8vfoHZ/IT66/it2OWoFpKDY8E+wKSBSRTvLfn9d6dcfHCRSTGxzv6FBcVzUxVD4XIly9dUvdtHhQCkgUkiwQkC0gWkOz5Eom0B+mm+jnXsGFDNWAiYXzoTrI9mss7MzNTQLKAZJFIQLKm2A6TCl+bvLwQ9UbZYXLi0jQEx1dDVIRd1Rc+cfy4euBlnaxIqw3jRo/BrZs3nZIqs2P79vD1MqtayPbQMEyZNBkf372LkgbBSvXYOMe2iI+JRX7e+jLWahaQLBJHclpamsqu46ygu5HgmOfomTNnsH37dixfvlzVER41ahSGDRuG4cOHOzR69GjMnDkTq1atwq5du3DhwgXcvXuXYJnXEI8EyYR8KSkpdA0bdq2kc5jQkMsnDGZfynoc8XcUl6Evq3HjxkyBbcjA9nfffaccstHR0WrZRdN9d+7cGUeOHKFz+VHipFLD0jUTCF++fJnLLbH4/aNHjxK2ewJI5nv68cVzhccF3eROSXX68ccfq3OzUaNGeh1mz4OJWr8CatbFb8Yu092oLofJBNisx/tC33GollbL6TCZ66zEa0A4pf1tpcIK6P53+HmY+rvSwuSgWvXx27HL8eTaM8ol/qN3DsO7VVeER0Z5FEwXkCwgOT46BtH2CMczyyu1aqtnHnfEpg0bOfHW0ZeG9epj7549D4XIly5eVP8vPGYhIFlAsoDkxB4Ckl0Nkr/cKSBZJKpMatKkyY+0c+2mJg4qOd+xISDZMSg1ceJEPjy5CCSLmFY2IS4erw8egntf3BOQ7KESCUjWFd4yvEwgOTW3JhIXpCKoegACfHzVsrKzsjjIQOjjlDSh06dMQWhgkMPpvG3rB6UeoMjXQAVTsOnr37hBA+zbuxe3bt1C6UJAskhAsu4WHTBgAO+5nAKQeV/HOr2EU2ynadOmyMjIQGxsLPugnLL+/v4ICAhwiJCO934Ea3TiMjXzm2++idWrVxPw8nrCa4enOZIJZpmtyIj2eU0jNKQb1Bkwj8ukuG25roaAb4JiptImqC4Ksbktue9btGjBFNgPFeEsobMRExvoosrPz0ebNm2QlZXF5ZdE7CP7ytTgOrh3G0jWITLPA6ba/uijj5ziDqP7nelL6cJm1q2iqaw9DSQTGL7ceTB+tPwQHcIuB8kUAeYzK0/g+Tf+jIDMRnRKG5jqugAU5vJsNoTbIxEWHatqAFfLqAP/V5rAr34L+DbIVvJr0ByBtRsiKLUWwmLjtf5EKMBtDw+rXC7d+0DdP7OxBpOX4qk1Z/DcyAUITkr3uL4KSBaQzOcUZuPIrFlLf3bhhF2CWpdmWWF5IratK6dnL5YwehhE5gQ+gcgCkssvAckCkm3pTYptq2uXLizY7nY1adjYcYH0ecknSlMVAyQgWUDy4ytJaf2afr4tWbIErgkByXpw5j0HdAQku1bRERE4euSIgGQPk0hAsr/Nh68OBSUFInV9zRIB5Jhx1eEf4Qdfb9ai90J4aAjGau6/m9rgtDPi0MFD6NW9h0pfzYG/yRMn4u5f7pYJCLCfBde7b06OGgT5qMx9F5AsEpBMqGv0ZHAOWjJNPl2dfE6lu5HAmmmPCY71tMxsXweNRcXPCLMI7Pgb/k34TMg3e/ZslZng22+/9RiQzPXr1auXEXWA6d5mLWOmMWb7Tt3/hJdvv/0217nc24w1dbOzs7nPirbF/aen1y5WnFTAPr333nuGgOR58+ZxPbns0kpP0+02kKzv+/r162Px4sUq5bszgpNIeK4OHjwYcXFxHg2RdRGwBtSsh9+MW44n3ORK1p3JT609i19PXAXvll0REp+ow9tSQWUdGuuy8/y0RyAsJh7Byenw00CxV9sc/KnvGPxu+Hz8avIa/GzOVvxk4R78ZNFeJf79i7c34TcTluMPg6eiavu+CKjTFCEJybDaI3QoXalgcoAGk59/ayYszdoR5AtI9kgJSGb887t/4rVX2zieYTiJd83q1XB28H6ibetXHe0Gau3OmDrtoQCZOn/uXPmzv0gISBaQLCCZCk+qW2xb/vcdDR4kOpKTKrIjuUNDn0eC5BoZGWjXtq1bZQ+3ykVSVGmlnWeb9Nn0//73v+HaEJCsOx3mz5/PlHluA8kWLzN8A0Irn/xDYDZ5PXCdw6oFM72ggGQPkkhAckROFGyd7fzbId8QC5LeTX8gPE5ZWwOhWWEI8FPnOWrEWzApJwDVfL3RplW2GmQwMv72t79hwbz5SEtOIQhF6+yW2LplS5lnthNIcRmOQRBfP7w9Y4Ya7Lhx/ToYApJFApLLVhu5T58+hl4DCO94L0cQSuirQzcdHJdlO+i/0ev98l6wS5cu2Lx5M68PbgfJ/A4h4PTp0wkFy739CMlbtWpFkOkSR/rrr7+uHKnljRs3bqB9+/aEsA/dj48S+0QncF5eniHH4qJFiwhH9fZLK7eBZB5TTAnOc4hQnS58ZxznvMf/4IMP9HrIFQIiUwqI2lWtZPxo6X7dlew2mPyE1v5PF3+IPw6cDL+GrRGcmArlULbaNBF6FgbFlP6egs02mya7chwHJ9fUAHATeLXujhf6jdPA8Dv42fydmvv5OBQw33gDVTZcv6+rRXRN/1z7/kn8fM4HeP71mTC37KwttwbCIxVQriQwOUxB+5DqyQiPjoHVZhWQ7LESkMz45ptv8Nbrr8NSYNyFRjhn1SBmmYjUxCRHW1H2CLy3Zm2xEPnc2bMqm4yEgGQByRVUApIFJDfOsDwKJLM/nrPNTaZ/a6oiElUWJSUlPa2dZ//WpGZquycEJOtx8+ZNtGvXzqUg2T80BtV7L0DGhEP8v7BSKn38AcR3nQW/IBuKrv+YUaMEJHuYRAKSlbt4RBwsFi/H+xaLCfFTExwAOXZsAvwjfeHnbUKA2YT+bfxwemWo4x7zal4YUmP8kRIfiyuXL6O8ceTwYfTu2VOBXg5WTBw/vtx1i9mv5OoJjnWMiYhE3rp1arDj2rVr5XDRCUgWiSOZ4Irpog2GyKwTzHquBKFOS8tM4FWvXj2sWbOGINytIJl94frm5uZy4mOZxck2nIizbNky1uZl204HyVFRURg0aJAa4C5vMB13165dCZLLs29Zs9tQkMxtadx2cz5I5jbgZIkOHTqwnrhT0s7zWGPd8XfffZdZn4qph+zZruSg9Ew8N3Ihnlh3vogr2T2prp/MvYCfLtyJ54fNQdUO/ZWTODi1FkKrJyI8Jh7W6Bil8Kg4hMVVR1h8Iqql11bf82rTEy/2GY3fjl6Mn83bhmdWncITeRc1MHyNYJjAunTryO9vvI4ntdcfrTiC50Yv0YByFwQnpBBwVw53criqmQxrBTh2BSQLSGZ8qt0jzZszp1CtYmb3/Pqrr2FksHxQSOAPnKBeZiZ279pVLEQ+e+aMmlQmISBZQLJxEpCcXK/YthImpXDgyu2y94ysVCDZ4uVdbHsxocFIDrC4VYFeXgKSRZVS2gDLS/q5xvpu7g0ByfoA39q1a5GQkOB0kOzjE4C0MXsqLUAuqpThW+HjU3ib2ULDBCR7oEQCkqnqc5LhF1og1bXJC4FJAQjw91b/To81Y/moIHzzofWB95l8v0OjAAT7+2HTxo0obdDNtHD+AqSnpDrcx1s2bzYkJRqXE1zgGl6/Tl3s2b1bDXZcvXKlnBBZQLJIQDIBEid7GFkLnSVgkpOT6Rx2OgSlmPZ3/fr1zBjkTkcy15kpt1kPulzauHEjASIhvEtc6YSsI0aMoAPJEJDcrVu3coBkAcl6CvecnBwcPHhQfWZ08FxhPWQ66IvWQ65wstnh3ao7fjp/B13J7ofJbD//Gp7QnMHPrD6Fn87bjl9PWYvfjZyPPw6YiBd7jMCL3YfiTxow/sOQGdr7C9XnP5u7Hc+sOIYn+PsNChwTTN9fn/Ks00VNaplKP1l6AH94fQb86mXRxetimCwSkCwgmcHr7/q8PMRGRjmecZjF6SMtq4cRMW3K1EKu5+5dunKyb7EQ+czp05J5UUCygGQByY8vSI54zEDyyFoR+KpTjFvV0uYnIFlUKRWhhX6ubdmyBW4PAcl6sFYYUzI6FSSHpmU9NhBZl71Jv6LbQXMsrcGmTe8bqnnz5nNwrxwgWSQSkEwlzE+Fn91XfebvrclsQr9XfXHyXeU+LpGmDwiEj8kLY0ePKlGKNdYm7tOrt3IfR9rsGD92HG7fvm0YxJkySa2vQz01OMGa7RzsuHz5st5HAckiAcnlcKIypTHPN6Pg1N69e1G3bl3nQ9AiMJlpoE+dOuXWGsncv5zgSAiamppaJvG3WiYkppt2iTuUbdBJzfq7DPeBZAHJCxYsYE1sdQwNHTqU6c3pGnZKPeQjR45gyJAhOriu0ICNrmSmkf7j4Kl4as1punAJUD1DKg31dU03FMR9av0lPLXuvNKTuRc1aHyVbmN+XtRx7FzXtPb67MxN8GrXB6HVk2GzCkwWkCxyJUjmxC2CW06OpVPYUVIsOAQf7t2LsgYzi3Rq3+EHFuLjgykTJxYLkKnTWl/+9a9/4bEMAcn/j733AIviXN+4TXJ67+3fojSRjjQbdhEFu0ax9xJj7z3G2Cu2E+yIYsXe0YgF0HgUNWLXJJZjLMmJKSfl+87zzT1f3r0WhAWW2X4/13UHszM75d3Z2Xfe33s/D0EyQTJBMkEyQbKRoihtMKeW+q4dPXpUnCcIklUg7VujRo1sApKrthzocSA5dlDKC+3QuXMXeWPQEMOFQVqCZIqyHiTX3xEnPv5epmWT+laW51nhVvU5j/49SEKr+Ein9u2QXrXIWsVrVq2WuAYNADilU4ck2bd3r6G1tD5//lx6de9hOh9/H19ZOH++abDjan6+ARCZIJkiSAa0MjKtNZzNEydOhBPZ7ucCB+esWbMwiOowkAzh3AFRtUFkq6XqSeN6sQV0V7WqARABrFFrOj8/nyDZwSB5xYoVeqr2GTNmoGwDjt0m9ZAPHjyo6iGra9vFFaa7an3adJVfLtkDEAs5D0zOyH8x3bRJ18zXsY/MHMo/XpctfxsyTQLrx0tVwlq7iSCZIBmBkhz/OHdOdwr379NHsC7kW8lLVq9cJWWNu3fuSKP69U3biQwLl83p6SVD5AsXVOYLhr1BMkWQzBrJ9gfJBMkEyRRFkEyQjHiuwYfJkyYbDpLDm/byOJBcvdfsF9qhR4+eTgiSKYogOXZlPdPrkF+linJoaZDV/c67u0Olcc3KUismSi5fviwIDDTgYbiKn59EhVeVmdNnyAd374rRcfvWbaR2M51LdNUI2aKBLuwfyr9yxRiHFkEyxdTWcMAa5uJFTT24kTGpD4DSUWm6z507Z3+QbD/ntTkMLosUiNDhIWrv1qhRQ+Lj4/WsPqdOnTLkvkqQbH3ACbZv3z44w5HZw/CyDSg18eDBA0nXoEKHDh3M6yG7harifKrVkP8eOFF+mJpTSphMwZ38w/Tz8oeJS8W/aUukCbeTO5kiSCZIRty/d8/0jDNj2ts6RMZ7oNEjR5Z6oi5qH4eaGSqaNIqTo0eOlAiRL5w/L1988YW4YBAkEyRTdCQTJBMkUxRBMgYSnjx54slCKrdyvf/ggYNSt1asJHXoaAhIDohq4HEgObh2i4J1oitWQio8DBQbKtQCNLZGMkURJEPemuYO9be67/nlqXDp09Zf/LwqSc2Yavo2O7ZvL3t371GDGoZH5pEjEhoYZDqHhPgmciwz0zTYceX991XdZYJkiiDZACjZpk0buX//vhgRT58+leTkZIfVWsU+AX2XLVsGKOpqILkkSIzXixTgMCApwDBSIiNNdcOGDXVI3KRJE0lISNBrSLdv31769OkjQ4YMQT1k1HLWweXdu3cNg5YEydYHrlnUF4dDzRbx6NEjeeedd8zrIVtwvLtuiuuABk3kD2+myMvbLxMklwEmv7IjX347c734NXtNg8mRdnInUwTJBMn4/UWmJfWss27tugLPQu1at5FnWv/KUixbshRjNeo9SG1dYj1k6LwGkZHFxUWCIJkgmSJIJkgmSKYoguR//vOfkpqaKj179lTOTANFIdVheUEyFNNjpsdA5KgO4144/+YJiWKLeO+99wwEyRRFkOxbUf+r+mPSrZmvfJNdtn7n7kWBMrKrvwT4eElYUKBMnzZNT5dmy1iSnAxgqo4dELXAIMj7ly8bDLAJkimC5H79+iFdvSHQ9datWwCISM3ssHNCmuYRI0YAyDk1SC4MjHFdKKcwpKAwoF/Tpk2ldevWOghOSkqSXr16Se/evWXgwIEyePBgGT16tJ4Kee7cuXpq5A0bNsj27dvl8OHDcuzYMcnNzYXrHJM08Rkh/TjSGxt/PyVIdurA54K082bprN1OupM2LFx82vVA/V9AUsLkUgpptl/W2uo3c9PFt1VnCY+MlnCbTwiiCJIJklVGFziD1TMPSgbVqVlL8F6oVvUakn8lv8iSCAP7DzCt5+flLTOnT8c2StR5rX/zHBOXGM4PkgmSKYJkgmSCZIoiSMbA3Zw5cyQ6OprA1wVAMhRYo6lEtR8j1XrOdktFthslAdENizz3o5mZBMkU5QIguWuijwxO8jP9v7cGk+tEVZLHmZZrJt/YHiqJdb3Ev2JFqazB6FaJTWX3zl364IYtA5kOFDCFKnv7yOxZswoMdly+dMng4yBIpgiS8RcQ8uuvvzYkbS6AZWJiItyODj2vTp06ydWrV50OJKuaxNiWqumM1OJwDHfu3FlPMT158mRZsGCBpKSkSFpamuzcuVP2798vJ0+e1PtLFy9e1IHgvXv39Aw8AMJwsCItJQaU8VnC2QrZPwiSnTkwceD06dNwpaO/7aawTaW4ri7/02+c/Hjt6TKmuKYAk3+ZvEt82veSsOgYCbd5mmuKIJkgGfHJJ58UePY5eeKE7kbG+6Eg/ypy8MABUYFJYU3iGpuWVw0NlQ1paaWCyOjbYCyS4aEgmSCZIkgmSIamNiZIpihXAcnXr1+HwwDbMykuLk6fKb548WJZt24dVQ4tWbLEQJBM9e7RUxDOD5IpiiC5R0sfve+4ZEwV8Tb7Hgf7VZRLm0KwrIDmDK0iof4VxQfreFWUibUD5LVwPwnx95cjhw/bHDjEN2xkOsaIkFBJ37CxwGDHJQ2cAI4YHwTJFEEyUhwDtBlRH/nQoUNw06ptO0QAtQCz2dnZTgOSlfMYTlC4jFu2bCkDBgyQWbNmydatWwGJ4RZGinGAYbSl4fVxCZIJks3qMOOY3Rq4IcV1UN2G8qdxi+SVLXlSYdf1ssBkas9NzdG9Xyol9dFgcjXCZIJkyg4gWf1+mj8DvXf2rAwbPMRscnBFSV64UIfMVUNCTa/H1W8ghw8dKh1EPndOPtWgNYMgmSDZBUSQTJBMkExRBMmXLl2SmjVritpely5d5MyZM8IwNF24ASCZwsPK3NmzMQhHkExRLgaSocNLgyTIt6JZrfNXZeusQLm1I1Sa1fMW/0r6MmkR4iPpiaHypJfen5NP+1STqXWD9HvAgnnzbQI1TmRlSXhwiOnYGjdsiBrJBQY7Lubl6QPftgmCZIog+e2338ZvvBGZBZBO2eEpc7H/Ro0ayYEDB5wCJON9SLeNyaKAx3//+9/l1KlT8uDBA9QltKdzmCCZIFllxEJJJdTQVtkD3FJw51VObCu/mbVBXs64QpBcVu2+Ib9YfkC8OvShM9lOIkgmSMbv1ZX3338B/i6cP18q+/gUOV7To1s3ycnOLjVERukPBkGyfUWQTJBcp1kpQDJB8pOjBMkUVVAEyR9//LE+mKS2tWzZMmG4Bkj28/aR0OBQXbNnznJrARxtTNugp0xSQZBMUa4HkqFr20MkNsLLtNxbg8m+WrrrIA0iT4gNkAudiu8/bm0eKoHe3tKzSxd5/tlzMSpSNJjiU7GSmLIedO8huTk5BQY78i5c0FO1Gh8EyRRBsqrLu2jRIsPgFGCdM4BkAMiMjAyHgmSAerRvw4YNZejQoQCiqE/sEdkVCJKdPzCRASnU69Wrh2vVbVNch1eNEO/XesovFu+WlzJYL7nM2g1n8l7x7tAbMNnGNZMpgmSCZJU5AvWLC0PgTenpEmHmQvat5CVvTZ2KZaWGyCiJwSBILpUIkgmS6wyiI5kgmSCZohwNkjGYpLazZs0aYbgOSA7095c6derqYhAkU5SrgOS3Xq8sVbwril9Ffbk0D/aRjSb3cck63zFS6gZWlno1asiN69fLnVZz8MA3TMfqV8lLpr817YXBjgvnz+sOR9sGQTJFkGzUhEakZH7nnXecAiSj7rAjQTLAXHR0tJ5xKD09HSmrdZjp/kGQ7Epx+/ZtpLZHliy3hslhUVFSsdtg+VlKZjnqJdOZ7J3Ul85kgmTK9iAZAeBbJAxGyaFWzVtI7Ro1Zd3adXit1IKhxaOCIJkgecuWLT9dsWJFHe0BZUhKSspb2t8lWmqgjZoOav8+q/29of2944nyfrXSfwiSLYPksaPGPjSqvZOTk+8SJFsGyT6VKv3Hnb5jFDV58uQH5QXJFy5cELWN4cOHC4MgmVEQJI8ePfoRvm+eLopqVL/BF0aBZL9K+uu6RtWsYlX/7n6PGOkaXUUCff1k3969Yk3cv3dfEuKbmI4lLDhE0lJTi4TIX2gpX01BkCyhgUHfue71TGmw9o6WQvperVq1vnW21NazZ88GMDXEkbxq1SqPdySjXQEsBw0ahBTWruRARpptyDNAMkEyrm88m+JaBaBy6xTXodVryH8PnCQ/Sj1tZb1kwuRfLtkjPq/1kPAowmRHqU+fPs+0TAIfsF/lOqoeFfO1NSBZTfbBc5ERwtiYY4Mg2d/H9/9xtetX46y3tb+XNB3X/p2h/V2l/Z2jMdmxmjquXLny/zRVMFLlenNaWtqvtANL0A50lqZsTd+ipsyLopAajiDZMkgeP2a8Ye29ePFiIUi2DJJ9vbzc6jtGUVOmTJHyguTx48fj/XApFJEymEGQTJA8duxY3m80UVR8ozgxDCR7FbyXdY/0k0e9Ysrcx/uXphn19LrJMmv6jDI57HKycyQyLNx0DA3q1pODBw68MNCBVG7Pnz8XRkGQHBYU7MLXMwXXL2oRx8bGirOB5DfffNMQeIg09IC32K6jQTLSSe/du9chILlGjRoyatQouXTpkrhCfPfdd3qtZrimL1++LHfv3vUQkEyQjPjmm2/k0KFD0rFjR9yb3NqZHFSnvvx15Gz54Yb36Ey2RrtuyK8W7xafdj0kLCqaaa4doP79+8vChQvZr3Ih1axWXawFyeibXb50qdwQ+eGDB+L4IEgO8Kvsrtf5Rxqv3aD97af1tapoqlAeWfUm7QD+V9MCTV+4Q6MSJBMkEyRTFEEyBmowEIX3Y+Y3gyDZEUGQTFEEyVCDAG+51iXKqr7erpZhEuLrI13at9fT2ZYUa1ev1o7B27Tvbp27SM7p7CIh8meaq9E+QZBMESTjWCZMmACYZAiQOnz4MGAdIKBDQXJCQoJkZ2fbFSSrdNbo41+7dk2cIQBQMRCNZ5Bvv/0WpQVQMkC/bz969Eiv2Xzu3DnRMv+JlnkJbjfZunWr9Q51gmSXjC+++EK2b98uLVq0UNe5m8LkUAms31j+PHah/HDjOQ0kW+VMpjM5eZf4vNaTzmSCZMq2IFndn9EPsRoi3793TxgEyfYUjMBaNunmIvKSpgplVVlWhgs5UNthqrbj78pxwB6pkkEyQfKEsRMMa+8lS5aUDJIJkt3qO0ZRcGyUByTfuHHD9H4M2DAIkl8MguRx48bxfqOJomwBkrtHVpamQT6m9UK9K0lWu6pW9fcudYqUBsGVJTYmWq68f6VosPT1NzJy2HDT/vC88ubkKUUOdPzj3DnLUJog2YWvZ2r58uUyffp0pwPJ0IABA+Szzz4zxNl65swZSUxMlJCQEIeeW+fOnQFz7QqS8bm2a9dOsrKy7JKSGO2twDCyOOD++fTpU70G4sOHD+XDDz+Uq1evysWLFyUnJ0cHnQcOHNCBoTbAJ9OmTdOhRKtWrfTrUqvlqNfK3bhxo1gfBMmuGo8fP8Z9Cm5+9wXJGvREmuuAhk3lz+MWaTDZWmcyncm/XLxHfDr0lrDoanaEyRR+rxctWsR+lQupvCAZ8eif/7QKIn+k9QOcKgiSXfIaLgefvaplme6hjX3/SFOF0qpUK2kb/6G28eSSDkIDzbIpPV1PU5R55Ii8++67clLrsGWfPi1ncnPxRfFY+VT0IkguASSnrkk1rL1PnzrlEJDs40Ig2c/b262+YxSF35/ygOTTp0+b3o8BnReDQZBMkLxq5UrebzRRVFK79oaD5JE1q8iTXtXk9Rh/07q+r1aUdQkhVvX5HvaMkV4xVaSKj4/syMh44R7csllztR/U+IUzuViI/OzZM2EUD5Kjq0a48PVMndUA684dOwDGnAYkKwF+Iq2xEXDz5s2b8sYbb0hQUJDD4HhUVJSMGDECUNVuIBn7rVOnDpznSPFtk/TTgP1wEd+5c0euXLkiubm5kpmZqY+NAf6iPvX8+fNlxowZMmnSJL0N+mr3D0D15s2bS1xcnMTExOjtg2sQ5wMICoAcHBysw/969erJpk2bxKogSLary9wWgc8NE160Wu7um+IawrVfP150Z3L6OStrJtOZ/IvlB8S7fS8Js5szmUJmk+MaB2G/ynUUV79BuUEy4uaNG2Xa7wcoU8FwKpAcXCXAJa/h986elZzsbDl18qRkHT8uRzMz5dDBg/rExNTU1NLA5tsa8w3VVKE0KnEFrcP7N22np4vb4cYNG3CAAHdO15gEyQTJQQTJpQTJFEWQvGfPHtP74ZJgECQz7A2SKYogWfXXFjUMEm+z+9uE2AD5tI91fb95DfS6yTJ18hR9gBvHHx0Radp23djasn/fvmIhMoAPgyCZINkx4LV+/fp6bVwjAt/l5ORkQEls2yHnAwi2dOlSBdvs5kgGsH3//ffFyEA6asDjs2fPSlpamrz11lt6+umkpCRp2bKlxMfH6w5S9HNxXQHc+vv74y8AMSAnoL658NngXCDlpFbgFiAZqa2dFCQTJH/99df6ZI38/Hy9rrUtAveC4cOHY8IBPgO3rZds7kz+QTqdyVZp90355ZK94tMezmRbwmSKIJkgGRPKLubllWqfmHDGIEi2l2DsPXb0KJ5zkPGmOJj8lbasi6YKJcniQm1DsRpEflRUeuqtW7bIiawst2hUgmSCZIJkiiJI3r17t+n9SHPNIEhmOAokUxRBMrS3VbgEVKpoel/7MD+53yPGqv7ffm1bYb4+Uj82tkA95E7tO1icEIx0mgzHgWSKILlGjRqybds2w0DXsWPH4H51GIRq27atnmIbYQeQrNoQA/yGupGRthppqefOnStt2rRRYA9wFlKQGG5iczBstQCYGzVqhPTXTgiSCZKRwhwZreA0R3pbfM/wfbOF+/2k5jrq2bOnREREuK0zGdJhcgM4k+fLj9LOaHCUzmSrnMlL94l3BzqTCZIpW4JkBMpYYAKupf3dvnVLGATJjnze2bd3L/pcxaW7XlZSquviFsCN3KCoWsjpGzciVbUVB0yQ7O/rR5BMkEyQTFEEyQyCZILkEkVRBMnQ+Y6RUt3Py/TeWpW9UP8Yy8qkp71jpF+0v2k7cChPGDvO4jnCacdwLEimCJKjo6Nl/PjxhqXLffDggUycOFFt357nghq/SO0Mt6ZdQXLjxo2Vk9eQ+Oabb+TcuXPSr18/bB+wEqDX3EVs+HWEbSckJMipU6ecCCQTJOPYP/nkE9m/f7/06NHDdM0NHjxYLly4oINfowMTInbt2oUJDGp/bu1MDqwbJ38bNl1+suaEvAQ4agVMpjN5j/jYNM01RZBMkIx4cP9+sftC+mv0WxgEyc6QDvuA1m/R2G9RMHmviLykqUJRKvJF7U3/q+lJ4Y3t3rWLNxqCZIJkgmSKIkhmECQTJFOUHUSQDH3UI1rahPqa3h/oVVEOtA7HslLpTrdoaRbsY3p/FT8/WflOisXze/jwoTAIkgmSnUMAYeg3GQVBT58+jdTLgIF2gVDYB9q1a9euOlxD2Askq/bLy8sTo+Kjjz6SsWPHwnFsF2e3+ow6duyI+stOApIJkjG5A9fCmjVr8H1SKePRNni2AVRCW9sEHHz66afYr3l2ATd2JodISI3a8n99RsrPlx/QYHK+tamu6Uxu113CI6Okqk3u+xRBMkEy4trVqy/s58b164TIBMlOp5Na32jtmjVFweQpmioUpRde0GbS/Vj7e9Z8A8ihjUF53mQIkgmSCZIpiiCZQZBMkExR9hNBMvRJn2oyupbpPinempbHBWOZRZ1uX1UifSqZ3lcjOlr27t5t8dzu378vDIJkgmTncSXXqVMHwM5QCJWamorztQtMxvabNWsmWuY71BW2K0jG54k0wEjTb2BqYdV22L7dnOlIm/zs2TMnAMkEybiOUQ95wYIFaAekMDdNmFCpyAF5AXvxfbNF3Lt3T2bNmiWxsbEeAJND9Tq/lToPlF8u2ikvb7tsBUwmTP713C3i36y9hEdESFU6kwmSKZuAZEzYy7twwbSPa9eu6b9zDIJkZ62hnJ6eXhgk/0djwU1KVSNZe0NKYRJ9zD4QmSD5jaUEyW4Okt8kSKYogmQGQTJBMkWVUQTJSivjQwCRTdsaUt0faauLXDctIVT8KlY0rduuVWvMPLZ4XnBXMZwLJFMEyRAcsF988YVhABb9sGXLlqHuLvZjOFBWzkzUcW3RogXMCQqo2RUkR0ZGIs2wYfWRP/vsM1m3bh22i3O02zWAa3P9+vVGpDgnSDYgtTSc9VOnTgXEBUQu9nuLfe/bt099dw2Pq1evypgxYyQmJsb9YXKYdn4RkeLbuov89u018sP0cwDJZQDK1Ms78uW3M9eLf2I7Ca8aQWcyQbINRJCsJuxh+1fz8wmRCZJdItV1WlpaYWfyJ1rfrJKmCuYy/x9A5NjCEHmPmrFOESQTJBMkUxRBMoMg2WFBkExRBMnQ6zH+6HOatoe01R90jy7gXh5fK8C0HBo9clSJ5/ShBhcYzgWSKYJkBe3i4+P1urxGBly6cCa3a9dOgahyA2UF0FRN5O7du+uw8fnz54KwJ0jGMaCPOWXKFPn6668Na7PZs2fjOrFbWnBA64EDBxox0Ycg2YCJBJmZmdK/f3+JioqC87jElOR9td+J3NxcuJhtkl4bqep79+5tPrnBbQXwCQU0SpC/jpyl1U3OkpcyrpYSJlNoo5c0/XZWmvg3aye6M5kAmCDZcBEkIz5+9Ei/RzMIkl1BOdnZsnLlyhfqJZeU2nqv+Rs2bdrEGwtBMkEyQTJFESQzCJKdJAiSKYogeVrdAGlcr57UiI4xbTPa10veS4qQDzWg3DrUx/S6v4+vLF+6rMTzuXvnjjCcEyRTBMkKJiKNLdyQRsbnn3+OVM26uxJ1XmvVqgUXMfZpDpYtSq2HYwVcwzY6dOgg8+bNA+hVENdhIBnnhlSTBoJku0Fk/MXnsnfvXgxIuytIhhMGnyPO2WlBMtr/XQ0Sde7cWV33pWonpCWfOHEini9tAhXgdtZKFKKGtvkEB7cVUjKjbnJozdryas+h8quFO+QHW/I0SHqdQLmUMPnlnVflNzPXi1+LDhIeFaXaliJIJkhmECR7sN49dgzwuECKaw0uB2qqoKT+AQViBbVySkqK5ObkuOjJEySHEyQTJLslSKYogmSCZIJkgmSKIkhuFhenp6lu3byFmKBxxYoSZVYPOToiEnCsxHO5ffu2MJwXJFMEyUqouYrfZABUIwMg7+nTpzpsS05O1t2vzZs3l/r16yN1L5yigGEmSAUBNsPFjD4cajgjRXbr1q1l2LBh+iAUjhOQWoUjQDKE45s8ebJRIBnOatSSQ5vYpbY0nOhIC67a0h1BMj7HQ4cOAZhju04Lkv/973/js8A1pZzIpRLWxTW6cOFCPP8Y/f1VTmkYgVCLXH1HsW+3r5sMNy3SNP9x4lL5ybrTmjs5H0C5BJhMoY1e2qXVTJ63RfybtpZw9btHESQTJDMIkj1aOzIywIXNtU5TBSX1D7iRU81XxBvZgATJBMkEyRRFkMwgSHa+IEimKIJkVdNoyKDBL9wPWzRrJieOZ5V4Hrdu3jRuUJsgmSJItilUxLHB2QhXrK0CjmeAxlOnTklGRoYsX75cHxwfP368vPHGG9KrVy89je6QIUP0Y5k5c6asWrVK9uzZI2e0drx37555Cl+Hg2Q4pEeOHInzMiyV8NmzZwHsAAhtlsocn3VCQgLaX33e7gqSVa1fTELAeZfWCQ+p9ewBknEN4VqH417ts0wwOTExUYe9gL62iEePHsmSJUswAcQjQLJKdQ2gHBxbV3MnD9fB6A/T37NUO5napYN2eWXHFfnJqnfFK6mvhEVGoz0pgmSCZAZBsocr+/Tpwq7kb7XyI/9ToEayNqvuv7HAHCQDxLEBCZIJkgmSKYogmUGQ7JxBkExRBMlKC+bNE18vL307w4cMAWAu8RxuXL/uGhCZIJkiSFbwDs5fOGIBtewCauHCBPj65JNP5OOPP5aHDx9C+r8//fRTLLPg9nU8SMbnCfiNYzWytvTcuXMBqQ110AI2AgDC7Yz0yficnz17JghnBck7d+4UI+LLL7+Uffv2oaY2XPBwwL9QcxsueLQ5BDd8jRo1UIcb7YXP2alBsoLenTp1wnOqTdK94/tw584dpHJH25hfm57hTg4LlypxifK3oW/Lz985LK9svcR01wV0TW+PlzPy5Uepp+UPU1dpELmPhNaoLeFVCYGdFyRTBMkMZM65c/s2JitiQp+NQTK1ZcsWxYdV1up5mipA+A/cyH3MV9i8ebObNwpBcm2XBMkEyQTJFEWQTJBMkEyQTFEEyS3i4184pszDR2TPrl2lOv7r167pD+UM1wHJFEGyEuoPA4zB+ev2YQBITkpKkps3bwrCKFfylStXZPDgwQCaJbpiLck8DTG2BafzlClT4O5WgN5pQXK9evUwdojPwbBav+hzwgkzZswY6dOnj3Tp0sWkAQMGyKhRo2T06NF6GwHmL1iwQPr3749nCbSj04JkSF2jgwYNkry8PPnuu+9sMtiObSNjgNlEB89xJ4cESVh0NfFt213+OGm5/GR1lryy7bJnO5QB0wGQd+TLD9POyK/npMurPYZKSGwdvdZ0hCZCYONBMkWQHN+wkb7tkIBATB4qsi8RGRaur5PYpImYByb74nUoPDgE77f4HFJYVfz8pFpklPTq3kNu37ot5tGudRt9HRyfCmv2Pej1gfo6vpW8JP9KvhSO+rXr6MvffustQYwdNRr/X6JGDh8hKi5fviytmreQKr46x1D7w77Rl7ERSKayjh8HHzbXbfPU1gDJ681XOHb0KBuOIJkgmSCZolwWJB88eJAgmSCZ4dIgmcrNyZFTJ08WuYwiSG7fsqVcOH/emmNHClEXhcgEyRRBsnJm9uvXDwNsJcAogmQIKaIPHDggRgYgPmDyW9oAaZMmTQDtFCxU7lOo2P9X7mO8DzWm27dvL9OmTZPDhw/LkydPRIWzgmQcO9od44dGX4MYtEYb3L9/H8dq0oMHD+AGh3SnNsDzv/71L1mzZg3aEG3r1CBZuc7x3IPrBu5hW/wW4/PIysqCuxv3M3WsHgSUQzU4GiohNWOlUqf+8ru3V8tP1p6Sl7e/j7TOyqXsKSmsNYB8RX648T351dwt8j/9xkpQ/TgdukeEECC7E0g+fvw4/jqRCJKXLVmKbes6WEQfRHELaEVKipjHtKlT1TII5Wctg2QLAiDNu3DBEki2at8KJEOtW7REn8xQkIz2MQfIsTVq4lxM/x8dESkPHzwgSLaR1q1bZw6S0f/5m6YKFbT/ACR/YL7wTG4uG40gmSCZIJlyWREkd+3aFR0ZgmSCZAZBsssKEBmuHNQfZN+8sAiSO7RuhVSzZT7u/CtXDEwJxiBIJkh2EExG6l+4NjExxNVgMuAZQKxdQDLaCqmSk5OTDb/34TyQ4hvpnfFZtG3bFvVpVapl7LuAcDxYBugZHx8PwKm7a9euXSu5ubmAogosugRIRvrkOXPm6ODXUQHX9vr166Vu3bquAJIVTMa9Bqki5enTp2KLAGTftm2btGjRAsfqeTA5HO7kEIkIC5GgOg2kYrch8tu318qP152SVwCUd5oDZcidAPI1kwP5R2m58uv5W+W/+4+XwEYJaBsNIAdLuNtdDwTJYDlbNm9GfVP265wEJN+/dx/b1jWw/wApHOPHjsMycATUuC8wUU05lRVIfa1NW4vPIVgf/QcIfR2UJcnYts20/3Gjx1gCyVbt2xwkQxvTNlgEyarfpNSyWXN9eUJ8kwKvq3Hcvr16i36cjeLk6ZMnpvevW7vWtM+Zb08nSLaRMrZvL+xKfk13JKNgsvkCEGdHHSRB8hI7g2SC5MeZBMkU5Y4gGcLABkEyQbIrB0EyQbLqn6Pd9+/fb173liJIVkCg1Md85f333cC9SJBMESQrkIdUyGPHjpX3te82IJcrBKDjrVu3IAVNbQ6SIyMj5Y033tD7nrYI3FdRNzonJ0evazx79myAZaS+1tMxA9oOHTpUh8YzZswAjNTrAZ8/fx6uW7unKP/oo4/01McBAQHlugYxmQGppp8/f+7Q6yk1NdWlQLJypLdq1UqfhPD555+LLQJgYtmyZdKwYUN1vB6pqho4hUs5sF5DDSgP0moDr5SfrjgqP9h8Xl7KuKKcu0ou6z6GXtL+/crmPPnZ6uPy21lp8r+9R0tAXKLWDlUJkD0AJENwtu7ZvRv9GrzuQBEkm0PbAL/K8uWXXxYAqlHhVfVlnTp0EPPYv2+fCTDj88S/oVs3b1kEyUUF0ltjefOmCRZAsnX7LgySw4KDAXwtgmSzAEhWab0tLq9VvYY8efykQN9v9cpVsj41FeMlBMk2EjLkmPFiGByWaKoAN3KS+QIQZzYYQTJBMkEy5eoiSIYwQ50gmSDZ1YMgmSDZfMLnu8eOYTlFkKwGIgCISzze9y9fdqN6qgTJFEGyOSRFjdhjx47Z3tFqAIhD7V/UtwVAs1dqayxr2rQp0lvbPCMDjgnnCTiIrBHo78K1/Omnn+I1LHP0hB5Ab3wGgJnlAoz4rqB28YcffkiQbGVt7J49e8qpU6dsWQ8bcAvHrNrHA1X1+/rJej1gCalZS3zadJW/jJgpv1q4Q360/oy8vO190SHyLnOgDDm/8xh/X95+RX608T35+ZJ98ucx88UrqbcEx9bVzls75+AgAmQPAslKa1avlsOHDmGZA0WQnJa6HtvXtXvnLlGRk51ten3zpk1iHt27dP0eMCfpvztB/lUUkC0TSP7HuXOmfQwZNMgCSLZu3wok169TVwIr++v/HjZ4iGEgefbMWVhuAvE4121btppDZdZItqGQ3aAQSL6oQPJ8swW80RAk21IEyfFRBMmUHUWQjBpyR44cIUgmSHaLIEgmSFbavHkz+jpsJ4Jk0+D2eQ2mFHesly9dIkQmSKbcECSb1+Rt166dPoHy7t27gGtOl8oaKXyPHj0qfbXvTL169SQjI8NuIBnto9yzgKieHqg/PH/+/HLXz8V7ExMT8axFkGyFVI1sZBVAinpbTXJALfURI0bgO6DayKNVVYPJkYCrkZFSpXFzzbU7Qn4/daX87J1DOox9afsVQGRVT9k53MqmY9Cdx/rxvbINtY/Pyc9SMrU60Gvkf/qOE//ENhIWXQ3wGA5kwl0PBslKG9LS5ERWFvt6DgLJmFDm5+WNfeipmlVMHD/elD76+WfPC0z0AlfAMqQqRwDO4v8jQsMw6ajI5xDsA+cAtUhsJg3q1hPfSl76ssrePpJ/Jd8CSLZu3wokJ73WXpYuXoJ/QwCQhoBkpOduEtcY6xSQ96sVpXNSR5wTQbJthT6P+f3k/9Uy5P0YIHm92Yu8wRAkEyQTJFOU24BkBSnz8vIIkgmS3SYIkgmSIdRP3rVrF1OXESTr8eTx4yKP89LFiwY4nRgEyQTJzq7g4GAAVR1KHT9+HLDQ4c5XgF4MBN68eVNWrFihp/INCgpCHWG7gmQF7fD5ArYDCLpgAJrC1Vxu4PjZZ59JWloa6jWXFyQDqqKcEH5nCJKtrJeMWtNz586VBw8e4Jq2Sep1nH+PHj0wydrysXuUSzkUwFWvoxxSo5b4tUiS/3l9gvxm5nrRU19vfE+DtZcVxAXAtVca7CL2p6Wt3nlVXtlyUX684Yz87O8H5fdvrdKOd5L4tOokodrxw3UdERwoVfXvgf47RhEkm7R92zbJzcmx83ERJCN6de+h78Pfx1eHxnpa66oReE1/jjCPJcnJBdZFZL17HK/p2rNrd1HPIUUKcBmQN+/CBUFYAMlW7dscJKMPUC+2Nv4fEBv/X26QrH7jly5erLZVQKFBwXLh/AWCZBtq44YNhcef/gqQvNf8RRZmJ0gmSCZIpig3AskQnA9I7UWQTJDsNkGQTJCshEHNQwcPenh7ESQjbt++XeAYL+bl6RDH/YMgmSJIhlSq4pYtW8rChQv13+/Hjx8DgAFQ2RUgf/HFF6jFq6eTHjlyJGAZIDKOEe1sd5CsBKh48uRJl7o34jzhasrOzpaDBw/qbVuewABvZmamebuVO7016nTjOAmSrfveNm7cGOehp0C3RWACAtLJoy0KOtEpAFjAVwVhg+o0EN/WneV/+4+TP0xJkV8u3iU/XfWuBpbPaWD5kryUkf897L2hgV7ourl7uYzCe/B+6AaE11HrWE+3/cpmDRyvz9Fcx0fkN/O2yJ/GLpBXuw+Vys3aS3DNWIkIDdaPOyIkhOmrCZJLFCZ07du7V947e9aOx0aQjJTW2Ae0fes2ycnOMf3/wQMHxDzq1Kylvw4n7uFDh3Ud2L8fcLfIesrqOQQQdGPaBpkza5aEBgaZgO4Hd+8KwgJItnrf5iAZcezoUdN5LV60yBCQbB7Xrl7V6zY3bRyP96ljIki2oZAFz/wekpKSEgCQnG3+4pncXLc4WYLkFgTJBMkEyRRBspmaNWuGQRiCZIJktwqCZIJkpbT16yXr+HHPbTOCZLjUkMYax4fZ567ouiNIpgiSDUiXC2CL82jTpo3MnDlTh4+3bt1CPxjgDW4Ym8BOAMpnz57p/WhAq/Hjx0tcXBxAmYJ8+OswkKxqSvfu3Rsw2SXukShLACC/adMm6dy5s4wePRqw0ZB0xy1atFDtVm5X8rx58/DZEyRbn6Ie6enxXbXZJAfUUIcTPSEhAfsr6hzoUgZYDvkezuL6joqWoAaNxfu1npr7d6L8ccJi+e2sNPn50n3yk7Un5YcbzsoPNv1DfrAlT17advl7yGwGiE3pqAuCY8gEi7deklc25cmP0s/Jj9NydTf0Lxft0BzHK+WvI2bJ//UeobulQ2LrSnhklA6NI4IAj4PpPC6TCJKV1qxZI0czM+10bATJ5rWGUYN48sSJJkftN19/Iypg7MTrFoS0zoDDFmskY2KXgr9VQ0Ll3r17FkCy9fsuDJIRPbt311+r4ueH87MaJCPN9qoVK2XG22+jv/JCuZQeXbuq2snIukGQbBshk0HhySi1Kmj/uGb+IhuKIJkgmSCZotwQJKsZ83g4J0gmSHa7IEgmSFbaunWr5GRne2CbESQj4FQDRP7yyy+luEhdt05StGulsFZo6dLXp6ZKxvbtANJFPphjcALrllKWUrBi0EKth2MqEXoUtf2VKSv0GfjvHjsmnz9/LiXFndu3dTfA9GnTpF/vPjJ+7DhZvnSZhd99gmSCZNd1JwOiIeV1w4YNZdCgQfpvxOHDh/Xv8f379wGWcc8AjMP3vdSAGeviPRgcBdRESt78/Hw5pn0Ply1bJv369QPYU/s3B3wKJKPfbXeQrMAn0vsizS9cuYBrOG9nBMhwk+fk5ABO6NmVMEGgW7du8uGHH0p5A3AadXPRbkZA0Pj4eNm4cSNgshXOZIJktb3XX38d17T++dsiHj16JIsXLy62vSiT4E7WFKKBmCC9pnKEJoDl4DoNxD+hrfi06ymv9houfxv0pvx5zDz53dQV8psZ6+XXizLkV4t36/rFsv3y85TDmo7Iz7V01L9aute07Ddz0vXaxn+cuFj+ogHj/+07Rip1HiC+rTpJQFxzCa1eE8eg9g1wrB1LCI6tXACZIkhWwj0bz5u2PTaCZPNaw0g3jXrD+PeYUaPEPIYMGqSALUDwC8IyaOb0GRZBMsK8ZnH7tu0sgGQr9m0BJAM0m61vNUi+f+++6f2A04VjYP8BqsY0xncJkm2kXTt3FgbJzeFIfmz2gg0PgCC5so+v3UEyQXIgQTJBMkWQXEBDhw7FgBFBMkGyOwZBMkGySjske3bv9rz6yQTJiBLrZ0aFV8X+SlTzpgn6vdo81q1di2WlFgb5iwukXDNfF/Wci4s1q1aXtC+cFyCxRYCOAZyi3os+/Mhhw1WdT4JkgmS3EmBuQEAAAKoOkJKSkmTChAl6rf1t27YBqOq/9VevXpW7d+8CMsINUkAAz1iGPvL58+clKytLr9OPMaQpU6bALYvax2g/7Av7tAgd4Qg2AiTjuMsCkhW0g1q3bi2rV69GWQCncSfjHgSAfPbsWUlOToar3PQZ4pgTExP1tkeUt04yAKyRaY6bNm0qK1euxPWDe7+9QDLqXpcEkg253jBYbSuQrK7L6OhomTRpEuqK22yCA7aNfSDdfFnPg2mwv4fLQYESGeivAd5g0Z+RIqI18FtdQmvE6s7h4Nr1JDi2jgQ2bCqVm7XV1F4C4ptLUN2G3y+rJyE1a0toNQ0WR8do74/Utx0RFCC6QgI12BRi03OhCJIh9AN2ZGTYODstQbJ5rWElTKg1/02GixevDx8yVIoIgF8sR31lTOizCJLxLJgQ38S0r62btxQFkq3fd9EgGYH02uUGyYi4Bg1M25g7ezZSW+P3CxOA9ec2vN6lYyemtrah9uzZU3iMqRscyd+ZpzdgQxEkEyQTJFOUO4NkaMaMGQTJBMluGwTJBMnmffsjhw97WNsRJCOMAMlQTGQkHIeGg2SAu8Lrjhs9plwgWWno4MFSOOBeLryecgSYa8rESQTJEEGy2wnnpVzKENytgJMAV0g/3aFDBz3l88CBA2X48OEAzYBNJo0dO1Z3S8Jt3LFjR71kTM2aNbFNbAvwGNsFzFP7KxaWAY7m5eUZ4to9fvw4jqPMQAzr4/wBBUeNGqXXcv7ggw/k+fPnapDULgFYCEAJZzRAPc4H9a3RxmhD8zaF4Exevnw5nOTIPGG1MHB86tQppLfGZ2gIBMV2APVx7Rw9ehTOabQnBrRt5lJG261bt65YkIxrs23btjgenHe5hN9CTNSLjY21GYDFOdSpU0eWLFkCpz/2a7hw7aDedv/+/TG5xJpzoeAIrqrp+/rK/7+CpGqwJqSdVtKdxN8L/y60rGoIpIFpHRqHYZuQ3c6DIkhWwiQg1MM1vn4yQbICu+bPX9Wjogv8LiIjlFoG6FxUIHOUWmf/vn0WQbLKIuVbyUstRz+jMEi2dt8WQTIm5tWIjikvSEZ2DnX8RQnpwjFeQpBsO6FvXPheMayC+Qtr16514AESJNceuNh+IHkOQTL08RGCZA8URZAM4TePIJkg2a2DIJkgWWlDWpqcPHHihX1QBMkHDxxAumddt27ekqv5+bJw/oICacn27t5TJEju1CEJ77Oo4gLu36IGBD7//PMSQTJmn6vjzb+SL5lHjkjXTp3Nt4XvT5Ez6aEVKSlI76nSuyK1XIH3Pnv6lCCZINkjpOCfAsxVqlTR5e/vL35+fgVUuXJltVyBaFVnFSpTWmnAaCPSMwOIbd++XapVq4ZtWw3ucP5wUw8bNgy1iOG4xj0CEBQOYUMhKMAxADi2jX1cu3ZNd4TD2T148GBp1KiRqnNdJBhFjefu3bvDSY460zh/q4T3YrIZnOTYl5Ep1SGcx8iRI/X2vHDhgn6vffLkCQA2YCYmGaEdAO0VaIbwb9NrWA5QjPXx24BBcKTOhmMbzxG4hi5duiSzZs0C3EV7FenGb9y4scyePbvc7bV582ZMsrB8vRnUhnCeo+409m20cC4og4LJIXAlM8U1RREkKyF7D8rFECQbH5isin1BKLFjHs2aNlXZlYrLLIXfPgVW8dxTIkhGzJ5pcgejnE9hkGztvi2CZMSeXbvLDZLVxOO2rVqZtqXqIrdq3kJ/BkQQJNtOGCcolMFgFEGyE4HkWr1nSP2phwxRWGxTgmSCZIJkiiDZgjCQdfDgQYJkgmS3D4JkgmSVugy1b3Nzckz7oAiSUQu5qBiggR61zuiRI4sEyX179RZrArWMMQig37cr+0u7Vq3VNlHvuCSQjEGTouH08BGmdRppUEjBn6cavFCvhwYFvzBIArARGhikaoNhAI0gmSDZ04TzLousBtcxMTEyefJkHQyWM+AQBYAFXC032FNwHH1SuIGnTp0qGzZskNOnT8v169fl3r17CoSqutKAzMUKy7EegPHTp0/R74XjGA4hPTV1eno6MiQBqsMRjmNQzm6APUttjOV4jimv0G6QTa4nnAsE6NqpUycZPXq07rROS0vTn7ngvD4L59u5cwDNSKuOOtuA+HgN0utDA7Lv3btXtmzZAvCtO3VnzpypX0OAun369EFK7ZLOQ51vedpLXfv2mOChZPmYDPjszc6HoiiCZCVMAsK4OkGyUwQDk8qQ1hoTyMwn9xEkEyR7NEj299UBnyHy9/MjSC4FSD6wJERy1hqjuJpeJYLkKY0Jkp1RqE2xdcsWysW1YP78Mg+CYTALKVMIkgmSGaUDydOnT7fTd5rC5Bj0zw0WBt2Rkspi6jKKIBlpptU6ALRGguQN69NM2xg2eAgAHv6tZqVbDZKfPH5SIAXawwcPTA7AILPfmFEjRuhQyDwAtTBIUbYgSFbKyc526fvt5k2bZOmSJUgp69RgBQDRKDkinS32idTMAIJGBBy906ZNU9s2zA0KmAuhf9q8eXPp0aOHDkIXLFigH/sO7Z516NAhpEsuVpisCtcn6hCj1jHA9BtvvIH61HCbwgmq3N9wzarPxCHXgi0/c+V2h6NdgWVcA4C/LVu2RJ1qpDqHMxrS01C3atUKryHttl5PG25xOI6joqLQVtiWcsmretylArOu8L0x/pgtn4tx50NRFCa3INW+A/oxtnhmBDSSnTt3YrIdQbLbBoMgmSDZJIJkguTAKgEW20EVaDdKBMmuqRNZWbgHUi6uKVOmlBkkQxg0vHPnDkEyQXLJQZCMNHi837iJAASPZmayH0CQ/ELaVfSLwoNDTOukb9hQJEhuEtcYdbWKU7Epx5onJJq2gX0B4IYEBJpeu3TxolUguXAK65zsHFEBVzVeM1dcgwZ6ujUcAxyECIJk4zMouIKWLVsGJxFgldOCZNQyBlBr2LAhnNPlFkCmIyAS3L5XrlwRIwL99wEDBgBQ2qqmNEAlYK+qVYzXATQxGRVtWKwATJXjU9VjxrYAP7EtVZfYUxyh6jxVTW20AdoDMB1Cu6Cd1f8XSqOO9yn4aZhDnqIoykCh7jgyL7jdM+PqVasweYogmUGQbC8RJFP+Pn4lguQRI0agZpchatSgAUEyQTJBMkEyQXIplZCQgPR1BMkEyR4XBMkUnM+AQOwPeCZIRqrniNAwCDDX3NELoW4WUkEXAZJLFp5JCgfgslpeLTIK4PoF0Dt21GirQXKnDh1M6+HaVvHVV18pgF2kcP5LkpMJkpUIkp1GCvbCubl+/Xq4YVHj1Frh/UhZid9zBTntdh61atWSuXPnqokb5Qncl5AGGa5VgEa7QFBVT1kJcNiSsL6ZLIBjilCYoiiCZOcWJolmHT9OkMwgSLaxCJIpDJCXCJJR+8Wo6NKpk0eDZKRXKw1I3rZ1qxw5fNgQNaxblyCZINlFRZAMNWvWjCCZINkjgyCZSkHqMg0unMnNZb/As0CyRbA6ZeIkHcAijALJ2KZaPn3aNFGBa0+9jjTUSDVtDUjunNTRtN7SxYsLgyccP9JnF3vMGEzAegTJBMnOBpKR9hd9JyMCIBd1ehs3bgwgao9zQLsiRbRhbuQvv/wSUBzuXwVpKYqiKIog2bZC2QaMuRMkMwiSHSmCZILkPXv22BkkEySjrpJRgR8egmSCZNcVQXJwYJB4v1oR92q9dqjtgyA5ODBQfCp5ia+XtwQQJBMkO1zUypUr8cDAvoEHgeThQ4bqbuCmjePVa0hrbZ7yuliQnPRae7man1+snj17VhBeff2NhAUHm96P1OqoYww9uH9fYiIj1TLUUbYKJDeqX9+03o6MDCkuPvroI30fA/r1w8AB1jdpz67dBMkEyU5XGxnpqC/he2lQPNC+dxMmTLC5K1lBZDiHMfiMiRpGxOPHj2XOnDnKjUxRFEVRBMl2UkpKChgG6icTJDMIkgmSPUcEyQTJBMkUQTJBcnhYWIFB5Jox1eTy5cvCMDbgMAOcABBo9H2nXcnf15cgmSDZSUSlpqbKu8eOsY/gQTWS//Of/8ibkyabXq/i5yeHDx22CJL79uotZYldO3eW9n4O17BVINkcCl84f8HkXLx7547uen748KEUjqdPnhRIiT1+zFiCZIJkp3Mlw3lrZAazb7/9Fr/vcAnjnFX9WUOPGYqIiJBWrVrpabk/++wzMSKQEv/ixYvSoUMH5aimKIqiKIJkO2v16tVy5MgRgmQGQTJBsmeIIJkgmSCZIkgmSA4OUg6pggrwq0wZIAAJuL0tQYPKPgTJBMnOJmrbtm3sJ3gISFZwpq0GfMxTTN+5fdswkNyxvd5mpdbFvLwygWR8Lmod1Hr+17/+JQhzQI5/FxXbt24zrdO6RUuCZIJkpwPJ0dHRMnXqVMMcvYh///vfcuLECRkwYIDUrl1buZPNVebjBJBWALlevXrSu3dvPQX18+fPxagAkE5LS5OYmBh1jBRFURRFkOwgpW/cCHcyQTKDIJkgmSJIJkgmSCZIptwcJAcGWQaalOHyrlhJ/H39TP/vT5BMkOxUorZt3Sq5OTnsJ3gQSEZ88MEHEljZ37S8Xes2cCuXGyRju2pCESBv8sKFqGH8gszrF48dNbrUIPnJ4yfSIrGZaZ2pU94UFUihrV7HIMHNmzeL7Meb124mSCZIdkaY3Lp1az0lNcJIZ/LVq1f1NujZs6c0atRIqlevDnCtwDJcvxal4HFUVJTunI6Pj5e+2ndjxYoVcuPGDb0mM8IoNzLqLHfr1k2Cg4MJLxwoiqIoiiB506ZNcvrUKTqSGQTJBMmUoSDZ13lAcqyngOQAyyAZNToJkimCZILkoMBA071IHxALCZXAKlWkil9lygD5468GJlAHOSggQGvfELQ72tkMJPsRJBMkO14UXF74XWT/wENBcmFYDK1PTS1yWWyNmkgDbVGqTMTc2bNN7+vepasUF3v37DF3RKMkwgsguXZN035l2OAh0q1zF2S/UMtRaxnvK+C6rF+7jmm5n5e3dOqQJCtTVsikCROkTs1a5ueL5wiCZIJkpwTJNbVrf+fOnYa6klVqe6SAB6DdsmWLLFiwQEaMGCFdunSR5s2bS5MmTSQuLq5INW7cWBITEyUpKUmGDRum1y3OyMiQ/Px8/btndHz66ad6+YVq1aoBYBNeOEAURVEUQTIY0jHjSyERJDMIkgmSqeAqJTuSt2/fTpBsuCO5isV2+Oijj+wKkic3jiRIJkh2fhEk4zXbiiJIJkh2MlFrVq+Ww4cOsV9AkKxH0mv6+ekKDQySjz/+2Bwkl1oH9u/XoVe1yCjTa6iVXFzAuVjV7Ldhw/o0c5BcoqpHRcvJEyekcORfyVewuVjBMb140SJBECQTJDsjSMbfUaNGqcnQNouvvvpK7t27J3l5eXLq1CnUP0R95heEifD7te94VlaWnD9/Xj788EM1icMmgftDTk6OdOzYkW5kB4iiKIoiSEa2kf379sl7Z8/aok9JkMwgSCZIpsKCgzVw52NxQH3VqlV2A8mB1QPEv1Zlh8svxNsmIBnCNgP8LYPkS0UPnhEkEyRTBMkeIYJkBkEytSIlRXbv2lWgrhVFkAyIBEewWm9g/wFWg+TMI0fMU0uX6FKcOnmKaf2E+CYWQTLcxXAod2j3GiCwDsCKi/v37svIYcOlsrdPgW0glXfzhEQ5dOCgqCBIJkh2RsGB26BBA6SRVN8jjwlMSEGa7NGjR9OJTFEURREk21eASDDA2bj0EUEygyCZIJnC4ALqgVkcUJ83b57dQLIz1gC1AUhGSlWL7XDixAm7guTwoEAJ9a7oUPm++ipBMkEyQTJBMkEygyCZki2bN0v26dPsC3gUSGagxuqD+/fxOVrITkSQTJDsnMLxIeX0mTNn5LvvvvOY7yy+q0i5jTrMBMkURVEUQbL9tHHjRmT8sUefkiCZQZBMkEzVq1Ub6dIsDqgjTZVR0bN7d4v7atK4sbRt1crhqlW9ulnqug3GgmStvUtKJ75161a7gmTUBnU8tCdIJkgmSCZIJkj25CBIptavXy/H332XfQDLIkhmECQTJDtliuvIyEgZNGgQahoDJru9E/mDDz7AZ4TPB/10O7QzRVEURREkr12zBpmF7NmnJEhmECQTJFMtE5vjwlSziItU586dxagYNnSoxcH7o0ePijPE8qXLTINqO7ZlGNrmSFvn6+VtsR2Sk5PFqOjXp0+JkwUSExKRUs+RQu247+vdBX7fVhTSeWYdP065uNasWUOQbGMRJBMkL1261E7faerI4cNGDwagjAoeCPjbbyaCZAZBMvvS7x47JukbN0r9+vWdHiSrFNfR0dGYiI5STXrtYHcMnNf169dl9uzZgMhOWxeZoiiKoiZPnozawQ7px9iiDvLePXscUQeZIJlBkEyQTPXq1gMXJgbNixtQx4Or3UByZmam24PksMAg8S7BIYwaS0bFnFmzSgTJmOnu6OjWuYve3rHVq/O76VaiNqWnEyTbWATJBMmrVq50wfsDHX6oabVzxw45k5vLtiVIZhAkU0WAcNwjUX/YFUCyciYjzfPrr78uubm5/x975x0W1fHvf36/23v//xt6jxiqvfeCvSDpxa5IxI5+Y78qkGaer71XRDoR7EbwVq/p3WtLj4VUWz73vAcPz7JhhxPPnl3A9+d5XivZ3TMzOztnsmdeZ2aktrZWfvnlF2kNgc9x/fp1qa6ulrlz50KaUyITQghpzmC8128rPnlTImPlzprqaqTrDyiSGRTJFMnkhUWL6gRFTKx2UP3GjRsUyV6ifXJyk7PAR48e7VORvGTJEopkQiiSCUUyRTLxmUhG33j6jTdYpxTJDIrkVgRFMjCXeU5LS5O9e/eqJaB/+umnFi2Qv//+e/nkk09k9+7dMnbsWHxG7onsMIQQQiiSd+zYgdnNZpoUyQ94UCST119//dci2Xj43nW5O2cyJ/v27q0b5I6M0g6qnzt3jiLZSwwZNKjJWeDt27e3cec2RTIhFMmEIpkiuXlCkbx161Y5euQI65IimUGR3EqhSDbBbF0s/Txz5ky1hdXnn38uP/74Y4uZoXz37l0lkM+fPy+FhYWSkZGhrtNjYmIw89qheiOEEEIokjdt2iSVhw41SI8imUGRTMrKyhr0FevWrZuIGckXzSfWr1vnYAFI0O+MgfKwcO2gelFRkXgjMmfMeOBF8pQJE5EuBJG2Lq5cueI9kfwQRTIhFMmEIpkimfhPJG8wvqPy8nJzTytCkcygSCatXCSjzJDJmLk7YMAAWbx4MWYRYIayWiL61q1bStY2p7hz546aPf3VV1/Je++9JwUFBdh2Sm33FR0dzaWsnYMQQghFMqSQFBcV4TcQ0miFUCRfu3ZNPvn4E0s0Frdv37Z8/He1td7MH78TrR6L1X0dEMmkpLjYfe/0MRDJ/+P6pHODTiQiJFTCgkO0g+pr1qyhSPYSOavrBvKiIiK0dXHy5EmKZBsQQpFMKJIpkon/RTL2QS44cEDO1NSw/iiSGRTJpFWLZP3sZNCrVy+ZPn26bN68WV3vfvDBB0raYrANAhcDdL4MyGzMksag4qVLl+Ts2bNqpsPq1avlqaeeki5dukhUVBTK7uAsZEIIIYQiGaumVp8+jWNbMRTJr7z0kuWxtMa2Rrl86bLV4zEeayv/n3/+WVwCv1ktH7tt61YHRDI5ePCg+80nvbG09VHXJ50bfCLJCQlNLn08fvx470zNz8zU5lNVVSXNIda+8mr9iV9U4F2RXFVZqdINDwnV1QVm7nhRJAfq8sLd4RTJhFAkE4pkimTiVZG8a+dOOXXyJOuNIplBkUwecJFsgmWhIWbxNz7Xs88+KytWrJBt27apsQCIXMxYxupcX3/9tZq5jOWlMZB48+ZNiGZXMKO5UczXMWsFg4CQxbW1tXL16lU1CHj58mX5+OOP1W8JrL6G/2fNnTtXRo8eLR06dMDvbpQT5XWuPgghhBCKZPwGsFRGimSKZIpkkr9/v7tIToBIznd9EoNWzhSADBk4GI0Td9h6HFTHUk7eiIXZ2drBe+w91BwiZ/Ua88SH+PV6nYcEBgFtXWAJLW/EmlWrJKiJGwVmz54t/o5H08ap+u7cvoP9OiaEIplQJFMkE7+JZMw0O1xVxfqiSGZQJBOK5EbB71pT1kZERKjn+vTpo0TupEmTZMGCBWpVtI0bN8r+/fsxS1itXlZdXS01NTWKfzPq59y5c/L222/LW2+9pXjzzTfV7wO8fubMGTl9+jT2Z1ayeNeuXWqVjOXLl6t9m5955hkZNmyYdO3aVc04joyMxPLVWIobOPj5CSGEEIpkTKB6vaIC7yUPqEhOHJctKU8sa0BMp1TLIjk2NVaSZqc04JHx8ZZF8vrsSNm1NLoBjw8OtSSSn04Mk419H27Aqh7RzotkjrG7r4AXBJG83vXJY0ePOlQA8vyMe40zJlY7sP7ll1+K3cjLy9PmsWXLFmkOsXTx4voT/9/OnPF6nbeJiUHa2rpITU0Vb8SrL7/SZF4TJkwQf8fIocNUOXt26263fgmhSCYUyRTJxPciGfvTSGlpKbekoUhmUCQTimTL4LPhpnbM/oXMDQ8Ph2BWMjc+Pl6SkpKkffv20rFjR7XUdLdu3SB/Vb0MHz5c0tLSZOzYsQAiGvsx43UTHIfjkY65bzPktZmPue+xD+qYEEIIoUhe94c/4PeNOd5O/CCScTPeoUOHvA1u6PtNIrnH0mO4Vm1A/PAMyyI5MbP+utb9+taSSL52rC2uexuwKiPCkkh+sWfMr66H30pPcFgkk+3btzfoT3bu3PkP2CN5vuuTZaWlDhWA7Nm5896evZHagfVjx46J3SguLNbmkZubK80hZs7INE98R+q8V7cedWIoVi/vsVeU3di0YaPKq61mxvmoUaPE3zGgbz+UE//aqltCKJIJRTJFMvG9SM7Pz5ea6mrWEUWyMCiSCUWyHakM8DfksjlzGbIXQDY3BmYSmzT2Oo6FmDZxz4sQQghxHopkjI9x1Vn/i+THH3/ckbG0efPmOSWSKZIJJixgKWvXPgXi7P8FGDMauphPgr179jhYEII9ksND9Xv2rl27VuwGTnZdHllZWYJoLssshwYFO1Lfk8dPqJMLUVHa+sDSXXbjYEEB8oIY8ZhP586dxd+RkpCoyjlm1CgbdUsIRTKhSKZIJr4UyTu2b5eTJ064v49QJDMokglFMiGEEEIoktUKpEePHLnvtCiSKZIpksmJ48fdl7WuMAgI2Ldv318Yf9w0X8DeOM4VhDwcFS0hDwVqO4PJkyeLN8K8+7cx0tPTpTlEnx49607YKGdO2I3r1tUJitAwbZ1jiUi7cfqNN1ReMdEx2rxqa2vFX3H37l0JDQxS5Zz03HhbdUsIRTKhSKZIJs6L5E2bNkllZeWvXicUyQyKZEKRTAghhBCKZIxrl5eVcesjimS7IpkimWAbNXeRPFeJZOMBVLu+CCHmVEHYSdWJ0zjN8sfYi8gboVtiGXsYNYeIDq/rNNonJzs2KID0QwKDtB3wtGnTxG58+OGHdSIjUr90+XvvvSf+Cuy/jTKCebPnOFLnhFAkE4pkimRin+rTp6W4uFj9lnF/jVAkUyRTJBOKZEIIIYRQJBccOCBnamq8khZFMkUyRTLZu3evq0jGjSodDQKwtDVE8krXF8vLyx0qCJk+dWpdA43Rz1q9cOGC2I2E+ARtHteuXRNEc5CaqQMGOVbnpqzW1QUGDezGjz/+qPKJDAvX5lVRUSH+CtRHfUecm6epN0IokglFMoMimRCKZOeDIplQJBNCCCGEIplQJFMkUyT7/7pow4YNriL5Z8MV/5lBQAAeDJHc31Ukb9682aHCkN27dqkGGhURoe0QSkpKxG507dJVm8fZs2fFn2HOFgaTJk50rM57dO1aJ4diY7X1ceXKFbEbyQkJEhYcos3ntddeE3/Fgf359XW+d/duTb0RQpFMKJJbe1AkE0KRzKBIJhTJhBBCCKFIpkimSKZIJhXl5e7LWp8wCADq4dixY39s/HvB9U2Hq6ocKhAJfuihJmXj8uXLxW6MGztOm0dBQYH4M0ypDl7McW52LCR1nbzXLzmN/QftxshhwyW4iT2ws7KyxF+Rs7p+cBMbp2vqjRCKZEKR3NqDIpkQimQGRTKhSCaEEEIIRTJFMkUyRTLZsmWLq0gG4wwCAB4U69atm+b6ph3btztUIBLfJg4yWdshjBo1SuxG7po12jxWrlwp/owXFi7yidREW0Ye4SF6eb9q1SqxG7OzspAXfkR4zGfIkCHir5g4frwqX1x0jL7eCKFIJhTJDIpkQiiS/RYUyYQiOS4uTtq0aaMF7+EAPiGkJWKhf2vRfRyhSCYUyTU1Ndji0uucO3eOIpki2dtgYrH7bOQLmIBsEADwoNi2bdtfGS9+6/rmI4cPO1AoMmLYsDqZ16aNx8H1+Ph4uX79utiJ8+fPawfwn3rqKfFnjBszRtVDWGCQ43Ue9LuHJNhAVx+jR48Wu7Fx/YYml9FOSEhQHbQ/okfXuk5+UL9+PBdJq2PPnj0UyRTJFMkUyYRQJLeGoEgmFMkoi46WM3uaQIZpoCwjD2RbbrX9G6FIJhTJvg+9SE5MWygpTyxrQGznVMsiOXZIrCTNTm7AIxMSLIvkjdlRsntZQ54YHGpJJD+dGC6b+sU2YHWPGIpkL/Mf//7vYvjhBiIZE48NAkzwYIJlrl9wfTMGC6tPn/ZywciqlSvrBryjorQ/mgzBL3YDP+I8pd+5c2fxZyS2fUTVQ3ybNs7PAm/btsmZwpD3tbW1YidOnjhh6bt96623xNdx8+ebEhoYpMo38bnxPBdJq6O0pIQi2SYUyQyKZEIokv0fFMmEIjkxMVHd+L148WKPzJkzR1JTUzFzD8eQ5gvGGjxCaUZaCOa4mQ7LMrpdu3aqj1uyZInHPm7u3Ln2+zhCKJIJRbIevUjWYUUk69GLZB0UyV7mYEGB+2zkbzHx2CDARD2YbNq06V/wJvclrmGkvVcwAjmvBs1Dw7QD7GvWrBG7kfBIvDaPS5cuiT/iyy+/rD/hB/bt73idjx4xSuUVExWtrY8Thgi2E59//rnKJzIsXJtPfn6++Dreefud+jpfvnQZz0XS6njj1CmKZA9QJDMokgmhSMZNhZWvH/IAgUykSG4OUCRDsnTo0EE2b96M60uPYFnD559/vplLFoo3tKsJEybItGnTZOrUqb9iypQpMm7cONw80GxnJhPOQk5JScEqfh7bMZ5/7rnnpHfv3k32SXi9e/fuah/GL774wmMf9/bbb0tWVhauz/k9EIpkQpFMkUyR7BCVlZXwwO4iebZBgCt4cAUzk3sbb77reiCkl3cLSNBAgx8K1A6wp6Wlid3o1bOXNg80FH/E0SNH6k/4zIwMx+s7NydH5RXehLzPzc21L+/j2kpYULA2H9xd6evYv3dffZ3v2rGT5yFplXTt2pUi+ddQJDMokgmhSMZgraYPJSOHDadIbjZQJHfs2FH27dsnurhy5Qpm7FGyNGOwrdWzzz6LPRKxKlmjnD17FkIN3zlvCiDNdi9j9I9r166F3PXYlo8fPy6TJ0+W2NjYJtPr2bOnFBQUNDkBZeHChZr0PEEIRTKhSL569ap8/NHHlmgsbt++bfn42hu1Xs3/zp07lo+9ceMGRbINsLqu4YPdRfLrIvL/DQJcUQ/uGG+e73awurPg386c8VIhycB+/e/tkxyn3Uv3u+++EzuRMT1DO4ifl5cn/oiXX3wRn1+xe/dux+sbbRd5hQTq5X16errYjUfTxiEvbT5jxowRX8eiBdkoF/aKbrVL1hOSMX06RbJ9KJIZFMmEUCT7GYpkQpG8Z88e0cXly5exvDVFcjOexZmUlCSZmZly7do18RQ3b97EtmbYeowimTRL0Mf06dNHioqKRBefffaZzJs3T2JiYrwikvG7JTs7myKZUCQTimQHgkGRXHnokLtExkzk/92wYcM/GgS4ox7cEZH/ZxxU7C6TsbQSLLX9gpKFCxagoUJeaAfZT506JXbi1IkT2vRxd6w/4rmnnzGlps/qPC46RuUZr1nuG8tJ/fDDD2InVi5fgXwgSDzmgwtKLNvgyxgxdKgqV6f27XkOkta8TzJuwqFIdoMimUGRTAhF8q1bt3BzpQfIu++8S5Hc+qFI9ikUyRhfmDFjhnzzzTdakXz48GHp1KkTRXKLhiLZXCWBIplQJBOKZAZFcvMF2xgXHDjQ2HLWPxm0NQhoDDw0iiGN/974t9o9QVjqwsJCqamutlFggqWdreyT/KIxc9dm4ALGY/rYe+nu3bvi60hOSKibkR0T47M6HzJokCV5X11dLXaitLikLp+oKG0+WMLKV4HvODo8QpXrsXHpPAcJZyVTJFMkMyiSCWn9IplhP1qhSCYUyYQimRD/i+SDBw9yaWtCkUwcgiL5cFWVJMUnWOLDDz+UxqJ9UrKl419uxF9VVVbayr9rx06Wjl2zahVFsnVUu9i+bZsniTzCIEArkjX8iZHQqwaNJS779++XUydP3mfBSVRYuIQGBmkH2R999FGxG/Ga9MEHH3wgvozPrlypH0gb2Le/z+p78e9/r/KMDAvX1sdLL70kduLChQuWbhLYvn27+Cree/fd+jpfvGhRqz6vCDlx/DguTCmSKZIZFMmEUCQzKJIJRTKhSCakmYnkrl27yquvvupxz2U8f/LkSZk2bRr7OEKRTO4DiuSy0lLLY2nwBo1FWFCwpeNXLF0mboEVIy3n39jKUDERkZaO/X32QopkC9c6FeXlWG0aXrcxzpszke2IZIUxCzndSPBHDxnJNsNiHywoUOtqc+9V6/S612HFxen3Sb5+/brYiX59+2oH8vPz88WXUV5WVn+yz5k5y6dyCXmGNCHvx44da1/et4mT0KBgbT4zZ84UX8XuXbvq63z3zl2t/twipLioSFKSkymSKZIZFMmEUCQzKJLJAyeSkQaAsGkE83XHPwfyQvlcwexCE/M5p8tlpqspi3t5NGWiSNa1LX2dOV8uE0/lup907rPdOPoZzXx05xUwy2LSHEQywHZvI0aMkIkTJ8qECRPcUc8//fTT0qtXL5TbK+1UV0/A6XaLtO2Xyfmy6MpzP+c3jkFaelReHt+P19zL30h5PfShFMmEIjm6w0Bp0zOtAZGJPSyL5PD4UHl4aGwDYofEWhbJ6QNCZNrYsAYM7RFsSSR3Cg+SickR7lAk669tVH9UUlKirifWr1/vSSBjsnCF657IOvBgiY0bNz5sJHwUGTTFpk2bZOfOnbJ3717J379fLVdSUlysBGJ5efk9SMa0aZaWWq6qqhI7sWjhQm36WCrGl7F86VLzZMedED6tc8wCR766Wdrx8fFy9epVsRNPPv448tHWe+/evcVXMTsrC+VRM+CLi4t5/pEHgq1bt0pycrIXRTKhSKZIzsnJYf9iQMiw1CEUyQxTJOMm0hbcnklpaakaw+jWrVuLFcnu0sRMC7+FISgB0sR/m+91Sni5potra2ynhdWC+vfvL6mpqTJy5EjF8OHDZeDAgUpQdenSBTfRu8oN/Ov1srRr107dMNCvXz+VN8pglmfYsGGqjH379pXu3btLSkrKb64rd8GHz5SRkdGUSMaYD74fdxlp4g8ZDCx9t67tC+LcXVDZFFxWhLB7u8F57LHt62is/aHNoD2YbQbtxGwzYMiQIWg3aMeYZYt2456W16WfKWKRH/IdNGhQg7aM8wxlQltH+d2/O+vtWAExDKELkWypT4qOjrb0vTUiAr0iBs28zLbQvn1787xH3UBgu/dBqEu0X0f6Rve6R7+Adol88T0NHTpUlWfUqFH4G8+hz8T5hfPMLJPX2pJ7WXB+ID/UD9oz6se9LZn9NOrITMdqHSFtTNjBipueeOyxx1T+aNfILy0tzfV1vNYgb5R78ODBkp6ebr4Hf+N7Rp3hfT4HbR8Ttfi7quXQrXMXx0VybW2tfPvtt17n+++//00iudPMnbhWbUDKU6ssi+S4R+NwTduA7sV9LIvk/975MK57G5D/r9GWRHJmu8avkYO9KJKRV0tsw/iOiwoLpaCgQPbv24eJhJjwCznclL/Fe742yBKR/28QYAU8/BYwOznByCTfyPCugZD7Jy8vT4ItLIG8ePFisRPnP/1Umz5+OPkwzIE3zAz2eZ1369z53v7FenlfUVEhduKlvLw6IRUbq83n0qVL4ovo06OnKk+7pGSee+SBYuXKlbgY8Y5IJhTJFMm4QGbfYkBI3169KZIZ9SI5Lia2BbdnsnbtWswkwsB+ixTJpnzE4DkG+iEjMjMz5YUXXpDVq1erz/faa6+pf9esWSPZ2dkyZcoUSBTkg89sCgHbogRpde7cWcm26dOnq7EMbB2FQa0DBw6o6+zjx48DNQu3sLBQfU5jnAnfAVbtgnCA1MHnue9ymcIFIgYiZNKkSagPNQazZcsWtU0bpNiRI0dUWY4dOyaVlZWYBIEJEeqG99zcXFVXU6dOhdxB+0CaHssEyQEJjc9uAiGzYsUKuXbtmniKW7duSXV1NUQIymoei+8H+SJNpO1V4Y+6hXBBPp6AnIQUdZWh+F7w/cyaNUuWLVsmL7/8stm28Le69po/f75MnjxZfXa0x/9j7z6g5CrO/O+382bnbEvTo5FmFBA44ZyzxbKYnBdWh2j8Agsma4lO4ISNwwFnW7AmHHJwNjlHLxmc3l3njLNB/a/PyCWuhu57b/f0JM1T5/zOpL73Vniqpru+9TxPt/ZlzDKwHSt9AnSph/sWgZbPfBtttNHoWB+R0nmx9Wz7Qigfl3Io/sd//Aew2hZEF21Gv7sPp4tsM+yCvYL+bCbroosuGrXtU045hcfP6JxbsWIFj1rjCWJmuxkX9DMWbGH58uWtQw89dPQ5yenH/GHL5lO2ZZvLbNzBanasPrx89av2l9oxsXWQTn/rd+3YZZddzJfKnMae59qxY2fcHNZhexkEgtP6up1c4z7q4vV1+4pdWNfcwzpn3h9//PHWIJDPGqSP8hqk74ytsTNXRRAAUK1j414bXUvaar5tv/321jhrHbtkMzb/2dSaOlmH/E4qPHPqmGOOsQYZA7Y5blvK/ye22Wab0ba6/wknnND64he/qH/Y85q6EFti3ycnOGGdPvbYY0ftaddddwV82VP2Du743AOSY4t2XXXVVR11dfKe8/+CjVhHPLv4N/3FXvKz2LI6XXnllcX7aJO5MhUREcwx68UMev8TetmLXzLhIHmnnXaakL00/wcCJPcPJHM8nEW2L4z1Xkn/kNToRvmbrpXeWM9PXz/m4b1XPMRQhwaapYuDf87jLc/xhqrkGT/+8Y9bk1H++Mc/rlmEXvCc5056f++x2x6rQzIsWDChXtqXX3bZaqixaFHpc7xpnejiQysbU58tN99i1s2xUMgHlwDJAZKjBEgOhQIkr4slQHIoQDI4YIMdbOBZ/Y1vfKN1xx13tH70ox+1fvrTn/o8uEY/+9nPXDeag/TLX/6yttv85jE2LkABToBc6sALwr1vS5uSP/zhD9d4zvzpT39q/fWvf23lsmrVqtHf/eEPf2j95je/Gd2TuOeeewCE0Xbsv//+wGAGyl33n7C4ABJwdMsttwi7qy68gzyTJ/BoHXJ54IEH7FeQ1/AidvA79xVbAQf1lWfQWs8EQERu8VowxnPBkxtuuGG0nZ2K52o7qOW6os4//3wwyb376pmsf9hOrmsbgZBAnud6PXAEJBkf4wQa6s9sW6K6+V3uM20H59gXiFZpX7k/gWrj36luYK6DEsCx++U9MwAbgLv55pvZOFtXp1w/fey+vItd95Dnmv9AszE0dtqgLezgvvvuW2MzxqtY7r///rXs2Jz7wQ9+oB7GD1hli+C353QdXlvfA5DAPdAHqn33u9/1HM97iC37ap6xY/VW/+9///utG2+80f4TSMdzMwNaegj8VF8gOvd5PgBiDpUUddFury+OmbE0bj6XA48+b3s+iNnR/lxz6qmnqktljuQMSK2DDjFY19xDnuU87/WFPtE3xbnnIMfvf/97fcl+rZ3movkORLtnBsrdQlte8IA44APUyvls7lib2aTnqpN6FO3JeFoz1dtrtQPQtZ6ZhyXzqfxgjbY44JMijoLF2mqNNk/a9o+if9h3sY/Yt3XNwRsHGhyCcWBCu9v1hUMd5mNVcWCD5zNArD7Fot923nln48xD3v8H9Ri7lrJvwD1A8toKBUgOkBwgWcrir6f3Zduk/62PSGr0ovzNuJQq8SwV+RtY/nb6uqpeI0KvTx8IGOwGFeDCG8XxlFe87OWl9/emZDKKHNraS5umDyVTsUkwyBu6OTihYae9ybLYDs+fP+Vhxb+aPsDlPv/PffedlfMsFHIiPkBygOQoAZJDoQDJUQIkh9YVkAxy+TwpJCjoB3yBD0BEneJ1YMb111/PA9cmPfDRLTABOYVvVl/AFoStU4dSsAoi3H777aPgbO+99+YdWMcz0PiBn8AF4AZe1K9LRV9pF3iir/Q5L+HcV74C6QCU+o8VuFdS/L3tdQDOpZdeKqyrML99sy9jxpvVMzoJZNJOMF9/Xnfdda1f//rX/lZ3HHOfAUQAmHH0/FKQ7DACO+pQL3Xg5c5WjQG7BemApgxUO44hMAnCFcPBA1O8r3laXpYcAoAyMM19xlncB/Bliw4E8PD1vFqhpbWP16i+dzgEEM3t66W4DhR0H1AYAFSfHFK+EG6aJz7w3osdd7Jlaw2vUmsbWzYOPF3LbBDkVJdK23c4gN3wOr7ppps8q3reV6+N7uWebDd7A1eqeLgBKLfXevfdd4Ps+mZcdbKemYfmLnhbx/M2e0Sby9qS1+jqeVxtT+pkvgDk7Mna2y6s9EEHHeQ1+tPeNjtey4sdKCce4eyCN7Y6Foq1UH9qs2dYh60FY/vIgYvZDpJDAZIDJAdIdqjzF0nnJL0jMdsXpa+PSmqMV/mbviqdbHpk+gf1lHTKaThV/sXpd8vSycUd0vdvT9/v86BCKezU50cNdmG556qQJ+Mpb9tzz+rw2ZNQjv/gh/Ii4c3YR6aiz0eGhn7t+fmDZid5szee8taN/y1tLjanPKx4MSf1u4455rDZOM9CobQxcEj6MP6bcYHkUIDkAMn+d59kToVCs10vet6GNwVIjpJB8qIFw7+cufYcSmFv903hRt+b3iv+YgaBZBvwQrKClzzFxgUEQBPAEpiu4/2ac4+CjKDerbfeCmJPSEQz8AGkAGo9lzpBSOFz7Z3w4tOufhf3BC+E+AUNivlLwV4havtYgL8JBcllBfCxTwTqjLM/QSEhaEEeHofqUAqSQaUOha2zOQc/eKvyPDUmtfqSbWSQnJ8nBDIoxbsR+J2Iou+AWfN16623rhofwJ3Xtbbp+77Xy/0AZR6lu6X/Y+BcnvP6xjpgfelnMW5FkMyznwd5SdFnlSDZnhpPXesEO+vzvHdPIZOFSzdvakFkkJetfe9735sQm7JeX3zxxbxyS6FpXqeNcf4/MRHrIoDLTnkSO8A/FiZbv8xVHv8ObmXbArN5rwtHLdqDAx1sf+XKlRkkr+Vpf16CYuyGZ/Xll1+eD3zQdADJ2n5OguGHzJz3P6ENliz9XoDkAMnuMTw47/cz0YYTb90labMEi1+Tvm6QfjcnHZr7p6TGRCh/M0UKJZt9WDLYB4aHyj1XncIdT7numutK7+/N82SU7bfZZvUiMWfg/qTGlGju4Cl1wk77wDSe8s6jj9ZWoKT0Od5wTWABtNUD1L4jqTFbFQqlk7j/GiB5XAqQHCDZxsDGSY3ZrlBoaE7ztADJuQRITmlkfpDUmJkKpfLwtAm8NOl7Mwgk89ACg3i69Q0u3XXXXTyqKmGyfpKzk6dpl59nwQMb/r7WhRS8r4VJBiHVq22f8UoEKUCWLr3qqCvAos+FhgWM1GcdBMnZW13Y277ZFzgNJoJ/6tErSOZtI4Qx0OpnpVuQnL3IhesGLCvGP4dBzmGsyUEHP9e2H6GK9bsIeOrQbmwAcvCUJ7f7T2Th+c87WT7onId7gkAy2N93kOzgiHDlQpkbn4kqxpfHu7DJ8gvrpw7QFjAVlptddrsmUldrg3DlvH31aae833LmW5fY6wSuiXmM7aNap9tC9gULFpjfQq3ndpsP2SbYnhDZa4Fk6485pnD2seYKOf6d73wnrwc5CsZUg2Sfkw9YvHjxE5IaM0Ih7wGvDpAcIDndg36Z1AiVa+orEbJwfZ/Rli0Q/hF6ozCekt/YdJLcKRNZfHBYMrKwpa2DAwPfS2pMhUaazTeoQxW892Z6POUrF335b3mSF5c+R96ciSreLOZFf2hu86SkxixWKGTz/9MBkgMkRwmQHAoFSJ7pJUByKEDyRBQb+7zo3v3ud8tl2hHaAmEAHDBWN1R1ztMKXIGwwv0CA/Yhqrz2AA3QnOehvhlbLxDM53dwoc5n5JzHl1ebsMG8+0BTXqnSVNX0sOSBx4MOBAF7eSr3HSRfccUVUwKS9bkQtP0seRx5FAO66tIDSAbGKMOmrkGyvhTu10Y8QFgGzNgmoMlezznnHJCL7YOioJnw6/JBsyfzoRIGyh3O23iM16b2y/vMy5R3rfp2NW+NlXnGfru51nzQl/Iwq4e+4X0rJHcfSz4A0A+QnPPCC93ZLfA2Ptqc8153BaDvvfdeY9cupLSfhSJnC7XWEHYlVPWdd97Zuuaaa6yJ5ACBNbhWGHltYSvmjFDoY+eT3+kjMLbG+DjoYR20HlrLrI/s2mGhujZlPMDttn20cOFCTlLW2Wy38jVnOG9c7X0XQbJQ3Orl+cYMKDbvtEn79ZdxYf+zACSHAiQHSA6QHCA5NDDw2VF4sXRp6SLhA8R4yotf9KLS+3vDM5HlhutvWLNA8ApOakydmn9N/zD8o+/YH950Vb4BrDhtqq0jCxZMWVjxyy699ME+bzZ3TGrMZoVC6cP5v6R5/4MAyQGSowRIDoUCJEcJkBxa10AyIOYzrM184JZAAL+r6+0GcNx8883A7EPyEmfYdeCBB4KxtTwebfTzGnQNbz0eciRnp5CmH/zgB0GLyvDJ6i9vbwpDnutSBKM8lkvb6N6g9VlnnWUDdjTv6XbbbdfacssthRsWdtXvhR/Wb7U8VEFBbVqyZAmw5Vr9kvseMAGLquoFaBWvywJQhB92b59RJgskl9aVl6w25ToCwyCUNta8h34zjtkLtiuQDBgZR89UegHJ6623nsMQcibzdiy1OWHbeemzV57UrnfIAih79atfPerVvNlmm7V23313cJlXf2lfmIv6nudxse207bbbCkGsTrVz5gJ9DkOAsnKlyxd97rnnCiVufOrci/e5OrELtmxsAEr26x7AJq9f4etrwdGxduw+xlweah6n4wXJ1qDDDjts9BBAjaIP1AGw5cXODvSV+WofVFvV25yusl2wV25pn4uKYNuBEoBTX1beA6R3oEb+a961bMsaZF1cvnx564ADDnBYwWewynDd+pzNsMExsF3OYm2uXH+sM6JLOMRgHQTErYvs0TqZcz2rd50+cuAiR48YJ0gGtdmz/VX3ZutsyToAthtHB5RmB0gOBUgOkBwgOUByqNlsbshoF48sLF0kUq6F1njKzv++U/X9J7CcmE4faifNnzt306TGFEm/37Ea3pcDI2/IxlPekMJrDVXkSRaOZ6LKB9//gTV9Pjw83ExqzHKFQkJcvy7NvVUBkgMkz/QSIDkUCpAcJUByKEByhko2+W2625i3mb5ixQoCAXilCbcK6tTyLAO7wAAAcywMENLavcqAgr8Bw0ANEAFQeY8NUoF4wJDvfQXlAJTjjjsOQCy9L4gJ/IAURbAKmFxyySVVbRINTB7OsXUh3+t73r/2RmpBKkATCHc/7QCVga/c9zw7gT1AvQQEgZXGDBzL15Kf5TcF34zDlIFkY8IDkFcucCPEb67jUUcd1Up5xuUv1We1wKXXnH322WypW5BcCTLBTjCPgCZ9z+b9jT1mj+Rddtmldd1115XeD0BlC6Av+8ghsYvyO/cjYFl4dVCypOgr+cVzm93HnAesK69VtEneXrYHYAObgLbw7tq3bNkyhySMDwBYeU+ADoxzeER9zBEwM9syYAECOxRSVtjIaaed1jrkkEPWsmP3Ofjgg81zEFif9QyS9ZU8vNYgY6pUrEPWMuG6hUXWNs/WV9Y3NmiN0ve8cSvvCXDK2Z7DSZM67bvvviIseGZpPzsMwyMfrHUYgV3ltZB8r3+AaVDZ+qEN7lvmTVysU+4nkJoNV4Wj/sAHPqAviusy5br5/Kef5BKvE4VCP4Bn6jFukGxP1tronjlHvYM3+UAKWzMXzPEAydNQAZLNGw5kfZdDZgGSAyQHSJ61mvv7+YODpRvu3gyOp1hkyu7vBNtEluU775zzI3sH9LCkxlRpfrN5rLosGhkp7RNvWsdTDthv/9UL7gbleZK9QZ6Ist3WW+ew1v9/hDV+UKFQmncfDZAcIDlKgOTQbFKA5PvSptzhK1Z0UOhTJ30yQPKMVIBkUI4HMagETri2GHnL9zbWHWAGU3jC2YivAkugM4CUAaZ6gaU+I1fBBOAKqARlvbemnFt5rNwftFBHsAkwKCsghNCpuV4EpAFhZQVoB6vT5rFr29bH79QV2ABHwUjQDtQYK7/n+c0LERByz7H3A814XQPOJR62PLJByLb3KNRtSkAyQGNMhCUG4gAuYaGL9sUueOsaFx6fYE9FAbhARtePCySzVQDTNRdeeOGo9/sJJ5wA1hFQlT10hUbP/Qzy59y6D5F7mldnnHEGGy7aS2necHYMlIGFZcWhB3O1CP14pPJ29fwq8Kc94Gf27gX7MuguAkB1MjeEYNdHVYCUV3+GemPnqMMbgGxZ4Z0O7HX4PJ3H2v16AsnqxYvd2FVFRNCPPG3ZgoMmnl+Etrmv8s/60tgJk1zlXQ9c2j9VH7J2aUuFxzabV/ccDr/jfNfWXEee82y6KhWhQx5AvWvz9Q4RVe01SgfA9rxeWzqs0foOhPf/o2pdNK+EALdOuOe4QbIDOjyRx4badzCEB7XonQ4ABEgevwIkz+BSAZJfsvvxrdf817lracNtDq0NktfffGnrVZ957Vp6xadeUxskn//hJa3vnbvBWvrEoYtrgeTdNhxp/c8OL3iI5gVIDpA8WzU4p3kxwy374EDekI5jAam8v5NuE1F8+MgLQArl/b9JjanU3Llzn57qsmp+sxzeO6XozU2v5cjDD6+VJ1l4rX6XP//pz61Fw8O5z09+sP2hUCiduP3HNPfuCZBcqQDJUWxu5jGzefPypEYoNMs1E0GyzcSSNTS05WabB0iecQqQDHYBCLzhgBCb/dlT0n2yMjQBYWy88+ZzbR2v5LwpT2BSFfzhNWzzn8dlrk9uV5nUkecnz8AyD16wRh3AS9dlkAymV4EyfQpgAbwZGmepZxY72HTTTYFH4YrbClgFm4HiYj7QopcicGZsKkCyHLXanmHOWLnvlIBk+zhCJzuAoH4l9kXaC0oBPHkMy9otxLDxcI+eQLL68ebmcb/nnnsChtq2JvS0r8AxyL3DDjuMwj8pzDyHp/fnP/95dW0roX55+4Jh2ltmt0UJ4Q5GVh3SNA/zPfQbYOrwRtV8N7e0JdtYhn9joXYeazYFWruuKhrBpZdeKlfyQ9rrHjxWeZGXFADRYRBwtlN/jQckZ09m41O5fv3whz8Ekb0+z+/SgwBeY300rysOyoDyOTS764SAZrNlIajNB+sFaJrrU7km5kM2AK41wtpadn8RDNh9Bsm8lKtyXfu7Ax32PnOo+Qyxs4p2xsvZAQ391Glt1O9srh8g2eEK3tDmOIBdaK96W/sdEgmQXKFQgOQqVYDkClWA5ApVgORKBUgOkDzrNH9gYKc6wFGejF7LfvvsW3l/XssTUa5OJ/a0j4bmDq5Maky9mj9Tn+d1OBmd5WRzr+Vte+xRK0+yk7v9LpdfdtmDfd5sLk9qhEKhNeKV/LKkBwIklypAchQf5ovjNi+pEQrNcs1EkMxzzmZAB4W23WrrAMkzSgGSQQveZIBGBiEV9c8wAMQS1rfy/uAISJI9isGBisPtvBJ5MmePuwwIKuV1xKtOiNkyT0PQGEDI9QIIq9rjOnXTr0ceeSRIqD2AMVADlAEa4KM+UhfQEZgpCrAvCpRu1+/Ze7KORzLIB9RWhK+efJCsv4AZ7TCedewrjyEApFR4UcoPaxy7BsnAkdy2bC3DueyVOxbG+T0V+9dY8gwdK+NAYFSGyGPlerbhNZ4NVoogKBQxOFoF0dkqL8tcR8/ldQoUVnmPai/wXNNWsh2CbkJT84rlqdtWQuPvsccerhkLWR0iAZLreCTr65I6jQskW494dFfahhD4QmDXzC2ePW/lBnaIVl936isHBawhbAg0F96dU07lmKtP3fWQ8mu1wfjp37LC8zwfzMjXAOpVaREcFgF0jR0b1gfgrXDp7mc+aGs+9DB2XbRmFtZEP+eoBaWhrf2PYfdjQbJ2FHMkO5Qk7LoxyeWWW26xfjskYh0ZHXNr/Lvf/e4Ayeu2AiQHSA6QHCA5JNRzCm99//DQ/NJNd3lOein+oT4vbdgPzy+/v9OTE1Hef1yOu580b96rkxpTrmbzVPVZvLA8N7UT0b32+fM3eM7qNlfkSfZBzpuofpb3vOvda/p88eDgnKRGKBRaS2DAcebg0vRhN8+XDQIkT5b09fQHyVF84M/h6H6Tvj4qIGJouitAcpTIkTwLFCBZeGoew15bG5ZkiGPz34Z7VShWEALIyXky99prr0pvPdfw5HVYGvTqRkCGnLRVUFhYW3sXOa8xiCf/cZ2i34AMUA2Q4uknZy14wcuOFx8PXIAeVAOuhDcuevrl3MpFsLiugWT1Ek4WYNfeLu4PrhvPKlsBhuTi1e5uQXKGeaAV+68VenrsPMhjKdx5hpXqDmw5XABU7ZxStPF23meffdSVjZoT8kTbK2Iz8thyygC0hHW3t1MXJKsHGMYGXVe6vyRsOMDXxXynHE3A/DUv24knsTaqy7QFyepVxz5y7nLAs8s55RpjY33p2E9sO+d7Vjd5w4HPqkMT1lz90+266Lm8matg9TXXXMMrP4NkB4Zq5Xpnd/Ism1PuIXy1qBVCu/Pa52XM3nk8C1mtfxzesY5kL+ZiyHBqB3IXLVrkAA/7571uHTZO1r41cwGIlmf8zjvvZE9sxPou97c5Au671vplnrLL/Hse/epoTfC8AMmzXgGSzYsvpDlcRzz+25WVX/hiresdABzH89umXzjl5JNrXXttWrcCJAdIno26Iym/ye0ooUe6LddcfXURaK7qdG9hf/oNNJW3bvxvni8R+l+SGtNBw3OGX6pOCwbnlfa3E8q9FAuZ+2etv7QcTgnJ1M/yr295Sx7zO5IaD1UoFEqnbB+T5t+t66+3tDBXlwbknRzJH6/Pc+SG6QeSo9gMs6GSQfIpkRs3FBpVgOQoAZKnXAGSQYWjjz46Q5aupJ1C2FZ5F9vYAxGANiAVuAOfqwAk0GtTvxeBBFVQBqAEq9Ur9xdIU+UJWubBKJS3jVT7LQAHqAJYXXbZZQAaYAgC8dYDPooQeZ0EyeoshG2vdeIxLpdsFSiUx9UhhW5AMs/d0047jcdkr/Urhh8HNUE3+WtBLIcK5FvmDakNvB/NFTZhY5x9q7v3yWzGnBA2uriXVhckZy9bdub60jYL7wu09dje3Md1YPu0BMm8XK1B1oiyctddd5l3HYBmpdSv9uvAZDmrK7zJ2UnP6yKb07fWiqqDGcKeF/vY2iIMeC+OMZ7Htq23otqoizkgaqP7micf+MAHzFkAuxjiv9SO9t9/f+MK+jqssSbVAFkz/W8C7L1GLnDre46kAay7Tgh9c9d8cODH7/ztrQkIspMAyaEAyVECJAdIXsc1952MV4jVso33008/vdVlEWKgCDQfKLu/sCb9LL/+9a9bchGvfn7ztqTG9NHAH9XrBc9/fmmfe+M2Ho/gOp7PQsn0q/zi5z+3ofU3eN/8SFKjrUKhkFyXz0+esX/Nc5V38lRA1QhtPQ1BchQb1MUxWxaeqKFQeCRHCZA8PRQg+e67785hgXt6jnChvM7KCjALFugXG/agAWAy1QVo/sIXvqCvSP1ySNqqPLA9ARX31Be8+3gwA6w77rhjztO8LoJk0IiHKojX0zNALDmQKw4E8EgEoLS9LkgWXhg06hkY5THjbW18eJTKs+oAAdsyJg4XALvUe6kGyfrX97y5KsC+wxJg1XjHv0rTFiQ73CpigL4oKfqSJ7l6TFg/5XZYq4XRBmunujgEI4dxMTcxb2aHIISw7mNxP212mMJBC9Ex2KcDJJ5fu2+t3RV93+73fq74fYDkdVgBkqMESA6QHBoZGXligsmrRuaX59MVVqfb8vrXvPZv3qkDf1qyeMkPyu5/0kkntfpZLrzggmKu3vdNqzypAwOXj8L7xeXw/jOf+cy4+jz9c/qzvAJlz9hll1361ufnnn1Osc83Lu+HUCj0nPWf8/4HD30smgqoGh7JQ9MMJEexUeoDcfZGvji8UEOh8EiOEiB52ihAsohWwtD2BB89441vfCOvztJn8NIFe3iMAcm8gEG2qS685Bywz6FrczhleyVXXHFFhWfy+AEKeCIHsByk8tuqw7oGkh1UAMt7qVMGbCeccELZI4TT5emdw7PXBcm8hNXNNb1CZB6U6seRol/2wjaETgfYugLJwv3K+10VgYA3bgbJE6jpCJJzznLhxHn3VtoHz9xcjwmSdpi30wYkc+LhyWst1F+5fselVIO33XZbfw7ZlKxj0hoAyiC+tU8fTZ0CJIcCJDsQdvIXV9ZSh3VVeOla1994w419f/6pX/pSrWuvu/baAMkBkmepBgZ+MiT89HM7b7x7U2CC1S3f/c53ChvtzSuXjIx82H06afny5a1+lkMPPiQ/f1WC5QNJjemieXPm7aJuw/P7m5v6e+mEbO7zwbnNq9KYfsv37lU2rj6o96McsP/+2Rv5r0NDQ/+S1OisUCgk72uap38yb3Ku+klWgOTwSJ5OxeaoDeMMkf+QNgPWT2qEQqFRBUiOEiB5qhUgGRgYF0gGhsDYCmCbQTLv22kDkjPgVqdi+4Em+xm8lXnJ8dDLnqX9LqAM4Akm59DM6xBIFsJWKFl16hmwgcRV4X7B3Je97GXdgGSe50LY9gIKtZ8nspyzwCeP89qe6TQ25C94pq/Ym5zSZ511Fg/QrkDybrvtVgWShRYWKWg8INm4VEmdpjVIFhVByP2KfMSiNajHRPbTtAPJDkTorwySc1ukMJTz+YILLhD22wGOiVoXHaTQH0JP+xypHgGSQ1OkAMnnn3de7b2029N7ynaFU1yd60XCHVvOO/fc2s+/7daHPn+9hYtqXXvkfx0+a0FygOQAyZ9dHVq1PEfnmWee2apbPnbCR9dMLuD08Y9//GOf+5znrioBmn09wfvKl70sQ+z7khrTSVtsscUjBucOPADeC79S1ufymdQtJ5144po+nz84uHt61qGj47qkPGTut771rb70+Utf+KLc55fV64tQKDR/cN5nzJuKudpHheSOj9DW06rYWLBBkz2RaVXSFhHKOBSK0NZRAiRPLwVIlg9TVKtevTJ5ZAJfVWBCPVwD2r7nPe8BCirz1/JmBdPuvPPOCRFoJ5dt9khuF7KYl+d73/ve1sqVK0ejrFxzzTXqBfwBgNoGCI4r5KvrL7/88tamm26qHusSSJaHVljcnu0LeARVK+Aom9JPXYW2vuiii1obb7xx12G385jw2mRHVSDN2ICjQiUbIyBVuO5Pf/rTcoePhlk+6qijRiEde9tss81aO+20E3vrJkeya8DPWu/Re/X0tN8F2L/qVa/iRd9R5tR0Bcna/q53vcvvS6+zPoDz6tFD/nhrnb4o7Suv8XoHb+XTtu5VhSa3/kzUmggQy12899576yf9tVZ/G/9NNtmktd9++42G5heNwvp/0003OQihTwF6h3TY/bggs3vI5/3a1742QPJEKRQgOUBygOQAyaHBwcEFPHcXLhgu3Xx/29ve1qpbNl62bI13avrxYUmNlAP0T2X398a8H+U79977IFBtNi9Nakw3Dc0duK1OburPfvaztdu92SZv/Zs38sD9YPWCZnPDUVAyXD6uPoSMt9xz9z0PhrUeGDg8qVGtUCg0PDzcTOvkfXn+mK/rr7e0zx9+Qs/z1Qnu9ddfywt58fBIX0EyL4FqhYRAk2fOQaZjjjnGxkjx/9Jf0qbDdkmNtRUKhQIkRwmQPMUKkMzbVsjS7B3XNSzZddddhR2s9Bg9MR2SXpI+Kztw/o53vMN7h8pQiuDPXnvtJV/nhMi95UTWjtxnRQFfIBJ44r0N0MtDEWQ57LDDRgHgpz71KR7ZACGwx8Nb3QGfrryuhRz2HmodA8nGWR/p467r5Zrtt9++CgjLSSzXsTpRNyCZN37XIJldOGTwpS99Sb9Xtp/ns3DSvNzlfl22bJm+Aw/ldTa+4NGaey9atAjglnu8K5Cc9ozYQCm840Gq3gCs67odc3Pg2GOPBcHbSjo3YaO32WYbr5+WOZL1NWgvV3lZ4QRyxBFHdO0R67XG1jPND33Srp/sDQqjnw9AnHrqqT5XVXrR65uJWhfNG3UyTvpp7LqonsYlpwF43etex6ZH/w/su+++o/1l/OULdwDkkksucYACpHaoxEGG2t77ivVUZMcIbR2aHgqQ/NLdPtR69WHnrKUNtzmkNkheuvnS1is/9dq1ddJraoPkcz+4Xuves5+zlj528KJaIHnXDUdaN2+34VgFSA6QHKIEH3/BQ7YMOHrD4rRYjXj4xbDWNyQ1kiyY15Xd35v5Lkq1Zy7Nm7drUmOaCUA6SP1GKsJbC81SpzjhbPxWt3vu/yQ1wOShNK7zm4Nlz/BmetzhZT510icfBMlzh16c1AiFQjXVbG4pp7n5M3kKWX+XLFzUT5AcHt/jlI2wpzzlKYcnNR6qUCgUIDlKgOSpVoBkoVTBJ7mOu/EazaFOeYxVwTQgRzhdIFnf8JwE/2qERPZeAgDqWmmvolLqU2wzMJLFozJ/b98ETC7CPs/we96Gb37zm9dAZoBHmOoPfvCD+p4HM5hYK18z7751LEey0N28ZHnZZgDVVVhr3rMVke6Ec+Y5akwmDSQDpRdffHFlrmOe5jz+2b3nuDarUyhoP8vdfOWVV9YGyRmqsqEy73j7RDfccIP+ys+tPd7s//DDDzc3RRTgMdpWvFNBSfefhiBZW/Sb/qv0iOU57tBAt2sje5dj2bzu1E+gKs9oXsnWIx6+5niFzVpvelkT9ac1r9baqI8oR5tstzbmNZG02zO03eEIXsQbbbSRww2juY732WefUcgsFLzxuv3222uF8dZPALU1MEByaBaDZGuiQxv9loMfXYHkV7xjpc+qa+nFy4+rDZKfs+NzfKZdS6895021QfINK9f3uXctnX7sklogeb+XtP+MPBQguREgOZRgxuDK1eGty0OrOo1YVT7xsY8/CBUHB/dIatDiocUbl93bhypvYsdbttlyq/z8B8DUpMY0k/yoj1G/FFa8Mry18GFV5TOf+nTRC3v/Bz2fm1/0u/ymrpO+/e1vj6vPt91q6+yB/rP0hu/hSY1QKFRf8wcGXpgO3Hx1ciBqKIUUbz3rWc9qPfNpTx/3vRYvXOjDKvUIUEPJM7/1hCc8ofWwhz2s1Wg0FjfalihRogRIjhIgeaoVIDmDG16GXl8HLnkNmCC8qTCrJQW44o0GsmbQx9uyMh0T+MpjD5TwPH1aUz6PA+O8jUslP64wve4P5ojYBnjzHu0k3tRgWO6jDAUBGtJGYEW9gXbASkqxCtgOuMoJqv49g2ShkKcbSFZ+9rOf8UrmvVjXvoAq4yG8cGXO4fPOO49NuW7CQXK+f526gWA8T3mzs5E8tnXmFjvkEVsXJJM+A95FAKiKEGA8QNi6MNnrzJdzzjkHAKzymjW/egbJvPMB61y3foNkX4Xkl+uX/ZStXbfeeqs5r289rzZwF+VByPWqkOyAu9erE5v93ve+Vxkq/oADDlCf7LVfSzk3tHlStiZutdVW5oQ1x3iZs9b50jVRRMQ99tjD/488LlQE1zkstnXRmLmnAyA85CvXRc+wPmtDgOTQbAXJDuD1fd8mOwAGSA6QHCA5tHjOnEV1wls7HVZVlr3pzX/LjTxwfw5rnZXCij5Qdv9rr722NZ7iA+SCBAg8P0Hau5Ia01h31cmLKg9TVdli0021mfTvo5IalJ6x1WrQsaj0GSeccMJ4clta5POG1ud631AJhUL//Pf//NZ//Id/WPXYf/mX1hMe97ikx/dJocc/9rGtf/mnf2495tGPbv393/89YNl6xCMe0Xrcvzy2q/s87SlPbT3z6c9ozXnWs0f1zGc+s/WMZzyjS4We/OQntx6XbPyRj3ykscj6ZiNKlCiVIJkWveDVazSy3vMDJAdInjSFAiTbrJeLGEwWdjfDgKJAhQxSAEShYa+44goHxytzHcv96ZC5OpHvjz/++FK46r4AdIYmdQGk1wBewseC1Z0kr6fwq4AziAOYOGQP3onI1km8OYW1zv1SDMNMue/1VwZX+++/P3g3oSDZ9Zdddpn2ADjFcaPcN1MCksE6Hui81wE84CzXKatYRwCeB6NxYiN1QrPnnLyT4JGcx5R9VnqZA7YZJFf1Zz4wIQR2bntdkJzvwSO+CnADpA6AmO/sJT/b16KK3tIA8Ec+8hGHTqrazCa0udfQ1nKPA+LmvbHpZCc9g+Ts7f6JT3zC3lfV+mUshHvOh1pK+8pc5U3OK9ucLCs89UUyyHWy9goF7bqydfHSSy8VrQGYzXWpZVdgr7765je/Wbo25jRFIDC4zPMe4C5bF9Wbl/7Y5xfXxFyXHJmC93xFKG+gWX2ECg+QPJEKBUgOkBwgOUByiDdpCrFauXCUnXS866671kzIdK9rkxpFLVm06Ptl95Y3aDzlrDPPfNDjbM7ge5Ma01VDc+ceop7DFeGtfVipyMVS8I4bvDWpkfW8wcHHpn9Sf1kwb6j0Gd7w9VrOOevs4phvbpx7VygUSuWjoFooNMt0X1KzESVKlI5lXjqwVxElIUBygOQZpADJaaN7/ZkIkjOgAPu8HsTbZJNNwFVAgXzvc6xcmAAMTz2gq5a3s5CmRQgAJIJl8l9Whd0Gbm10Zu/JsSpCHPcVqpEnMxhVVn7729+2Tj75ZB7J4BQo6udaYByIAYA8s5PckwCr97///VXhmYUKBnl6DW0NDgo5LPylvgJjSYhZ8M7v3Mf9Jh0kZ5jMA5N3Nm9EsF+dHCrQR+oJpAFkgCUQZ/yrgCiotsMOO+jzSQPJngEs8qisgmDyHNubKQLQsfI3X/XD7rvv3jr//PMz4OwKJLu/sMcAHTutqJtIecZPtADzO8P4XFf2Yow5n3gd+6rMb2tdEIq4TV7h7OGqfmX3Me76wLh6frZl9qJ9bCYfRugJJBehe51IfuauEOXWMfPeepghbj4gwGP2LW95C9t2cKay/62dwjwXgbs+A/cdjqhYg9iu8OQZsOYxayfjal45PGOdqfSStt5qD+BrfnAOqloXAWH/F4xHVX3IAQZzo6qf1LfogT0LxA4OnGyQHAqQHCA5QHKA5BCweSJDXq/CQ/bjH/94q1N5/3EPLgiDg4PbJjWKWrRg0RFl93bidDzl/3vbXvn5q5rN5lOTGtNVPIcHhbeem8Nb9+ap/bETPrq6zdRs7pXUKMjYfj2p9P4k7E0vZe+3vz1vZv05hSf956RG7wqFQqn8Y9LKAIuhWaQfJ72pUVqiRIkyNDD0ohRt5kfrBkiOEiA5QPJ6qaSN4O9Oe5BcAQTAFWFfP//5z4umReCssLY+x4K0IF7lvUACsADIzMCFstcpkCL8bxVw4fnp+QC3nJvgUs7TCXjIYbrtttuOQtRzzz0XOKr0/tUOoCpDPPdz/S9/+cvKdnnNWWedBRoBSwCg67PAHflBATgH6wHNyv4S/lle5XbhcwE9IXCF/K3qb3CTZ/WJJ57YOumkk0ZDK59++unGEOgC0yYeJFeAIflR2cXKlSvBJ3bAg1yfyisLpGXQWAUIQXqHAdRnskCyMaqVI1n54Q9/yHbZp3qulVfW9+yF/QLCcuTyUnXIIZduPZLdF3gzZ6pszt/1tVzMX/ziF+XrFWnAoQXhnIUs5lHN090hhsr7WTvkFM7zvU2/sRnPMq/L6mVshcjWd9mWXQdCiwhgzhm3nkGyuoDa7K4OtLdmAO/sls26H690fXXQQQfxilc3IL0S1irmKYBvvIqg3QEedgWmV80jc8Wa4fABO2Zf7Imsiw5mWIOMqwMXxqdq/Tj11FPB+rXCrPsdmF5VeCYbJ4BbfayDrs8Cze0PCwtvXO1XVsxzfa6fsmfzbNE7Jh0khwIkB0gOkBwgOZTyRT5z3ty5q0bmLyhbOLzB6Pim8BUvfdnqyZCgYlJjrOQsfs4GG6wqu7+Ti70UpyQ3WG/p3zZWBn6a1JjuSuG/b1HfJYsWl/a5nC+dyptf/4acn/ivY0OJ09DA4D6rDwgsKX2GN/FdFicNC33e/HL/NldCoVAqmyZdvw7Dw1Do10knJT2xESVKlLpFbv2HF5Xy7O8UIDlA8sxSKJWHJQ+ueQnm3DmTQXLBgxTkFZIZOAUhfFbs6rM8qAK0qAuNzTO79dZb8+513yrgBWQAkMI/g6O8VgmE5EksJKt9BzCksl3f/e535aBd41WY6wQKAy51ir4RDhxwB1NBwFwnoOmMM84AesBfz6ysE+9sG8XtABzYxKPQ8+oUkA4cI+MGgjkcoL95GU4lSC6OqbGSyox9GV82U7d4PYC42Wab5TGcLJBsjMBfcwvwq5Uj+mtf+xrbGN20NydoxYoVQDj7BZDB2q7mGJDMs1+bx+QNBvSyV35te2arwN0999xjLgF9+rkO1PcaXuT6va03MmWva2CYXVYUfaFe2ZbB0ww7tVEI955Bcq6POTc2jHhFndis6A36SF9ZT3jy1rIFhb0Dz/pizLqo7xwEEBnSs2odBLjxxhsdwhA6no1ZixwcAbatQQ4zVM4t4+dAgf6wLha93B0qYBeeV1HYnPoYE+ug+uQ6WSscRrIW+b9Sa110CMbBBHWZRdotHHpmlgIkB0gOkLyugOSQBe37STlERkd5gzG2XHnFFQ96I89pfiWp0U4JaP6q7N5O+fVSrrj88mKI51OSGtNe8+btqr7DQ+Whp53Oa/fm+db/ubXY51e36++hoaF5/l51QMBJwG6LMS/xhh6/QqFQ9lD2zdA6pFDoabyxGuMuUaJEWTAw8O/TFST/PG2Iv/aVrwp10N57vT1A8ixW2nx/RvrcfctMAslgBVjUx+KeIAtPOJ9723rZZi9gnrbAA9hZF1ADSuqc5WfPrAO7wDER2XhEj/WS5sXH2zGn/qpbH5/reZHm+vgZmPK8OgVABMc79BUJBQ269wxtQfipAsme38+S89busssuReg1maGtPRecBBHr2gmACMg5NGA85J019trDVrruN/NGqPmx7SfhtB1mAD0nuqi7dr3nPe8RdrqjDecw7dYF0LpXZxOAFMjvB0i2BvEsFkLfGjTRxTphXHjmtoHt+s46IEc0D1/rWt3DI9Yd609ei4BtNlT30A9gbA1Uh3Ze5OBv7fWmWJ9cJ4cC9HGdOnkNm3LgQuSJ3FezRFulPd/HJDVmiEIBkgMkB0heV0BySN7eOh6y3jSMLf+59z7FsNIbdgrpvHjBwv8uu/d2223X6qUcdcSRaxaD/PyZoLQI/VmdnRwu6RcnKR/a5sOPWNPmtIm4SVKjjXg/31b1DG/SfVjpphyTwhfl5y9sNucmNSZPoVAoFAqFQqFpDJJtyHpOqIO23GzzAMmzV1IdPT3phpkEkgGds88+e9Sjrg8FKADK5PrkcQdElIUIBi7kVAVJJxR6ARvqJRSs/Jwgcrs+k6uXBx3gDGZMYHF/3qrAWPas7QiX9CXva9Cxl+cARVMBkvW7vgRM+1B4pvLuFR4XBMz9NakgOY+Ja3kT99tuzSEerzxJyw4jOKzBe7VTH9iHY1vmeCWQ7L2uvGeFcAd3sw2X9ptw3Dywta2H52lTXzySc32EhNaPVySnCvBzogpvXXXfaqut7ON17KccdhuYFwIbEJ6okvM/a7//F+1syVoppYC0AWzdejIJBxP8D7Eeq8Nsyo98f7KNzceA5FCAZGkHHEDpu0TFCJAcIDlAcmiNhJ5O4ZbvXzCv3EPWKS8fYnK5L50YWzw8kkMs/yyp0UlCaKd/eqXhrZ3q7ba8/MUv8fykub9PaswU8d5W70UjC0v73CnRsW+Kn7/Bc1Z7I88d+F1So5PmDQy+1+uWrlee/9ppx27Kq1/+ihzK/OakxuQqFAqFQqFQKDSdQTKPlOU77xzqIBsgAZJnrxKoeEr6XPz16QSS5cstKQCjHJS8dIEp3mk9b/6DhSCfnKHyYQIQdYCcA9Dbb789r1yemn2FOdrzf//3f60LL7zQ4XlAtrRe+g1ofu9738vjs9pbu3evWm0VghhE9lzqVCfgSU5nobd5YHcNkkH0qQDJDrafeeaZxhYcBTR7Bpa8f3lG7rjjjmymbX/l3+29996VIFm+24033rgnkJzDELNbey4OYlSD0cpxAht5W/PaBR1L54L/xx/60IcA9Y71k3PXa66//noQ3jP6BvuMrfzJKSUHW6iCyOQ11gbzy3rDHrqdz72AZKnlOo6zOr34xS+Wa1goZXbWV3hrrstZLfS08Pn6iMpsK3smywt//vnnmzv99Ji2hgjLbT4Jm8+GOh76yb8HwOXNv/vuu9VlQg7WsFG2Kj+3qBGePYsgsq+/Stoo5d19dOQejhzJU1ECJAdIDpAc4r16OYPeYP31S6HjZz/72VYun0vfu4YGBwY+lNQo09Il6/2h+t71y03pQ1t+/vxm86tJjZki3tO8uIUUr/JK9sE9lwvOP79+KO95817idQuHh0vvv+eee9btcgtu4fnNo5Mak6tQKBQKhUKh0HQEyVEiR3K1QiMjI09Mn8G+OE1AMhABEAMjbSW0Lphmwx4IEtZZDmLghSdpHagjXKlD49/4xjdGgdU222yTcw935d2ZvYEBQLDi2muvVY8cDrXr3LsgF1DL2/qYY45pbb755j6b16pX9lIElj71qU/xVOQdWh6ataJOrnO9+2gbgCOkLs/DGgDO38EuoYyBNF6gvAO11b2BPc8h3xsbY+jvYDg4w6sQfJtUkOxwgfyovFXBoa9+9avAfq1w5NridfoMTONV+uY3v7kSuvvK6/XrX/86G29n9xlKu18GjD3DZLaljTw72WwOVV0zBLCwvzww1dd9HBhwCIR98Nxt2wZiA+AkmNopfHwGt3vssYdDC+rIQ5wtspGubVhdhVwGu3nI77zzzjkccleeow5rsAeRCNiD+oC3bMKzsvzsuf7mNWzeuGWQnG2wUx/5PdsXIrnkAIn6Wx8cLLAOmmNy+XK28Wzj2RVkZ7dswbNPOeWUPNc72W4pwHXQ5MgjjwS5eSiXhkMvSzegTnIqy629cuVKB37kH852UmXr+s+4uU7/qIv+KZ/LFXXK9m+c2AJ7FrLe+juLIDLl3OJ3Jzt8VaPReGRjxpQoswkkL3nZxq3137DdWlr8wtfXBsnDL1jQWrrZ0rW16dLaIHnHf53f2nu7BWtps9fPrwWSXzkyr7Xni0bGKl8bIDkrQHJoeM7wS4HNRSMjpdDRm6tc3vDa12bP2Aee+tSn/mNSo0yLFi68oJ/hrd/9zneumczDAwNvSmrMJPHirgPvvanMZfttts0e2KvSJsRAUqOT0ssflk7N/C9YXXJ/J3Vr5zL5QApHlPt8QbO5flJjchUKhUKhUCgUKoLkDbZOoV/f/fJRbXjAiwIkB0gOTWMNDg4+Nm0IHzsdQDIBuj6HC4nYSWApQAkS+OwIivGGBGABT550NvmzwBkeaTfddBMo6MA4wMdbzX1s/vcMANQhA2WA6uijj3Z/3sTq4hA2L8a16kMgHKgGbFxyySUghxDWwhsLP6wf3LsnuM2DGdgDckDlCy64ANwDmTwz16Ftne64447ROl155ZWt89JG7Kc//Wnem3IdamNuc1fgEuxatmzZaDhwbbSfALJefPHFngN6+149eQILvQzOGKOcF3pSQTIP2+OPPx70W2Nf+sBhATDylltuKY4p+/Izb2Le16Ah4Miz1vVV9c+QDrgGT0GvTrbPE78kL3VtOyF2wlPa3BEFgKeuNmhL2/njb8bqnHPOaX3sYx8TWhjoz/UhkE8I79I28Cb1bO2umldex4N6xYoVxs0hElCRt3rbeUXqaoyKdT344IN5t4PdPc/3HNbewRPewurDs9iaou88D/QG14VU5vXNZoSd32233VyrXdYc87O0j3j1mjN1Yak55oAB+xFeWnoAc8pcth6a253muygGgOjnPve5Ubtdvnw5+Fo+1yvq5KCDNYw9WK/VCQh2eEduZ+PXrk5sTZ28Rj+6hn2ymbz+dHkAQBvsMxbroq0iUVij2TVP5wzxx4qdWcf9/7BWG3MHEoyTiI1sqgRszwqP5CtS21+Q2OTDG1GmXQmQXK0KkFyhCpBcoQqQXKkAyUUFSA4NDPwEdMz/oNrJm1flivSGLU+koWbzsqRGlRY/61lDVeGtvXGoW175spflOvxxJvb3gjmD71H/4aH5paDXh1zl3nvuXdPnaazuSmpUaf7c5odXh7deWp2LuUZ5/Wtem8Na3xVhJUOhUCgUCoWmNrR1mQIkB0ieloocyf+QPhPvNo02p/OmfKnGQjHAbqONNhoFbWDsIYcc0jr00ENHvxKICZDwlHvVq16VAUN3UKIC5mRg4f5vTZuzIB2ws99++xXrQsAWT2aebAAXyCFUa76Peo2nLvl6np2AprrwDM5901bACAion8B8uQhd756161VRJ+nJAFP3BvI9Z9ttt9UHfgee8dwEZ4qhoKfEIxkMU2dfjSuQpc76sTimvgpBDhaCjLw4gb0MQ/tp99TPOZYPYxgT46ANPFHH2isHDn8zXuYZ78uiXXTTBq/pxm6y9zWobF451CCKnT5vZ8fqal4B0OoKdLtXP+a764v1ectb3uKggb7LtuxnXsL+xt6A47H5hd2n936qWIMAXHPInOLlbp0BrYt9ZHzN97e//e0OwACiDm2w8/pzvbqvivNHnfSN51lnOtXJ37zGa80l17rPeOqUr8/Q3Vz2v4AtdbT5XCd7ztZxMFufWqOMabHP81jNQpBMp6U+WJDUmDkKresg+aspUs2Gz3t+LTlg1668+AUb1rr+wx/6UL+fjy/VuvZ9xx4bIDkrQHJoaM7gfzHqJYsWdwSOTqwKLbLxsmV5AqwSpjmpUUfrr7e0NLz1Jz7xiVadcuMNOay1DZWBryc1ZppSex8lN7U2bPiCF3TMS+2knnL4ihVFb+Ada3irCKP9ijrhrb1Jqyp33nFHsc/fNTX9FgqFQqFQKBQSGSZAcoDkmaeQz4BpU/GNaVP4z9N/47oapPAiJT8XvT3BDLl2/W2CPcfUIz+P/JzrUlSxThMGIzLI8YyUFzH3TTt5rfrkOmWQOlH94xlF+R1VgLTJBclj662eY8Y093Put1z3GTl3tCHXvziHivbs+8luX7bP+vNqQm2Ycn062XIRMk52fxXHM8/5OmvQRI9frpO1aGxd8uvya3IfTnhdyM/t+qc4t/P6WQ20ZxVIXpXA/AdTXz0jqTFzFFoHQXKUAMkBkkPl4Zadqv3vFBZpoze/5cFNqmbzuqRGXS0eWXhuGdB04rFOOerwI9bUYaTZfENSYyZqcE7zy9owf3CeE9EP6Y/PfOYzGZyvCfOQxui+pP/H3l0Ax22uaxx/w8yZG85KWpntgKQwU5mZQ2UOlLlNyszM3N57D5SZmZmZ2zB5k9jfedbjdb+j0Wo3G1jw88z8yuBVwOP955MkHbglVFP8O7/YZvjtraHuuUJhw6+8+TskR+zBIERE2UFEREWRojLbNHfVWZY1iyGZITl3EdYUv1i43HXdL6Bg3uAOCqU587Fo+PGEyqmQnOIa8sdOFuTv92P+mM+XnxP5/Sn0NPJK/PHRCOxdQPIHMSRzDMkMyYUpGp2AALkmahgq/rxk3BIZqupOtOLv+60yTbMHSLpKSkp6p7q9dfz5KmGrqalRw7whieczLwfJV24Xt5MdMVbXxWTTUpNx6+jttt2u7oTwVVdeqR647351wrHHqfLihutfa0ciM0DSZhiXxP/dqsqq0JB8ww03pHUrcfz3PgXJMURERERk2xUMyQzJlPOnknvBA57n8Q3yQMSQTEREvucj/+x53va2bbcCyRvEkFy4Y0hmSCbDMEptw/w+Hi1BBcBfj3xUWlraDWRd4TYlS8OC5rx581TYXnj+ee1jsf4fJJ/ZEXsmXksNqBRqcYL5CZB1UYxbj9ed3C4qDg3J8WckJdvbb7/d8HEUmeYpIERERESUUxiSGZLzAJWUlHRwHOd0z/Nq8HtFlAJDMhERQ/JbQzARaQr5M44hufDHkMyQTPHTxlY/a5ZlGBdHDeNW2zQvhMO6d+/eASRTOOl8c1jQHDdunKqurlbJdtQRRzSE1eLi4oEg+c6Mx17D+CLa36gJDvfmItuyTgPJSMT4Kv7f8jwvNCbHg7FvvuczQzRqgxBRblBKdYcJMBO2hgpoD0IFpy2017QASSAiYkhmSKY8gNNEeObj9q7rLk7j9tZEDMlERAzJN+Pn1QgI8RnJHEMyQzI1Cq7rtsBPpKG3t37ooYdU0JYuWarKE7fZNow/QApJ375929i2Pa7u9tWI9gjM21X0rugPsj5swzo9fs0qKypCQ/IpCMb+xWIx5QwYmHjz6jUQIsoupdRW8Dj8GnYDBxgNQgXjddB3FAgREUMyQzLlD6yJ53klruu+w5BMKTAkExHRKs/zDorf0QSE8sr6hGSOIZkhmaiqovKnsKA5bdo0FbS77riz4WQsQutFIJRaSUmJiWtWWxK1Q0PyiBEj1OLFi5W+h/79779va20YB4EQUXYopZrArbAu+19oDkIMyUSUfQzJHEMyVVVVdXFd9xqGZCrkkDxp0qSUIXnBggXqiiuuYEgmoiR4Ghk+wJ1MRopIM+B4a2uOIZkhmRoPhM2Z/ojp9/nnnyv/ttt668QPvJoePXq0A0kP2Yb5XPzauY4Tet3vuOMOpW/6flMT13yla1mdQIgoO5RS50AmOx+EGJKJqPGEZDvOMBtEDYZkhmTKpdtbO46zEywDRVSIIXn8+PHqoosuUj/++KP6/vvvA33wwQdq/vz5ASGZiIg8z1uL31+HxyP3A6G8s74hmWNIZkgmGlg1YE1Y0Lz66quVvnffeVd7bnDkLRBKX7FhTI1fu/LSstCQPHXqVJVY/Au+ItOqfzPSvBOEiLJDKdULqkHfErgE9oUtYTrcA2tBXy1vc10Q/he+0kwDISJKKCsr6xWNGDWgkmFIZkim7MOaIJyZiG3P8lQyFarhw4erPffcU5166qnxx2gFOu6449Suu+6qcNrO9+8TEQGfjfwHYuQ+FRUV7UEo/zAkcwzJDMm0nsrLyh4JC5qzZ89W+uYcPSvxg662zLImg1D6cNvqNvGfsGzDVJ7nJb3uo0aNUomdd865f9/W2jTHghBRdiilDgZ9a6EUJMA08G8+CBERFbaoae4XNYx7/KxIpIYhmSGZcgdCcjvHcWa5rlvNmFzgGEKIiDLAnz89z3sSv+hsoIg0BY63tuYYkhmSqfHp0qVLJ3wyrE0WNM8//3yV2Ddff63KiooTt+n7C4QyYBiXxK9hVWVl0pC88847q/hisZgaMthR+OfB/BCEiLJHKXUB6HsTJMTHoO8ZkBSaQxVsDv1AMtASXBgLrUFyRMavLYQJm0NfkAx0hbGwGZRCG5CNqA2MhSEZfttYsDn0BCGi/IKYHGNIZkim3IE1cxzHhfcZ26iQb3GdBt+/R0TEiAxL8fPjHJxq7QpCPJHMMSQzJFOjVV5a/k5QzBw2bJj67LPPVG1trXrj9dfV5AkTFf55ANM8HITWXYlpluAa1hZZlsKtUQJD8l133aXiu+aqqxuuuW1Yh4AQUfYopY4A/8aCJDEZDoS9YQcYA5JEGTwLq0DfAngQeoGkEIEnIAaJVcPzUA7NIab5AsTnS4hpdgMJcAnENLdvpNf2FsQ0vaENXAs/g77f4ZA0Y+7F8Av4txpuCTlp/hLENEeAaLaHmOYaEJgCr8FqSGwNPJFGVG8B58IC0PcFHAwC16b+dvMjIoZkjiGZLMvq5LruGRDjG+dEREQJDMn1j79weBqZJ5I5hmSGZGr0XNdtUVlevvq/YqbjqNEjRqrxo8eoAeUV2jPdwDC+AKHM2YZxU/xaluKEtz8mz5s3T3315Zfq4gsvVMVWNHEC/KvxxvjWIESUPUqpYeBfDdwL20F7kAwcBCshbH/A1iBJbAsLIdl+h0Gg7zsQn+9B3x4gAa4AfXdvpNf2Puhz4HUI2w0gSQyHLyDVViYJ//7/91Egmh0DPpYDYA0k2wKoAgnQF16DsJ0DN6X+dvMjIoZkjiGZsGau6zrYxzyVTEREVI+PBFiG38/laWSeSOYYkhmSiepV9u7dr8iy/ygrKVGlxcXKNsyGcKyzI+Z7ffv2bQNCmSstLe0WfxOq7pqaliorLqlTYhcFXfM1RaY5BYSIsqv+RO8bkGwxeBqOhUHQBCSFLSFoy8C/WhgA4tMHVkCq/bWJQ3LGry0kJP/pu941Sf5blSA+3WABpLslYK1nSP4T9K2CoN0PEuDfaV7DFZmFZCJiSOYYkqmkpKSD67onO46zgjEZiIiIGJKf9jxvqIg0A44nkjmGZIZkooRiw9gBseSpulPHEeO3aMT8BX/8udXf/Fc0Gp0AsmHQgGj0f2zDeDQo2EPDG1X49hgLkjOIGJMNWAjp7Bs4HrqE3LL4M9D3pBYve8P/gb7/B/G5Dfy7AmxoC1PgG1CbMCRvqNf2fpLrujO0h+5wCfh3HojPdQHx9VDoDQbMgeWg76pMQ7Jvl0ElNIUR8Bvoqwk40T4R/HsKbBDoAVdCYhmEZCJiSOYYkglrOnDgwArP855zXbe2Ub+BTkRExIj8KxyEP+4EQvmNIZljSGZIJsp7FZbVP2qaB9qGdXpRxLwccfkcBPw5iM0jTz/99KYguYWIlFKD4Q1Id59CPxCfQ0DfT9A2jSDrgNQzoRb0XQni0wtWbsKQnPlrCw/JMfBANE0D/rkHQTRFUAP69gXxOc93svkDaJlhSPZ/LLpdwL9KEM3ToO9LaA7icxtDMlF+heQp5bY6fkx5SltX2WrEEE+devLJWUZ45A9DcoGL33nLcZxpruv+2mhPJRMRETEir4ab8cdREWkCXIGcSB7mDWmkX8vQ5AkTGZIZkomIiDa5RLzcGx6CGKTaD1AGorkd9F0AEuAk0Hc6SL1tQN9q6A0S4PJNGJIzf23hIfk+kADXgr7HQDR7gL5foVmS218fBCOgA4gm05BcBuLTE/wbDqL5DfQdAhKgmCGZKD9EI8ZKUJS/7IjxNQgVHqxJZWVlj8GDB1/hOE6s0cVkIiIihuRaeHfQoEFTbNtuBUL5LxoxX+LXMVTHMH8HoXC8CERERBuBUqoj7AE3wjeQbJ+AaF4DfbfB3gGuBH03gtSbBfq+Akliv00YkjN/beEh+QyQAOeAvqdBNKeCvudB0pZ5SK5Jcoq4Ffg3BqReO/BvFEgSy3M/JBORFTEujkaMajtirlknhrk6J9DSIsM4CoQKE9bccRwXe6nRhWQiIiKeRv4JZpWWlnYDocJgW9bedsRY0Mi/jqGIucw2rBNAKBwvAhER0SZQf0L0aqhJEQsXQCZ7FKSeP8Q+C5LEhAxC8l5hJ4BDQnLmry08JM8BCXBaipB8J+i7dxOF5GUgSdSGfN+oAv8MkCQ+W/+QTERERPFbXONU8h6u637WOGIyERERIzIs9jzvkqqqKktEmkL64ziO4ziO4ziOU0p1hYmwN8yFC6EVSIhdwb/DQOpVQyb7wP9MX21PgiSxTQYheSpIgFtThOTMX1t4SD4UJMDJKULy3aDv35soJP8BksTakJBcDv4NBEni5w0TkomIiMiyrE6IyUd6nvc9YzIREVHBn0Re6XneHYMwEWkOjWYcx/2nvXsAjiRt4DD+bpKz7bv1bn1zl5lGsnW2bdu2bdu2bdu2bduc77mqQ1fXhNt70fNU/eqdZdKl4F+dtuIyMzNzSE6Rb2GEVjTUGFT3aeUu0v0wazukCH9ZH9leRGjBpp0YktdDqOGS2kNyoddWa0jeqJND8l7I9hhCC8YpcEj+qJND8tjItzhCDWPiD4dkSZKK8ffzktM03T9Jki8dkyVJvZMjMn7D7WmazuVzkSVpcOjkP5QkSdVqtQ6fIdv1CK0YAmpxCL0R2fZHqGG0Dvy46h8xMUIN17ZjSH4F2bZDqOHx2kNyoddW5JC8CrJ9j4laGGW/wwe4BUdihi4YkgM+RrbzEGrYCOSQXBRJkqi+XC4PrVQqx6Vp+pljcq8jSfJO5F9wVxRFyw4bNmw8hL5OkkbiH0uSpL+eq5tvXzQg5EyCh5GvGeEvOyDbexgXIWN0vIKPcDuOwdwIf5kMP9d4n0LObKA2h+SHkO1ChJw58VutIbngaytySB6OP5BtT4ScdZHtO4zbRUPyscj2GxZAyGjERw7JkiQVj+rTNB0Wx/FxSZJ85ZgsSVKvuhP53nK5vHSpVBoXQaUgSSPxjyVJUrVabcLPyPcSDsXG2B6n4FvkexAhY3x8iWy3YCoETIDDkW9mhIwjke0P7IPpMDHWxrftHJLPQb51MCbGwTL4GPkuKP7aChySgdOR70g0/+VwfItsZyN00ZA8oMbf+Q3nYyechR9Ao2JIliRJVB9F0f8qlcopSZJ87ZgsSVLPH5F5/SAj8opF34ksSd6RLEmSdyVvgc70HRKEnO2R7w+8jp+R7wqEnInxJTpW7SF5adTqB/yIv/s4PyQXfm3FD8mT4yu0t08xsMuGZGBPtNWb+HLUDMmSJIkakiSZKY7jYzk/7LFjsiRJjsg/4nZ+vQKPMRsfoWiS5DOSJUnyeckH42e0t8+wIEIL1m/n/3crxkaoYRl8h5Z6D4e3OSQDB6G1LsOS+SG54GsrfkgGZsebaKsfMRtClw7JwNb4A7V6GlPhbWRbAaEYkiSJ6pMkGYS94zh+BdUeRZIkR+Svoii6FAs3NjaOg4AMSVJB/5EkSapWq/1xDn5ES72BwzExQhtmxl34CvnexBYYA6EVJTxRYxC9HlNi5XYNycD+eBfZXsHeqMMC+SG5uGsrfkjOGQ8n4Qvk+wUnYzqErhySc+bF0XgcX+BG7IQJEGoMyYsgFEeSJFG/JEmmiqJoY84H4zj+qXvfnSxJkgPyX+cbnMdyjujfv/+YCJAk5RT8H0qSpGq12g9TY1asguUwNwYidNIALIF5MQj1CB0wEebHzBgdAWjfkJwzEIthKoScwq6tC02DhbAwhmNMhB7oY2SbA6F4kiTpz+cpViqVReM4Pi9Jkg+67ZgsSZIj8g+c9/Jxe8umpqaBIYQG5DIzMzMzM7NODMldTKficVyBI7EVJkfImQS/I9uUCKOGJEmihnK5XIqiaMckSe6N4/hHzqqjsiRJ3WJA/oPz1SiKTsPCAwcOnAChdZKkvnvxkiSp5w3J2h35TsZYCH+ZEDcg22sIkiRp1KMxmpubhzIqbxbH8QP4vmvHZEmS/DHWURSdwDk3Jg0h1MHMzHpVZmZm5h3JasbvyPczHsdz+An5lkf4b0iSJOpXqVQmbGxsjGnrNE1vjaLoJ+9QliTpvxuQ8VIURcdg4VKpND1GR2g/SVLfvXhJktQzh2TtjI50PYIkSeqaQXnmmWeeuEJJkmwRx/FtnJ9xVosZlSVJUu7j6rd4HkemaboIvzf94MGDx0CQJHVc3714SZLUc4dkLY370VrvYS3UIUiSpK5DdX8+izFJkpniOF41TdOTOR/h119i5EZlSZIcj39J0/Q1Xl/F6+05Z4uiaOr+/fuPGULoBzMzMzMzM+vEj7YeHZNkTITQY2gQ5sYq2AbrYl4MRANC9yFJkqjf1FNPPXbC3VFRFKVxHK/GuXelUrmW8y1+/yvOn7LfHIckScqMxvgeH0dR9DC/dzyvN03TdH4MK5VKE4cQ6mFmZmZmZmZmZmZm1iPrlyTJ2I2NjZM3NzcPZUyepVwur8jv7YRTcVWapndxPo7H4jh+hvN1zrd7OUmS3krT9Lm/PgY+wev7OW/G+VEUHci5HhbgdZmPo9PyeoIQwugwMzMzMzMzMzMzM+t19Zt22mnHGj58+CRRFM0wYsSIIQzLpaa/YmyeOY7jhbBYbyZJEh//OOPZ0zRtbqIZZ5yxMUmS4YzGA0ql0pR8jBw/hNAAMzMbxf0frafpICEIhIkAAAAASUVORK5CYII=)\n",
        "\n",
        "샘플에서 DNA를 추출하고 DNA를 무작위로 자른 후 sequencer가 읽을 수 있도록 library를 제작합니다. 이후 이를 sequencer에 넣고 DNA 서열을 읽어서 A, C, G, T의 서열이 나오게 됩니다.\n",
        "\n",
        "Sequencer가 library를 한 번 읽는 단위를 리드(read)라고 합니다. 한 샘플을 Sequencer에서 서열을 읽게 되면 수백만개의 리드가 나오게 되는데 이렇게 리드로 구성된 파일을 FASTQ 파일이라고 합니다.\n",
        "\n",
        "FASTQ는 네 줄이 하나의 리드로 구성됩니다. 첫 번째 줄은 헤더, 두 번째 줄은 서열, 세 번째 줄은 구분자, 네 번째 줄은 각 서열에 대한 퀄리티 줄입니다.\n",
        "\n",
        "이렇게 나온 FASTQ 파일을 reference sequence, 즉 기준이 되는 서열에 mapping (alignment)을 합니다. 이렇게 mapping이 된 파일을 BAM 파일이라고 합니다. BAM 파일에서 기준 서열과 다른 서열을 찾아낼 수 있는데, 이를 변이를 찾아내는 과정, 즉 variant calling 이라고 합니다.\n",
        "\n",
        "다시 정리해서 말하자면 샘플 -> sequencer -> FASTQ -> (reference sequence에 mapping) -> BAM -> (variant calling) -> VCF"
      ],
      "metadata": {
        "id": "MrRYtW9TsipX"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Sequencing read 확인\n"
      ],
      "metadata": {
        "id": "mpjLpVAqs-Fs"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### FASTQ 파일\n",
        "샘플을 Sequencer로 읽게 되면 수백만개의 리드들이 나오게 되는데, 이를 저장한 파일 포맷을 FASTQ 라고 합니다. FASTQ 파일을 살펴보겠습니다. data 디렉터리에 들어가서 fastq.gz 파일을 열어서 보겠습니다.\n",
        "\n",
        "모든 data는 /content/drive/MyDrive/AMC_NGS1/02_linux/data 에서 확인할 수 있습니다."
      ],
      "metadata": {
        "id": "tIe3_KSccPMR"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "cd(\"/content/drive/MyDrive/AMC_NGS1/02_linux/data\")\n",
        "!cd /content/drive/MyDrive/AMC_NGS1/02_linux/data"
      ],
      "metadata": {
        "id": "dxBbY1M-ter1"
      },
      "execution_count": 41,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!ls -al"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "T6uOxgdMtKvV",
        "outputId": "76af5cd4-5ac7-48f5-9aa6-cfae54019588"
      },
      "execution_count": 42,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "total 72744\n",
            "-rw------- 1 root root 16013759 Aug 20 02:14 normal_R1.fastq.gz\n",
            "-rw------- 1 root root 16698858 Aug 20 02:14 normal_R2.fastq.gz\n",
            "-rw------- 1 root root  4368573 Aug 20 00:54 sample_1.fastq.gz\n",
            "-rw------- 1 root root  4376708 Aug 20 00:54 sample_2.fastq.gz\n",
            "-rw------- 1 root root 16120236 Aug 20 02:18 tumor_R1.fastq.gz\n",
            "-rw------- 1 root root 16909873 Aug 20 02:18 tumor_R2.fastq.gz\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!zless --help"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "l6lVrG6gt5uF",
        "outputId": "6720cab2-8802-44e8-a3e5-5b7afd824beb"
      },
      "execution_count": 43,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Usage: /usr/bin/zless [OPTION]... [FILE]...\n",
            "Like 'less', but operate on the uncompressed contents of any compressed FILEs.\n",
            "\n",
            "Options are the same as for 'less'.\n",
            "\n",
            "Report bugs to <bug-gzip@gnu.org>.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cd /content/drive/MyDrive/AMC_NGS1/02_linux/data\n",
        "# zless sample_1.fastq.gz # 터미널에서만 실행"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "L1qCrTnTanuT",
        "outputId": "9676c8db-d2d9-4e2e-896b-627517dbc42b"
      },
      "execution_count": 44,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b7\u001b[?47h\u001b[?1h\u001b=\r@SRR1518133.318\r\n",
            "CCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAGCATAAAAAAAATAATGTTTTTCATAA\r\n",
            "+\r\n",
            "CCCFFFEDDFDDFHGBGII<EHIJGIIIIJIIEIJ@F?CFFEGIGIJGIEDEGIE@HGIHDCG@CEHEEHFBCFED\r\n",
            "@SRR1518133.319\r\n",
            "TGTTGTAACAAAATTGGGATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAGCATA\r\n",
            "+\r\n",
            "@@@FFDDDHHGHHJJGGGHIJIIIECE<CEHCECGBFB?D9?DFGEGIIJ@FEHIJIJGBGHCHGGIIJIEGGIJC\r\n",
            "@SRR1518133.320\r\n",
            "TCGATGTTGTAACAAAATTGGGATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAG\r\n",
            "+\r\n",
            "B@@FDDDFHGHGFGFFHHBHGIEIIIJJIJIIHGHBHCFFGHIIIFHIIBGGHIBGCGEFHGH8=FHEBGFDBGI>\r\n",
            "@SRR1518133.321\r\n",
            "ATGCTTTCGATGTTGTAACAAAATTGGGATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCT\r\n",
            "+\r\n",
            "CCCFFFFFHHGHFHGGIIJJGIIIJGIIDGGIIJIEGIJIIGID>FHHHJJJJIEIGIJJEFGGIIGIHGIIIIIG\r\n",
            "@SRR1518133.322\r\n",
            "ATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAGCATAAAAAAAATAATGTTTTTC\r\n",
            "+\r\n",
            "CEECFFFFFEFHHGHEJJIIJJJJIIIIIIJJJIJIIGGDIIIIGHJJJJJJJJJJIJJJJJJHHFHFFFFFFCBC\r\n",
            "@SRR1518133.323\r\n",
            "TTGGGATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAGCATAAAAAAAATAATGT\r\n",
            "+\n",
            "\u001b[7msample_1.fastq.gz\u001b[m\u001b[K\u0007\u001b[H\u001b[2J\u001b[H\u001b[H\u001b[2J\u001b[H@SRR1518133.318\n",
            "CCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAGCATAAAAAAAATAATGTTTTTCATAA\n",
            "+\n",
            "CCCFFFEDDFDDFHGBGII<EHIJGIIIIJIIEIJ@F?CFFEGIGIJGIEDEGIE@HGIHDCG@CEHEEHFBCFED\n",
            "@SRR1518133.319\n",
            "TGTTGTAACAAAATTGGGATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAGCATA\n",
            "+\n",
            "@@@FFDDDHHGHHJJGGGHIJIIIECE<CEHCECGBFB?D9?DFGEGIIJ@FEHIJIJGBGHCHGGIIJIEGGIJC\n",
            "@SRR1518133.320\n",
            "TCGATGTTGTAACAAAATTGGGATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAG\n",
            "+\n",
            "B@@FDDDFHGHGFGFFHHBHGIEIIIJJIJIIHGHBHCFFGHIIIFHIIBGGHIBGCGEFHGH8=FHEBGFDBGI>\n",
            "@SRR1518133.321\n",
            "ATGCTTTCGATGTTGTAACAAAATTGGGATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCT\n",
            "+\n",
            "CCCFFFFFHHGHFHGGIIJJGIIIJGIIDGGIIJIEGIJIIGID>FHHHJJJJIEIGIJJEFGGIIGIHGIIIIIG\n",
            "@SRR1518133.322\n",
            "ATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAGCATAAAAAAAATAATGTTTTTC\n",
            "+\n",
            "CEECFFFFFEFHHGHEJJIIJJJJIIIIIIJJJIJIIGGDIIIIGHJJJJJJJJJJIJJJJJJHHFHFFFFFFCBC\n",
            "@SRR1518133.323\n",
            "TTGGGATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAGCATAAAAAAAATAATGT\n",
            "+\n",
            ":\u001b[K\u0007\u001b[H\u001b[2J\u001b[H\u001b[H\u001b[2J\u001b[H@SRR1518133.318\n",
            "CCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAGCATAAAAAAAATAATGTTTTTCATAA\n",
            "+\n",
            "CCCFFFEDDFDDFHGBGII<EHIJGIIIIJIIEIJ@F?CFFEGIGIJGIEDEGIE@HGIHDCG@CEHEEHFBCFED\n",
            "@SRR1518133.319\n",
            "TGTTGTAACAAAATTGGGATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAGCATA\n",
            "+\n",
            "@@@FFDDDHHGHHJJGGGHIJIIIECE<CEHCECGBFB?D9?DFGEGIIJ@FEHIJIJGBGHCHGGIIJIEGGIJC\n",
            "@SRR1518133.320\n",
            "TCGATGTTGTAACAAAATTGGGATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAG\n",
            "+\n",
            "B@@FDDDFHGHGFGFFHHBHGIEIIIJJIJIIHGHBHCFFGHIIIFHIIBGGHIBGCGEFHGH8=FHEBGFDBGI>\n",
            "@SRR1518133.321\n",
            "ATGCTTTCGATGTTGTAACAAAATTGGGATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCT\n",
            "+\n",
            "CCCFFFFFHHGHFHGGIIJJGIIIJGIIDGGIIJIEGIJIIGID>FHHHJJJJIEIGIJJEFGGIIGIHGIIIIIG\n",
            "@SRR1518133.322\n",
            "ATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAGCATAAAAAAAATAATGTTTTTC\n",
            "+\n",
            "CEECFFFFFEFHHGHEJJIIJJJJIIIIIIJJJIJIIGGDIIIIGHJJJJJJJJJJIJJJJJJHHFHFFFFFFCBC\n",
            "@SRR1518133.323\n",
            "TTGGGATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAGCATAAAAAAAATAATGT\n",
            "+\n",
            ":\u001b[K\u0007\u001b[H\u001b[2J\u001b[H\u001b[H\u001b[2J\u001b[H@SRR1518133.318\n",
            "CCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAGCATAAAAAAAATAATGTTTTTCATAA\n",
            "+\n",
            "CCCFFFEDDFDDFHGBGII<EHIJGIIIIJIIEIJ@F?CFFEGIGIJGIEDEGIE@HGIHDCG@CEHEEHFBCFED\n",
            "@SRR1518133.319\n",
            "TGTTGTAACAAAATTGGGATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAGCATA\n",
            "+\n",
            "@@@FFDDDHHGHHJJGGGHIJIIIECE<CEHCECGBFB?D9?DFGEGIIJ@FEHIJIJGBGHCHGGIIJIEGGIJC\n",
            "@SRR1518133.320\n",
            "TCGATGTTGTAACAAAATTGGGATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAG\n",
            "+\n",
            "B@@FDDDFHGHGFGFFHHBHGIEIIIJJIJIIHGHBHCFFGHIIIFHIIBGGHIBGCGEFHGH8=FHEBGFDBGI>\n",
            "@SRR1518133.321\n",
            "ATGCTTTCGATGTTGTAACAAAATTGGGATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCT\n",
            "+\n",
            "CCCFFFFFHHGHFHGGIIJJGIIIJGIIDGGIIJIEGIJIIGID>FHHHJJJJIEIGIJJEFGGIIGIHGIIIIIG\n",
            "@SRR1518133.322\n",
            "ATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAGCATAAAAAAAATAATGTTTTTC\n",
            "+\n",
            "CEECFFFFFEFHHGHEJJIIJJJJIIIIIIJJJIJIIGGDIIIIGHJJJJJJJJJJIJJJJJJHHFHFFFFFFCBC\n",
            "@SRR1518133.323\n",
            "TTGGGATCACCTAAACTGAGTCCAGCTGGCTAACTCTAAATATATGTGTATCTTTTCAGCATAAAAAAAATAATGT\n",
            "+\n",
            "\u001b[K\u001b[?1l\u001b>\u001b[2J\u001b[?47l\u001b8"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# terminal 에서만 실행\n",
        "# cd /content/drive/MyDrive/AMC_NGS1/02_linux/data\n",
        "# zless normal_R1.fastq.gz"
      ],
      "metadata": {
        "id": "10J1dLrCtDl9"
      },
      "execution_count": 127,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "- `zless` or `less` 사용법\n",
        "  - space bar 사용시 다음 장으로 넘어감\n",
        "  - 맨 끝 줄의 \":\" 뒤에 q 입력시 종료"
      ],
      "metadata": {
        "id": "7HWaEAKmusqY"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# terminal 에서만 실행해보기 파일 한개 열어보고 종료 후 새로운 파일 열기\n",
        "# zless normal_R2.fastq.gz\n",
        "# zless tumor_R1.fastq.gz\n",
        "# zless tumor_R2.fastq.gz"
      ],
      "metadata": {
        "id": "QFAXWAxntbSf"
      },
      "execution_count": 128,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "### FASTQ 파일 구조\n",
        "- ppt 자료에서 설명"
      ],
      "metadata": {
        "id": "fsAQpiDXbl-N"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "- 우리가 샘플로 사용하게 될 fastq 파일은 tumor, normal tissue 유래의 sequencing data에서 21번 염색체의 read만 추출한 fastq 파일입니다.  \n",
        "\n",
        "- fastq 파일에 들어있는 전체 리드의 개수는 몇 개 일까요?\n",
        "\n",
        "  - fastq 파일의 전체 라인의 개수에 [[Quiz-1]] 를 나누게 되면 리드의 개수가 됩니다."
      ],
      "metadata": {
        "id": "25Bvui4EGO80"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wc --help"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "iKEdAiXovDpm",
        "outputId": "72b654e9-e0c9-4cd5-f329-c247ea48eedc"
      },
      "execution_count": 129,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Usage: wc [OPTION]... [FILE]...\n",
            "  or:  wc [OPTION]... --files0-from=F\n",
            "Print newline, word, and byte counts for each FILE, and a total line if\n",
            "more than one FILE is specified.  A word is a non-zero-length sequence of\n",
            "characters delimited by white space.\n",
            "\n",
            "With no FILE, or when FILE is -, read standard input.\n",
            "\n",
            "The options below may be used to select which counts are printed, always in\n",
            "the following order: newline, word, character, byte, maximum line length.\n",
            "  -c, --bytes            print the byte counts\n",
            "  -m, --chars            print the character counts\n",
            "  -l, --lines            print the newline counts\n",
            "      --files0-from=F    read input from the files specified by\n",
            "                           NUL-terminated names in file F;\n",
            "                           If F is - then read names from standard input\n",
            "  -L, --max-line-length  print the maximum display width\n",
            "  -w, --words            print the word counts\n",
            "      --help     display this help and exit\n",
            "      --version  output version information and exit\n",
            "\n",
            "GNU coreutils online help: <https://www.gnu.org/software/coreutils/>\n",
            "Full documentation <https://www.gnu.org/software/coreutils/wc>\n",
            "or available locally via: info '(coreutils) wc invocation'\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!zless tumor_R1.fastq.gz | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "hVjpfZVru-6w",
        "outputId": "3c9b1261-8151-4346-f5a8-bc24fad2b9c5"
      },
      "execution_count": 45,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "1070884\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!zless tumor_R2.fastq.gz | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "SvaHjM8ju_Do",
        "outputId": "b33fc59f-ab02-44d8-a827-b22ebacf87a8"
      },
      "execution_count": 46,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "1070884\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!expr 1070884 / [[Quiz-1]]"
      ],
      "metadata": {
        "id": "9IQmibriGs1_"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "총 267,721개의 리드가 tumor 유래의 fastq 파일에 들어있습니다.\n"
      ],
      "metadata": {
        "id": "h7U_gHfdHXi0"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        " ### FASTQC\n",
        " 이제 시퀀싱된 리드의 quality 체크를 해보겠습니다."
      ],
      "metadata": {
        "id": "6sp1UXvndXCE"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "FastQC는 [[설치 방법 2. 프로그램 홈페이지에서 패키지를 다운받는 방법]] 에 해당합니다. 다음 페이지에 접속하여 링크를 복사할 수 있습니다. 그 후의 방법은 gatk 와 동일합니다.\n",
        "\n",
        "링크: https://www.bioinformatics.babraham.ac.uk/projects/fastqc/"
      ],
      "metadata": {
        "id": "zm0d_35NdiNY"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!cd /content/drive/MyDrive/AMC_NGS1/02_linux/tools\n",
        "cd(\"/content/drive/MyDrive/AMC_NGS1/02_linux/tools\")"
      ],
      "metadata": {
        "id": "9MWHAuyjH4j-"
      },
      "execution_count": 47,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.12.1.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "xGcPshIBH0y-",
        "outputId": "a6c97b33-5c8d-4edb-aca9-2481d01d842e"
      },
      "execution_count": 48,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-08-20 23:40:40--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.12.1.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 11709692 (11M) [application/zip]\n",
            "Saving to: ‘fastqc_v0.12.1.zip’\n",
            "\n",
            "fastqc_v0.12.1.zip  100%[===================>]  11.17M  36.5MB/s    in 0.3s    \n",
            "\n",
            "2025-08-20 23:40:40 (36.5 MB/s) - ‘fastqc_v0.12.1.zip’ saved [11709692/11709692]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!unzip fastqc_v0.12.1.zip\n",
        "!rm fastqc_v0.12.1.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "-xznDMcQIV_7",
        "outputId": "176f5229-ba9f-440b-f4e6-fb573df0412f"
      },
      "execution_count": 49,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  fastqc_v0.12.1.zip\n",
            "   creating: FastQC/\n",
            "  inflating: FastQC/htsjdk.jar       \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/LICENSE          \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/Templates/\n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageSaver/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageSaver/SVGImageSaver.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageSaver/SVGGenerator.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageSaver/SVGGenerator$SVGGraphics.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cd /content/drive/MyDrive/AMC_NGS1/02_linux/tools/FastQC/\n",
        "cd(\"/content/drive/MyDrive/AMC_NGS1/02_linux/tools/FastQC/\")"
      ],
      "metadata": {
        "id": "RAfLtzt_Inl_"
      },
      "execution_count": 50,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!ls -al"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "OqIQA4DIta9W",
        "outputId": "a6f9b4dd-bb21-4849-ded4-4e13a6849e97"
      },
      "execution_count": 51,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "total 11031\n",
            "-rw------- 1 root root 9267374 Aug  3  2022 cisd-jhdf5.jar\n",
            "drwx------ 2 root root    4096 Mar  1  2023 Configuration\n",
            "-rwx------ 1 root root   16438 Feb 16  2023 fastqc\n",
            "-rw------- 1 root root    2238 Aug  3  2022 fastqc_icon.ico\n",
            "drwx------ 5 root root    4096 Mar  1  2023 Help\n",
            "-rw------- 1 root root 1799577 Feb 16  2023 htsjdk.jar\n",
            "-rw------- 1 root root    6477 Aug  3  2022 INSTALL.txt\n",
            "-rw------- 1 root root   50147 Aug  3  2022 jbzip2-0.9.jar\n",
            "-rw------- 1 root root   35147 Feb 16  2023 LICENSE\n",
            "-rw------- 1 root root   11766 Aug  3  2022 LICENSE_JHDF5.txt\n",
            "-rw------- 1 root root   35147 Aug  3  2022 LICENSE.txt\n",
            "drwx------ 3 root root    4096 Mar  1  2023 net\n",
            "drwx------ 3 root root    4096 Mar  1  2023 org\n",
            "-rw------- 1 root root    1044 Aug  3  2022 README.md\n",
            "-rw------- 1 root root    2242 Aug  3  2022 README.txt\n",
            "-rw------- 1 root root   39770 Mar  1  2023 RELEASE_NOTES.txt\n",
            "-rw------- 1 root root     104 Aug  3  2022 run_fastqc.bat\n",
            "drwx------ 3 root root    4096 Mar  1  2023 Templates\n",
            "drwx------ 3 root root    4096 Mar  1  2023 uk\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!./fastqc --help"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9obao99rIWFs",
        "outputId": "4b19f8f9-99f6-44a2-bcc5-9f277077745e"
      },
      "execution_count": 52,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "            FastQC - A high throughput sequence QC analysis tool\n",
            "\n",
            "SYNOPSIS\n",
            "\n",
            "\tfastqc seqfile1 seqfile2 .. seqfileN\n",
            "\n",
            "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \n",
            "           [-c contaminant file] seqfile1 .. seqfileN\n",
            "\n",
            "DESCRIPTION\n",
            "\n",
            "    FastQC reads a set of sequence files and produces from each one a quality\n",
            "    control report consisting of a number of different modules, each one of \n",
            "    which will help to identify a different potential type of problem in your\n",
            "    data.\n",
            "    \n",
            "    If no files to process are specified on the command line then the program\n",
            "    will start as an interactive graphical application.  If files are provided\n",
            "    on the command line then the program will run with no user interaction\n",
            "    required.  In this mode it is suitable for inclusion into a standardised\n",
            "    analysis pipeline.\n",
            "    \n",
            "    The options for the program as as follows:\n",
            "    \n",
            "    -h --help       Print this help file and exit\n",
            "    \n",
            "    -v --version    Print the version of the program and exit\n",
            "    \n",
            "    -o --outdir     Create all output files in the specified output directory.\n",
            "                    Please note that this directory must exist as the program\n",
            "                    will not create it.  If this option is not set then the \n",
            "                    output file for each sequence file is created in the same\n",
            "                    directory as the sequence file which was processed.\n",
            "                    \n",
            "    --casava        Files come from raw casava output. Files in the same sample\n",
            "                    group (differing only by the group number) will be analysed\n",
            "                    as a set rather than individually. Sequences with the filter\n",
            "                    flag set in the header will be excluded from the analysis.\n",
            "                    Files must have the same names given to them by casava\n",
            "                    (including being gzipped and ending with .gz) otherwise they\n",
            "                    won't be grouped together correctly.\n",
            "                    \n",
            "    --nano          Files come from nanopore sequences and are in fast5 format. In\n",
            "                    this mode you can pass in directories to process and the program\n",
            "                    will take in all fast5 files within those directories and produce\n",
            "                    a single output file from the sequences found in all files.                    \n",
            "                    \n",
            "    --nofilter      If running with --casava then don't remove read flagged by\n",
            "                    casava as poor quality when performing the QC analysis.\n",
            "                   \n",
            "    --extract       If set then the zipped output file will be uncompressed in\n",
            "                    the same directory after it has been created. If --delete is \n",
            "                    also specified then the zip file will be removed after the \n",
            "                    contents are unzipped. \n",
            "                    \n",
            "    -j --java       Provides the full path to the java binary you want to use to\n",
            "                    launch fastqc. If not supplied then java is assumed to be in\n",
            "                    your path.\n",
            "                   \n",
            "    --noextract     Do not uncompress the output file after creating it.  You\n",
            "                    should set this option if you do not wish to uncompress\n",
            "                    the output when running in non-interactive mode.\n",
            "                    \n",
            "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\n",
            "                    show data for every base in the read.  WARNING: Using this\n",
            "                    option will cause fastqc to crash and burn if you use it on\n",
            "                    really long reads, and your plots may end up a ridiculous size.\n",
            "                    You have been warned!\n",
            "                    \n",
            "    --min_length    Sets an artificial lower limit on the length of the sequence\n",
            "                    to be shown in the report.  As long as you set this to a value\n",
            "                    greater or equal to your longest read length then this will be\n",
            "                    the sequence length used to create your read groups.  This can\n",
            "                    be useful for making directly comaparable statistics from \n",
            "                    datasets with somewhat variable read lengths.\n",
            "\n",
            "    --dup_length    Sets a length to which the sequences will be truncated when \n",
            "                    defining them to be duplicates, affecting the duplication and\n",
            "                    overrepresented sequences plot.  This can be useful if you have\n",
            "                    long reads with higher levels of miscalls, or contamination with\n",
            "                    adapter dimers containing UMI sequences.\n",
            "\n",
            "                    \n",
            "    -f --format     Bypasses the normal sequence file format detection and\n",
            "                    forces the program to use the specified format.  Valid\n",
            "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\n",
            "                    \n",
            "\n",
            "    --memory        Sets the base amount of memory, in Megabytes, used to process \n",
            "                    each file.  Defaults to 512MB.  You may need to increase this if\n",
            "                    you have a file with very long sequences in it.\n",
            "                \n",
            "    --svg           Save the graphs in the report in SVG format.\n",
            "\n",
            "    -t --threads    Specifies the number of files which can be processed\n",
            "                    simultaneously.  Each thread will be allocated 250MB of\n",
            "                    memory so you shouldn't run more threads than your\n",
            "                    available memory will cope with, and not more than\n",
            "                    6 threads on a 32 bit machine\n",
            "                  \n",
            "    -c              Specifies a non-default file which contains the list of\n",
            "    --contaminants  contaminants to screen overrepresented sequences against.\n",
            "                    The file must contain sets of named contaminants in the\n",
            "                    form name[tab]sequence.  Lines prefixed with a hash will\n",
            "                    be ignored.\n",
            "\n",
            "    -a              Specifies a non-default file which contains the list of\n",
            "    --adapters      adapter sequences which will be explicity searched against\n",
            "                    the library. The file must contain sets of named adapters\n",
            "                    in the form name[tab]sequence.  Lines prefixed with a hash\n",
            "                    will be ignored.\n",
            "                    \n",
            "    -l              Specifies a non-default file which contains a set of criteria\n",
            "    --limits        which will be used to determine the warn/error limits for the\n",
            "                    various modules.  This file can also be used to selectively \n",
            "                    remove some modules from the output all together.  The format\n",
            "                    needs to mirror the default limits.txt file found in the\n",
            "                    Configuration folder.\n",
            "                    \n",
            "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\n",
            "                    module. Specified Kmer length must be between 2 and 10. Default\n",
            "                    length is 7 if not specified.\n",
            "                    \n",
            "   -q --quiet       Suppress all progress messages on stdout and only report errors.\n",
            "   \n",
            "   -d --dir         Selects a directory to be used for temporary files written when\n",
            "                    generating report images. Defaults to system temp directory if\n",
            "                    not specified.\n",
            "                    \n",
            "BUGS\n",
            "\n",
            "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\n",
            "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\n",
            "                   \n",
            "    \n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "fastq 파일들이 모여있는 directory (/content/drive/MyDrive/AMC_NGS1/02_linux/data)로 이동 후 `FastQC`를 수행합니다. 기본 방식은 다음과 같습니다.\n",
        "\n",
        "```\n",
        "# Example\n",
        "fastqc \\\n",
        "  readfile1 \\\n",
        "  readfile2 \\\n",
        "  readfile3 \\\n",
        "  readfile4 \\\n",
        "  -o output_directory\n",
        "  ```"
      ],
      "metadata": {
        "id": "mXC4VXMAeQl1"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!cd /content/drive/MyDrive/AMC_NGS1/02_linux/data\n",
        "cd(\"/content/drive/MyDrive/AMC_NGS1/02_linux/data/\")"
      ],
      "metadata": {
        "id": "AImAodelIWIN"
      },
      "execution_count": 53,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir fastqc"
      ],
      "metadata": {
        "id": "Pbgu65B3fq4P"
      },
      "execution_count": 54,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 40s\n",
        "!/content/drive/MyDrive/AMC_NGS1/02_linux/tools/FastQC/fastqc \\\n",
        "  normal_R1.fastq.gz \\\n",
        "  normal_R2.fastq.gz \\\n",
        "  tumor_R1.fastq.gz \\\n",
        "  tumor_R2.fastq.gz \\\n",
        "  -o fastqc"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "RTUE70MuIWKt",
        "outputId": "cbbae967-afc0-4ccc-9bed-cc02b8e06993"
      },
      "execution_count": 55,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "application/gzip\n",
            "application/gzip\n",
            "application/gzip\n",
            "application/gzip\n",
            "Started analysis of normal_R1.fastq.gz\n",
            "Approx 5% complete for normal_R1.fastq.gz\n",
            "Approx 10% complete for normal_R1.fastq.gz\n",
            "Approx 15% complete for normal_R1.fastq.gz\n",
            "Approx 20% complete for normal_R1.fastq.gz\n",
            "Approx 25% complete for normal_R1.fastq.gz\n",
            "Approx 30% complete for normal_R1.fastq.gz\n",
            "Approx 35% complete for normal_R1.fastq.gz\n",
            "Approx 40% complete for normal_R1.fastq.gz\n",
            "Approx 45% complete for normal_R1.fastq.gz\n",
            "Approx 50% complete for normal_R1.fastq.gz\n",
            "Approx 55% complete for normal_R1.fastq.gz\n",
            "Approx 60% complete for normal_R1.fastq.gz\n",
            "Approx 65% complete for normal_R1.fastq.gz\n",
            "Approx 70% complete for normal_R1.fastq.gz\n",
            "Approx 75% complete for normal_R1.fastq.gz\n",
            "Approx 80% complete for normal_R1.fastq.gz\n",
            "Approx 85% complete for normal_R1.fastq.gz\n",
            "Approx 90% complete for normal_R1.fastq.gz\n",
            "Approx 95% complete for normal_R1.fastq.gz\n",
            "Analysis complete for normal_R1.fastq.gz\n",
            "Started analysis of normal_R2.fastq.gz\n",
            "Approx 5% complete for normal_R2.fastq.gz\n",
            "Approx 10% complete for normal_R2.fastq.gz\n",
            "Approx 15% complete for normal_R2.fastq.gz\n",
            "Approx 20% complete for normal_R2.fastq.gz\n",
            "Approx 25% complete for normal_R2.fastq.gz\n",
            "Approx 30% complete for normal_R2.fastq.gz\n",
            "Approx 35% complete for normal_R2.fastq.gz\n",
            "Approx 40% complete for normal_R2.fastq.gz\n",
            "Approx 45% complete for normal_R2.fastq.gz\n",
            "Approx 50% complete for normal_R2.fastq.gz\n",
            "Approx 55% complete for normal_R2.fastq.gz\n",
            "Approx 60% complete for normal_R2.fastq.gz\n",
            "Approx 65% complete for normal_R2.fastq.gz\n",
            "Approx 70% complete for normal_R2.fastq.gz\n",
            "Approx 75% complete for normal_R2.fastq.gz\n",
            "Approx 80% complete for normal_R2.fastq.gz\n",
            "Approx 85% complete for normal_R2.fastq.gz\n",
            "Approx 90% complete for normal_R2.fastq.gz\n",
            "Approx 95% complete for normal_R2.fastq.gz\n",
            "Analysis complete for normal_R2.fastq.gz\n",
            "Started analysis of tumor_R1.fastq.gz\n",
            "Approx 5% complete for tumor_R1.fastq.gz\n",
            "Approx 10% complete for tumor_R1.fastq.gz\n",
            "Approx 15% complete for tumor_R1.fastq.gz\n",
            "Approx 20% complete for tumor_R1.fastq.gz\n",
            "Approx 25% complete for tumor_R1.fastq.gz\n",
            "Approx 30% complete for tumor_R1.fastq.gz\n",
            "Approx 35% complete for tumor_R1.fastq.gz\n",
            "Approx 40% complete for tumor_R1.fastq.gz\n",
            "Approx 45% complete for tumor_R1.fastq.gz\n",
            "Approx 50% complete for tumor_R1.fastq.gz\n",
            "Approx 55% complete for tumor_R1.fastq.gz\n",
            "Approx 60% complete for tumor_R1.fastq.gz\n",
            "Approx 65% complete for tumor_R1.fastq.gz\n",
            "Approx 70% complete for tumor_R1.fastq.gz\n",
            "Approx 75% complete for tumor_R1.fastq.gz\n",
            "Approx 80% complete for tumor_R1.fastq.gz\n",
            "Approx 85% complete for tumor_R1.fastq.gz\n",
            "Approx 90% complete for tumor_R1.fastq.gz\n",
            "Approx 95% complete for tumor_R1.fastq.gz\n",
            "Analysis complete for tumor_R1.fastq.gz\n",
            "Started analysis of tumor_R2.fastq.gz\n",
            "Approx 5% complete for tumor_R2.fastq.gz\n",
            "Approx 10% complete for tumor_R2.fastq.gz\n",
            "Approx 15% complete for tumor_R2.fastq.gz\n",
            "Approx 20% complete for tumor_R2.fastq.gz\n",
            "Approx 25% complete for tumor_R2.fastq.gz\n",
            "Approx 30% complete for tumor_R2.fastq.gz\n",
            "Approx 35% complete for tumor_R2.fastq.gz\n",
            "Approx 40% complete for tumor_R2.fastq.gz\n",
            "Approx 45% complete for tumor_R2.fastq.gz\n",
            "Approx 50% complete for tumor_R2.fastq.gz\n",
            "Approx 55% complete for tumor_R2.fastq.gz\n",
            "Approx 60% complete for tumor_R2.fastq.gz\n",
            "Approx 65% complete for tumor_R2.fastq.gz\n",
            "Approx 70% complete for tumor_R2.fastq.gz\n",
            "Approx 75% complete for tumor_R2.fastq.gz\n",
            "Approx 80% complete for tumor_R2.fastq.gz\n",
            "Approx 85% complete for tumor_R2.fastq.gz\n",
            "Approx 90% complete for tumor_R2.fastq.gz\n",
            "Approx 95% complete for tumor_R2.fastq.gz\n",
            "Analysis complete for tumor_R2.fastq.gz\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "fastqc 결과를 확인합니다.\n",
        "\n",
        "/content/drive/MyDrive/AMC_NGS1/02_linux/data/fastqc 에 4개의 fastq 파일의 quality check 결과가 생성되어 있습니다. 아래의 `html` 링크를 그대로 클릭하면 우리가 읽을 수 없는 형태로 되어있습니다. html 파일 다운로드 후 google chrome으로 연결하면 제대로 볼 수 있습니다.\n",
        "\n",
        "- tumor read1: /content/drive/MyDrive/AMC_NGS1/02_linux/data/fastqc/tumor_R1_fastqc.html\n",
        "\n",
        "- tumor read2: /content/drive/MyDrive/AMC_NGS1/02_linux/data/fastqc/tumor_R2_fastqc.html\n",
        "\n",
        "- normal read1: /content/drive/MyDrive/AMC_NGS1/02_linux/data/fastqc/normal_R1_fastqc.html\n",
        "\n",
        "- normal read2: /content/drive/MyDrive/AMC_NGS1/02_linux/data/fastqc/normal_R1_fastqc.html"
      ],
      "metadata": {
        "id": "qUbAebtlf1zj"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### FASTQC 결과 해석\n",
        "\n",
        "ppt 파일에서 함께 확인"
      ],
      "metadata": {
        "id": "Fib0niCvhF7-"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Mapping (Alignment)"
      ],
      "metadata": {
        "id": "18uIb-SfhT-3"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "BWA 툴을 사용하여 기준서열 (reference sequence) 에 리드들을 mapping 하는 과정입니다."
      ],
      "metadata": {
        "id": "mo7eY2l-Md7Q"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Reference sequence 란?"
      ],
      "metadata": {
        "id": "JcOQ0qTThi6m"
      }
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "SfTROowOhpfs"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Mapping 수행\n",
        "\n",
        "모든 분석은 /content/drive/MyDrive/AMC_NGS1/02_linux/data 에서 수행"
      ],
      "metadata": {
        "id": "x4Exxaayhtj6"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "우리가 가지고 있는 fastq reads\n",
        "- tumor tissue 유래 paired end 1쌍 (R1, R2)\n",
        "- normal tissue 유래 paried end 1쌍 (R1, R2)"
      ],
      "metadata": {
        "id": "TSb_UXash9Cs"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!cd /content/drive/MyDrive/AMC_NGS1/02_linux/data\n",
        "cd(\"/content/drive/MyDrive/AMC_NGS1/02_linux/data\")"
      ],
      "metadata": {
        "id": "sJWUMN70Mseq"
      },
      "execution_count": 56,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 0s\n",
        "!bwa mem -t 1 -R \"@RG\\tID:normal\\tSM:normal\\tPL:platform\" \\\n",
        "  ../resource/reference/hg38.chr21.fa \\\n",
        "  normal_R1.fastq.gz \\\n",
        "  normal_R2.fastq.gz > normal.sam 2> normal.bwa.log"
      ],
      "metadata": {
        "id": "hlS_v95vGJXV"
      },
      "execution_count": 57,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!ls -al"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8hnMuDmz3F1D",
        "outputId": "8b4c62ea-946a-4bd5-8770-1998589bbd0b"
      },
      "execution_count": 58,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "total 72748\n",
            "drwx------ 2 root root     4096 Aug 20 23:41 fastqc\n",
            "-rw------- 1 root root       59 Aug 20 23:41 normal.bwa.log\n",
            "-rw------- 1 root root 16013759 Aug 20 02:14 normal_R1.fastq.gz\n",
            "-rw------- 1 root root 16698858 Aug 20 02:14 normal_R2.fastq.gz\n",
            "-rw------- 1 root root        0 Aug 20 23:41 normal.sam\n",
            "-rw------- 1 root root  4368573 Aug 20 00:54 sample_1.fastq.gz\n",
            "-rw------- 1 root root  4376708 Aug 20 00:54 sample_2.fastq.gz\n",
            "-rw------- 1 root root 16120236 Aug 20 02:18 tumor_R1.fastq.gz\n",
            "-rw------- 1 root root 16909873 Aug 20 02:18 tumor_R2.fastq.gz\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cat normal.bwa.log"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "W_2CW0l8uYPE",
        "outputId": "269528e8-ff4b-467c-cf34-c7ca8410c2b9"
      },
      "execution_count": 59,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[E::bwa_idx_load_from_disk] fail to locate the index files\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "에러 해결: https://github.com/jts/methylation-analysis/issues/13"
      ],
      "metadata": {
        "id": "NUuZuLeziZll"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!cd /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference\n",
        "cd(\"/content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference\") # 노트북에서만 실행"
      ],
      "metadata": {
        "id": "y-cagj1wGJa7"
      },
      "execution_count": 60,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 60s\n",
        "!bwa index hg38.chr21.fa"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "vqEI_EhyGJdg",
        "outputId": "bb7d36be-ab1e-419b-adce-6018cdc59fdf"
      },
      "execution_count": 61,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bwa_index] Pack FASTA... 0.43 sec\n",
            "[bwa_index] Construct BWT for the packed sequence...\n",
            "[BWTIncCreate] textLength=93419966, availableWord=18573208\n",
            "[BWTIncConstructFromPacked] 10 iterations done. 30636734 characters processed.\n",
            "[BWTIncConstructFromPacked] 20 iterations done. 56597342 characters processed.\n",
            "[BWTIncConstructFromPacked] 30 iterations done. 79667134 characters processed.\n",
            "[bwt_gen] Finished constructing BWT in 37 iterations.\n",
            "[bwa_index] 55.92 seconds elapse.\n",
            "[bwa_index] Update BWT... 0.41 sec\n",
            "[bwa_index] Pack forward-only FASTA... 0.44 sec\n",
            "[bwa_index] Construct SA from BWT and Occ... 16.95 sec\n",
            "[main] Version: 0.7.17-r1188\n",
            "[main] CMD: bwa index hg38.chr21.fa\n",
            "[main] Real time: 75.893 sec; CPU: 74.157 sec\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cd /content/drive/MyDrive/AMC_NGS1/02_linux/data\n",
        "cd(\"/content/drive/MyDrive/AMC_NGS1/02_linux/data\")"
      ],
      "metadata": {
        "id": "cqrXhe0kGJf-"
      },
      "execution_count": 62,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 65s\n",
        "!bwa mem \\\n",
        "  -t 1 \\\n",
        "  -R \"@RG\\tID:normal\\tSM:normal\\tPL:platform\" \\\n",
        "  ../resource/reference/hg38.chr21.fa \\\n",
        "  normal_R1.fastq.gz \\\n",
        "  normal_R2.fastq.gz > normal.sam 2> normal.bwa.log"
      ],
      "metadata": {
        "id": "sXHdcw-UNUKJ"
      },
      "execution_count": 63,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# 터미널에서 실행\n",
        "# less normal.sam"
      ],
      "metadata": {
        "id": "XyAGAqRqvtiA"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "**Header 정보 |** @SQ     SN:chr21        LN:46709983\n",
        "\n",
        "**Header 정보 |** @RG     ID:normal       SM:normal       PL:platform\n",
        "\n",
        "**Header 정보 |**  @PG     ID:bwa  PN:bwa  VN:0.7.17-r1188 CL:bwa mem -t 1 -R\n",
        "@RG\\tID:normal\\tSM:normal\\tPL:platform ../resource/reference/hg38.chr21.fa normal_R1.fastq.gz normal_R2.fastq.gz\n",
        "\n",
        "**Read 정보 |** normal#21#998579#1      99      chr21   20998580        60      100M          =       20998796        316     CCTTCTCCCTGCTGGGGTTGCTTGTCAGTAGCGGGCAAGGTAGGAGTGTGGCGCTTTATTGCATTTACTTTCCCTCCCCCTTCCCCCCGGCCAAGAGAGG        102302000331;3333;23133320233330*33/233333333333333/313232333/3;3;3/333000;11/00;;01//103*1032323233  NM:i:2  MD:Z:6A77A15    MC:Z:100M     AS:i:90 XS:i:0  RG:Z:normal\n",
        "\n",
        "**Read 정보 |** normal#21#998579#1      147     chr21   20998796        60      100M          =       20998580        -316    GTGGAGACCCTTAAATAAGAGAAGGGAAAATACAGAATTGGAGAGGGGAATCGGTGAGCAGGATATCCAGGGAAACAGCTTCTGCTTCTTCTTTTTTTTT        030/3133332332;22303032/11;;23033312;22/130300102;23/000;/*;0030;3333/112333203333013333233332333320  NM:i:2  MD:Z:76C6T16    MC:Z:100M     AS:i:90 XS:i:0  RG:Z:normal\n"
      ],
      "metadata": {
        "id": "XmbHIrDPv9nA"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Mapping 정보 확인 (sam 파일의 구조 확인)\n",
        "\n",
        "ppt 에서 함께 확인"
      ],
      "metadata": {
        "id": "_DEBJ_M9iwoF"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### sam to bam"
      ],
      "metadata": {
        "id": "FMKxBPvzjLI7"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "기본적으로 BWA mem 커맨드를 사용하여 나오게되는 파일은 sam 파일입니다. 실제 연구 또는 업무에서는 sam 파일은 크기가 너무 크므로 이를 binary 형태로 압축한 bam 파일을 많이 사용합니다.  \n",
        "`samtools` 프로그램을 사용하여 sam 파일을 bam 파일로 변환해 봅시다."
      ],
      "metadata": {
        "id": "KkTAh_PhNp7S"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 10s\n",
        "!apt-get install samtools"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "CzXG_rsgNyxX",
        "outputId": "7e6925d8-f515-4a57-b999-0b20514349f0"
      },
      "execution_count": 64,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "samtools is already the newest version (1.13-4).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 35 not upgraded.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools --help"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "l7DFrJ0cN68G",
        "outputId": "a007a062-1274-49e2-9c0d-127740321d5b"
      },
      "execution_count": 65,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Program: samtools (Tools for alignments in the SAM format)\n",
            "Version: 1.13 (using htslib 1.13+ds)\n",
            "\n",
            "Usage:   samtools <command> [options]\n",
            "\n",
            "Commands:\n",
            "  -- Indexing\n",
            "     dict           create a sequence dictionary file\n",
            "     faidx          index/extract FASTA\n",
            "     fqidx          index/extract FASTQ\n",
            "     index          index alignment\n",
            "\n",
            "  -- Editing\n",
            "     calmd          recalculate MD/NM tags and '=' bases\n",
            "     fixmate        fix mate information\n",
            "     reheader       replace BAM header\n",
            "     targetcut      cut fosmid regions (for fosmid pool only)\n",
            "     addreplacerg   adds or replaces RG tags\n",
            "     markdup        mark duplicates\n",
            "     ampliconclip   clip oligos from the end of reads\n",
            "\n",
            "  -- File operations\n",
            "     collate        shuffle and group alignments by name\n",
            "     cat            concatenate BAMs\n",
            "     merge          merge sorted alignments\n",
            "     mpileup        multi-way pileup\n",
            "     sort           sort alignment file\n",
            "     split          splits a file by read group\n",
            "     quickcheck     quickly check if SAM/BAM/CRAM file appears intact\n",
            "     fastq          converts a BAM to a FASTQ\n",
            "     fasta          converts a BAM to a FASTA\n",
            "     import         Converts FASTA or FASTQ files to SAM/BAM/CRAM\n",
            "\n",
            "  -- Statistics\n",
            "     bedcov         read depth per BED region\n",
            "     coverage       alignment depth and percent coverage\n",
            "     depth          compute the depth\n",
            "     flagstat       simple stats\n",
            "     idxstats       BAM index stats\n",
            "     phase          phase heterozygotes\n",
            "     stats          generate stats (former bamcheck)\n",
            "     ampliconstats  generate amplicon specific stats\n",
            "\n",
            "  -- Viewing\n",
            "     flags          explain BAM flags\n",
            "     tview          text alignment viewer\n",
            "     view           SAM<->BAM<->CRAM conversion\n",
            "     depad          convert padded BAM to unpadded BAM\n",
            "\n",
            "  -- Misc\n",
            "     help [cmd]     display this help message or help for [cmd]\n",
            "     version        detailed version information\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 4s\n",
        "!samtools view -Sb normal.sam > normal.bam"
      ],
      "metadata": {
        "id": "0NtL71xQNsny"
      },
      "execution_count": 66,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "이렇게 sequencer에서 나온 리드들을 기준서열에 mapping 하였습니다. 우리가 mapping한 리드를 눈으로 살펴 보겠습니다"
      ],
      "metadata": {
        "id": "CHTC5fFSOLUN"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# 터미널에서 실행\n",
        "# less normal.bam"
      ],
      "metadata": {
        "id": "3NNSxv5Aw06V"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view -H normal.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Y78IPME1OK4p",
        "outputId": "ea93fec4-aa32-4ea1-b62e-07e0a87a692c"
      },
      "execution_count": 67,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "@SQ\tSN:chr21\tLN:46709983\n",
            "@RG\tID:normal\tSM:normal\tPL:platform\n",
            "@PG\tID:bwa\tPN:bwa\tVN:0.7.17-r1188\tCL:bwa mem -t 1 -R @RG\\tID:normal\\tSM:normal\\tPL:platform ../resource/reference/hg38.chr21.fa normal_R1.fastq.gz normal_R2.fastq.gz\n",
            "@PG\tID:samtools\tPN:samtools\tPP:bwa\tVN:1.13\tCL:samtools view -Sb normal.sam\n",
            "@PG\tID:samtools.1\tPN:samtools\tPP:samtools\tVN:1.13\tCL:samtools view -H normal.bam\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view normal.bam | head -10"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Yw6HG5mkOZ-m",
        "outputId": "3268ae24-66f4-4246-a61b-5c000a312932"
      },
      "execution_count": 68,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "normal#21#998579#1\t99\tchr21\t20998580\t60\t100M\t=\t20998796\t316\tCCTTCTCCCTGCTGGGGTTGCTTGTCAGTAGCGGGCAAGGTAGGAGTGTGGCGCTTTATTGCATTTACTTTCCCTCCCCCTTCCCCCCGGCCAAGAGAGG\t102302000331;3333;23133320233330*33/233333333333333/313232333/3;3;3/333000;11/00;;01//103*1032323233\tNM:i:2\tMD:Z:6A77A15\tMC:Z:100M\tAS:i:90\tXS:i:0\tRG:Z:normal\n",
            "normal#21#998579#1\t147\tchr21\t20998796\t60\t100M\t=\t20998580\t-316\tGTGGAGACCCTTAAATAAGAGAAGGGAAAATACAGAATTGGAGAGGGGAATCGGTGAGCAGGATATCCAGGGAAACAGCTTCTGCTTCTTCTTTTTTTTT\t030/3133332332;22303032/11;;23033312;22/130300102;23/000;/*;0030;3333/112333203333013333233332333320\tNM:i:2\tMD:Z:76C6T16\tMC:Z:100M\tAS:i:90\tXS:i:0\tRG:Z:normal\n",
            "normal#21#998572#2\t99\tchr21\t20998573\t60\t100M\t=\t20998746\t273\tCTCCTCTCCTTCTACCTGCTGGGGTTGCTTGTCAGTAGCGGGCAAGGTCGGAGTGTTGCGCTTTATTGCATTTACTTTCCCTCCCCCTTCCACCCGGCCA\t/201302/0330;30033133333;230;333033333033300333203333333;*03133;223300;2;3/33211/210001;2/13/0133//3\tNM:i:2\tMD:Z:48A7G43\tMC:Z:46M1I54M\tAS:i:90\tXS:i:0\tRG:Z:normal\n",
            "normal#21#998572#2\t147\tchr21\t20998746\t60\t46M1I54M\t=\t20998573\t-273\tTTATTATTTTCGTTCTTTCTCCTAGCTGTCCCTAAAATTGTATCTGTTTGTGTGGAGACCCTTAAATAAGAGAAGGGAAAATACTCAATTGGAGAGGGGAA\t23323333323/3233323233320300333*2;3;;30022332/32303/300;1333322;333230;/321/032233;323;3330030;00/223\tNM:i:3\tMD:Z:83A0G15\tMC:Z:100M\tAS:i:83\tXS:i:0\tRG:Z:normal\n",
            "normal#21#998620#3\t99\tchr21\t20998621\t60\t7M1D64M1D27M\t=\t20998812\t291\tAGGAGTGGGCGCTTTATTGCATTTACTTTCCGTCCCCCTTCCACCCGGCCAAGAGAGGCAAAGAGGGAGGCCAGGAAGAATGAAATGAAAGAACTAAA\t223333333/30;323;;303;3320;231032001/033102/103*0033323333/33333333233//03333*22;33032302333302333\tNM:i:3\tMD:Z:7^T24C39^G27\tMC:Z:100M\tAS:i:79\tXS:i:0\tRG:Z:normal\n",
            "normal#21#998620#3\t147\tchr21\t20998812\t60\t100M\t=\t20998621\t-291\tAAGAGAAGGGAAAATACAGAATTGGAGAGGGGAATCGTTGAGCAGGATATCCAGGGAAACCGCTTCTTCTTCTTCTTTTTTTTTCCTCTGGGTTTATGGC\t;2/30;300/2333233;1;3031021;0//13;3303302/3301233233;100;;33*1322*33323333322323023333333//0003;0123\tNM:i:2\tMD:Z:37G54A7\tMC:Z:7M1D64M1D27M\tAS:i:90\tXS:i:23\tRG:Z:normal\n",
            "normal#21#998545#4\t99\tchr21\t20998546\t60\t20M2D78M\t=\t20999032\t586\tCGGTGTCACGTCCTGAACATGCCTCCTCCTCTCCTTCTACCTGCTGGGGTTGCTTGTCAGTAGCGGGCAAGGTAGGAGTGTGGCGCTTTATTGCATTT\t0233*203/330/333303330031/;/03130/331331133033333;331;;3;033223/3330203333332333333130;3;333303332\tNM:i:2\tMD:Z:20^GA78\tMC:Z:100M\tAS:i:90\tXS:i:0\tRG:Z:normal\n",
            "normal#21#998545#4\t147\tchr21\t20999032\t60\t100M\t=\t20998546\t-586\tTCATTTGCAAAATATCTGCTGTGTGCAAGACGTTCGATTCTTGGAGACTCTTTTTTTTTTCCACCCCTTTCCCAAAACCCTAATTCCTGACTATGAAGTG\t2332330*3;220;233/33120303;3033033302303201020232332223223303333333232333;2;23332;333330/;3333133002\tNM:i:0\tMD:Z:100\tMC:Z:20M2D78M\tAS:i:100\tXS:i:19\tRG:Z:normal\n",
            "normal#21#998602#5\t99\tchr21\t20998603\t60\t100M\t=\t20998647\t141\tGTCAGTATCGGGCAAGGTAGGAGTGTGGCGCTTTATTGCATTTACTTTCCCTCCCCCTTCCACCCGGCGAAGAGAGGCAAAGAGGGAGGCGCAGGAAGAA\t32123333/33303333;03333;333303/;333;33/3;2;31323000;1011/;200310133033332323302033333*3330*023322303\tNM:i:2\tMD:Z:7G60C31\tMC:Z:3S97M\tAS:i:90\tXS:i:0\tRG:Z:normal\n",
            "normal#21#998602#5\t147\tchr21\t20998647\t60\t3S97M\t=\t20998603\t-141\tTGTCTTTCACTCCCCCTTCCACCCGGCCAAGAGAGGCAAAGAGGGAGGCGCAGGAAGAATGAAAGGAAAGAACTAAAGTTACAGTTATTGCTGTGGTTGT\t203*3333333333**3333;3331/3333021;/0333;/2/10;0/313300;31332132;11323/2333;33003;33133230/*003003002\tNM:i:3\tMD:Z:5C55T29T5\tMC:Z:100M\tAS:i:82\tXS:i:0\tRG:Z:normal\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools flagstat normal.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NVIghALhOnl2",
        "outputId": "813fc9cf-5ec6-4be2-c6d2-f13f66ee36e1"
      },
      "execution_count": 69,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "533472 + 0 in total (QC-passed reads + QC-failed reads)\n",
            "533472 + 0 primary\n",
            "0 + 0 secondary\n",
            "0 + 0 supplementary\n",
            "0 + 0 duplicates\n",
            "0 + 0 primary duplicates\n",
            "533472 + 0 mapped (100.00% : N/A)\n",
            "533472 + 0 primary mapped (100.00% : N/A)\n",
            "533472 + 0 paired in sequencing\n",
            "266736 + 0 read1\n",
            "266736 + 0 read2\n",
            "533472 + 0 properly paired (100.00% : N/A)\n",
            "533472 + 0 with itself and mate mapped\n",
            "0 + 0 singletons (0.00% : N/A)\n",
            "0 + 0 with mate mapped to a different chr\n",
            "0 + 0 with mate mapped to a different chr (mapQ>=5)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Sort"
      ],
      "metadata": {
        "id": "JoYE2Lzfjd1B"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 5s\n",
        "!samtools sort \\\n",
        "  -@ 1 \\\n",
        "  -o normal.s.bam \\\n",
        "  normal.bam"
      ],
      "metadata": {
        "id": "2-qhvBWBjv4j"
      },
      "execution_count": 70,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools index normal.s.bam"
      ],
      "metadata": {
        "id": "gWfvApCD0U_y"
      },
      "execution_count": 71,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "How to check <normal.s.bam> is sorted?"
      ],
      "metadata": {
        "id": "jA-VZNQijjHZ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view -H normal.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "RP2ArJ4JkQHG",
        "outputId": "42dd1fd6-28ce-40c9-cf75-616d749a3380"
      },
      "execution_count": 72,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "@SQ\tSN:chr21\tLN:46709983\n",
            "@RG\tID:normal\tSM:normal\tPL:platform\n",
            "@PG\tID:bwa\tPN:bwa\tVN:0.7.17-r1188\tCL:bwa mem -t 1 -R @RG\\tID:normal\\tSM:normal\\tPL:platform ../resource/reference/hg38.chr21.fa normal_R1.fastq.gz normal_R2.fastq.gz\n",
            "@PG\tID:samtools\tPN:samtools\tPP:bwa\tVN:1.13\tCL:samtools view -Sb normal.sam\n",
            "@PG\tID:samtools.1\tPN:samtools\tPP:samtools\tVN:1.13\tCL:samtools view -H normal.bam\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view -H normal.s.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sJGHGHwEkHTD",
        "outputId": "95a68f4e-55a2-4c63-e812-4a5f0e1196e0"
      },
      "execution_count": 73,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "@HD\tVN:1.6\tSO:coordinate\n",
            "@SQ\tSN:chr21\tLN:46709983\n",
            "@RG\tID:normal\tSM:normal\tPL:platform\n",
            "@PG\tID:bwa\tPN:bwa\tVN:0.7.17-r1188\tCL:bwa mem -t 1 -R @RG\\tID:normal\\tSM:normal\\tPL:platform ../resource/reference/hg38.chr21.fa normal_R1.fastq.gz normal_R2.fastq.gz\n",
            "@PG\tID:samtools\tPN:samtools\tPP:bwa\tVN:1.13\tCL:samtools view -Sb normal.sam\n",
            "@PG\tID:samtools.1\tPN:samtools\tPP:samtools\tVN:1.13\tCL:samtools sort -@ 1 -o normal.s.bam normal.bam\n",
            "@PG\tID:samtools.2\tPN:samtools\tPP:samtools.1\tVN:1.13\tCL:samtools view -H normal.s.bam\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "read들을 한 눈에 볼 수 있는 방법\n",
        "\n",
        "- 사용법\n",
        "  - \"/\" 입력 후 좌표 입력 \"chr21:20998580\"\n",
        "  - 종료시 \"q\" 입력"
      ],
      "metadata": {
        "id": "iANdkN821Rcb"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# 터미널에서만 실행\n",
        "# samtools tview normal.s.bam"
      ],
      "metadata": {
        "id": "PMmO4ACS1GoL"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Mark duplication"
      ],
      "metadata": {
        "id": "uanQu43dkbK4"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "이번에는 리드에서 duplication, 즉 중복이 있는 리드를 표기하는 방법에 대해 알아보겠습니다.  \n",
        "여기서 중복이라고 말하는 것은 무엇일까요? Sequencing을 진행하게 되면 필연적으로 PCR (Polymerase Chain Reaction) 과정에 의해 중복 리드가 발생하게 되는데요, 이러한 중복 리드들은 variant calling 과정에서 영향을 주게 됩니다. 그래서 duplication을 마킹하여 variant calling 때 영향을 주지 않도록 만들어야만 합니다."
      ],
      "metadata": {
        "id": "KRLAtGyyxudl"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#runtime: 23s\n",
        "!java -jar \\\n",
        "  /content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar \\\n",
        "  MarkDuplicates \\\n",
        "  --REMOVE_DUPLICATES false \\\n",
        "  --INPUT normal.s.bam \\\n",
        "  --OUTPUT normal.s.md.bam \\\n",
        "  --CREATE_INDEX true \\\n",
        "  --METRICS_FILE normal.md.metrics"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "hRc26EEhkam5",
        "outputId": "ea7d7adc-4152-4714-99f3-5e0443edee5e"
      },
      "execution_count": 74,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "23:46:08.765 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "[Wed Aug 20 23:46:08 UTC 2025] MarkDuplicates --INPUT normal.s.bam --OUTPUT normal.s.md.bam --METRICS_FILE normal.md.metrics --REMOVE_DUPLICATES false --CREATE_INDEX true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --ASSUME_SORTED false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 2 --MAX_RECORDS_IN_RAM 500000 --CREATE_MD5_FILE false --GA4GH_CLIENT_SECRETS client_secrets.json --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
            "[Wed Aug 20 23:46:09 UTC 2025] Executing as root@49a1eeb69d60 on Linux 6.1.123+ amd64; OpenJDK 64-Bit Server VM 11.0.28+6-post-Ubuntu-1ubuntu122.04.1; Deflater: Intel; Inflater: Intel; Provider GCS is available; Picard version: Version:4.3.0.0\n",
            "INFO\t2025-08-20 23:46:09\tMarkDuplicates\tStart of doWork freeMemory: 71151816; totalMemory: 98566144; maxMemory: 3403677696\n",
            "INFO\t2025-08-20 23:46:09\tMarkDuplicates\tReading input file and constructing read end information.\n",
            "INFO\t2025-08-20 23:46:09\tMarkDuplicates\tWill retain up to 12332165 data points before spilling to disk.\n",
            "WARNING\t2025-08-20 23:46:09\tAbstractOpticalDuplicateFinderCommandLineProgram\tA field field parsed out of a read name was expected to contain an integer and did not. Read name: normal#21#251697#3938. Cause: String 'normal#21#251697#3938' did not start with a parsable number.\n",
            "INFO\t2025-08-20 23:46:16\tMarkDuplicates\tRead 533472 records. 0 pairs never matched.\n",
            "INFO\t2025-08-20 23:46:16\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 400232832; totalMemory: 586153984; maxMemory: 3403677696\n",
            "INFO\t2025-08-20 23:46:16\tMarkDuplicates\tWill retain up to 106364928 duplicate indices before spilling to disk.\n",
            "INFO\t2025-08-20 23:46:18\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
            "INFO\t2025-08-20 23:46:18\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
            "INFO\t2025-08-20 23:46:19\tMarkDuplicates\tSorting list of duplicate records.\n",
            "INFO\t2025-08-20 23:46:19\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 591055664; totalMemory: 1470103552; maxMemory: 3403677696\n",
            "INFO\t2025-08-20 23:46:19\tMarkDuplicates\tMarking 284 records as duplicates.\n",
            "INFO\t2025-08-20 23:46:19\tMarkDuplicates\tFound 0 optical duplicate clusters.\n",
            "INFO\t2025-08-20 23:46:19\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
            "INFO\t2025-08-20 23:46:26\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
            "INFO\t2025-08-20 23:46:26\tMarkDuplicates\tDuplicate Index cleanup.\n",
            "INFO\t2025-08-20 23:46:26\tMarkDuplicates\tGetting Memory Stats.\n",
            "INFO\t2025-08-20 23:46:26\tMarkDuplicates\tBefore output close freeMemory: 78595768; totalMemory: 109051904; maxMemory: 3403677696\n",
            "INFO\t2025-08-20 23:46:26\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
            "INFO\t2025-08-20 23:46:26\tMarkDuplicates\tAfter output close freeMemory: 70070496; totalMemory: 98566144; maxMemory: 3403677696\n",
            "[Wed Aug 20 23:46:26 UTC 2025] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.30 minutes.\n",
            "Runtime.totalMemory()=98566144\n",
            "Tool returned:\n",
            "0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# notebook에서만 실행\n",
        "import pandas as pd\n",
        "from IPython.display import display\n",
        "\n",
        "file_path = \"normal.md.metrics\"\n",
        "\n",
        "df = pd.read_csv(\n",
        "    file_path,\n",
        "    sep=\"\\t\",\n",
        "    comment=\"#\",           # '#'으로 시작하는 주석 줄은 무시\n",
        "    skip_blank_lines=True,\n",
        "    nrows=1                # 첫 번째 데이터 행만 읽기\n",
        ")\n",
        "\n",
        "df = df[['PERCENT_DUPLICATION',\n",
        "         'READ_PAIRS_EXAMINED',\n",
        "         'READ_PAIR_DUPLICATES',\n",
        "         'READ_PAIR_OPTICAL_DUPLICATES']]\n",
        "\n",
        "# 데이터 확인\n",
        "df_t = df.T\n",
        "df_t.columns = [\"Value\"]\n",
        "\n",
        "display(df)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 109
        },
        "id": "8nIDq_A7mGkh",
        "outputId": "8ce130f8-0244-4498-e9b5-8f91c4e15015"
      },
      "execution_count": 75,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "   PERCENT_DUPLICATION  READ_PAIRS_EXAMINED  READ_PAIR_DUPLICATES  \\\n",
              "0             0.000532               266736                   142   \n",
              "\n",
              "   READ_PAIR_OPTICAL_DUPLICATES  \n",
              "0                             0  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-2b5af4d7-83ac-4142-a95b-667bad4861fa\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>PERCENT_DUPLICATION</th>\n",
              "      <th>READ_PAIRS_EXAMINED</th>\n",
              "      <th>READ_PAIR_DUPLICATES</th>\n",
              "      <th>READ_PAIR_OPTICAL_DUPLICATES</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>0.000532</td>\n",
              "      <td>266736</td>\n",
              "      <td>142</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-2b5af4d7-83ac-4142-a95b-667bad4861fa')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-2b5af4d7-83ac-4142-a95b-667bad4861fa button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-2b5af4d7-83ac-4142-a95b-667bad4861fa');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "  <div id=\"id_f869a30c-d40a-4b24-a147-23e0d74a287b\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('df')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_f869a30c-d40a-4b24-a147-23e0d74a287b button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('df');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df",
              "summary": "{\n  \"name\": \"df\",\n  \"rows\": 1,\n  \"fields\": [\n    {\n      \"column\": \"PERCENT_DUPLICATION\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": 0.000532,\n        \"max\": 0.000532,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          0.000532\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"READ_PAIRS_EXAMINED\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": 266736,\n        \"max\": 266736,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          266736\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"READ_PAIR_DUPLICATES\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": 142,\n        \"max\": 142,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          142\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"READ_PAIR_OPTICAL_DUPLICATES\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": 0,\n        \"max\": 0,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "How to find the duplicated reads?\n",
        "\n",
        "https://broadinstitute.github.io/picard/explain-flags.html\n",
        "\n",
        "You can find the duplicated reads using the flag \"[[Quiz-2]]\""
      ],
      "metadata": {
        "id": "awF-MkctwGvf"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view -f [[Quiz-2]] normal.s.md.bam | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ffrdXaguv1cc",
        "outputId": "492968d8-cc01-4b36-af75-95d745b93dad"
      },
      "execution_count": 77,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "284\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view -f [[Quiz-2]] normal.s.md.bam | head -2"
      ],
      "metadata": {
        "id": "Lf8ieONmxFxD"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Base recalibration\n"
      ],
      "metadata": {
        "id": "548R62Dfxhxr"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Calculate the recalibration table"
      ],
      "metadata": {
        "id": "kfCflczJx0j2"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#runtime: 60s\n",
        "!java -jar /content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar \\\n",
        "  BaseRecalibrator \\\n",
        "  -I normal.s.md.bam \\\n",
        "  -R /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.fa \\\n",
        "  --known-sites /content/drive/MyDrive/AMC_NGS1/02_linux/resource/knownsites/hg38_v0_Homo_sapiens_assembly38.known_indels.chr21.vcf.gz \\\n",
        "  -L chr21 \\\n",
        "  -O normal.recal_data.table"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3ATMnnbX2DtN",
        "outputId": "9da2dd50-3cc6-4e30-bc1f-78b1e67519cd"
      },
      "execution_count": 78,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "23:47:09.752 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "23:47:10.038 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
            "23:47:10.039 INFO  BaseRecalibrator - The Genome Analysis Toolkit (GATK) v4.3.0.0\n",
            "23:47:10.039 INFO  BaseRecalibrator - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "23:47:10.040 INFO  BaseRecalibrator - Executing as root@49a1eeb69d60 on Linux v6.1.123+ amd64\n",
            "23:47:10.040 INFO  BaseRecalibrator - Java runtime: OpenJDK 64-Bit Server VM v11.0.28+6-post-Ubuntu-1ubuntu122.04.1\n",
            "23:47:10.040 INFO  BaseRecalibrator - Start Date/Time: August 20, 2025 at 11:47:09 PM UTC\n",
            "23:47:10.040 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
            "23:47:10.040 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
            "23:47:10.041 INFO  BaseRecalibrator - HTSJDK Version: 3.0.1\n",
            "23:47:10.041 INFO  BaseRecalibrator - Picard Version: 2.27.5\n",
            "23:47:10.041 INFO  BaseRecalibrator - Built for Spark Version: 2.4.5\n",
            "23:47:10.041 INFO  BaseRecalibrator - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "23:47:10.042 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "23:47:10.042 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "23:47:10.042 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "23:47:10.043 INFO  BaseRecalibrator - Deflater: IntelDeflater\n",
            "23:47:10.043 INFO  BaseRecalibrator - Inflater: IntelInflater\n",
            "23:47:10.043 INFO  BaseRecalibrator - GCS max retries/reopens: 20\n",
            "23:47:10.043 INFO  BaseRecalibrator - Requester pays: disabled\n",
            "23:47:10.043 INFO  BaseRecalibrator - Initializing engine\n",
            "23:47:10.370 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/drive/MyDrive/AMC_NGS1/02_linux/resource/knownsites/hg38_v0_Homo_sapiens_assembly38.known_indels.chr21.vcf.gz\n",
            "23:47:10.543 INFO  IntervalArgumentCollection - Processing 46709983 bp from intervals\n",
            "23:47:10.589 INFO  BaseRecalibrator - Done initializing engine\n",
            "23:47:10.590 WARN  BaseRecalibrator - This tool has only been well tested on ILLUMINA-based sequencing data. For other data use at your own risk.\n",
            "23:47:10.599 INFO  BaseRecalibrationEngine - The covariates being used here: \n",
            "23:47:10.599 INFO  BaseRecalibrationEngine - \tReadGroupCovariate\n",
            "23:47:10.599 INFO  BaseRecalibrationEngine - \tQualityScoreCovariate\n",
            "23:47:10.602 INFO  BaseRecalibrationEngine - \tContextCovariate\n",
            "23:47:10.602 INFO  BaseRecalibrationEngine - \tCycleCovariate\n",
            "23:47:10.617 INFO  ProgressMeter - Starting traversal\n",
            "23:47:10.617 INFO  ProgressMeter -        Current Locus  Elapsed Minutes       Reads Processed     Reads/Minute\n",
            "23:47:20.627 INFO  ProgressMeter -       chr21:36248527              0.2                221000        1324807.7\n",
            "23:47:27.096 INFO  BaseRecalibrator - 27492 read(s) filtered by: MappingQualityNotZeroReadFilter \n",
            "0 read(s) filtered by: MappingQualityAvailableReadFilter \n",
            "0 read(s) filtered by: MappedReadFilter \n",
            "0 read(s) filtered by: NotSecondaryAlignmentReadFilter \n",
            "278 read(s) filtered by: NotDuplicateReadFilter \n",
            "0 read(s) filtered by: PassesVendorQualityCheckReadFilter \n",
            "0 read(s) filtered by: WellformedReadFilter \n",
            "27770 total reads filtered\n",
            "23:47:27.097 INFO  ProgressMeter -       chr21:46661772              0.3                505702        1841148.1\n",
            "23:47:27.097 INFO  ProgressMeter - Traversal complete. Processed 505702 total reads in 0.3 minutes.\n",
            "23:47:27.229 INFO  BaseRecalibrator - Calculating quantized quality scores...\n",
            "23:47:27.263 INFO  BaseRecalibrator - Writing recalibration report...\n",
            "23:48:02.709 INFO  BaseRecalibrator - ...done!\n",
            "23:48:02.710 INFO  BaseRecalibrator - BaseRecalibrator was able to recalibrate 505702 reads\n",
            "23:48:02.710 INFO  BaseRecalibrator - Shutting down engine\n",
            "[August 20, 2025 at 11:48:02 PM UTC] org.broadinstitute.hellbender.tools.walkers.bqsr.BaseRecalibrator done. Elapsed time: 0.88 minutes.\n",
            "Runtime.totalMemory()=482344960\n",
            "Tool returned:\n",
            "SUCCESS\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# terminal 에서 실행\n",
        "# less normal.recal_data.table"
      ],
      "metadata": {
        "id": "AbATUdq50oI-"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#runtime: 26s\n",
        "!java -jar /content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar \\\n",
        "  ApplyBQSR \\\n",
        "  -I normal.s.md.bam \\\n",
        "  -R /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.fa \\\n",
        "  --bqsr-recal-file normal.recal_data.table \\\n",
        "  -L chr21 \\\n",
        "  -O normal.s.md.recal.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "gA_3pYoP2vWh",
        "outputId": "86e2dd3f-ff16-411c-8e29-42dc20af0cfa"
      },
      "execution_count": 79,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "23:49:05.287 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "23:49:05.858 INFO  ApplyBQSR - ------------------------------------------------------------\n",
            "23:49:05.859 INFO  ApplyBQSR - The Genome Analysis Toolkit (GATK) v4.3.0.0\n",
            "23:49:05.860 INFO  ApplyBQSR - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "23:49:05.861 INFO  ApplyBQSR - Executing as root@49a1eeb69d60 on Linux v6.1.123+ amd64\n",
            "23:49:05.861 INFO  ApplyBQSR - Java runtime: OpenJDK 64-Bit Server VM v11.0.28+6-post-Ubuntu-1ubuntu122.04.1\n",
            "23:49:05.861 INFO  ApplyBQSR - Start Date/Time: August 20, 2025 at 11:49:05 PM UTC\n",
            "23:49:05.861 INFO  ApplyBQSR - ------------------------------------------------------------\n",
            "23:49:05.861 INFO  ApplyBQSR - ------------------------------------------------------------\n",
            "23:49:05.862 INFO  ApplyBQSR - HTSJDK Version: 3.0.1\n",
            "23:49:05.862 INFO  ApplyBQSR - Picard Version: 2.27.5\n",
            "23:49:05.863 INFO  ApplyBQSR - Built for Spark Version: 2.4.5\n",
            "23:49:05.863 INFO  ApplyBQSR - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "23:49:05.863 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "23:49:05.863 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "23:49:05.863 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "23:49:05.864 INFO  ApplyBQSR - Deflater: IntelDeflater\n",
            "23:49:05.864 INFO  ApplyBQSR - Inflater: IntelInflater\n",
            "23:49:05.864 INFO  ApplyBQSR - GCS max retries/reopens: 20\n",
            "23:49:05.864 INFO  ApplyBQSR - Requester pays: disabled\n",
            "23:49:05.864 INFO  ApplyBQSR - Initializing engine\n",
            "23:49:06.330 INFO  IntervalArgumentCollection - Processing 46709983 bp from intervals\n",
            "23:49:06.336 INFO  ApplyBQSR - Done initializing engine\n",
            "23:49:06.371 WARN  ApplyBQSR - This tool has only been well tested on ILLUMINA-based sequencing data. For other data use at your own risk.\n",
            "23:49:06.372 INFO  ProgressMeter - Starting traversal\n",
            "23:49:06.372 INFO  ProgressMeter -        Current Locus  Elapsed Minutes       Reads Processed     Reads/Minute\n",
            "23:49:16.380 INFO  ProgressMeter -       chr21:36408887              0.2                246000        1474967.5\n",
            "23:49:25.688 INFO  ApplyBQSR - 0 read(s) filtered by: WellformedReadFilter \n",
            "\n",
            "23:49:25.688 INFO  ProgressMeter -       chr21:46663346              0.3                533472        1657174.2\n",
            "23:49:25.688 INFO  ProgressMeter - Traversal complete. Processed 533472 total reads in 0.3 minutes.\n",
            "23:49:25.745 INFO  ApplyBQSR - Shutting down engine\n",
            "[August 20, 2025 at 11:49:25 PM UTC] org.broadinstitute.hellbender.tools.walkers.bqsr.ApplyBQSR done. Elapsed time: 0.34 minutes.\n",
            "Runtime.totalMemory()=391118848\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!ls -al normal.sam normal*.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YZsM-IQ40r5d",
        "outputId": "54b24227-24aa-464f-82ea-2879de87adaf"
      },
      "execution_count": 80,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "-rw------- 1 root root  46988789 Aug 20 23:45 normal.bam\n",
            "-rw------- 1 root root 175050433 Aug 20 23:44 normal.sam\n",
            "-rw------- 1 root root  44554897 Aug 20 23:45 normal.s.bam\n",
            "-rw------- 1 root root  56650422 Aug 20 23:46 normal.s.md.bam\n",
            "-rw------- 1 root root  67113885 Aug 20 23:49 normal.s.md.recal.bam\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Pipeline (normal tissue)"
      ],
      "metadata": {
        "id": "Ulh6WqhE3wwx"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!cd /content/drive/MyDrive/AMC_NGS1/02_linux/data\n",
        "cd(\"/content/drive/MyDrive/AMC_NGS1/02_linux/data\")"
      ],
      "metadata": {
        "id": "fGak0nWK3V9k"
      },
      "execution_count": 81,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 120~180s\n",
        "!bwa mem -t 1 -R \"@RG\\tID:normal\\tSM:normal\\tPL:platform\" \\\n",
        "  ../resource/reference/hg38.chr21.fa \\\n",
        "  normal_R1.fastq.gz \\\n",
        "  normal_R2.fastq.gz > normal.sam 2> normal.bwa.log\n",
        "\n",
        "!samtools view -Sb normal.sam > normal.bam\n",
        "\n",
        "!samtools sort \\\n",
        "  -@ 1 \\\n",
        "  -o normal.s.bam \\\n",
        "  normal.bam\n",
        "\n",
        "!java -jar \\\n",
        "  /content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar \\\n",
        "  MarkDuplicates \\\n",
        "  --REMOVE_DUPLICATES false \\\n",
        "  --INPUT normal.s.bam \\\n",
        "  --OUTPUT normal.s.md.bam \\\n",
        "  --CREATE_INDEX true \\\n",
        "  --METRICS_FILE normal.md.metrics\n",
        "\n",
        "!java -jar /content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar \\\n",
        "  BaseRecalibrator \\\n",
        "  -I normal.s.md.bam \\\n",
        "  -R /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.fa \\\n",
        "  --known-sites /content/drive/MyDrive/AMC_NGS1/02_linux/resource/knownsites/hg38_v0_Homo_sapiens_assembly38.known_indels.chr21.vcf.gz \\\n",
        "  -L chr21 \\\n",
        "  -O normal.recal_data.table\n",
        "\n",
        "!java -jar /content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar \\\n",
        "  ApplyBQSR \\\n",
        "  -I normal.s.md.bam \\\n",
        "  -R /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.fa \\\n",
        "  --bqsr-recal-file normal.recal_data.table \\\n",
        "  -L chr21 \\\n",
        "  -O normal.s.md.recal.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "09b5804a-1c09-4599-a399-b75f8cbf82c7",
        "id": "AT4WJ9wd3V9l"
      },
      "execution_count": 82,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "23:53:36.275 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "[Wed Aug 20 23:53:36 UTC 2025] MarkDuplicates --INPUT normal.s.bam --OUTPUT normal.s.md.bam --METRICS_FILE normal.md.metrics --REMOVE_DUPLICATES false --CREATE_INDEX true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --ASSUME_SORTED false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 2 --MAX_RECORDS_IN_RAM 500000 --CREATE_MD5_FILE false --GA4GH_CLIENT_SECRETS client_secrets.json --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
            "[Wed Aug 20 23:53:36 UTC 2025] Executing as root@49a1eeb69d60 on Linux 6.1.123+ amd64; OpenJDK 64-Bit Server VM 11.0.28+6-post-Ubuntu-1ubuntu122.04.1; Deflater: Intel; Inflater: Intel; Provider GCS is available; Picard version: Version:4.3.0.0\n",
            "INFO\t2025-08-20 23:53:36\tMarkDuplicates\tStart of doWork freeMemory: 71151680; totalMemory: 98566144; maxMemory: 3403677696\n",
            "INFO\t2025-08-20 23:53:36\tMarkDuplicates\tReading input file and constructing read end information.\n",
            "INFO\t2025-08-20 23:53:36\tMarkDuplicates\tWill retain up to 12332165 data points before spilling to disk.\n",
            "WARNING: BAM index file /content/drive/MyDrive/AMC_NGS1/02_linux/data/normal.s.bam.bai is older than BAM /content/drive/MyDrive/AMC_NGS1/02_linux/data/normal.s.bam\n",
            "WARNING\t2025-08-20 23:53:36\tAbstractOpticalDuplicateFinderCommandLineProgram\tA field field parsed out of a read name was expected to contain an integer and did not. Read name: normal#21#251697#3938. Cause: String 'normal#21#251697#3938' did not start with a parsable number.\n",
            "INFO\t2025-08-20 23:53:45\tMarkDuplicates\tRead 533472 records. 0 pairs never matched.\n",
            "INFO\t2025-08-20 23:53:45\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 414912768; totalMemory: 600834048; maxMemory: 3403677696\n",
            "INFO\t2025-08-20 23:53:45\tMarkDuplicates\tWill retain up to 106364928 duplicate indices before spilling to disk.\n",
            "INFO\t2025-08-20 23:53:47\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
            "INFO\t2025-08-20 23:53:47\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
            "INFO\t2025-08-20 23:53:47\tMarkDuplicates\tSorting list of duplicate records.\n",
            "INFO\t2025-08-20 23:53:47\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 588958384; totalMemory: 1468006400; maxMemory: 3403677696\n",
            "INFO\t2025-08-20 23:53:47\tMarkDuplicates\tMarking 284 records as duplicates.\n",
            "INFO\t2025-08-20 23:53:47\tMarkDuplicates\tFound 0 optical duplicate clusters.\n",
            "WARNING: BAM index file /content/drive/MyDrive/AMC_NGS1/02_linux/data/normal.s.bam.bai is older than BAM /content/drive/MyDrive/AMC_NGS1/02_linux/data/normal.s.bam\n",
            "INFO\t2025-08-20 23:53:47\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
            "INFO\t2025-08-20 23:53:52\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
            "INFO\t2025-08-20 23:53:52\tMarkDuplicates\tDuplicate Index cleanup.\n",
            "INFO\t2025-08-20 23:53:52\tMarkDuplicates\tGetting Memory Stats.\n",
            "INFO\t2025-08-20 23:53:52\tMarkDuplicates\tBefore output close freeMemory: 78057760; totalMemory: 109051904; maxMemory: 3403677696\n",
            "INFO\t2025-08-20 23:53:52\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
            "INFO\t2025-08-20 23:53:52\tMarkDuplicates\tAfter output close freeMemory: 73216176; totalMemory: 101711872; maxMemory: 3403677696\n",
            "[Wed Aug 20 23:53:52 UTC 2025] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.28 minutes.\n",
            "Runtime.totalMemory()=101711872\n",
            "Tool returned:\n",
            "0\n",
            "23:53:59.479 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "23:53:59.761 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
            "23:53:59.762 INFO  BaseRecalibrator - The Genome Analysis Toolkit (GATK) v4.3.0.0\n",
            "23:53:59.762 INFO  BaseRecalibrator - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "23:53:59.763 INFO  BaseRecalibrator - Executing as root@49a1eeb69d60 on Linux v6.1.123+ amd64\n",
            "23:53:59.763 INFO  BaseRecalibrator - Java runtime: OpenJDK 64-Bit Server VM v11.0.28+6-post-Ubuntu-1ubuntu122.04.1\n",
            "23:53:59.763 INFO  BaseRecalibrator - Start Date/Time: August 20, 2025 at 11:53:58 PM UTC\n",
            "23:53:59.763 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
            "23:53:59.763 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
            "23:53:59.765 INFO  BaseRecalibrator - HTSJDK Version: 3.0.1\n",
            "23:53:59.765 INFO  BaseRecalibrator - Picard Version: 2.27.5\n",
            "23:53:59.765 INFO  BaseRecalibrator - Built for Spark Version: 2.4.5\n",
            "23:53:59.765 INFO  BaseRecalibrator - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "23:53:59.766 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "23:53:59.766 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "23:53:59.766 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "23:53:59.767 INFO  BaseRecalibrator - Deflater: IntelDeflater\n",
            "23:53:59.767 INFO  BaseRecalibrator - Inflater: IntelInflater\n",
            "23:53:59.767 INFO  BaseRecalibrator - GCS max retries/reopens: 20\n",
            "23:53:59.767 INFO  BaseRecalibrator - Requester pays: disabled\n",
            "23:53:59.767 INFO  BaseRecalibrator - Initializing engine\n",
            "23:54:00.146 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/drive/MyDrive/AMC_NGS1/02_linux/resource/knownsites/hg38_v0_Homo_sapiens_assembly38.known_indels.chr21.vcf.gz\n",
            "23:54:00.329 INFO  IntervalArgumentCollection - Processing 46709983 bp from intervals\n",
            "23:54:00.383 INFO  BaseRecalibrator - Done initializing engine\n",
            "23:54:00.384 WARN  BaseRecalibrator - This tool has only been well tested on ILLUMINA-based sequencing data. For other data use at your own risk.\n",
            "23:54:00.388 INFO  BaseRecalibrationEngine - The covariates being used here: \n",
            "23:54:00.389 INFO  BaseRecalibrationEngine - \tReadGroupCovariate\n",
            "23:54:00.389 INFO  BaseRecalibrationEngine - \tQualityScoreCovariate\n",
            "23:54:00.392 INFO  BaseRecalibrationEngine - \tContextCovariate\n",
            "23:54:00.392 INFO  BaseRecalibrationEngine - \tCycleCovariate\n",
            "23:54:00.406 INFO  ProgressMeter - Starting traversal\n",
            "23:54:00.406 INFO  ProgressMeter -        Current Locus  Elapsed Minutes       Reads Processed     Reads/Minute\n",
            "23:54:10.442 INFO  ProgressMeter -       chr21:41889306              0.2                323000        1931240.7\n",
            "23:54:15.372 INFO  BaseRecalibrator - 27492 read(s) filtered by: MappingQualityNotZeroReadFilter \n",
            "0 read(s) filtered by: MappingQualityAvailableReadFilter \n",
            "0 read(s) filtered by: MappedReadFilter \n",
            "0 read(s) filtered by: NotSecondaryAlignmentReadFilter \n",
            "278 read(s) filtered by: NotDuplicateReadFilter \n",
            "0 read(s) filtered by: PassesVendorQualityCheckReadFilter \n",
            "0 read(s) filtered by: WellformedReadFilter \n",
            "27770 total reads filtered\n",
            "23:54:15.373 INFO  ProgressMeter -       chr21:46661772              0.2                505702        2027268.0\n",
            "23:54:15.373 INFO  ProgressMeter - Traversal complete. Processed 505702 total reads in 0.2 minutes.\n",
            "23:54:15.459 INFO  BaseRecalibrator - Calculating quantized quality scores...\n",
            "23:54:15.476 INFO  BaseRecalibrator - Writing recalibration report...\n",
            "23:54:48.442 INFO  BaseRecalibrator - ...done!\n",
            "23:54:48.442 INFO  BaseRecalibrator - BaseRecalibrator was able to recalibrate 505702 reads\n",
            "23:54:48.443 INFO  BaseRecalibrator - Shutting down engine\n",
            "[August 20, 2025 at 11:54:48 PM UTC] org.broadinstitute.hellbender.tools.walkers.bqsr.BaseRecalibrator done. Elapsed time: 0.83 minutes.\n",
            "Runtime.totalMemory()=601882624\n",
            "Tool returned:\n",
            "SUCCESS\n",
            "23:54:53.534 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "23:54:54.026 INFO  ApplyBQSR - ------------------------------------------------------------\n",
            "23:54:54.027 INFO  ApplyBQSR - The Genome Analysis Toolkit (GATK) v4.3.0.0\n",
            "23:54:54.027 INFO  ApplyBQSR - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "23:54:54.027 INFO  ApplyBQSR - Executing as root@49a1eeb69d60 on Linux v6.1.123+ amd64\n",
            "23:54:54.027 INFO  ApplyBQSR - Java runtime: OpenJDK 64-Bit Server VM v11.0.28+6-post-Ubuntu-1ubuntu122.04.1\n",
            "23:54:54.028 INFO  ApplyBQSR - Start Date/Time: August 20, 2025 at 11:54:53 PM UTC\n",
            "23:54:54.028 INFO  ApplyBQSR - ------------------------------------------------------------\n",
            "23:54:54.028 INFO  ApplyBQSR - ------------------------------------------------------------\n",
            "23:54:54.029 INFO  ApplyBQSR - HTSJDK Version: 3.0.1\n",
            "23:54:54.031 INFO  ApplyBQSR - Picard Version: 2.27.5\n",
            "23:54:54.032 INFO  ApplyBQSR - Built for Spark Version: 2.4.5\n",
            "23:54:54.032 INFO  ApplyBQSR - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "23:54:54.032 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "23:54:54.032 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "23:54:54.032 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "23:54:54.033 INFO  ApplyBQSR - Deflater: IntelDeflater\n",
            "23:54:54.033 INFO  ApplyBQSR - Inflater: IntelInflater\n",
            "23:54:54.033 INFO  ApplyBQSR - GCS max retries/reopens: 20\n",
            "23:54:54.033 INFO  ApplyBQSR - Requester pays: disabled\n",
            "23:54:54.033 INFO  ApplyBQSR - Initializing engine\n",
            "23:54:54.424 INFO  IntervalArgumentCollection - Processing 46709983 bp from intervals\n",
            "23:54:54.434 INFO  ApplyBQSR - Done initializing engine\n",
            "23:54:54.496 WARN  ApplyBQSR - This tool has only been well tested on ILLUMINA-based sequencing data. For other data use at your own risk.\n",
            "23:54:54.496 INFO  ProgressMeter - Starting traversal\n",
            "23:54:54.497 INFO  ProgressMeter -        Current Locus  Elapsed Minutes       Reads Processed     Reads/Minute\n",
            "23:55:04.540 INFO  ProgressMeter -       chr21:33263466              0.2                175000        1045608.4\n",
            "23:55:14.545 INFO  ProgressMeter -       chr21:46213018              0.3                492000        1472539.5\n",
            "23:55:15.743 INFO  ApplyBQSR - 0 read(s) filtered by: WellformedReadFilter \n",
            "\n",
            "23:55:15.744 INFO  ProgressMeter -       chr21:46663346              0.4                533472        1506557.5\n",
            "23:55:15.744 INFO  ProgressMeter - Traversal complete. Processed 533472 total reads in 0.4 minutes.\n",
            "23:55:15.830 INFO  ApplyBQSR - Shutting down engine\n",
            "[August 20, 2025 at 11:55:15 PM UTC] org.broadinstitute.hellbender.tools.walkers.bqsr.ApplyBQSR done. Elapsed time: 0.37 minutes.\n",
            "Runtime.totalMemory()=351272960\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Pipeline (tumor tissue)"
      ],
      "metadata": {
        "id": "RrcruP6p4qk7"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 120s\n",
        "!bwa mem -t 1 -R \"@RG\\tID:tumor\\tSM:tumor\\tPL:platform\" \\\n",
        "  ../resource/reference/hg38.chr21.fa \\\n",
        "  tumor_R1.fastq.gz \\\n",
        "  tumor_R2.fastq.gz > tumor.sam 2> tumor.bwa.log\n",
        "\n",
        "!samtools view -Sb tumor.sam > tumor.bam\n",
        "\n",
        "!samtools sort \\\n",
        "  -@ 1 \\\n",
        "  -o tumor.s.bam \\\n",
        "  tumor.bam\n",
        "\n",
        "!java -jar \\\n",
        "  /content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar \\\n",
        "  MarkDuplicates \\\n",
        "  --REMOVE_DUPLICATES false \\\n",
        "  --INPUT tumor.s.bam \\\n",
        "  --OUTPUT tumor.s.md.bam \\\n",
        "  --CREATE_INDEX true \\\n",
        "  --METRICS_FILE tumor.md.metrics\n",
        "\n",
        "!java -jar /content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar \\\n",
        "  BaseRecalibrator \\\n",
        "  -I tumor.s.md.bam \\\n",
        "  -R /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.fa \\\n",
        "  --known-sites /content/drive/MyDrive/AMC_NGS1/02_linux/resource/knownsites/hg38_v0_Homo_sapiens_assembly38.known_indels.chr21.vcf.gz \\\n",
        "  -L chr21 \\\n",
        "  -O tumor.recal_data.table\n",
        "\n",
        "!java -jar /content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar \\\n",
        "  ApplyBQSR \\\n",
        "  -I tumor.s.md.bam \\\n",
        "  -R /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.fa \\\n",
        "  --bqsr-recal-file tumor.recal_data.table \\\n",
        "  -L chr21 \\\n",
        "  -O tumor.s.md.recal.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0b679e39-56b8-449f-ff31-e96d46231212",
        "id": "iDENy3WO4qk7"
      },
      "execution_count": 83,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "23:58:21.778 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "[Wed Aug 20 23:58:21 UTC 2025] MarkDuplicates --INPUT tumor.s.bam --OUTPUT tumor.s.md.bam --METRICS_FILE tumor.md.metrics --REMOVE_DUPLICATES false --CREATE_INDEX true --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS false --REMOVE_SEQUENCING_DUPLICATES false --TAGGING_POLICY DontTag --CLEAR_DT true --DUPLEX_UMI false --FLOW_MODE false --FLOW_QUALITY_SUM_STRATEGY false --USE_END_IN_UNPAIRED_READS false --USE_UNPAIRED_CLIPPED_END false --UNPAIRED_END_UNCERTAINTY 0 --FLOW_SKIP_FIRST_N_FLOWS 0 --FLOW_Q_IS_KNOWN_END false --FLOW_EFFECTIVE_QUALITY_THRESHOLD 15 --ADD_PG_TAG_TO_READS true --ASSUME_SORTED false --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 2 --MAX_RECORDS_IN_RAM 500000 --CREATE_MD5_FILE false --GA4GH_CLIENT_SECRETS client_secrets.json --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
            "[Wed Aug 20 23:58:22 UTC 2025] Executing as root@49a1eeb69d60 on Linux 6.1.123+ amd64; OpenJDK 64-Bit Server VM 11.0.28+6-post-Ubuntu-1ubuntu122.04.1; Deflater: Intel; Inflater: Intel; Provider GCS is available; Picard version: Version:4.3.0.0\n",
            "INFO\t2025-08-20 23:58:22\tMarkDuplicates\tStart of doWork freeMemory: 66957024; totalMemory: 94371840; maxMemory: 3403677696\n",
            "INFO\t2025-08-20 23:58:22\tMarkDuplicates\tReading input file and constructing read end information.\n",
            "INFO\t2025-08-20 23:58:22\tMarkDuplicates\tWill retain up to 12332165 data points before spilling to disk.\n",
            "WARNING\t2025-08-20 23:58:23\tAbstractOpticalDuplicateFinderCommandLineProgram\tA field field parsed out of a read name was expected to contain an integer and did not. Read name: tumor#21#251703#13093. Cause: String 'tumor#21#251703#13093' did not start with a parsable number.\n",
            "INFO\t2025-08-20 23:58:30\tMarkDuplicates\tRead 535442 records. 0 pairs never matched.\n",
            "INFO\t2025-08-20 23:58:30\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 439865600; totalMemory: 625999872; maxMemory: 3403677696\n",
            "INFO\t2025-08-20 23:58:30\tMarkDuplicates\tWill retain up to 106364928 duplicate indices before spilling to disk.\n",
            "INFO\t2025-08-20 23:58:32\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
            "INFO\t2025-08-20 23:58:32\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
            "INFO\t2025-08-20 23:58:33\tMarkDuplicates\tSorting list of duplicate records.\n",
            "INFO\t2025-08-20 23:58:33\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1052428200; totalMemory: 1931476992; maxMemory: 3403677696\n",
            "INFO\t2025-08-20 23:58:33\tMarkDuplicates\tMarking 272 records as duplicates.\n",
            "INFO\t2025-08-20 23:58:33\tMarkDuplicates\tFound 0 optical duplicate clusters.\n",
            "INFO\t2025-08-20 23:58:33\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
            "INFO\t2025-08-20 23:58:41\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
            "INFO\t2025-08-20 23:58:41\tMarkDuplicates\tDuplicate Index cleanup.\n",
            "INFO\t2025-08-20 23:58:41\tMarkDuplicates\tGetting Memory Stats.\n",
            "INFO\t2025-08-20 23:58:41\tMarkDuplicates\tBefore output close freeMemory: 78489744; totalMemory: 109051904; maxMemory: 3403677696\n",
            "INFO\t2025-08-20 23:58:41\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
            "INFO\t2025-08-20 23:58:41\tMarkDuplicates\tAfter output close freeMemory: 73214976; totalMemory: 101711872; maxMemory: 3403677696\n",
            "[Wed Aug 20 23:58:41 UTC 2025] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 0.33 minutes.\n",
            "Runtime.totalMemory()=101711872\n",
            "Tool returned:\n",
            "0\n",
            "23:58:45.480 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "23:58:45.750 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
            "23:58:45.750 INFO  BaseRecalibrator - The Genome Analysis Toolkit (GATK) v4.3.0.0\n",
            "23:58:45.751 INFO  BaseRecalibrator - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "23:58:45.751 INFO  BaseRecalibrator - Executing as root@49a1eeb69d60 on Linux v6.1.123+ amd64\n",
            "23:58:45.751 INFO  BaseRecalibrator - Java runtime: OpenJDK 64-Bit Server VM v11.0.28+6-post-Ubuntu-1ubuntu122.04.1\n",
            "23:58:45.751 INFO  BaseRecalibrator - Start Date/Time: August 20, 2025 at 11:58:45 PM UTC\n",
            "23:58:45.751 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
            "23:58:45.751 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
            "23:58:45.753 INFO  BaseRecalibrator - HTSJDK Version: 3.0.1\n",
            "23:58:45.753 INFO  BaseRecalibrator - Picard Version: 2.27.5\n",
            "23:58:45.753 INFO  BaseRecalibrator - Built for Spark Version: 2.4.5\n",
            "23:58:45.753 INFO  BaseRecalibrator - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "23:58:45.753 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "23:58:45.753 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "23:58:45.754 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "23:58:45.754 INFO  BaseRecalibrator - Deflater: IntelDeflater\n",
            "23:58:45.754 INFO  BaseRecalibrator - Inflater: IntelInflater\n",
            "23:58:45.754 INFO  BaseRecalibrator - GCS max retries/reopens: 20\n",
            "23:58:45.754 INFO  BaseRecalibrator - Requester pays: disabled\n",
            "23:58:45.755 INFO  BaseRecalibrator - Initializing engine\n",
            "23:58:46.132 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/drive/MyDrive/AMC_NGS1/02_linux/resource/knownsites/hg38_v0_Homo_sapiens_assembly38.known_indels.chr21.vcf.gz\n",
            "23:58:46.289 INFO  IntervalArgumentCollection - Processing 46709983 bp from intervals\n",
            "23:58:46.343 INFO  BaseRecalibrator - Done initializing engine\n",
            "23:58:46.346 WARN  BaseRecalibrator - This tool has only been well tested on ILLUMINA-based sequencing data. For other data use at your own risk.\n",
            "23:58:46.355 INFO  BaseRecalibrationEngine - The covariates being used here: \n",
            "23:58:46.355 INFO  BaseRecalibrationEngine - \tReadGroupCovariate\n",
            "23:58:46.355 INFO  BaseRecalibrationEngine - \tQualityScoreCovariate\n",
            "23:58:46.358 INFO  BaseRecalibrationEngine - \tContextCovariate\n",
            "23:58:46.358 INFO  BaseRecalibrationEngine - \tCycleCovariate\n",
            "23:58:46.371 INFO  ProgressMeter - Starting traversal\n",
            "23:58:46.372 INFO  ProgressMeter -        Current Locus  Elapsed Minutes       Reads Processed     Reads/Minute\n",
            "23:58:56.395 INFO  ProgressMeter -       chr21:36364069              0.2                228000        1364997.0\n",
            "23:59:02.813 INFO  BaseRecalibrator - 27141 read(s) filtered by: MappingQualityNotZeroReadFilter \n",
            "0 read(s) filtered by: MappingQualityAvailableReadFilter \n",
            "0 read(s) filtered by: MappedReadFilter \n",
            "0 read(s) filtered by: NotSecondaryAlignmentReadFilter \n",
            "258 read(s) filtered by: NotDuplicateReadFilter \n",
            "0 read(s) filtered by: PassesVendorQualityCheckReadFilter \n",
            "0 read(s) filtered by: WellformedReadFilter \n",
            "27399 total reads filtered\n",
            "23:59:02.814 INFO  ProgressMeter -       chr21:46664653              0.3                508043        1854058.8\n",
            "23:59:02.814 INFO  ProgressMeter - Traversal complete. Processed 508043 total reads in 0.3 minutes.\n",
            "23:59:02.899 INFO  BaseRecalibrator - Calculating quantized quality scores...\n",
            "23:59:02.939 INFO  BaseRecalibrator - Writing recalibration report...\n",
            "23:59:38.832 INFO  BaseRecalibrator - ...done!\n",
            "23:59:38.832 INFO  BaseRecalibrator - BaseRecalibrator was able to recalibrate 508043 reads\n",
            "23:59:38.832 INFO  BaseRecalibrator - Shutting down engine\n",
            "[August 20, 2025 at 11:59:38 PM UTC] org.broadinstitute.hellbender.tools.walkers.bqsr.BaseRecalibrator done. Elapsed time: 0.89 minutes.\n",
            "Runtime.totalMemory()=622854144\n",
            "Tool returned:\n",
            "SUCCESS\n",
            "23:59:44.135 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "23:59:44.368 INFO  ApplyBQSR - ------------------------------------------------------------\n",
            "23:59:44.369 INFO  ApplyBQSR - The Genome Analysis Toolkit (GATK) v4.3.0.0\n",
            "23:59:44.369 INFO  ApplyBQSR - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "23:59:44.370 INFO  ApplyBQSR - Executing as root@49a1eeb69d60 on Linux v6.1.123+ amd64\n",
            "23:59:44.370 INFO  ApplyBQSR - Java runtime: OpenJDK 64-Bit Server VM v11.0.28+6-post-Ubuntu-1ubuntu122.04.1\n",
            "23:59:44.370 INFO  ApplyBQSR - Start Date/Time: August 20, 2025 at 11:59:44 PM UTC\n",
            "23:59:44.370 INFO  ApplyBQSR - ------------------------------------------------------------\n",
            "23:59:44.370 INFO  ApplyBQSR - ------------------------------------------------------------\n",
            "23:59:44.371 INFO  ApplyBQSR - HTSJDK Version: 3.0.1\n",
            "23:59:44.371 INFO  ApplyBQSR - Picard Version: 2.27.5\n",
            "23:59:44.371 INFO  ApplyBQSR - Built for Spark Version: 2.4.5\n",
            "23:59:44.371 INFO  ApplyBQSR - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "23:59:44.372 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "23:59:44.372 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "23:59:44.372 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "23:59:44.372 INFO  ApplyBQSR - Deflater: IntelDeflater\n",
            "23:59:44.372 INFO  ApplyBQSR - Inflater: IntelInflater\n",
            "23:59:44.372 INFO  ApplyBQSR - GCS max retries/reopens: 20\n",
            "23:59:44.372 INFO  ApplyBQSR - Requester pays: disabled\n",
            "23:59:44.373 INFO  ApplyBQSR - Initializing engine\n",
            "23:59:44.673 INFO  IntervalArgumentCollection - Processing 46709983 bp from intervals\n",
            "23:59:44.678 INFO  ApplyBQSR - Done initializing engine\n",
            "23:59:44.711 WARN  ApplyBQSR - This tool has only been well tested on ILLUMINA-based sequencing data. For other data use at your own risk.\n",
            "23:59:44.711 INFO  ProgressMeter - Starting traversal\n",
            "23:59:44.712 INFO  ProgressMeter -        Current Locus  Elapsed Minutes       Reads Processed     Reads/Minute\n",
            "23:59:54.725 INFO  ProgressMeter -       chr21:33858831              0.2                210000        1258489.8\n",
            "00:00:03.721 INFO  ApplyBQSR - 0 read(s) filtered by: WellformedReadFilter \n",
            "\n",
            "00:00:03.721 INFO  ProgressMeter -       chr21:46663441              0.3                535442        1690068.9\n",
            "00:00:03.721 INFO  ProgressMeter - Traversal complete. Processed 535442 total reads in 0.3 minutes.\n",
            "00:00:03.782 INFO  ApplyBQSR - Shutting down engine\n",
            "[August 21, 2025 at 12:00:03 AM UTC] org.broadinstitute.hellbender.tools.walkers.bqsr.ApplyBQSR done. Elapsed time: 0.33 minutes.\n",
            "Runtime.totalMemory()=312475648\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!rm normal.sam normal.s.md.bai normal.s.bam.bai tumor.s.md.bai normal.bwa.log normal.bam normal.s.bam normal.md.metrics normal.recal_data.table normal.s.md.bam\n",
        "!rm tumor.sam tumor.s.md.bai tumor.s.bam.bai tumor.s.md.bai tumor.bwa.log tumor.bam tumor.s.bam tumor.md.metrics tumor.recal_data.table tumor.s.md.bam"
      ],
      "metadata": {
        "id": "Joz8Yeu65OPJ"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!ls -al"
      ],
      "metadata": {
        "id": "jHiE1eKa5hl9"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Variant calling\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "-_6hwDRQ53B1"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Somatic vs. Germline mutation calling"
      ],
      "metadata": {
        "id": "BohbUoNq6izT"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "ppt에서 설명"
      ],
      "metadata": {
        "id": "s_yAPb2y6njH"
      }
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "8TZrVv-H69NX"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Mutect2를 이용한 somatic mutation calling\n",
        "\n",
        "참조문서: https://gatk.broadinstitute.org/hc/en-us/articles/360037593851-Mutect2"
      ],
      "metadata": {
        "id": "B0psijY06wcL"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Tumor with matched normal\n",
        "Given a matched normal, **Mutect2 is designed to call somatic variants only**. The tool includes logic to **skip emitting variants** that are clearly present in the germline based on provided evidence, e.g. in the matched normal. This is done at an early stage to avoid spending computational resources on germline events. If the variant's **germline status is borderline**, then Mutect2 will emit the variant to the callset for subsequent filtering by **FilterMutectCalls** and review.\n",
        "\n",
        "```\n",
        "gatk Mutect2 \\\n",
        "     -R reference.fa \\\n",
        "     -I tumor.bam \\\n",
        "     -I normal.bam \\\n",
        "     -normal normal_sample_name \\\n",
        "     --germline-resource af-only-gnomad.vcf.gz \\\n",
        "     -O somatic.vcf.gz\n",
        "```\n"
      ],
      "metadata": {
        "id": "JIZD7syy61JN"
      }
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "72dYdYDA7ZSy"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#runtime: 30s\n",
        "!java -jar /content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar \\\n",
        "  Mutect2 \\\n",
        "  -R /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.fa \\\n",
        "  -I tumor.s.md.recal.bam \\\n",
        "  -I normal.s.md.recal.bam \\\n",
        "  -normal normal \\\n",
        "  --germline-resource /content/drive/MyDrive/AMC_NGS1/02_linux/resource/knownsites/af-only-gnomad.hg38.chr21.vcf.gz \\\n",
        "  -L chr21q \\\n",
        "  -O somatic.vcf.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "w3-MCXW_56l0",
        "outputId": "3a6c8968-9d82-48d7-fdf2-6fd078d81917"
      },
      "execution_count": 84,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "00:01:42.453 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "00:01:42.732 INFO  Mutect2 - ------------------------------------------------------------\n",
            "00:01:42.733 INFO  Mutect2 - The Genome Analysis Toolkit (GATK) v4.3.0.0\n",
            "00:01:42.734 INFO  Mutect2 - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "00:01:42.734 INFO  Mutect2 - Executing as root@49a1eeb69d60 on Linux v6.1.123+ amd64\n",
            "00:01:42.734 INFO  Mutect2 - Java runtime: OpenJDK 64-Bit Server VM v11.0.28+6-post-Ubuntu-1ubuntu122.04.1\n",
            "00:01:42.734 INFO  Mutect2 - Start Date/Time: August 21, 2025 at 12:01:42 AM UTC\n",
            "00:01:42.735 INFO  Mutect2 - ------------------------------------------------------------\n",
            "00:01:42.735 INFO  Mutect2 - ------------------------------------------------------------\n",
            "00:01:42.737 INFO  Mutect2 - HTSJDK Version: 3.0.1\n",
            "00:01:42.737 INFO  Mutect2 - Picard Version: 2.27.5\n",
            "00:01:42.737 INFO  Mutect2 - Built for Spark Version: 2.4.5\n",
            "00:01:42.737 INFO  Mutect2 - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "00:01:42.738 INFO  Mutect2 - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "00:01:42.738 INFO  Mutect2 - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "00:01:42.738 INFO  Mutect2 - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "00:01:42.738 INFO  Mutect2 - Deflater: IntelDeflater\n",
            "00:01:42.738 INFO  Mutect2 - Inflater: IntelInflater\n",
            "00:01:42.738 INFO  Mutect2 - GCS max retries/reopens: 20\n",
            "00:01:42.738 INFO  Mutect2 - Requester pays: disabled\n",
            "00:01:42.738 INFO  Mutect2 - Initializing engine\n",
            "00:01:43.138 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/drive/MyDrive/AMC_NGS1/02_linux/resource/knownsites/af-only-gnomad.hg38.chr21.vcf.gz\n",
            "00:01:43.295 INFO  IntervalArgumentCollection - Processing 46709983 bp from intervals\n",
            "00:01:43.338 INFO  Mutect2 - Done initializing engine\n",
            "00:01:43.402 INFO  NativeLibraryLoader - Loading libgkl_utils.so from jar:file:/content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar!/com/intel/gkl/native/libgkl_utils.so\n",
            "00:01:43.412 INFO  NativeLibraryLoader - Loading libgkl_pairhmm_omp.so from jar:file:/content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar!/com/intel/gkl/native/libgkl_pairhmm_omp.so\n",
            "00:01:43.452 INFO  IntelPairHmm - Flush-to-zero (FTZ) is enabled when running PairHMM\n",
            "00:01:43.453 INFO  IntelPairHmm - Available threads: 2\n",
            "00:01:43.454 INFO  IntelPairHmm - Requested threads: 4\n",
            "00:01:43.454 WARN  IntelPairHmm - Using 2 available threads, but 4 were requested\n",
            "00:01:43.454 INFO  PairHMM - Using the OpenMP multi-threaded AVX-accelerated native PairHMM implementation\n",
            "00:01:43.571 INFO  ProgressMeter - Starting traversal\n",
            "00:01:43.571 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Regions Processed   Regions/Minute\n",
            "00:01:54.780 INFO  ProgressMeter -        chr21:6120035              0.2                 20410         109388.1\n",
            "00:02:05.610 INFO  ProgressMeter -        chr21:6501739              0.4                 21710          59104.3\n",
            "00:02:15.883 INFO  ProgressMeter -       chr21:10542784              0.5                 35200          65374.9\n",
            "00:02:26.747 INFO  ProgressMeter -       chr21:10578950              0.7                 35350          49125.7\n",
            "00:02:36.886 INFO  ProgressMeter -       chr21:13620116              0.9                 45520          51227.6\n",
            "00:02:46.949 INFO  ProgressMeter -       chr21:14208250              1.1                 47550          45015.6\n",
            "00:02:58.003 INFO  ProgressMeter -       chr21:14965992              1.2                 50180          40450.9\n",
            "00:03:08.041 INFO  ProgressMeter -       chr21:15835742              1.4                 53160          37760.2\n",
            "00:03:18.189 INFO  ProgressMeter -       chr21:18247446              1.6                 61290          38866.2\n",
            "00:03:28.450 INFO  ProgressMeter -       chr21:18401034              1.7                 61910          35418.0\n",
            "00:03:38.564 INFO  ProgressMeter -       chr21:25588658              1.9                 85950          44846.2\n",
            "00:03:48.594 INFO  ProgressMeter -       chr21:25762966              2.1                 86630          41574.8\n",
            "00:03:59.504 INFO  ProgressMeter -       chr21:26841957              2.3                 90340          39875.8\n",
            "00:04:09.591 INFO  ProgressMeter -       chr21:28945164              2.4                 97470          40050.7\n",
            "00:04:19.788 INFO  ProgressMeter -       chr21:29028302              2.6                 97840          37578.5\n",
            "00:04:29.817 INFO  ProgressMeter -       chr21:29067623              2.8                 98060          35390.9\n",
            "00:04:39.953 INFO  ProgressMeter -       chr21:29651561              2.9                100130          34061.3\n",
            "00:04:50.062 INFO  ProgressMeter -       chr21:30430828              3.1                102820          33080.4\n",
            "00:05:00.066 INFO  ProgressMeter -       chr21:31038187              3.3                104940          32043.7\n",
            "00:05:10.414 INFO  ProgressMeter -       chr21:31674080              3.4                107200          31096.2\n",
            "00:05:20.828 INFO  ProgressMeter -       chr21:32270187              3.6                109320          30191.0\n",
            "00:05:31.007 INFO  ProgressMeter -       chr21:32362546              3.8                109740          28950.6\n",
            "00:05:41.035 INFO  ProgressMeter -       chr21:32581661              4.0                110560          27935.2\n",
            "00:05:51.042 INFO  ProgressMeter -       chr21:32671997              4.1                110970          26905.1\n",
            "00:06:02.125 INFO  ProgressMeter -       chr21:32766358              4.3                111400          25851.7\n",
            "00:06:12.180 INFO  ProgressMeter -       chr21:33327399              4.5                113360          25322.0\n",
            "00:06:22.614 INFO  ProgressMeter -       chr21:33506412              4.7                114080          24529.6\n",
            "00:06:32.789 INFO  ProgressMeter -       chr21:33560267              4.8                114370          23726.8\n",
            "00:06:42.825 INFO  ProgressMeter -       chr21:33624442              5.0                114680          22993.3\n",
            "00:06:52.936 INFO  ProgressMeter -       chr21:33837464              5.2                115510          22402.7\n",
            "00:07:03.155 INFO  ProgressMeter -       chr21:34095490              5.3                116440          21860.9\n",
            "00:07:14.448 INFO  ProgressMeter -       chr21:34615312              5.5                118240          21441.2\n",
            "00:07:24.656 INFO  ProgressMeter -       chr21:36041045              5.7                123090          21652.7\n",
            "00:07:34.734 INFO  ProgressMeter -       chr21:36230487              5.9                123820          21156.0\n",
            "00:07:44.779 INFO  ProgressMeter -       chr21:36264734              6.0                124010          20599.2\n",
            "00:07:54.998 INFO  ProgressMeter -       chr21:36387956              6.2                124550          20119.7\n",
            "00:08:05.455 INFO  ProgressMeter -       chr21:36768751              6.4                125950          19788.8\n",
            "00:08:15.647 INFO  ProgressMeter -       chr21:37133146              6.5                127290          19479.4\n",
            "00:08:30.525 INFO  ProgressMeter -       chr21:37198175              6.8                127620          18815.9\n",
            "00:08:40.826 INFO  ProgressMeter -       chr21:38055944              7.0                130590          18778.4\n",
            "00:08:51.135 INFO  ProgressMeter -       chr21:38823223              7.1                133240          18697.9\n",
            "00:09:01.596 INFO  ProgressMeter -       chr21:39227193              7.3                134690          18449.7\n",
            "00:09:12.169 INFO  ProgressMeter -       chr21:39407260              7.5                135440          18115.1\n",
            "00:09:22.258 INFO  ProgressMeter -       chr21:39867511              7.6                137060          17928.6\n",
            "00:09:32.322 INFO  ProgressMeter -       chr21:40168812              7.8                138140          17681.9\n",
            "00:09:42.523 INFO  ProgressMeter -       chr21:41323634              8.0                142110          17802.6\n",
            "00:09:53.116 INFO  ProgressMeter -       chr21:41438106              8.2                142580          17475.0\n",
            "00:10:03.861 INFO  ProgressMeter -       chr21:41742298              8.3                143690          17232.8\n",
            "00:10:13.953 INFO  ProgressMeter -       chr21:41864895              8.5                144230          16955.6\n",
            "00:10:24.090 INFO  ProgressMeter -       chr21:42090673              8.7                145110          16726.8\n",
            "00:10:35.385 INFO  ProgressMeter -       chr21:42290371              8.9                145890          16459.5\n",
            "00:10:45.524 INFO  ProgressMeter -       chr21:42409631              9.0                146390          16206.9\n",
            "00:10:55.979 INFO  ProgressMeter -       chr21:42559902              9.2                147010          15967.6\n",
            "00:11:06.246 INFO  ProgressMeter -       chr21:42854262              9.4                148100          15792.4\n",
            "00:11:16.327 INFO  ProgressMeter -       chr21:43418812              9.5                150090          15722.9\n",
            "00:11:26.910 INFO  ProgressMeter -       chr21:43746461              9.7                151290          15561.1\n",
            "00:11:37.084 INFO  ProgressMeter -       chr21:44081066              9.9                152550          15421.7\n",
            "00:11:47.223 INFO  ProgressMeter -       chr21:44257377             10.1                153250          15232.3\n",
            "00:11:57.405 INFO  ProgressMeter -       chr21:44318985             10.2                153570          15010.9\n",
            "00:12:07.557 INFO  ProgressMeter -       chr21:44400869             10.4                153970          14805.1\n",
            "00:12:17.716 INFO  ProgressMeter -       chr21:44551391             10.6                154590          14626.6\n",
            "00:12:28.340 INFO  ProgressMeter -       chr21:44659619             10.7                155050          14428.4\n",
            "00:12:39.240 INFO  ProgressMeter -       chr21:44901918             10.9                155990          14274.6\n",
            "00:12:49.369 INFO  ProgressMeter -       chr21:45221780             11.1                157130          14160.2\n",
            "00:12:59.738 INFO  ProgressMeter -       chr21:45479207             11.3                158080          14027.3\n",
            "00:13:10.246 INFO  ProgressMeter -       chr21:45511702             11.4                158280          13830.1\n",
            "00:13:20.632 INFO  ProgressMeter -       chr21:45985035             11.6                159960          13768.7\n",
            "00:13:30.671 INFO  ProgressMeter -       chr21:46125058             11.8                160560          13624.1\n",
            "00:13:41.356 INFO  ProgressMeter -       chr21:46210527             12.0                160950          13453.9\n",
            "00:13:52.021 INFO  ProgressMeter -       chr21:46267382             12.1                161240          13280.8\n",
            "00:14:02.425 INFO  ProgressMeter -       chr21:46347864             12.3                161610          13123.8\n",
            "00:14:13.417 INFO  ProgressMeter -       chr21:46433447             12.5                162020          12964.3\n",
            "00:14:23.823 INFO  ProgressMeter -       chr21:46550218             12.7                162510          12825.5\n",
            "00:14:33.156 INFO  Mutect2 - 55712 read(s) filtered by: MappingQualityReadFilter \n",
            "0 read(s) filtered by: MappingQualityAvailableReadFilter \n",
            "0 read(s) filtered by: MappingQualityNotZeroReadFilter \n",
            "0 read(s) filtered by: MappedReadFilter \n",
            "0 read(s) filtered by: NotSecondaryAlignmentReadFilter \n",
            "536 read(s) filtered by: NotDuplicateReadFilter \n",
            "0 read(s) filtered by: PassesVendorQualityCheckReadFilter \n",
            "0 read(s) filtered by: NonChimericOriginalAlignmentReadFilter \n",
            "0 read(s) filtered by: NonZeroReferenceLengthAlignmentReadFilter \n",
            "0 read(s) filtered by: ReadLengthReadFilter \n",
            "0 read(s) filtered by: GoodCigarReadFilter \n",
            "0 read(s) filtered by: WellformedReadFilter \n",
            "56248 total reads filtered\n",
            "00:14:33.156 INFO  ProgressMeter -       chr21:46708488             12.8                163154          12720.2\n",
            "00:14:33.156 INFO  ProgressMeter - Traversal complete. Processed 163154 total regions in 12.8 minutes.\n",
            "00:14:33.201 INFO  VectorLoglessPairHMM - Time spent in setup for JNI call : 0.264307028\n",
            "00:14:33.208 INFO  PairHMM - Total compute time in PairHMM computeLogLikelihoods() : 99.61991500100001\n",
            "00:14:33.208 INFO  SmithWatermanAligner - Total compute time in java Smith-Waterman : 138.47 sec\n",
            "00:14:33.212 INFO  Mutect2 - Shutting down engine\n",
            "[August 21, 2025 at 12:14:33 AM UTC] org.broadinstitute.hellbender.tools.walkers.mutect.Mutect2 done. Elapsed time: 12.85 minutes.\n",
            "Runtime.totalMemory()=1269825536\n",
            "Tool returned:\n",
            "SUCCESS\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 6s\n",
        "!java -jar /content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar \\\n",
        "  FilterMutectCalls \\\n",
        "  -R /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.fa \\\n",
        "  -V somatic.vcf.gz \\\n",
        "  -O filtered.vcf.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "i4345Np86CtP",
        "outputId": "239ed1f9-b811-49b5-9967-0f31a2f142f6"
      },
      "execution_count": 85,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "00:17:38.584 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/drive/MyDrive/AMC_NGS1/02_linux/tools/gatk-4.3.0.0/gatk-package-4.3.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "00:17:38.859 INFO  FilterMutectCalls - ------------------------------------------------------------\n",
            "00:17:38.860 INFO  FilterMutectCalls - The Genome Analysis Toolkit (GATK) v4.3.0.0\n",
            "00:17:38.860 INFO  FilterMutectCalls - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "00:17:38.860 INFO  FilterMutectCalls - Executing as root@49a1eeb69d60 on Linux v6.1.123+ amd64\n",
            "00:17:38.860 INFO  FilterMutectCalls - Java runtime: OpenJDK 64-Bit Server VM v11.0.28+6-post-Ubuntu-1ubuntu122.04.1\n",
            "00:17:38.861 INFO  FilterMutectCalls - Start Date/Time: August 21, 2025 at 12:17:38 AM UTC\n",
            "00:17:38.861 INFO  FilterMutectCalls - ------------------------------------------------------------\n",
            "00:17:38.861 INFO  FilterMutectCalls - ------------------------------------------------------------\n",
            "00:17:38.862 INFO  FilterMutectCalls - HTSJDK Version: 3.0.1\n",
            "00:17:38.862 INFO  FilterMutectCalls - Picard Version: 2.27.5\n",
            "00:17:38.862 INFO  FilterMutectCalls - Built for Spark Version: 2.4.5\n",
            "00:17:38.862 INFO  FilterMutectCalls - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "00:17:38.863 INFO  FilterMutectCalls - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "00:17:38.863 INFO  FilterMutectCalls - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "00:17:38.863 INFO  FilterMutectCalls - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "00:17:38.863 INFO  FilterMutectCalls - Deflater: IntelDeflater\n",
            "00:17:38.863 INFO  FilterMutectCalls - Inflater: IntelInflater\n",
            "00:17:38.863 INFO  FilterMutectCalls - GCS max retries/reopens: 20\n",
            "00:17:38.864 INFO  FilterMutectCalls - Requester pays: disabled\n",
            "00:17:38.864 INFO  FilterMutectCalls - Initializing engine\n",
            "00:17:39.167 INFO  FeatureManager - Using codec VCFCodec to read file file:///content/drive/MyDrive/AMC_NGS1/02_linux/data/somatic.vcf.gz\n",
            "00:17:39.187 WARN  IntelInflater - Zero Bytes Written : 0\n",
            "00:17:39.212 WARN  IntelInflater - Zero Bytes Written : 0\n",
            "00:17:39.224 INFO  FilterMutectCalls - Done initializing engine\n",
            "00:17:39.424 INFO  ProgressMeter - Starting traversal\n",
            "00:17:39.425 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
            "00:17:39.426 INFO  FilterMutectCalls - Starting pass 0 through the variants\n",
            "00:17:39.441 WARN  IntelInflater - Zero Bytes Written : 0\n",
            "00:17:39.907 INFO  FilterMutectCalls - Finished pass 0 through the variants\n",
            "00:17:39.957 INFO  FilterMutectCalls - Starting pass 1 through the variants\n",
            "00:17:39.964 WARN  IntelInflater - Zero Bytes Written : 0\n",
            "00:17:40.127 INFO  FilterMutectCalls - Finished pass 1 through the variants\n",
            "00:17:40.132 INFO  FilterMutectCalls - Starting pass 2 through the variants\n",
            "00:17:40.138 WARN  IntelInflater - Zero Bytes Written : 0\n",
            "00:17:40.247 INFO  FilterMutectCalls - Finished pass 2 through the variants\n",
            "00:17:40.247 INFO  FilterMutectCalls - Starting pass 3 through the variants\n",
            "00:17:40.252 WARN  IntelInflater - Zero Bytes Written : 0\n",
            "00:17:40.464 INFO  FilterMutectCalls - Finished pass 3 through the variants\n",
            "00:17:40.496 INFO  FilterMutectCalls - No variants filtered by: AllowAllVariantsVariantFilter\n",
            "00:17:40.497 INFO  FilterMutectCalls - 0 read(s) filtered by: AllowAllReadsReadFilter \n",
            "\n",
            "00:17:40.497 INFO  ProgressMeter -             unmapped              0.0                   236          13209.0\n",
            "00:17:40.497 INFO  ProgressMeter - Traversal complete. Processed 236 total variants in 0.0 minutes.\n",
            "00:17:40.518 INFO  FilterMutectCalls - Shutting down engine\n",
            "[August 21, 2025 at 12:17:40 AM UTC] org.broadinstitute.hellbender.tools.walkers.mutect.filtering.FilterMutectCalls done. Elapsed time: 0.03 minutes.\n",
            "Runtime.totalMemory()=349175808\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "최종 vcf 확인"
      ],
      "metadata": {
        "id": "VCD3q1Dn_bkx"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 8s\n",
        "!apt-get install bcftools"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "jh1C-Pcw_yMh",
        "outputId": "8539c1a5-6088-4663-ef0c-9f899c0f7309"
      },
      "execution_count": 86,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "Suggested packages:\n",
            "  python3-numpy python3-matplotlib texlive-latex-recommended\n",
            "The following NEW packages will be installed:\n",
            "  bcftools\n",
            "0 upgraded, 1 newly installed, 0 to remove and 35 not upgraded.\n",
            "Need to get 697 kB of archives.\n",
            "After this operation, 2,360 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 bcftools amd64 1.13-1 [697 kB]\n",
            "Fetched 697 kB in 0s (6,182 kB/s)\n",
            "Selecting previously unselected package bcftools.\n",
            "(Reading database ... 126495 files and directories currently installed.)\n",
            "Preparing to unpack .../bcftools_1.13-1_amd64.deb ...\n",
            "Unpacking bcftools (1.13-1) ...\n",
            "Setting up bcftools (1.13-1) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bcftools stats filtered.vcf.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "SduQ50zw_upy",
        "outputId": "bc24042a-80e1-484d-b541-07a3ad1d0d5e"
      },
      "execution_count": 87,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "# This file was produced by bcftools stats (1.13+htslib-1.13+ds) and can be plotted using plot-vcfstats.\n",
            "# The command line was:\tbcftools stats  filtered.vcf.gz\n",
            "#\n",
            "# Definition of sets:\n",
            "# ID\t[2]id\t[3]tab-separated file names\n",
            "ID\t0\tfiltered.vcf.gz\n",
            "# SN, Summary numbers:\n",
            "#   number of records   .. number of data rows in the VCF\n",
            "#   number of no-ALTs   .. reference-only sites, ALT is either \".\" or identical to REF\n",
            "#   number of SNPs      .. number of rows with a SNP\n",
            "#   number of MNPs      .. number of rows with a MNP, such as CC>TT\n",
            "#   number of indels    .. number of rows with an indel\n",
            "#   number of others    .. number of rows with other type, for example a symbolic allele or\n",
            "#                          a complex substitution, such as ACT>TCGA\n",
            "#   number of multiallelic sites     .. number of rows with multiple alternate alleles\n",
            "#   number of multiallelic SNP sites .. number of rows with multiple alternate alleles, all SNPs\n",
            "# \n",
            "#   Note that rows containing multiple types will be counted multiple times, in each\n",
            "#   counter. For example, a row with a SNP and an indel increments both the SNP and\n",
            "#   the indel counter.\n",
            "# \n",
            "# SN\t[2]id\t[3]key\t[4]value\n",
            "SN\t0\tnumber of samples:\t2\n",
            "SN\t0\tnumber of records:\t59\n",
            "SN\t0\tnumber of no-ALTs:\t0\n",
            "SN\t0\tnumber of SNPs:\t58\n",
            "SN\t0\tnumber of MNPs:\t0\n",
            "SN\t0\tnumber of indels:\t1\n",
            "SN\t0\tnumber of others:\t0\n",
            "SN\t0\tnumber of multiallelic sites:\t0\n",
            "SN\t0\tnumber of multiallelic SNP sites:\t0\n",
            "# TSTV, transitions/transversions:\n",
            "# TSTV\t[2]id\t[3]ts\t[4]tv\t[5]ts/tv\t[6]ts (1st ALT)\t[7]tv (1st ALT)\t[8]ts/tv (1st ALT)\n",
            "TSTV\t0\t1\t57\t0.02\t1\t57\t0.02\n",
            "# SiS, Singleton stats:\n",
            "# SiS\t[2]id\t[3]allele count\t[4]number of SNPs\t[5]number of transitions\t[6]number of transversions\t[7]number of indels\t[8]repeat-consistent\t[9]repeat-inconsistent\t[10]not applicable\n",
            "SiS\t0\t1\t58\t1\t57\t1\t0\t0\t1\n",
            "# AF, Stats by non-reference allele frequency:\n",
            "# AF\t[2]id\t[3]allele frequency\t[4]number of SNPs\t[5]number of transitions\t[6]number of transversions\t[7]number of indels\t[8]repeat-consistent\t[9]repeat-inconsistent\t[10]not applicable\n",
            "AF\t0\t0.000000\t58\t1\t57\t1\t0\t0\t1\n",
            "# QUAL, Stats by quality\n",
            "# QUAL\t[2]id\t[3]Quality\t[4]number of SNPs\t[5]number of transitions (1st ALT)\t[6]number of transversions (1st ALT)\t[7]number of indels\n",
            "QUAL\t0\t.\t58\t1\t57\t1\n",
            "# IDD, InDel distribution:\n",
            "# IDD\t[2]id\t[3]length (deletions negative)\t[4]number of sites\t[5]number of genotypes\t[6]mean VAF\n",
            "IDD\t0\t-1\t1\t0\t.\n",
            "# ST, Substitution types:\n",
            "# ST\t[2]id\t[3]type\t[4]count\n",
            "ST\t0\tA>C\t4\n",
            "ST\t0\tA>G\t0\n",
            "ST\t0\tA>T\t10\n",
            "ST\t0\tC>A\t5\n",
            "ST\t0\tC>G\t9\n",
            "ST\t0\tC>T\t0\n",
            "ST\t0\tG>A\t0\n",
            "ST\t0\tG>C\t10\n",
            "ST\t0\tG>T\t9\n",
            "ST\t0\tT>A\t6\n",
            "ST\t0\tT>C\t1\n",
            "ST\t0\tT>G\t4\n",
            "# DP, Depth distribution\n",
            "# DP\t[2]id\t[3]bin\t[4]number of genotypes\t[5]fraction of genotypes (%)\t[6]number of sites\t[7]fraction of sites (%)\n",
            "DP\t0\t12\t0\t0.000000\t2\t3.389831\n",
            "DP\t0\t13\t0\t0.000000\t2\t3.389831\n",
            "DP\t0\t14\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t16\t0\t0.000000\t2\t3.389831\n",
            "DP\t0\t17\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t18\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t19\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t20\t0\t0.000000\t4\t6.779661\n",
            "DP\t0\t21\t0\t0.000000\t2\t3.389831\n",
            "DP\t0\t22\t0\t0.000000\t5\t8.474576\n",
            "DP\t0\t23\t0\t0.000000\t2\t3.389831\n",
            "DP\t0\t24\t0\t0.000000\t3\t5.084746\n",
            "DP\t0\t25\t0\t0.000000\t3\t5.084746\n",
            "DP\t0\t26\t0\t0.000000\t6\t10.169492\n",
            "DP\t0\t27\t0\t0.000000\t3\t5.084746\n",
            "DP\t0\t28\t0\t0.000000\t3\t5.084746\n",
            "DP\t0\t29\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t31\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t32\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t33\t0\t0.000000\t2\t3.389831\n",
            "DP\t0\t34\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t36\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t39\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t42\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t43\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t45\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t46\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t49\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t51\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t56\t0\t0.000000\t2\t3.389831\n",
            "DP\t0\t66\t0\t0.000000\t1\t1.694915\n",
            "DP\t0\t74\t0\t0.000000\t1\t1.694915\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# SNV만 추출\n",
        "!bcftools view -v snps filtered.vcf.gz -o somatic.snvs.vcf\n",
        "\n",
        "# indel만 추출\n",
        "!bcftools view -v indels filtered.vcf.gz -o somatic.indels.vcf"
      ],
      "metadata": {
        "id": "1s53pDJ2_wAs"
      },
      "execution_count": 89,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bcftools view -H filtered.vcf.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9WwD9E5tB64X",
        "outputId": "c6142c0e-02b7-4a49-dcea-871f21d3dcd9"
      },
      "execution_count": 91,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "chr21\t10590479\t.\tG\tT\t.\tbase_qual;haplotype;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=48,1|2,0;DP=51;ECNT=2;GERMQ=79;MBQ=18,16;MFRL=321,314;MMQ=60,60;MPOS=20;NALOD=1.36;NLOD=6.62;POPAF=6;TLOD=3.32\tGT:AD:AF:DP:F1R2:F2R1:FAD:PGT:PID:PS:SB\t0|0:23,0:0.042:23:0,0:0,0:22,0:0|1:10590479_G_T:10590479:22,1,0,0\t0|1:26,2:0.1:28:0,0:0,0:26,2:0|1:10590479_G_T:10590479:26,0,2,0\n",
            "chr21\t10590485\t.\tG\tT\t.\tbase_qual;haplotype;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=51,2|3,0;DP=56;ECNT=2;GERMQ=80;MBQ=19,18;MFRL=321,327;MMQ=60,60;MPOS=17;NALOD=1.38;NLOD=6.92;POPAF=6;TLOD=3.66\tGT:AD:AF:DP:F1R2:F2R1:FAD:PGT:PID:PS:SB\t0|0:25,0:0.04:25:6,0:0,0:23,0:0|1:10590479_G_T:10590479:23,2,0,0\t0|1:28,3:0.112:31:5,1:0,0:28,3:0|1:10590479_G_T:10590479:28,0,3,0\n",
            "chr21\t14517382\t.\tT\tG\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,17|0,2;DP=20;ECNT=1;GERMQ=3;MBQ=15,18;MFRL=486,379;MMQ=60,60;MPOS=43;NALOD=1.06;NLOD=3;POPAF=6;TLOD=4.8\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:10,0:0.081:10:0,0:0,0:10,0:0,10,0,0\t0/1:7,2:0.272:9:0,0:0,0:7,2:0,7,0,2\n",
            "chr21\t18257491\t.\tA\tT\t.\tbase_qual\tAS_FilterStatus=base_qual;AS_SB_TABLE=0,21|0,3;DP=25;ECNT=1;GERMQ=5;MBQ=16,19;MFRL=449,421;MMQ=60,60;MPOS=31;NALOD=1.13;NLOD=3.45;POPAF=6;TLOD=8.15\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:13,0:0.07:13:1,0:0,0:13,0:0,13,0,0\t0/1:8,3:0.321:11:1,1:0,0:8,3:0,8,0,3\n",
            "chr21\t18266615\t.\tT\tC\t.\tweak_evidence\tAS_FilterStatus=weak_evidence;AS_SB_TABLE=0,11|0,1;DP=12;ECNT=1;GERMQ=1;MBQ=19,20;MFRL=537,468;MMQ=60,60;MPOS=44;NALOD=0.954;NLOD=2.41;POPAF=6;TLOD=3.2\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:8,0:0.1:8:2,0:0,0:8,0:0,8,0,0\t0/1:3,1:0.333:4:0,1:0,0:3,1:0,3,0,1\n",
            "chr21\t21286586\t.\tG\tC\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,19|0,4;DP=27;ECNT=1;GERMQ=2;MBQ=17,18;MFRL=327,277;MMQ=60,60;MPOS=10;NALOD=1.11;NLOD=3.03;POPAF=6;TLOD=4.36\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:11,0:0.074:11:3,0:0,0:11,0:0,11,0,0\t0/1:8,4:0.364:12:1,1:0,0:8,4:0,8,0,4\n",
            "chr21\t25892233\t.\tA\tT\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,19|0,2;DP=22;ECNT=1;GERMQ=4;MBQ=15,18;MFRL=485,451;MMQ=60,60;MPOS=26;NALOD=1.06;NLOD=2.94;POPAF=6;TLOD=3.19\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:11,0:0.081:11:0,0:0,0:11,0:0,11,0,0\t0/1:8,2:0.269:10:0,1:0,0:8,2:0,8,0,2\n",
            "chr21\t26845315\t.\tG\tC\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,24|0,3;DP=28;ECNT=1;GERMQ=8;MBQ=15,19;MFRL=499,477;MMQ=60,60;MPOS=31;NALOD=1.13;NLOD=3.36;POPAF=6;TLOD=3.65\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:13,0:0.069:13:0,0:0,0:13,0:0,13,0,0\t0/1:11,3:0.271:14:0,1:0,0:11,3:0,11,0,3\n",
            "chr21\t26966228\t.\tGC\tG\t.\tbase_qual;haplotype\tAS_FilterStatus=base_qual;AS_SB_TABLE=14,5|3,0;DP=29;ECNT=2;GERMQ=2;MBQ=18,17;MFRL=331,428;MMQ=60,60;MPOS=26;NALOD=1.12;NLOD=3.31;POPAF=6;TLOD=7.75\tGT:AD:AF:DP:F1R2:F2R1:FAD:PGT:PID:PS:SB\t0|0:10,0:0.071:10:0,0:0,0:11,0:0|1:26966228_GC_G:26966228:7,3,0,0\t0|1:9,3:0.281:12:0,0:0,0:9,3:0|1:26966228_GC_G:26966228:7,2,3,0\n",
            "chr21\t26966236\t.\tC\tA\t.\tbase_qual;haplotype\tAS_FilterStatus=base_qual;AS_SB_TABLE=14,7|3,0;DP=24;ECNT=2;GERMQ=7;MBQ=18,18;MFRL=331,428;MMQ=60,60;MPOS=26;NALOD=1.08;NLOD=3.31;POPAF=6;TLOD=7.67\tGT:AD:AF:DP:F1R2:F2R1:FAD:PGT:PID:PS:SB\t0|0:11,0:0.077:11:1,0:0,0:11,0:0|1:26966228_GC_G:26966228:7,4,0,0\t0|1:10,3:0.267:13:3,0:0,0:10,3:0|1:26966228_GC_G:26966228:7,3,3,0\n",
            "chr21\t28985320\t.\tG\tC\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,18|0,2;DP=20;ECNT=1;GERMQ=12;MBQ=17,19;MFRL=501,462;MMQ=60,60;MPOS=26;NALOD=1.2;NLOD=4.5;POPAF=6;TLOD=3.92\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:15,0:0.059:15:3,0:0,0:15,0:0,15,0,0\t0/1:3,2:0.428:5:0,0:0,0:3,2:0,3,0,2\n",
            "chr21\t29056769\t.\tC\tA\t.\tPASS\tAS_FilterStatus=SITE;AS_SB_TABLE=0,24|0,4;DP=28;ECNT=1;GERMQ=8;MBQ=14,20;MFRL=299,345;MMQ=60,60;MPOS=13;NALOD=1.18;NLOD=3.87;POPAF=6;TLOD=4.96\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:15,0:0.063:15:0,0:0,0:15,0:0,15,0,0\t0/1:9,4:0.35:13:0,3:0,0:9,4:0,9,0,4\n",
            "chr21\t29321284\t.\tT\tA\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=30,1|5,0;DP=36;ECNT=1;GERMQ=16;MBQ=15,19;MFRL=322,319;MMQ=60,60;MPOS=24;NALOD=1.19;NLOD=4.18;POPAF=6;TLOD=4.85\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:16,0:0.061:16:4,0:0,0:15,0:15,1,0,0\t0/1:15,5:0.262:20:5,0:0,0:15,5:15,0,5,0\n",
            "chr21\t31691870\t.\tC\tA\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=21,1|5,0;DP=31;ECNT=1;GERMQ=5;MBQ=17,17;MFRL=230,399;MMQ=60,60;MPOS=30;NALOD=1.16;NLOD=3.62;POPAF=6;TLOD=3.3\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:13,0:0.066:13:0,0:0,0:13,0:13,0,0,0\t0/1:9,5:0.364:14:0,0:0,0:9,5:8,1,5,0\n",
            "chr21\t31696122\t.\tC\tG\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=24,0|3,0;DP=27;ECNT=1;GERMQ=14;MBQ=18,18;MFRL=334,260;MMQ=60,60;MPOS=8;NALOD=1.17;NLOD=4.16;POPAF=6;TLOD=3.34\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:14,0:0.063:14:4,0:0,0:14,0:14,0,0,0\t0/1:10,3:0.265:13:1,0:0,0:10,3:10,0,3,0\n",
            "chr21\t32634897\t.\tA\tC\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=50,2|2,0;DP=56;ECNT=1;GERMQ=90;MBQ=18,18;MFRL=346,459;MMQ=60,60;MPOS=38;NALOD=1.55;NLOD=9.89;POPAF=6;TLOD=3.27\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:35,0:0.028:35:2,0:0,0:35,0:33,2,0,0\t0/1:17,2:0.144:19:0,0:0,0:17,2:17,0,2,0\n",
            "chr21\t32745927\t.\tC\tA\t.\tweak_evidence\tAS_FilterStatus=weak_evidence;AS_SB_TABLE=0,10|0,2;DP=12;ECNT=1;GERMQ=1;MBQ=19,22;MFRL=331,356;MMQ=60,60;MPOS=23;NALOD=0.954;NLOD=2.4;POPAF=6;TLOD=3.77\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:8,0:0.1:8:4,0:0,0:8,0:0,8,0,0\t0/1:2,2:0.499:4:0,2:0,0:2,2:0,2,0,2\n",
            "chr21\t33343263\t.\tT\tA\t.\tbase_qual;haplotype;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=32,0|5,0;DP=39;ECNT=2;GERMQ=15;MBQ=17,15;MFRL=420,323;MMQ=60,60;MPOS=15;NALOD=1.06;NLOD=3.72;POPAF=6;TLOD=4\tGT:AD:AF:DP:F1R2:F2R1:FAD:PGT:PID:PS:SB\t0|0:19,2:0.08:21:6,0:0,0:19,2:0|1:33343263_T_A:33343263:19,0,2,0\t0|1:13,3:0.222:16:2,0:0,0:13,3:0|1:33343263_T_A:33343263:13,0,3,0\n",
            "chr21\t33343279\t.\tC\tG\t.\thaplotype;weak_evidence\tAS_FilterStatus=weak_evidence;AS_SB_TABLE=38,1|3,0;DP=42;ECNT=2;GERMQ=50;MBQ=18,20;MFRL=410,323;MMQ=60,60;MPOS=15;NALOD=1.38;NLOD=6.85;POPAF=6;TLOD=4.1\tGT:AD:AF:DP:F1R2:F2R1:FAD:PGT:PID:PS:SB\t0|0:24,0:0.04:24:2,0:0,0:23,0:0|1:33343263_T_A:33343263:23,1,0,0\t0|1:15,3:0.199:18:2,2:0,0:15,3:0|1:33343263_T_A:33343263:15,0,3,0\n",
            "chr21\t33467020\t.\tC\tG\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=29,1|2,0;DP=32;ECNT=1;GERMQ=32;MBQ=17,17;MFRL=296,471;MMQ=60,60;MPOS=46;NALOD=1.25;NLOD=5.02;POPAF=6;TLOD=3.31\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:17,0:0.053:17:1,0:0,0:17,0:17,0,0,0\t0/1:13,2:0.186:15:1,0:0,0:12,2:12,1,2,0\n",
            "chr21\t33543146\t.\tG\tT\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=18,0|3,0;DP=23;ECNT=1;GERMQ=2;MBQ=18,16;MFRL=447,442;MMQ=60,60;MPOS=46;NALOD=1.05;NLOD=2.94;POPAF=6;TLOD=3.59\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:10,0:0.082:10:0,0:0,0:10,0:10,0,0,0\t0/1:8,3:0.296:11:0,1:0,0:8,3:8,0,3,0\n",
            "chr21\t33836476\t.\tC\tG\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=19,0|2,0;DP=22;ECNT=1;GERMQ=8;MBQ=17,16;MFRL=351,382;MMQ=60,60;MPOS=38;NALOD=1.16;NLOD=3.92;POPAF=6;TLOD=3.38\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:14,0:0.064:14:1,0:0,0:14,0:14,0,0,0\t0/1:5,2:0.331:7:0,0:0,0:5,2:5,0,2,0\n",
            "chr21\t33903551\t.\tC\tG\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=22,2|4,0;DP=28;ECNT=1;GERMQ=5;MBQ=17,18;MFRL=385,465;MMQ=60,60;MPOS=12;NALOD=1.03;NLOD=2.73;POPAF=6;TLOD=3.28\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:10,0:0.087:10:1,0:0,0:10,0:10,0,0,0\t0/1:14,4:0.248:18:1,0:0,0:14,4:12,2,4,0\n",
            "chr21\t34525736\t.\tA\tT\t.\tbase_qual\tAS_FilterStatus=base_qual;AS_SB_TABLE=0,26|0,4;DP=34;ECNT=1;GERMQ=9;MBQ=15,19;MFRL=481,488;MMQ=60,60;MPOS=39;NALOD=1.14;NLOD=3.49;POPAF=6;TLOD=5.01\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:13,0:0.068:13:1,0:0,0:13,0:0,13,0,0\t0/1:13,4:0.28:17:1,1:0,0:13,4:0,13,0,4\n",
            "chr21\t36228137\t.\tT\tG\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,28|0,3;DP=33;ECNT=1;GERMQ=15;MBQ=18,18;MFRL=410,461;MMQ=60,60;MPOS=29;NALOD=0.928;NLOD=2.42;POPAF=6;TLOD=3.3\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:11,1:0.103:12:0,0:0,0:11,1:0,11,0,1\t0/1:17,2:0.147:19:0,0:0,0:17,2:0,17,0,2\n",
            "chr21\t37072216\t.\tG\tT\t.\tweak_evidence\tAS_FilterStatus=weak_evidence;AS_SB_TABLE=20,0|2,0;DP=22;ECNT=1;GERMQ=4;MBQ=14,21;MFRL=411,221;MMQ=60,60;MPOS=41;NALOD=1.02;NLOD=2.64;POPAF=6;TLOD=3.19\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:10,0:0.088:10:0,0:0,0:10,0:10,0,0,0\t0/1:10,2:0.235:12:0,2:0,0:10,2:10,0,2,0\n",
            "chr21\t37160241\t.\tA\tT\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,13|0,1;DP=14;ECNT=1;GERMQ=2;MBQ=21,18;MFRL=537,567;MMQ=60,60;MPOS=33;NALOD=0.997;NLOD=2.66;POPAF=6;TLOD=3.02\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:9,0:0.092:9:8,0:0,0:9,0:0,9,0,0\t0/1:4,1:0.286:5:4,0:0,0:4,1:0,4,0,1\n",
            "chr21\t37481253\t.\tG\tC\t.\tweak_evidence\tAS_FilterStatus=weak_evidence;AS_SB_TABLE=0,19|0,3;DP=22;ECNT=1;GERMQ=5;MBQ=18,20;MFRL=474,550;MMQ=60,60;MPOS=18;NALOD=1.11;NLOD=3.55;POPAF=6;TLOD=4.41\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:12,0:0.072:12:0,0:0,0:12,0:0,12,0,0\t0/1:7,3:0.333:10:0,3:0,0:7,3:0,7,0,3\n",
            "chr21\t38810344\t.\tG\tC\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,12|0,2;DP=16;ECNT=1;GERMQ=1;MBQ=17,15;MFRL=318,354;MMQ=60,60;MPOS=33;NALOD=1;NLOD=2.43;POPAF=6;TLOD=3.52\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:9,0:0.092:9:1,0:0,0:9,0:0,9,0,0\t0/1:3,2:0.418:5:0,0:0,0:3,2:0,3,0,2\n",
            "chr21\t40044559\t.\tT\tG\t.\tbase_qual\tAS_FilterStatus=base_qual;AS_SB_TABLE=0,17|0,3;DP=20;ECNT=1;GERMQ=1;MBQ=18,16;MFRL=433,420;MMQ=60,60;MPOS=24;NALOD=0.941;NLOD=2.21;POPAF=6;TLOD=5.17\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:8,0:0.103:8:0,0:0,0:8,0:0,8,0,0\t0/1:9,3:0.292:12:0,0:0,0:9,3:0,9,0,3\n",
            "chr21\t40312083\t.\tC\tG\t.\tbase_qual;haplotype\tAS_FilterStatus=base_qual;AS_SB_TABLE=21,1|2,0;DP=24;ECNT=2;GERMQ=16;MBQ=18,19;MFRL=405,366;MMQ=60,60;MPOS=24;NALOD=1.2;NLOD=4.51;POPAF=6;TLOD=6.44\tGT:AD:AF:DP:F1R2:F2R1:FAD:PGT:PID:PS:SB\t0|0:15,0:0.059:15:3,0:0,0:15,0:0|1:40312083_C_G:40312083:15,0,0,0\t0|1:7,2:0.273:9:1,0:0,0:7,2:0|1:40312083_C_G:40312083:6,1,2,0\n",
            "chr21\t40312091\t.\tC\tG\t.\tbase_qual;haplotype\tAS_FilterStatus=base_qual;AS_SB_TABLE=23,1|2,0;DP=26;ECNT=2;GERMQ=19;MBQ=18,19;MFRL=405,366;MMQ=60,60;MPOS=32;NALOD=1.2;NLOD=4.51;POPAF=6;TLOD=6.18\tGT:AD:AF:DP:F1R2:F2R1:FAD:PGT:PID:PS:SB\t0|0:15,0:0.059:15:1,0:0,0:15,0:0|1:40312083_C_G:40312083:15,0,0,0\t0|1:9,2:0.231:11:3,0:0,0:9,2:0|1:40312083_C_G:40312083:8,1,2,0\n",
            "chr21\t40692893\t.\tA\tC\t.\tbase_qual\tAS_FilterStatus=base_qual;AS_SB_TABLE=14,4|2,0;DP=24;ECNT=1;GERMQ=5;MBQ=18,19;MFRL=297,382;MMQ=60,60;MPOS=43;NALOD=1.1;NLOD=3.19;POPAF=6;TLOD=6.43\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:11,0:0.074:11:1,0:0,0:11,0:9,2,0,0\t0/1:7,2:0.274:9:1,0:0,0:7,2:5,2,2,0\n",
            "chr21\t41243757\t.\tA\tT\t.\tbase_qual\tAS_FilterStatus=base_qual;AS_SB_TABLE=0,19|0,2;DP=22;ECNT=1;GERMQ=9;MBQ=21,18;MFRL=398,329;MMQ=60,60;MPOS=43;NALOD=1.15;NLOD=3.91;POPAF=6;TLOD=5.3\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:13,0:0.067:13:12,0:0,0:13,0:0,13,0,0\t0/1:6,2:0.298:8:6,0:0,0:6,2:0,6,0,2\n",
            "chr21\t41357578\t.\tC\tA\t.\tweak_evidence\tAS_FilterStatus=weak_evidence;AS_SB_TABLE=0,12|0,1;DP=13;ECNT=1;GERMQ=1;MBQ=19,20;MFRL=465,421;MMQ=60,60;MPOS=43;NALOD=0.954;NLOD=2.41;POPAF=6;TLOD=3.02\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:8,0:0.1:8:0,0:0,0:8,0:0,8,0,0\t0/1:4,1:0.286:5:0,1:0,0:4,1:0,4,0,1\n",
            "chr21\t41432093\t.\tT\tA\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=42,1|5,0;DP=49;ECNT=1;GERMQ=19;MBQ=15,19;MFRL=364,377;MMQ=60,60;MPOS=41;NALOD=-0.5331;NLOD=3.57;POPAF=6;TLOD=3.19\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:25,1:0.076:26:0,0:0,0:25,1:25,0,1,0\t0/1:18,4:0.218:22:0,0:0,0:17,4:17,1,4,0\n",
            "chr21\t41471070\t.\tG\tC\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,23|0,3;DP=26;ECNT=1;GERMQ=6;MBQ=15,19;MFRL=395,459;MMQ=60,60;MPOS=28;NALOD=1.02;NLOD=3.01;POPAF=6;TLOD=4.27\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:14,1:0.085:15:0,0:0,0:14,1:0,14,0,1\t0/1:9,2:0.25:11:0,1:0,0:9,2:0,9,0,2\n",
            "chr21\t41749252\t.\tG\tT\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=46,12|6,0;DP=66;ECNT=1;GERMQ=70;MBQ=19,15;MFRL=266,365;MMQ=60,60;MPOS=37;NALOD=1.3;NLOD=6.51;POPAF=6;TLOD=4.13\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:30,2:0.048:32:6,0:0,0:27,2:23,7,2,0\t0/1:28,4:0.124:32:2,0:0,0:27,4:23,5,4,0\n",
            "chr21\t41811153\t.\tT\tG\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,15|0,2;DP=17;ECNT=1;GERMQ=5;MBQ=17,16;MFRL=407,430;MMQ=60,60;MPOS=31;NALOD=1.13;NLOD=3.54;POPAF=6;TLOD=3.8\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:13,0:0.069:13:0,0:0,0:13,0:0,13,0,0\t0/1:2,2:0.517:4:0,0:0,0:2,2:0,2,0,2\n",
            "chr21\t41838382\t.\tA\tT\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,15|0,3;DP=19;ECNT=1;GERMQ=1;MBQ=16,17;MFRL=459,457;MMQ=60,60;MPOS=24;NALOD=1.03;NLOD=2.73;POPAF=6;TLOD=3.91\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:10,0:0.087:10:3,0:0,0:10,0:0,10,0,0\t0/1:5,3:0.387:8:1,0:0,0:5,3:0,5,0,3\n",
            "chr21\t41953693\t.\tG\tT\t.\tweak_evidence\tAS_FilterStatus=weak_evidence;AS_SB_TABLE=13,5|2,0;DP=21;ECNT=1;GERMQ=3;MBQ=14,21;MFRL=412,475;MMQ=60,60;MPOS=9;NALOD=1.02;NLOD=2.66;POPAF=6;TLOD=4.38\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:10,0:0.089:10:0,0:0,0:9,0:6,4,0,0\t0/1:8,2:0.249:10:0,2:0,0:8,2:7,1,2,0\n",
            "chr21\t42526234\t.\tA\tT\t.\tbase_qual\tAS_FilterStatus=base_qual;AS_SB_TABLE=0,18|0,2;DP=20;ECNT=1;GERMQ=2;MBQ=16,18;MFRL=450,408;MMQ=60,60;MPOS=20;NALOD=0.944;NLOD=2.26;POPAF=6;TLOD=5.4\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:8,0:0.102:8:1,0:0,0:8,0:0,8,0,0\t0/1:10,2:0.218:12:0,0:0,0:10,2:0,10,0,2\n",
            "chr21\t42564378\t.\tG\tC\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,11|0,2;DP=13;ECNT=1;GERMQ=1;MBQ=19,19;MFRL=553,554;MMQ=60,60;MPOS=22;NALOD=0.95;NLOD=2.35;POPAF=6;TLOD=3.04\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:8,0:0.101:8:1,0:0,0:8,0:0,8,0,0\t0/1:3,2:0.451:5:0,0:0,0:3,2:0,3,0,2\n",
            "chr21\t42759161\t.\tT\tA\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=17,7|2,0;DP=26;ECNT=2;GERMQ=17;MBQ=16,19;MFRL=278,284;MMQ=60,60;MPOS=29;NALOD=1.14;NLOD=3.86;POPAF=6;TLOD=3.76\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:13,0:0.067:13:1,0:0,0:13,0:11,2,0,0\t0/1:11,2:0.202:13:0,0:0,0:11,2:6,5,2,0\n",
            "chr21\t42759174\t.\tT\tA\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=16,5|2,0;DP=25;ECNT=2;GERMQ=11;MBQ=20,18;MFRL=276,274;MMQ=60,60;MPOS=41;NALOD=1.11;NLOD=3.6;POPAF=6;TLOD=3.71\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:12,0:0.072:12:10,0:0,0:12,0:10,2,0,0\t0/1:9,2:0.23:11:4,0:0,0:9,2:6,3,2,0\n",
            "chr21\t42762125\t.\tT\tA\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=28,0|5,0;DP=33;ECNT=1;GERMQ=7;MBQ=16,18;MFRL=325,447;MMQ=60,60;MPOS=45;NALOD=0.969;NLOD=3.16;POPAF=6;TLOD=3.37\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:17,2:0.092:19:0,0:0,0:17,2:17,0,2,0\t0/1:11,3:0.259:14:0,0:0,0:11,3:11,0,3,0\n",
            "chr21\t43683596\t.\tG\tC\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,22|0,4;DP=26;ECNT=1;GERMQ=5;MBQ=19,18;MFRL=343,360;MMQ=60,60;MPOS=26;NALOD=1.07;NLOD=3.2;POPAF=6;TLOD=3.68\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:11,0:0.078:11:4,0:0,0:11,0:0,11,0,0\t0/1:11,4:0.297:15:1,0:0,0:11,4:0,11,0,4\n",
            "chr21\t43978056\t.\tC\tG\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=21,0|3,0;DP=25;ECNT=2;GERMQ=5;MBQ=18,15;MFRL=359,461;MMQ=60,60;MPOS=13;NALOD=1.04;NLOD=2.95;POPAF=6;TLOD=4.04\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:10,0:0.084:10:1,0:0,0:10,0:10,0,0,0\t0/1:11,3:0.237:14:1,0:0,0:11,3:11,0,3,0\n",
            "chr21\t43978060\t.\tA\tC\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=21,0|4,0;DP=26;ECNT=2;GERMQ=3;MBQ=18,18;MFRL=381,251;MMQ=60,60;MPOS=15;NALOD=1.03;NLOD=2.81;POPAF=6;TLOD=4.63\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:10,0:0.086:10:0,0:0,0:10,0:10,0,0,0\t0/1:11,4:0.274:15:0,0:0,0:11,4:11,0,4,0\n",
            "chr21\t44296398\t.\tC\tG\t.\tbase_qual\tAS_FilterStatus=base_qual;AS_SB_TABLE=19,0|4,0;DP=23;ECNT=1;GERMQ=7;MBQ=18,19;MFRL=394,377;MMQ=60,60;MPOS=29;NALOD=1.17;NLOD=4.05;POPAF=6;TLOD=6.12\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:14,0:0.063:14:0,0:0,0:14,0:14,0,0,0\t0/1:5,4:0.451:9:0,1:0,0:5,4:5,0,4,0\n",
            "chr21\t44323980\t.\tA\tC\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=34,3|5,0;DP=46;ECNT=1;GERMQ=12;MBQ=15,18;MFRL=359,332;MMQ=60,60;MPOS=22;NALOD=0.19;NLOD=2.41;POPAF=6;TLOD=4\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:17,1:0.104:18:1,0:0,0:17,1:15,2,1,0\t0/1:20,4:0.209:24:0,0:0,0:20,4:19,1,4,0\n",
            "chr21\t44400371\t.\tG\tT\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=58,8|5,0;DP=74;ECNT=1;GERMQ=87;MBQ=19,15;MFRL=364,304;MMQ=60,60;MPOS=28;NALOD=1.37;NLOD=6.99;POPAF=4.61;TLOD=3.91\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:27,1:0.041:28:4,0:0,0:26,1:24,3,1,0\t0/1:39,4:0.09:43:12,0:0,0:38,4:34,5,4,0\n",
            "chr21\t44558103\t.\tG\tT\t.\tweak_evidence\tAS_FilterStatus=weak_evidence;AS_SB_TABLE=38,2|3,0;DP=43;ECNT=1;GERMQ=42;MBQ=14,20;MFRL=354,317;MMQ=60,60;MPOS=24;NALOD=1.3;NLOD=5.19;POPAF=6;TLOD=3.21\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:21,0:0.048:21:0,0:0,0:20,0:20,1,0,0\t0/1:19,3:0.186:22:0,3:0,0:18,3:18,1,3,0\n",
            "chr21\t44575409\t.\tA\tT\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,14|0,2;DP=16;ECNT=1;GERMQ=3;MBQ=15,19;MFRL=550,548;MMQ=60,60;MPOS=28;NALOD=1.09;NLOD=3.15;POPAF=6;TLOD=3.22\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:12,0:0.075:12:1,0:0,0:12,0:0,12,0,0\t0/1:2,2:0.517:4:0,0:0,0:2,2:0,2,0,2\n",
            "chr21\t45984818\t.\tA\tT\t.\tbase_qual\tAS_FilterStatus=base_qual;AS_SB_TABLE=0,18|0,2;DP=21;ECNT=1;GERMQ=4;MBQ=15,17;MFRL=465,502;MMQ=60,60;MPOS=28;NALOD=1.05;NLOD=2.87;POPAF=6;TLOD=5.06\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:10,0:0.082:10:2,0:0,0:10,0:0,10,0,0\t0/1:8,2:0.255:10:1,0:0,0:8,2:0,8,0,2\n",
            "chr21\t46213040\t.\tG\tT\t.\tweak_evidence\tAS_FilterStatus=weak_evidence;AS_SB_TABLE=23,0|3,0;DP=27;ECNT=1;GERMQ=5;MBQ=14,21;MFRL=332,488;MMQ=60,60;MPOS=29;NALOD=0.995;NLOD=2.34;POPAF=6;TLOD=3.96\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:9,0:0.093:9:0,0:0,0:9,0:9,0,0,0\t0/1:14,3:0.223:17:0,3:0,0:14,3:14,0,3,0\n",
            "chr21\t46277962\t.\tG\tC\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,22|0,3;DP=26;ECNT=1;GERMQ=5;MBQ=17,18;MFRL=349,406;MMQ=60,60;MPOS=38;NALOD=1.07;NLOD=2.94;POPAF=6;TLOD=3.19\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:11,0:0.078:11:0,0:0,0:11,0:0,11,0,0\t0/1:11,3:0.261:14:1,0:0,0:11,3:0,11,0,3\n",
            "chr21\t46335080\t.\tA\tT\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,38|0,4;DP=45;ECNT=1;GERMQ=45;MBQ=20,18;MFRL=461,571;MMQ=60,60;MPOS=30;NALOD=1.37;NLOD=6.6;POPAF=6;TLOD=3.02\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:22,0:0.041:22:18,0:0,0:22,0:0,22,0,0\t0/1:16,4:0.21:20:13,0:0,0:16,4:0,16,0,4\n",
            "chr21\t46599729\t.\tG\tC\t.\tbase_qual;weak_evidence\tAS_FilterStatus=weak_evidence,base_qual;AS_SB_TABLE=0,16|0,2;DP=18;ECNT=1;GERMQ=1;MBQ=18,19;MFRL=356,357;MMQ=60,60;MPOS=36;NALOD=0.947;NLOD=2.31;POPAF=6;TLOD=3.04\tGT:AD:AF:DP:F1R2:F2R1:FAD:SB\t0/0:8,0:0.102:8:0,0:0,0:8,0:0,8,0,0\t0/1:8,2:0.255:10:1,0:0,0:8,2:0,8,0,2\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bcftools view -H filtered.vcf.gz | cut -f 7"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "V1Bkk5UkCAPk",
        "outputId": "edea9505-d944-4131-c957-5b3e647bed41"
      },
      "execution_count": 92,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "base_qual;haplotype;weak_evidence\n",
            "base_qual;haplotype;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual\n",
            "weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;haplotype\n",
            "base_qual;haplotype\n",
            "base_qual;weak_evidence\n",
            "PASS\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "weak_evidence\n",
            "base_qual;haplotype;weak_evidence\n",
            "haplotype;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual\n",
            "base_qual;weak_evidence\n",
            "weak_evidence\n",
            "base_qual;weak_evidence\n",
            "weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual\n",
            "base_qual;haplotype\n",
            "base_qual;haplotype\n",
            "base_qual\n",
            "base_qual\n",
            "weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "weak_evidence\n",
            "base_qual\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual\n",
            "weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n",
            "base_qual;weak_evidence\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bcftools view -f 'PASS' filtered.vcf.gz -o somatic.pass.vcf"
      ],
      "metadata": {
        "id": "GBoKFMeXCDwl"
      },
      "execution_count": 95,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Annotation using vcf2maf"
      ],
      "metadata": {
        "id": "K0EzSVkh6C9v"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Docker install"
      ],
      "metadata": {
        "id": "gxrFHsUYTiWp"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "jupyter notebook에서 docker 사용하는 방법: https://github.com/indigo-dc/udocker"
      ],
      "metadata": {
        "id": "vBUkhurM9tUS"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 12s\n",
        "!pip install udocker\n",
        "!udocker --allow-root install"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "63JVxiGe6sSD",
        "outputId": "a7173312-8ba2-4119-cf95-9322af8ddefa"
      },
      "execution_count": 97,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting udocker\n",
            "  Using cached udocker-1.3.17-py2.py3-none-any.whl.metadata (37 kB)\n",
            "Using cached udocker-1.3.17-py2.py3-none-any.whl (119 kB)\n",
            "Installing collected packages: udocker\n",
            "Successfully installed udocker-1.3.17\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "dockerhub 등에서 필요한 패키지 검색 후 다운로드\n",
        "\n",
        "https://hub.docker.com/u/oghzzang\n"
      ],
      "metadata": {
        "id": "NgFox2PPK7HM"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 57s\n",
        "!udocker --allow-root pull oghzzang/vcf2maf:v1.6.18_chr21"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "v6BT6i3KH1fG",
        "outputId": "224e1d76-d4fc-42fa-c784-82a4bd4d4e3d"
      },
      "execution_count": 117,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Info: downloading layer sha256:8bb6fb2df1a7e2501927127b5babd8e7fb5c0c2bbd2bed41d8238cc15121b7be\n",
            "Info: downloading layer sha256:3971f078252b77ce63ae67ab86cd97e58deffe0890d707a5fa332f2a8fbedef9\n",
            "Info: downloading layer sha256:270761053b031626108f9054bbbf515578437905518e918bbe2db5e25d287bd2\n",
            "Info: downloading layer sha256:0102a72864f7c35282e8a1ac6f422b2e80630a8c53c3936f4464916e13ce7888\n",
            "Info: downloading layer sha256:43f67a011cd010e97c2f2524a68698afc8b6d081314a624bc5e2bd07b78f3eb4\n",
            "Info: downloading layer sha256:a64d47c52cd0c32bb765c00dba404b769dacda19668cf1f50caf24ef817ada5c\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Run vcf2maf"
      ],
      "metadata": {
        "id": "o8KaGNzWTqol"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#runtime: 55s\n",
        "!udocker \\\n",
        "  --allow-root run \\\n",
        "  --rm oghzzang/vcf2maf:v1.6.18_chr21 \\\n",
        "  perl vcf2maf.pl --help"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "llgDhr-_8_2A",
        "outputId": "5cb83acb-3ae6-4b77-f2f0-de559aee7f45"
      },
      "execution_count": 106,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING ebdac9fd-524d-373f-9acf-9606c1f85919                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: perl\n",
            "Usage:\n",
            "     perl vcf2maf.pl --help\n",
            "     perl vcf2maf.pl --input-vcf WD4086.vcf --output-maf WD4086.maf --tumor-id WD4086 --normal-id NB4086\n",
            "\n",
            "Options:\n",
            "     --input-vcf      Path to input file in VCF format\n",
            "     --output-maf     Path to output MAF file\n",
            "     --tmp-dir        Folder to retain intermediate VCFs after runtime [Default: Folder containing input VCF]\n",
            "     --tumor-id       Tumor_Sample_Barcode to report in the MAF [TUMOR]\n",
            "     --normal-id      Matched_Norm_Sample_Barcode to report in the MAF [NORMAL]\n",
            "     --vcf-tumor-id   Tumor sample ID used in VCF's genotype columns [--tumor-id]\n",
            "     --vcf-normal-id  Matched normal ID used in VCF's genotype columns [--normal-id]\n",
            "     --custom-enst    List of custom ENST IDs that override canonical selection\n",
            "     --vep-path       Folder containing the vep script [~/vep]\n",
            "     --vep-data       VEP's base cache/plugin directory [~/.vep]\n",
            "     --vep-forks      Number of forked processes to use when running VEP [4]\n",
            "     --buffer-size    Number of variants VEP loads at a time; Reduce this for low memory systems [5000]\n",
            "     --any-allele     When reporting co-located variants, allow mismatched variant alleles too\n",
            "     --inhibit-vep    Skip running VEP, but extract VEP annotation in VCF if found\n",
            "     --online         Use useastdb.ensembl.org instead of local cache (supports only GRCh38 VCFs listing <100 events)\n",
            "     --ref-fasta      Reference FASTA file [~/.vep/homo_sapiens/95_GRCh37/Homo_sapiens.GRCh37.75.dna.primary_assembly.fa.gz]\n",
            "     --filter-vcf     A VCF for FILTER tag common_variant. Set to 0 to disable [~/.vep/ExAC_nonTCGA.r0.3.1.sites.vep.vcf.gz]\n",
            "     --max-filter-ac  Use tag common_variant if the filter-vcf reports a subpopulation AC higher than this [10]\n",
            "     --species        Ensembl-friendly name of species (e.g. mus_musculus for mouse) [homo_sapiens]\n",
            "     --ncbi-build     NCBI reference assembly of variants MAF (e.g. GRCm38 for mouse) [GRCh37]\n",
            "     --cache-version  Version of offline cache to use with VEP (e.g. 75, 84, 91) [Default: Installed version]\n",
            "     --maf-center     Variant calling center to report in MAF [.]\n",
            "     --retain-info    Comma-delimited names of INFO fields to retain as extra columns in MAF []\n",
            "     --retain-fmt     Comma-delimited names of FORMAT fields to retain as extra columns in MAF []\n",
            "     --min-hom-vaf    If GT undefined in VCF, minimum allele fraction to call a variant homozygous [0.7]\n",
            "     --remap-chain    Chain file to remap variants to a different assembly before running VEP\n",
            "     --help           Print a brief help message and quit\n",
            "     --man            Print the detailed manual\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# runtime: 49s\n",
        "!udocker \\\n",
        "  --allow-root run \\\n",
        "  -v /content/drive/MyDrive/AMC_NGS1/02_linux:/content/drive/MyDrive/AMC_NGS1/02_linux \\\n",
        "  --rm oghzzang/vcf2maf:v1.6.18_chr21 \\\n",
        "  perl vcf2maf.pl \\\n",
        "  --input-vcf  /content/drive/MyDrive/AMC_NGS1/02_linux/data/somatic.pass.vcf \\\n",
        "  --output-maf /content/drive/MyDrive/AMC_NGS1/02_linux/data/somatic.pass.maf \\\n",
        "  --ref-fasta  /content/drive/MyDrive/AMC_NGS1/02_linux/resource/reference/hg38.chr21.fa \\\n",
        "  --vep-data   /.vep \\\n",
        "  --vep-path   /usr/local/bin \\\n",
        "  --ncbi-build GRCh38 \\\n",
        "  --cache-version 112 \\\n",
        "  --tumor-id tumor \\\n",
        "  --normal-id normal"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NFBHkW2f-doz",
        "outputId": "4dd7c508-a9ff-4fcb-d268-6b7e2bf849a7"
      },
      "execution_count": 121,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 89149a74-60ef-3a4a-974b-54e6b29faaf5                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: perl\n",
            "STATUS: Running VEP and writing to: /content/drive/MyDrive/AMC_NGS1/02_linux/data/somatic.pass.vep.vcf\n",
            "Possible precedence issue with control flow operator at /usr/local/lib/site_perl/5.26.2/Bio/DB/IndexedBase.pm line 805.\n",
            "WARNING: Unrecognized biotype \"protein_coding_CDS_not_defined\". Assigning lowest priority!\n",
            "WARNING: Unrecognized biotype \"protein_coding_CDS_not_defined\". Assigning lowest priority!\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "ls -al somatic.pass.vep.vcf somatic.pass.maf"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "7SQS2KO5UPVn",
        "outputId": "679bbb52-35dd-43b1-8a5e-8707f7daa6c8"
      },
      "execution_count": 122,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "-rw------- 1 root root  2550 Aug 21 01:49 somatic.pass.maf\n",
            "-rw------- 1 root root 19635 Aug 21 01:49 somatic.pass.vep.vcf\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "maf 파일 확인"
      ],
      "metadata": {
        "id": "Va6BNBC-WVbu"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "/content/drive/MyDrive/AMC_NGS1/02_linux/data/somatic.pass.maf"
      ],
      "metadata": {
        "id": "sX2Y4sTrUP8d"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# notebook에서만 실행\n",
        "import pandas as pd\n",
        "from IPython.display import display\n",
        "\n",
        "file_path = \"somatic.pass.maf\"\n",
        "\n",
        "df = pd.read_csv(\n",
        "    file_path,\n",
        "    sep=\"\\t\",\n",
        "    comment=\"#\",           # '#'으로 시작하는 주석 줄은 무시\n",
        "    skip_blank_lines=True\n",
        ")\n",
        "\n",
        "display(df)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 130
        },
        "id": "aHRUUhW3WJ6c",
        "outputId": "8bd7f9f4-080c-43b3-c932-bcf240cf525c"
      },
      "execution_count": 123,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "  Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  Start_Position  \\\n",
              "0        CCT8           10694      .     GRCh38      chr21        29056769   \n",
              "\n",
              "   End_Position Strand Variant_Classification Variant_Type  ... ExAC_AC_AN  \\\n",
              "0      29056769      +                 Intron          SNP  ...        NaN   \n",
              "\n",
              "  ExAC_AC_AN_AFR ExAC_AC_AN_AMR ExAC_AC_AN_EAS  ExAC_AC_AN_FIN ExAC_AC_AN_NFE  \\\n",
              "0            NaN            NaN            NaN             NaN            NaN   \n",
              "\n",
              "  ExAC_AC_AN_OTH ExAC_AC_AN_SAS ExAC_FILTER   vcf_pos  \n",
              "0            NaN            NaN         NaN  29056769  \n",
              "\n",
              "[1 rows x 124 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-32a16f56-a3be-4d6c-b313-de19e1b11d11\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Hugo_Symbol</th>\n",
              "      <th>Entrez_Gene_Id</th>\n",
              "      <th>Center</th>\n",
              "      <th>NCBI_Build</th>\n",
              "      <th>Chromosome</th>\n",
              "      <th>Start_Position</th>\n",
              "      <th>End_Position</th>\n",
              "      <th>Strand</th>\n",
              "      <th>Variant_Classification</th>\n",
              "      <th>Variant_Type</th>\n",
              "      <th>...</th>\n",
              "      <th>ExAC_AC_AN</th>\n",
              "      <th>ExAC_AC_AN_AFR</th>\n",
              "      <th>ExAC_AC_AN_AMR</th>\n",
              "      <th>ExAC_AC_AN_EAS</th>\n",
              "      <th>ExAC_AC_AN_FIN</th>\n",
              "      <th>ExAC_AC_AN_NFE</th>\n",
              "      <th>ExAC_AC_AN_OTH</th>\n",
              "      <th>ExAC_AC_AN_SAS</th>\n",
              "      <th>ExAC_FILTER</th>\n",
              "      <th>vcf_pos</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>CCT8</td>\n",
              "      <td>10694</td>\n",
              "      <td>.</td>\n",
              "      <td>GRCh38</td>\n",
              "      <td>chr21</td>\n",
              "      <td>29056769</td>\n",
              "      <td>29056769</td>\n",
              "      <td>+</td>\n",
              "      <td>Intron</td>\n",
              "      <td>SNP</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>29056769</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>1 rows × 124 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-32a16f56-a3be-4d6c-b313-de19e1b11d11')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-32a16f56-a3be-4d6c-b313-de19e1b11d11 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-32a16f56-a3be-4d6c-b313-de19e1b11d11');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "  <div id=\"id_a289021a-af7b-4193-9ff0-de38926764b4\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('df')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_a289021a-af7b-4193-9ff0-de38926764b4 button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('df');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df"
            }
          },
          "metadata": {}
        }
      ]
    }
  ]
}